// @pulumi/azure-native/types/enums/azurearcdata/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlManagedInstanceSkuTier = exports.PostgresInstanceSkuTier = exports.ExtendedLocationTypes = exports.AccountProvisioningMode = exports.v20220301preview = exports.v20211101 = exports.v20210801 = exports.v20210701preview = exports.v20210601preview = void 0;
// Export sub-modules:
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20210701preview = require("./v20210701preview");
exports.v20210701preview = v20210701preview;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
const v20211101 = require("./v20211101");
exports.v20211101 = v20211101;
const v20220301preview = require("./v20220301preview");
exports.v20220301preview = v20220301preview;
exports.AccountProvisioningMode = {
    Automatic: "automatic",
    Manual: "manual",
};
exports.ExtendedLocationTypes = {
    CustomLocation: "CustomLocation",
};
exports.PostgresInstanceSkuTier = {
    Hyperscale: "Hyperscale",
};
exports.SqlManagedInstanceSkuTier = {
    GeneralPurpose: "GeneralPurpose",
    BusinessCritical: "BusinessCritical",
};
