// @pulumi/azure-native/cognitiveservices/v20160201preview/cognitiveServicesAccount.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CognitiveServicesAccount = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Cognitive Services Account is an Azure resource representing the provisioned account, its type, location and SKU.
 *
 * @deprecated Version v20160201preview will be removed in the next major version of the provider. Upgrade to version v20170418 or later.
 */
class CognitiveServicesAccount extends pulumi.CustomResource {
    /**
     * Create a CognitiveServicesAccount resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160201preview will be removed in the next major version of the provider. Upgrade to version v20170418 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("CognitiveServicesAccount is deprecated: Version v20160201preview will be removed in the next major version of the provider. Upgrade to version v20170418 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sku === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sku'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["endpoint"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["endpoint"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cognitiveservices:CognitiveServicesAccount" }, { type: "azure-native:cognitiveservices/v20170418:CognitiveServicesAccount" }, { type: "azure-native:cognitiveservices/v20210430:CognitiveServicesAccount" }, { type: "azure-native:cognitiveservices/v20211001:CognitiveServicesAccount" }, { type: "azure-native:cognitiveservices/v20220301:CognitiveServicesAccount" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(CognitiveServicesAccount.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing CognitiveServicesAccount resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("CognitiveServicesAccount is deprecated: Version v20160201preview will be removed in the next major version of the provider. Upgrade to version v20170418 or later.");
        return new CognitiveServicesAccount(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CognitiveServicesAccount.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CognitiveServicesAccount.__pulumiType;
    }
}
exports.CognitiveServicesAccount = CognitiveServicesAccount;
/** @internal */
CognitiveServicesAccount.__pulumiType = 'azure-native:cognitiveservices/v20160201preview:CognitiveServicesAccount';
