// @pulumi/azure-native/dbforpostgresql/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210615privatepreview = exports.v20210601preview = exports.v20210601 = exports.v20210410privatepreview = exports.v20201105preview = exports.v20201005privatepreview = exports.v20200214privatepreview = exports.v20200214preview = exports.v20200101privatepreview = exports.v20200101 = exports.v20180601privatepreview = exports.v20180601 = exports.v20171201preview = exports.v20171201 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./configuration"), exports);
__exportStar(require("./database"), exports);
__exportStar(require("./firewallRule"), exports);
__exportStar(require("./getConfiguration"), exports);
__exportStar(require("./getDatabase"), exports);
__exportStar(require("./getFirewallRule"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getServer"), exports);
__exportStar(require("./getServerAdministrator"), exports);
__exportStar(require("./getServerKey"), exports);
__exportStar(require("./getServerSecurityAlertPolicy"), exports);
__exportStar(require("./getVirtualNetworkRule"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./server"), exports);
__exportStar(require("./serverAdministrator"), exports);
__exportStar(require("./serverKey"), exports);
__exportStar(require("./serverSecurityAlertPolicy"), exports);
__exportStar(require("./virtualNetworkRule"), exports);
// Export enums:
__exportStar(require("../types/enums/dbforpostgresql"), exports);
// Export sub-modules:
const v20171201 = require("./v20171201");
exports.v20171201 = v20171201;
const v20171201preview = require("./v20171201preview");
exports.v20171201preview = v20171201preview;
const v20180601 = require("./v20180601");
exports.v20180601 = v20180601;
const v20180601privatepreview = require("./v20180601privatepreview");
exports.v20180601privatepreview = v20180601privatepreview;
const v20200101 = require("./v20200101");
exports.v20200101 = v20200101;
const v20200101privatepreview = require("./v20200101privatepreview");
exports.v20200101privatepreview = v20200101privatepreview;
const v20200214preview = require("./v20200214preview");
exports.v20200214preview = v20200214preview;
const v20200214privatepreview = require("./v20200214privatepreview");
exports.v20200214privatepreview = v20200214privatepreview;
const v20201005privatepreview = require("./v20201005privatepreview");
exports.v20201005privatepreview = v20201005privatepreview;
const v20201105preview = require("./v20201105preview");
exports.v20201105preview = v20201105preview;
const v20210410privatepreview = require("./v20210410privatepreview");
exports.v20210410privatepreview = v20210410privatepreview;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20210615privatepreview = require("./v20210615privatepreview");
exports.v20210615privatepreview = v20210615privatepreview;
// Import resources to register:
const configuration_1 = require("./configuration");
const database_1 = require("./database");
const firewallRule_1 = require("./firewallRule");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const server_1 = require("./server");
const serverAdministrator_1 = require("./serverAdministrator");
const serverKey_1 = require("./serverKey");
const serverSecurityAlertPolicy_1 = require("./serverSecurityAlertPolicy");
const virtualNetworkRule_1 = require("./virtualNetworkRule");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dbforpostgresql:Configuration":
                return new configuration_1.Configuration(name, undefined, { urn });
            case "azure-native:dbforpostgresql:Database":
                return new database_1.Database(name, undefined, { urn });
            case "azure-native:dbforpostgresql:FirewallRule":
                return new firewallRule_1.FirewallRule(name, undefined, { urn });
            case "azure-native:dbforpostgresql:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:dbforpostgresql:Server":
                return new server_1.Server(name, undefined, { urn });
            case "azure-native:dbforpostgresql:ServerAdministrator":
                return new serverAdministrator_1.ServerAdministrator(name, undefined, { urn });
            case "azure-native:dbforpostgresql:ServerKey":
                return new serverKey_1.ServerKey(name, undefined, { urn });
            case "azure-native:dbforpostgresql:ServerSecurityAlertPolicy":
                return new serverSecurityAlertPolicy_1.ServerSecurityAlertPolicy(name, undefined, { urn });
            case "azure-native:dbforpostgresql:VirtualNetworkRule":
                return new virtualNetworkRule_1.VirtualNetworkRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dbforpostgresql", _module);
