// @pulumi/azure-native/network/v20180301preview/zone.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Zone = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Describes a DNS zone.
 *
 * @deprecated Version v20180301preview will be removed in the next major version of the provider. Upgrade to version v20180501 or later.
 */
class Zone extends pulumi.CustomResource {
    /**
     * Create a Zone resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180301preview will be removed in the next major version of the provider. Upgrade to version v20180501 or later. */
    constructor(name, args, opts) {
        var _a;
        pulumi.log.warn("Zone is deprecated: Version v20180301preview will be removed in the next major version of the provider. Upgrade to version v20180501 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["registrationVirtualNetworks"] = args ? args.registrationVirtualNetworks : undefined;
            resourceInputs["resolutionVirtualNetworks"] = args ? args.resolutionVirtualNetworks : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["zoneName"] = args ? args.zoneName : undefined;
            resourceInputs["zoneType"] = (_a = (args ? args.zoneType : undefined)) !== null && _a !== void 0 ? _a : "Public";
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["maxNumberOfRecordSets"] = undefined /*out*/;
            resourceInputs["maxNumberOfRecordsPerRecordSet"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nameServers"] = undefined /*out*/;
            resourceInputs["numberOfRecordSets"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maxNumberOfRecordSets"] = undefined /*out*/;
            resourceInputs["maxNumberOfRecordsPerRecordSet"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nameServers"] = undefined /*out*/;
            resourceInputs["numberOfRecordSets"] = undefined /*out*/;
            resourceInputs["registrationVirtualNetworks"] = undefined /*out*/;
            resourceInputs["resolutionVirtualNetworks"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["zoneType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network:Zone" }, { type: "azure-native:network/v20150504preview:Zone" }, { type: "azure-native:network/v20160401:Zone" }, { type: "azure-native:network/v20170901:Zone" }, { type: "azure-native:network/v20171001:Zone" }, { type: "azure-native:network/v20180501:Zone" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Zone.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Zone resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Zone is deprecated: Version v20180301preview will be removed in the next major version of the provider. Upgrade to version v20180501 or later.");
        return new Zone(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Zone.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Zone.__pulumiType;
    }
}
exports.Zone = Zone;
/** @internal */
Zone.__pulumiType = 'azure-native:network/v20180301preview:Zone';
