// @pulumi/azure-native/types/enums/streamanalytics/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.OutputStartMode = exports.OutputErrorPolicy = exports.JsonOutputSerializationFormat = exports.EventsOutOfOrderPolicy = exports.EventSerializationType = exports.Encoding = exports.CompatibilityLevel = exports.ClusterSkuName = exports.v20211001preview = exports.v20200301preview = exports.v20200301 = exports.v20170401preview = exports.v20160301 = void 0;
// Export sub-modules:
const v20160301 = require("./v20160301");
exports.v20160301 = v20160301;
const v20170401preview = require("./v20170401preview");
exports.v20170401preview = v20170401preview;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20211001preview = require("./v20211001preview");
exports.v20211001preview = v20211001preview;
exports.ClusterSkuName = {
    /**
     * The default SKU.
     */
    Default: "Default",
};
exports.CompatibilityLevel = {
    CompatibilityLevel_1_0: "1.0",
};
exports.Encoding = {
    UTF8: "UTF8",
};
exports.EventSerializationType = {
    Csv: "Csv",
    Avro: "Avro",
    Json: "Json",
};
exports.EventsOutOfOrderPolicy = {
    Adjust: "Adjust",
    Drop: "Drop",
};
exports.JsonOutputSerializationFormat = {
    LineSeparated: "LineSeparated",
    Array: "Array",
};
exports.OutputErrorPolicy = {
    Stop: "Stop",
    Drop: "Drop",
};
exports.OutputStartMode = {
    JobStartTime: "JobStartTime",
    CustomTime: "CustomTime",
    LastOutputEventTime: "LastOutputEventTime",
};
exports.SkuName = {
    Standard: "Standard",
};
