// @pulumi/azure-native/types/enums/search/v20200801preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.SharedPrivateLinkResourceStatus = exports.SharedPrivateLinkResourceProvisioningState = exports.PublicNetworkAccess = exports.PrivateLinkServiceConnectionStatus = exports.IdentityType = exports.HostingMode = void 0;
exports.HostingMode = {
    Default: "default",
    HighDensity: "highDensity",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.PrivateLinkServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "enabled",
    Disabled: "disabled",
};
exports.SharedPrivateLinkResourceProvisioningState = {
    Updating: "Updating",
    Deleting: "Deleting",
    Failed: "Failed",
    Succeeded: "Succeeded",
    Incomplete: "Incomplete",
};
exports.SharedPrivateLinkResourceStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.SkuName = {
    Free: "free",
    Basic: "basic",
    Standard: "standard",
    Standard2: "standard2",
    Standard3: "standard3",
    Storage_optimized_l1: "storage_optimized_l1",
    Storage_optimized_l2: "storage_optimized_l2",
};
