// @pulumi/azure-native/storagesync/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20200901 = exports.v20200301 = exports.v20191001 = exports.v20190601 = exports.v20190301 = exports.v20190201 = exports.v20181001 = exports.v20180701 = exports.v20180402 = exports.v20170605preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./cloudEndpoint"), exports);
__exportStar(require("./getCloudEndpoint"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getRegisteredServer"), exports);
__exportStar(require("./getServerEndpoint"), exports);
__exportStar(require("./getStorageSyncService"), exports);
__exportStar(require("./getSyncGroup"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./registeredServer"), exports);
__exportStar(require("./serverEndpoint"), exports);
__exportStar(require("./storageSyncService"), exports);
__exportStar(require("./syncGroup"), exports);
// Export enums:
__exportStar(require("../types/enums/storagesync"), exports);
// Export sub-modules:
const v20170605preview = require("./v20170605preview");
exports.v20170605preview = v20170605preview;
const v20180402 = require("./v20180402");
exports.v20180402 = v20180402;
const v20180701 = require("./v20180701");
exports.v20180701 = v20180701;
const v20181001 = require("./v20181001");
exports.v20181001 = v20181001;
const v20190201 = require("./v20190201");
exports.v20190201 = v20190201;
const v20190301 = require("./v20190301");
exports.v20190301 = v20190301;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
// Import resources to register:
const cloudEndpoint_1 = require("./cloudEndpoint");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const registeredServer_1 = require("./registeredServer");
const serverEndpoint_1 = require("./serverEndpoint");
const storageSyncService_1 = require("./storageSyncService");
const syncGroup_1 = require("./syncGroup");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:storagesync:CloudEndpoint":
                return new cloudEndpoint_1.CloudEndpoint(name, undefined, { urn });
            case "azure-native:storagesync:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:storagesync:RegisteredServer":
                return new registeredServer_1.RegisteredServer(name, undefined, { urn });
            case "azure-native:storagesync:ServerEndpoint":
                return new serverEndpoint_1.ServerEndpoint(name, undefined, { urn });
            case "azure-native:storagesync:StorageSyncService":
                return new storageSyncService_1.StorageSyncService(name, undefined, { urn });
            case "azure-native:storagesync:SyncGroup":
                return new syncGroup_1.SyncGroup(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "storagesync", _module);
