// @pulumi/azure-native/types/enums/storagecache/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UsernameSource = exports.StorageTargetType = exports.ProvisioningStateType = exports.NfsAccessRuleScope = exports.NfsAccessRuleAccess = exports.CacheIdentityType = exports.v20220101 = exports.v20210901 = exports.v20210501 = exports.v20210301 = exports.v20201001 = exports.v20200301 = exports.v20191101 = exports.v20190801preview = void 0;
// Export sub-modules:
const v20190801preview = require("./v20190801preview");
exports.v20190801preview = v20190801preview;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
const v20210901 = require("./v20210901");
exports.v20210901 = v20210901;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
exports.CacheIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
};
exports.NfsAccessRuleAccess = {
    No: "no",
    Ro: "ro",
    Rw: "rw",
};
exports.NfsAccessRuleScope = {
    Default: "default",
    Network: "network",
    Host: "host",
};
exports.ProvisioningStateType = {
    Succeeded: "Succeeded",
    Failed: "Failed",
    Cancelled: "Cancelled",
    Creating: "Creating",
    Deleting: "Deleting",
    Updating: "Updating",
};
exports.StorageTargetType = {
    Nfs3: "nfs3",
    Clfs: "clfs",
    Unknown: "unknown",
    BlobNfs: "blobNfs",
};
exports.UsernameSource = {
    AD: "AD",
    LDAP: "LDAP",
    File: "File",
    None: "None",
};
