// @pulumi/azure-native/azurestackhci/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220301 = exports.v20220101 = exports.v20210901 = exports.v20210101preview = exports.v20201001 = exports.v20200301preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./arcSetting"), exports);
__exportStar(require("./cluster"), exports);
__exportStar(require("./extension"), exports);
__exportStar(require("./getArcSetting"), exports);
__exportStar(require("./getCluster"), exports);
__exportStar(require("./getExtension"), exports);
// Export enums:
__exportStar(require("../types/enums/azurestackhci"), exports);
// Export sub-modules:
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20210101preview = require("./v20210101preview");
exports.v20210101preview = v20210101preview;
const v20210901 = require("./v20210901");
exports.v20210901 = v20210901;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
const v20220301 = require("./v20220301");
exports.v20220301 = v20220301;
// Import resources to register:
const arcSetting_1 = require("./arcSetting");
const cluster_1 = require("./cluster");
const extension_1 = require("./extension");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azurestackhci:ArcSetting":
                return new arcSetting_1.ArcSetting(name, undefined, { urn });
            case "azure-native:azurestackhci:Cluster":
                return new cluster_1.Cluster(name, undefined, { urn });
            case "azure-native:azurestackhci:Extension":
                return new extension_1.Extension(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azurestackhci", _module);
