// @pulumi/azure-native/management/v20191101/managementGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The management group details.
 *
 * @deprecated Version v20191101 will be removed in the next major version of the provider. Upgrade to version v20200501 or later.
 */
class ManagementGroup extends pulumi.CustomResource {
    /**
     * Create a ManagementGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20191101 will be removed in the next major version of the provider. Upgrade to version v20200501 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ManagementGroup is deprecated: Version v20191101 will be removed in the next major version of the provider. Upgrade to version v20200501 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["details"] = args ? args.details : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["groupId"] = args ? args.groupId : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["children"] = undefined /*out*/;
            resourceInputs["path"] = undefined /*out*/;
            resourceInputs["roles"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["children"] = undefined /*out*/;
            resourceInputs["details"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["path"] = undefined /*out*/;
            resourceInputs["roles"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:management:ManagementGroup" }, { type: "azure-native:management/v20171101preview:ManagementGroup" }, { type: "azure-native:management/v20180101preview:ManagementGroup" }, { type: "azure-native:management/v20180301preview:ManagementGroup" }, { type: "azure-native:management/v20200201:ManagementGroup" }, { type: "azure-native:management/v20200501:ManagementGroup" }, { type: "azure-native:management/v20201001:ManagementGroup" }, { type: "azure-native:management/v20210401:ManagementGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagementGroup.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ManagementGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ManagementGroup is deprecated: Version v20191101 will be removed in the next major version of the provider. Upgrade to version v20200501 or later.");
        return new ManagementGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagementGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagementGroup.__pulumiType;
    }
}
exports.ManagementGroup = ManagementGroup;
/** @internal */
ManagementGroup.__pulumiType = 'azure-native:management/v20191101:ManagementGroup';
