// @pulumi/azure-native/migrate/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210801 = exports.v20210101 = exports.v20200501 = exports.v20191001preview = exports.v20191001 = exports.v20180901preview = exports.v20180202 = exports.v20171111preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./assessment"), exports);
__exportStar(require("./getAssessment"), exports);
__exportStar(require("./getGroup"), exports);
__exportStar(require("./getHyperVCollector"), exports);
__exportStar(require("./getImportCollector"), exports);
__exportStar(require("./getMigrateProject"), exports);
__exportStar(require("./getMoveCollection"), exports);
__exportStar(require("./getMoveResource"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getProject"), exports);
__exportStar(require("./getProjectKeys"), exports);
__exportStar(require("./getServerCollector"), exports);
__exportStar(require("./getSolution"), exports);
__exportStar(require("./getSolutionConfig"), exports);
__exportStar(require("./getVMwareCollector"), exports);
__exportStar(require("./group"), exports);
__exportStar(require("./hyperVCollector"), exports);
__exportStar(require("./importCollector"), exports);
__exportStar(require("./migrateProject"), exports);
__exportStar(require("./moveCollection"), exports);
__exportStar(require("./moveResource"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./project"), exports);
__exportStar(require("./serverCollector"), exports);
__exportStar(require("./solution"), exports);
__exportStar(require("./vmwareCollector"), exports);
// Export enums:
__exportStar(require("../types/enums/migrate"), exports);
// Export sub-modules:
const v20171111preview = require("./v20171111preview");
exports.v20171111preview = v20171111preview;
const v20180202 = require("./v20180202");
exports.v20180202 = v20180202;
const v20180901preview = require("./v20180901preview");
exports.v20180901preview = v20180901preview;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191001preview = require("./v20191001preview");
exports.v20191001preview = v20191001preview;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
// Import resources to register:
const assessment_1 = require("./assessment");
const group_1 = require("./group");
const hyperVCollector_1 = require("./hyperVCollector");
const importCollector_1 = require("./importCollector");
const migrateProject_1 = require("./migrateProject");
const moveCollection_1 = require("./moveCollection");
const moveResource_1 = require("./moveResource");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const project_1 = require("./project");
const serverCollector_1 = require("./serverCollector");
const solution_1 = require("./solution");
const vmwareCollector_1 = require("./vmwareCollector");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:migrate:Assessment":
                return new assessment_1.Assessment(name, undefined, { urn });
            case "azure-native:migrate:Group":
                return new group_1.Group(name, undefined, { urn });
            case "azure-native:migrate:HyperVCollector":
                return new hyperVCollector_1.HyperVCollector(name, undefined, { urn });
            case "azure-native:migrate:ImportCollector":
                return new importCollector_1.ImportCollector(name, undefined, { urn });
            case "azure-native:migrate:MigrateProject":
                return new migrateProject_1.MigrateProject(name, undefined, { urn });
            case "azure-native:migrate:MoveCollection":
                return new moveCollection_1.MoveCollection(name, undefined, { urn });
            case "azure-native:migrate:MoveResource":
                return new moveResource_1.MoveResource(name, undefined, { urn });
            case "azure-native:migrate:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:migrate:Project":
                return new project_1.Project(name, undefined, { urn });
            case "azure-native:migrate:ServerCollector":
                return new serverCollector_1.ServerCollector(name, undefined, { urn });
            case "azure-native:migrate:Solution":
                return new solution_1.Solution(name, undefined, { urn });
            case "azure-native:migrate:VMwareCollector":
                return new vmwareCollector_1.VMwareCollector(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "migrate", _module);
