// @pulumi/azure-native/types/enums/elasticsan/v20211120preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolumeCreateOption = exports.Tier = exports.StorageTargetType = exports.Name = exports.EncryptionType = exports.Action = void 0;
exports.Action = {
    Allow: "Allow",
};
exports.EncryptionType = {
    /**
     * Volume is encrypted at rest with Platform managed key. It is the default encryption type. This is not a valid encryption type for disk encryption sets.
     */
    EncryptionAtRestWithPlatformKey: "EncryptionAtRestWithPlatformKey",
    /**
     * Volume is encrypted at rest with Customer managed key that can be changed and revoked by a customer.
     */
    EncryptionAtRestWithCustomerKey: "EncryptionAtRestWithCustomerKey",
    /**
     * Volume is encrypted at rest with 2 layers of encryption. One of the keys is Customer managed and the other key is Platform managed.
     */
    EncryptionAtRestWithPlatformAndCustomerKeys: "EncryptionAtRestWithPlatformAndCustomerKeys",
};
exports.Name = {
    /**
     * Standard locally redundant storage
     */
    Standard_LRS: "Standard_LRS",
    /**
     * Premium locally redundant storage
     */
    Premium_LRS: "Premium_LRS",
    /**
     * Standard zone redundant storage
     */
    Standard_ZRS: "Standard_ZRS",
};
exports.StorageTargetType = {
    Iscsi: "Iscsi",
    None: "None",
};
exports.Tier = {
    /**
     * Hero
     */
    Hero: "Hero",
    /**
     * Satellite
     */
    Satellite: "Satellite",
    /**
     * Hub
     */
    Hub: "Hub",
};
exports.VolumeCreateOption = {
    None: "None",
    FromVolume: "FromVolume",
    FromDiskSnapshot: "FromDiskSnapshot",
    Export: "Export",
};
