// @pulumi/azure-native/hybriddata/v20160601/jobDefinition.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Job Definition.
 *
 * @deprecated Version v20160601 will be removed in the next major version of the provider. Upgrade to version v20190601 or later.
 */
class JobDefinition extends pulumi.CustomResource {
    /**
     * Create a JobDefinition resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160601 will be removed in the next major version of the provider. Upgrade to version v20190601 or later. */
    constructor(name, args, opts) {
        var _a;
        pulumi.log.warn("JobDefinition is deprecated: Version v20160601 will be removed in the next major version of the provider. Upgrade to version v20190601 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.dataManagerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataManagerName'");
            }
            if ((!args || args.dataServiceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataServiceName'");
            }
            if ((!args || args.dataSinkId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataSinkId'");
            }
            if ((!args || args.dataSourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataSourceId'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.state === undefined) && !opts.urn) {
                throw new Error("Missing required property 'state'");
            }
            resourceInputs["customerSecrets"] = args ? args.customerSecrets : undefined;
            resourceInputs["dataManagerName"] = args ? args.dataManagerName : undefined;
            resourceInputs["dataServiceInput"] = args ? args.dataServiceInput : undefined;
            resourceInputs["dataServiceName"] = args ? args.dataServiceName : undefined;
            resourceInputs["dataSinkId"] = args ? args.dataSinkId : undefined;
            resourceInputs["dataSourceId"] = args ? args.dataSourceId : undefined;
            resourceInputs["jobDefinitionName"] = args ? args.jobDefinitionName : undefined;
            resourceInputs["lastModifiedTime"] = args ? args.lastModifiedTime : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["runLocation"] = args ? args.runLocation : undefined;
            resourceInputs["schedules"] = args ? args.schedules : undefined;
            resourceInputs["state"] = args ? args.state : undefined;
            resourceInputs["userConfirmation"] = (_a = (args ? args.userConfirmation : undefined)) !== null && _a !== void 0 ? _a : "NotRequired";
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["customerSecrets"] = undefined /*out*/;
            resourceInputs["dataServiceInput"] = undefined /*out*/;
            resourceInputs["dataSinkId"] = undefined /*out*/;
            resourceInputs["dataSourceId"] = undefined /*out*/;
            resourceInputs["lastModifiedTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["runLocation"] = undefined /*out*/;
            resourceInputs["schedules"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["userConfirmation"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:hybriddata:JobDefinition" }, { type: "azure-native:hybriddata/v20190601:JobDefinition" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(JobDefinition.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing JobDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("JobDefinition is deprecated: Version v20160601 will be removed in the next major version of the provider. Upgrade to version v20190601 or later.");
        return new JobDefinition(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of JobDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === JobDefinition.__pulumiType;
    }
}
exports.JobDefinition = JobDefinition;
/** @internal */
JobDefinition.__pulumiType = 'azure-native:hybriddata/v20160601:JobDefinition';
