// @pulumi/azure-native/types/enums/automation/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WindowsUpdateClasses = exports.TokenType = exports.TagOperators = exports.SourceType = exports.SkuNameEnum = exports.ScheduleFrequency = exports.ScheduleDay = exports.RunbookTypeEnum = exports.ResourceIdentityType = exports.OperatingSystemType = exports.LinuxUpdateClasses = exports.EncryptionKeySourceType = exports.ContentSourceType = exports.v20210622 = exports.v20200113preview = exports.v20190601 = exports.v20180630 = exports.v20180115 = exports.v20170515preview = exports.v20151031 = void 0;
// Export sub-modules:
const v20151031 = require("./v20151031");
exports.v20151031 = v20151031;
const v20170515preview = require("./v20170515preview");
exports.v20170515preview = v20170515preview;
const v20180115 = require("./v20180115");
exports.v20180115 = v20180115;
const v20180630 = require("./v20180630");
exports.v20180630 = v20180630;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20200113preview = require("./v20200113preview");
exports.v20200113preview = v20200113preview;
const v20210622 = require("./v20210622");
exports.v20210622 = v20210622;
exports.ContentSourceType = {
    EmbeddedContent: "embeddedContent",
    Uri: "uri",
};
exports.EncryptionKeySourceType = {
    Microsoft_Automation: "Microsoft.Automation",
    Microsoft_Keyvault: "Microsoft.Keyvault",
};
exports.LinuxUpdateClasses = {
    Unclassified: "Unclassified",
    Critical: "Critical",
    Security: "Security",
    Other: "Other",
};
exports.OperatingSystemType = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.RunbookTypeEnum = {
    Script: "Script",
    Graph: "Graph",
    PowerShellWorkflow: "PowerShellWorkflow",
    PowerShell: "PowerShell",
    GraphPowerShellWorkflow: "GraphPowerShellWorkflow",
    GraphPowerShell: "GraphPowerShell",
};
exports.ScheduleDay = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.ScheduleFrequency = {
    OneTime: "OneTime",
    Day: "Day",
    Hour: "Hour",
    Week: "Week",
    Month: "Month",
    /**
     * The minimum allowed interval for Minute schedules is 15 minutes.
     */
    Minute: "Minute",
};
exports.SkuNameEnum = {
    Free: "Free",
    Basic: "Basic",
};
exports.SourceType = {
    VsoGit: "VsoGit",
    VsoTfvc: "VsoTfvc",
    GitHub: "GitHub",
};
exports.TagOperators = {
    All: "All",
    Any: "Any",
};
exports.TokenType = {
    PersonalAccessToken: "PersonalAccessToken",
    Oauth: "Oauth",
};
exports.WindowsUpdateClasses = {
    Unclassified: "Unclassified",
    Critical: "Critical",
    Security: "Security",
    UpdateRollup: "UpdateRollup",
    FeaturePack: "FeaturePack",
    ServicePack: "ServicePack",
    Definition: "Definition",
    Tools: "Tools",
    Updates: "Updates",
};
