// @pulumi/azure-native/storagecache/v20200301/storageTarget.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageTarget = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Type of the Storage Target.
 *
 * @deprecated Version v20200301 will be removed in the next major version of the provider. Upgrade to version v20210301 or later.
 */
class StorageTarget extends pulumi.CustomResource {
    /**
     * Create a StorageTarget resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200301 will be removed in the next major version of the provider. Upgrade to version v20210301 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("StorageTarget is deprecated: Version v20200301 will be removed in the next major version of the provider. Upgrade to version v20210301 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.cacheName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cacheName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.targetType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetType'");
            }
            resourceInputs["cacheName"] = args ? args.cacheName : undefined;
            resourceInputs["clfs"] = args ? args.clfs : undefined;
            resourceInputs["junctions"] = args ? args.junctions : undefined;
            resourceInputs["nfs3"] = args ? args.nfs3 : undefined;
            resourceInputs["provisioningState"] = args ? args.provisioningState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["storageTargetName"] = args ? args.storageTargetName : undefined;
            resourceInputs["targetType"] = args ? args.targetType : undefined;
            resourceInputs["unknown"] = args ? args.unknown : undefined;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["clfs"] = undefined /*out*/;
            resourceInputs["junctions"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nfs3"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["targetType"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["unknown"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storagecache:StorageTarget" }, { type: "azure-native:storagecache/v20190801preview:StorageTarget" }, { type: "azure-native:storagecache/v20191101:StorageTarget" }, { type: "azure-native:storagecache/v20201001:StorageTarget" }, { type: "azure-native:storagecache/v20210301:StorageTarget" }, { type: "azure-native:storagecache/v20210501:StorageTarget" }, { type: "azure-native:storagecache/v20210901:StorageTarget" }, { type: "azure-native:storagecache/v20220101:StorageTarget" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(StorageTarget.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing StorageTarget resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("StorageTarget is deprecated: Version v20200301 will be removed in the next major version of the provider. Upgrade to version v20210301 or later.");
        return new StorageTarget(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of StorageTarget.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StorageTarget.__pulumiType;
    }
}
exports.StorageTarget = StorageTarget;
/** @internal */
StorageTarget.__pulumiType = 'azure-native:storagecache/v20200301:StorageTarget';
