// @pulumi/azure-native/types/enums/app/v20220101preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnauthenticatedClientActionV2 = exports.Type = exports.StorageType = exports.ManagedServiceIdentityType = exports.IngressTransportMethod = exports.ForwardProxyConvention = exports.CookieExpirationConvention = exports.ClientCredentialMethod = exports.BindingType = exports.AppProtocol = exports.ActiveRevisionsMode = exports.AccessMode = void 0;
exports.AccessMode = {
    ReadOnly: "ReadOnly",
    ReadWrite: "ReadWrite",
};
exports.ActiveRevisionsMode = {
    Multiple: "multiple",
    Single: "single",
};
exports.AppProtocol = {
    Http: "http",
    Grpc: "grpc",
};
exports.BindingType = {
    Disabled: "Disabled",
    SniEnabled: "SniEnabled",
};
exports.ClientCredentialMethod = {
    ClientSecretPost: "ClientSecretPost",
};
exports.CookieExpirationConvention = {
    FixedTime: "FixedTime",
    IdentityProviderDerived: "IdentityProviderDerived",
};
exports.ForwardProxyConvention = {
    NoProxy: "NoProxy",
    Standard: "Standard",
    Custom: "Custom",
};
exports.IngressTransportMethod = {
    Auto: "auto",
    Http: "http",
    Http2: "http2",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.StorageType = {
    AzureFile: "AzureFile",
    EmptyDir: "EmptyDir",
};
exports.Type = {
    Liveness: "liveness",
    Readiness: "readiness",
    Startup: "startup",
};
exports.UnauthenticatedClientActionV2 = {
    RedirectToLoginPage: "RedirectToLoginPage",
    AllowAnonymous: "AllowAnonymous",
    Return401: "Return401",
    Return403: "Return403",
};
