// @pulumi/azure-native/types/enums/datashare/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Type = exports.TriggerKind = exports.SynchronizationSettingKind = exports.SynchronizationMode = exports.ShareKind = exports.RecurrenceInterval = exports.OutputType = exports.DataSetMappingKind = exports.DataSetKind = exports.v20210801 = exports.v20201001preview = exports.v20200901 = exports.v20191101 = exports.v20181101preview = void 0;
// Export sub-modules:
const v20181101preview = require("./v20181101preview");
exports.v20181101preview = v20181101preview;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20201001preview = require("./v20201001preview");
exports.v20201001preview = v20201001preview;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
exports.DataSetKind = {
    Blob: "Blob",
    Container: "Container",
    BlobFolder: "BlobFolder",
    AdlsGen2FileSystem: "AdlsGen2FileSystem",
    AdlsGen2Folder: "AdlsGen2Folder",
    AdlsGen2File: "AdlsGen2File",
    AdlsGen1Folder: "AdlsGen1Folder",
    AdlsGen1File: "AdlsGen1File",
    KustoCluster: "KustoCluster",
    KustoDatabase: "KustoDatabase",
    SqlDBTable: "SqlDBTable",
    SqlDWTable: "SqlDWTable",
    SynapseWorkspaceSqlPoolTable: "SynapseWorkspaceSqlPoolTable",
};
exports.DataSetMappingKind = {
    Blob: "Blob",
    Container: "Container",
    BlobFolder: "BlobFolder",
    AdlsGen2FileSystem: "AdlsGen2FileSystem",
    AdlsGen2Folder: "AdlsGen2Folder",
    AdlsGen2File: "AdlsGen2File",
    KustoCluster: "KustoCluster",
    KustoDatabase: "KustoDatabase",
    SqlDBTable: "SqlDBTable",
    SqlDWTable: "SqlDWTable",
    SynapseWorkspaceSqlPoolTable: "SynapseWorkspaceSqlPoolTable",
};
exports.OutputType = {
    Csv: "Csv",
    Parquet: "Parquet",
};
exports.RecurrenceInterval = {
    Hour: "Hour",
    Day: "Day",
};
exports.ShareKind = {
    CopyBased: "CopyBased",
    InPlace: "InPlace",
};
exports.SynchronizationMode = {
    Incremental: "Incremental",
    FullSync: "FullSync",
};
exports.SynchronizationSettingKind = {
    ScheduleBased: "ScheduleBased",
};
exports.TriggerKind = {
    ScheduleBased: "ScheduleBased",
};
exports.Type = {
    SystemAssigned: "SystemAssigned",
};
