// @pulumi/azure-native/analysisservices/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20170801beta = exports.v20170801 = exports.v20170714 = exports.v20160516 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getServerDetails"), exports);
__exportStar(require("./listServerGatewayStatus"), exports);
__exportStar(require("./serverDetails"), exports);
// Export enums:
__exportStar(require("../types/enums/analysisservices"), exports);
// Export sub-modules:
const v20160516 = require("./v20160516");
exports.v20160516 = v20160516;
const v20170714 = require("./v20170714");
exports.v20170714 = v20170714;
const v20170801 = require("./v20170801");
exports.v20170801 = v20170801;
const v20170801beta = require("./v20170801beta");
exports.v20170801beta = v20170801beta;
// Import resources to register:
const serverDetails_1 = require("./serverDetails");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:analysisservices:ServerDetails":
                return new serverDetails_1.ServerDetails(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "analysisservices", _module);
