// @pulumi/azure-native/types/enums/elastic/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagAction = exports.ProvisioningState = exports.MonitoringStatus = exports.ManagedIdentityTypes = exports.v20211001preview = exports.v20210901preview = exports.v20200701preview = exports.v20200701 = void 0;
// Export sub-modules:
const v20200701 = require("./v20200701");
exports.v20200701 = v20200701;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
const v20211001preview = require("./v20211001preview");
exports.v20211001preview = v20211001preview;
exports.ManagedIdentityTypes = {
    SystemAssigned: "SystemAssigned",
};
exports.MonitoringStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ProvisioningState = {
    Accepted: "Accepted",
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
    Deleted: "Deleted",
    NotSpecified: "NotSpecified",
};
exports.TagAction = {
    Include: "Include",
    Exclude: "Exclude",
};
