// @pulumi/azure-native/types/enums/hardwaresecuritymodules/v20181031preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = void 0;
exports.SkuName = {
    /**
     * The dedicated HSM is a Safenet Luna Network HSM A790 device.
     */
    SafeNet_Luna_Network_HSM_A790: "SafeNet Luna Network HSM A790",
    /**
     * The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 1 local master key which supports up to 60 calls per second.
     */
    PayShield10K_LMK1_CPS60: "payShield10K_LMK1_CPS60",
    /**
     * The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 1 local master key which supports up to 250 calls per second.
     */
    PayShield10K_LMK1_CPS250: "payShield10K_LMK1_CPS250",
    /**
     * The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 1 local master key which supports up to 2500 calls per second.
     */
    PayShield10K_LMK1_CPS2500: "payShield10K_LMK1_CPS2500",
    /**
     * The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 2 local master keys which supports up to 60 calls per second.
     */
    PayShield10K_LMK2_CPS60: "payShield10K_LMK2_CPS60",
    /**
     * The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 2 local master keys which supports up to 250 calls per second.
     */
    PayShield10K_LMK2_CPS250: "payShield10K_LMK2_CPS250",
    /**
     * The dedicated HSM is a payShield 10K, model PS10-D, 10Gb Ethernet Hardware Platform device with 2 local master keys which supports up to 2500 calls per second.
     */
    PayShield10K_LMK2_CPS2500: "payShield10K_LMK2_CPS2500",
};
