// @pulumi/azure-native/types/enums/keyvault/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VaultProvisioningState = exports.StoragePermissions = exports.SkuName = exports.SkuFamily = exports.SecretPermissions = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.NetworkRuleBypassOptions = exports.NetworkRuleAction = exports.ManagedHsmSkuName = exports.ManagedHsmSkuFamily = exports.KeyPermissions = exports.JsonWebKeyType = exports.JsonWebKeyOperation = exports.JsonWebKeyCurveName = exports.CreateMode = exports.CertificatePermissions = exports.ActionsRequired = exports.v20211101preview = exports.v20211001 = exports.v20210601preview = exports.v20210401preview = exports.v20200401preview = exports.v20190901 = exports.v20180214preview = exports.v20180214 = exports.v20161001 = exports.v20150601 = void 0;
// Export sub-modules:
const v20150601 = require("./v20150601");
exports.v20150601 = v20150601;
const v20161001 = require("./v20161001");
exports.v20161001 = v20161001;
const v20180214 = require("./v20180214");
exports.v20180214 = v20180214;
const v20180214preview = require("./v20180214preview");
exports.v20180214preview = v20180214preview;
const v20190901 = require("./v20190901");
exports.v20190901 = v20190901;
const v20200401preview = require("./v20200401preview");
exports.v20200401preview = v20200401preview;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
const v20211101preview = require("./v20211101preview");
exports.v20211101preview = v20211101preview;
exports.ActionsRequired = {
    None: "None",
};
exports.CertificatePermissions = {
    All: "all",
    Get: "get",
    List: "list",
    Delete: "delete",
    Create: "create",
    Import: "import",
    Update: "update",
    Managecontacts: "managecontacts",
    Getissuers: "getissuers",
    Listissuers: "listissuers",
    Setissuers: "setissuers",
    Deleteissuers: "deleteissuers",
    Manageissuers: "manageissuers",
    Recover: "recover",
    Purge: "purge",
    Backup: "backup",
    Restore: "restore",
};
exports.CreateMode = {
    Recover: "recover",
    Default: "default",
};
exports.JsonWebKeyCurveName = {
    P_256: "P-256",
    P_384: "P-384",
    P_521: "P-521",
    P_256K: "P-256K",
};
exports.JsonWebKeyOperation = {
    Encrypt: "encrypt",
    Decrypt: "decrypt",
    Sign: "sign",
    Verify: "verify",
    WrapKey: "wrapKey",
    UnwrapKey: "unwrapKey",
    Import: "import",
};
exports.JsonWebKeyType = {
    EC: "EC",
    EC_HSM: "EC-HSM",
    RSA: "RSA",
    RSA_HSM: "RSA-HSM",
};
exports.KeyPermissions = {
    All: "all",
    Encrypt: "encrypt",
    Decrypt: "decrypt",
    WrapKey: "wrapKey",
    UnwrapKey: "unwrapKey",
    Sign: "sign",
    Verify: "verify",
    Get: "get",
    List: "list",
    Create: "create",
    Update: "update",
    Import: "import",
    Delete: "delete",
    Backup: "backup",
    Restore: "restore",
    Recover: "recover",
    Purge: "purge",
};
exports.ManagedHsmSkuFamily = {
    B: "B",
};
exports.ManagedHsmSkuName = {
    Standard_B1: "Standard_B1",
    Custom_B32: "Custom_B32",
};
exports.NetworkRuleAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.NetworkRuleBypassOptions = {
    AzureServices: "AzureServices",
    None: "None",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SecretPermissions = {
    All: "all",
    Get: "get",
    List: "list",
    Set: "set",
    Delete: "delete",
    Backup: "backup",
    Restore: "restore",
    Recover: "recover",
    Purge: "purge",
};
exports.SkuFamily = {
    A: "A",
};
exports.SkuName = {
    Standard: "standard",
    Premium: "premium",
};
exports.StoragePermissions = {
    All: "all",
    Get: "get",
    List: "list",
    Delete: "delete",
    Set: "set",
    Update: "update",
    Regeneratekey: "regeneratekey",
    Recover: "recover",
    Purge: "purge",
    Backup: "backup",
    Restore: "restore",
    Setsas: "setsas",
    Listsas: "listsas",
    Getsas: "getsas",
    Deletesas: "deletesas",
};
exports.VaultProvisioningState = {
    Succeeded: "Succeeded",
    RegisteringDns: "RegisteringDns",
};
