// @pulumi/azure-native/documentdb/v20151106/databaseAccountCassandraKeyspace.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseAccountCassandraKeyspace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure Cosmos DB Cassandra keyspace.
 *
 * @deprecated Version v20151106 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.
 */
class DatabaseAccountCassandraKeyspace extends pulumi.CustomResource {
    /**
     * Create a DatabaseAccountCassandraKeyspace resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20151106 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("DatabaseAccountCassandraKeyspace is deprecated: Version v20151106 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["keyspaceName"] = args ? args.keyspaceName : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resource"] = args ? args.resource : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20150401:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20150408:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20160319:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20160331:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20190801:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20191212:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20200301:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20200401:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20200601preview:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20200901:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20210115:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20210301preview:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20210315:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20210401preview:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20210415:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20210515:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20210615:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20210701preview:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20211015:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20211015preview:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20211115preview:DatabaseAccountCassandraKeyspace" }, { type: "azure-native:documentdb/v20220215preview:DatabaseAccountCassandraKeyspace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DatabaseAccountCassandraKeyspace.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DatabaseAccountCassandraKeyspace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("DatabaseAccountCassandraKeyspace is deprecated: Version v20151106 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        return new DatabaseAccountCassandraKeyspace(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DatabaseAccountCassandraKeyspace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatabaseAccountCassandraKeyspace.__pulumiType;
    }
}
exports.DatabaseAccountCassandraKeyspace = DatabaseAccountCassandraKeyspace;
/** @internal */
DatabaseAccountCassandraKeyspace.__pulumiType = 'azure-native:documentdb/v20151106:DatabaseAccountCassandraKeyspace';
