// @pulumi/azure-native/documentdb/v20200601preview/gremlinResourceGremlinGraph.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GremlinResourceGremlinGraph = void 0;
const pulumi = require("@pulumi/pulumi");
const types_1 = require("../../types");
const utilities = require("../../utilities");
/**
 * An Azure Cosmos DB Gremlin graph.
 *
 * @deprecated Version v20200601preview will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.
 */
class GremlinResourceGremlinGraph extends pulumi.CustomResource {
    /**
     * Create a GremlinResourceGremlinGraph resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200601preview will be removed in the next major version of the provider. Upgrade to version v20210301preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("GremlinResourceGremlinGraph is deprecated: Version v20200601preview will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["graphName"] = args ? args.graphName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resource"] = args ? (args.resource ? pulumi.output(args.resource).apply(types_1.input.documentdb.v20200601preview.gremlinGraphResourceArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["options"] = undefined /*out*/;
            resourceInputs["resource"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20150401:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20150408:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20151106:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20160319:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20160331:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20190801:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20191212:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20200301:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20200401:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20200901:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20210115:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20210301preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20210315:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20210401preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20210415:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20210515:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20210615:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20210701preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20211015:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20211015preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20211115preview:GremlinResourceGremlinGraph" }, { type: "azure-native:documentdb/v20220215preview:GremlinResourceGremlinGraph" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GremlinResourceGremlinGraph.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing GremlinResourceGremlinGraph resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("GremlinResourceGremlinGraph is deprecated: Version v20200601preview will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        return new GremlinResourceGremlinGraph(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GremlinResourceGremlinGraph.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GremlinResourceGremlinGraph.__pulumiType;
    }
}
exports.GremlinResourceGremlinGraph = GremlinResourceGremlinGraph;
/** @internal */
GremlinResourceGremlinGraph.__pulumiType = 'azure-native:documentdb/v20200601preview:GremlinResourceGremlinGraph';
