// @pulumi/azure-native/documentdb/v20200601preview/mongoDBResourceMongoDBDatabase.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MongoDBResourceMongoDBDatabase = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure Cosmos DB MongoDB database.
 *
 * @deprecated Version v20200601preview will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.
 */
class MongoDBResourceMongoDBDatabase extends pulumi.CustomResource {
    /**
     * Create a MongoDBResourceMongoDBDatabase resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200601preview will be removed in the next major version of the provider. Upgrade to version v20210301preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("MongoDBResourceMongoDBDatabase is deprecated: Version v20200601preview will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resource"] = args ? args.resource : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["options"] = undefined /*out*/;
            resourceInputs["resource"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20150401:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20150408:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20151106:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20160319:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20160331:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20190801:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20191212:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20200301:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20200401:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20200901:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20210115:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20210301preview:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20210315:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20210401preview:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20210415:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20210515:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20210615:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20210701preview:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20211015:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20211015preview:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20211115preview:MongoDBResourceMongoDBDatabase" }, { type: "azure-native:documentdb/v20220215preview:MongoDBResourceMongoDBDatabase" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MongoDBResourceMongoDBDatabase.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MongoDBResourceMongoDBDatabase resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("MongoDBResourceMongoDBDatabase is deprecated: Version v20200601preview will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        return new MongoDBResourceMongoDBDatabase(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MongoDBResourceMongoDBDatabase.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MongoDBResourceMongoDBDatabase.__pulumiType;
    }
}
exports.MongoDBResourceMongoDBDatabase = MongoDBResourceMongoDBDatabase;
/** @internal */
MongoDBResourceMongoDBDatabase.__pulumiType = 'azure-native:documentdb/v20200601preview:MongoDBResourceMongoDBDatabase';
