// @pulumi/azure-native/fluidrelay/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220215 = exports.v20210910preview = exports.v20210830preview = exports.v20210615preview = exports.v20210312preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./fluidRelayServer"), exports);
__exportStar(require("./getFluidRelayServer"), exports);
__exportStar(require("./getFluidRelayServerKeys"), exports);
// Export enums:
__exportStar(require("../types/enums/fluidrelay"), exports);
// Export sub-modules:
const v20210312preview = require("./v20210312preview");
exports.v20210312preview = v20210312preview;
const v20210615preview = require("./v20210615preview");
exports.v20210615preview = v20210615preview;
const v20210830preview = require("./v20210830preview");
exports.v20210830preview = v20210830preview;
const v20210910preview = require("./v20210910preview");
exports.v20210910preview = v20210910preview;
const v20220215 = require("./v20220215");
exports.v20220215 = v20220215;
// Import resources to register:
const fluidRelayServer_1 = require("./fluidRelayServer");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:fluidrelay:FluidRelayServer":
                return new fluidRelayServer_1.FluidRelayServer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "fluidrelay", _module);
