// @pulumi/azure-native/consumption/v20180630/budgetByResourceGroupName.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BudgetByResourceGroupName = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A budget resource.
 *
 * @deprecated Version v20180630 will be removed in the next major version of the provider. Upgrade to version v20191001 or later.
 */
class BudgetByResourceGroupName extends pulumi.CustomResource {
    /**
     * Create a BudgetByResourceGroupName resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180630 will be removed in the next major version of the provider. Upgrade to version v20191001 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("BudgetByResourceGroupName is deprecated: Version v20180630 will be removed in the next major version of the provider. Upgrade to version v20191001 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.amount === undefined) && !opts.urn) {
                throw new Error("Missing required property 'amount'");
            }
            if ((!args || args.category === undefined) && !opts.urn) {
                throw new Error("Missing required property 'category'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.timeGrain === undefined) && !opts.urn) {
                throw new Error("Missing required property 'timeGrain'");
            }
            if ((!args || args.timePeriod === undefined) && !opts.urn) {
                throw new Error("Missing required property 'timePeriod'");
            }
            resourceInputs["amount"] = args ? args.amount : undefined;
            resourceInputs["budgetName"] = args ? args.budgetName : undefined;
            resourceInputs["category"] = args ? args.category : undefined;
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["filters"] = args ? args.filters : undefined;
            resourceInputs["notifications"] = args ? args.notifications : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["timeGrain"] = args ? args.timeGrain : undefined;
            resourceInputs["timePeriod"] = args ? args.timePeriod : undefined;
            resourceInputs["currentSpend"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["amount"] = undefined /*out*/;
            resourceInputs["category"] = undefined /*out*/;
            resourceInputs["currentSpend"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["filters"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["notifications"] = undefined /*out*/;
            resourceInputs["timeGrain"] = undefined /*out*/;
            resourceInputs["timePeriod"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:consumption/v20180131:BudgetByResourceGroupName" }, { type: "azure-native:consumption/v20180331:BudgetByResourceGroupName" }, { type: "azure-native:consumption/v20180831:BudgetByResourceGroupName" }, { type: "azure-native:consumption/v20181001:BudgetByResourceGroupName" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BudgetByResourceGroupName.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing BudgetByResourceGroupName resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("BudgetByResourceGroupName is deprecated: Version v20180630 will be removed in the next major version of the provider. Upgrade to version v20191001 or later.");
        return new BudgetByResourceGroupName(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BudgetByResourceGroupName.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BudgetByResourceGroupName.__pulumiType;
    }
}
exports.BudgetByResourceGroupName = BudgetByResourceGroupName;
/** @internal */
BudgetByResourceGroupName.__pulumiType = 'azure-native:consumption/v20180630:BudgetByResourceGroupName';
