// @pulumi/azure-native/logic/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20190501 = exports.v20180701preview = exports.v20160601 = exports.v20150801preview = exports.v20150201preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getIntegrationAccount"), exports);
__exportStar(require("./getIntegrationAccountAgreement"), exports);
__exportStar(require("./getIntegrationAccountAssembly"), exports);
__exportStar(require("./getIntegrationAccountBatchConfiguration"), exports);
__exportStar(require("./getIntegrationAccountCertificate"), exports);
__exportStar(require("./getIntegrationAccountMap"), exports);
__exportStar(require("./getIntegrationAccountPartner"), exports);
__exportStar(require("./getIntegrationAccountSchema"), exports);
__exportStar(require("./getIntegrationAccountSession"), exports);
__exportStar(require("./getIntegrationServiceEnvironment"), exports);
__exportStar(require("./getIntegrationServiceEnvironmentManagedApi"), exports);
__exportStar(require("./getRosettaNetProcessConfiguration"), exports);
__exportStar(require("./getWorkflow"), exports);
__exportStar(require("./getWorkflowAccessKey"), exports);
__exportStar(require("./integrationAccount"), exports);
__exportStar(require("./integrationAccountAgreement"), exports);
__exportStar(require("./integrationAccountAssembly"), exports);
__exportStar(require("./integrationAccountBatchConfiguration"), exports);
__exportStar(require("./integrationAccountCertificate"), exports);
__exportStar(require("./integrationAccountMap"), exports);
__exportStar(require("./integrationAccountPartner"), exports);
__exportStar(require("./integrationAccountSchema"), exports);
__exportStar(require("./integrationAccountSession"), exports);
__exportStar(require("./integrationServiceEnvironment"), exports);
__exportStar(require("./integrationServiceEnvironmentManagedApi"), exports);
__exportStar(require("./listIntegrationAccountAgreementContentCallbackUrl"), exports);
__exportStar(require("./listIntegrationAccountAssemblyContentCallbackUrl"), exports);
__exportStar(require("./listIntegrationAccountCallbackUrl"), exports);
__exportStar(require("./listIntegrationAccountKeyVaultKeys"), exports);
__exportStar(require("./listIntegrationAccountMapContentCallbackUrl"), exports);
__exportStar(require("./listIntegrationAccountPartnerContentCallbackUrl"), exports);
__exportStar(require("./listIntegrationAccountSchemaContentCallbackUrl"), exports);
__exportStar(require("./listWorkflowAccessKeySecretKeys"), exports);
__exportStar(require("./listWorkflowCallbackUrl"), exports);
__exportStar(require("./listWorkflowRunActionExpressionTraces"), exports);
__exportStar(require("./listWorkflowRunActionRepetitionExpressionTraces"), exports);
__exportStar(require("./listWorkflowTriggerCallbackUrl"), exports);
__exportStar(require("./listWorkflowVersionTriggerCallbackUrl"), exports);
__exportStar(require("./rosettaNetProcessConfiguration"), exports);
__exportStar(require("./workflow"), exports);
__exportStar(require("./workflowAccessKey"), exports);
// Export enums:
__exportStar(require("../types/enums/logic"), exports);
// Export sub-modules:
const v20150201preview = require("./v20150201preview");
exports.v20150201preview = v20150201preview;
const v20150801preview = require("./v20150801preview");
exports.v20150801preview = v20150801preview;
const v20160601 = require("./v20160601");
exports.v20160601 = v20160601;
const v20180701preview = require("./v20180701preview");
exports.v20180701preview = v20180701preview;
const v20190501 = require("./v20190501");
exports.v20190501 = v20190501;
// Import resources to register:
const integrationAccount_1 = require("./integrationAccount");
const integrationAccountAgreement_1 = require("./integrationAccountAgreement");
const integrationAccountAssembly_1 = require("./integrationAccountAssembly");
const integrationAccountBatchConfiguration_1 = require("./integrationAccountBatchConfiguration");
const integrationAccountCertificate_1 = require("./integrationAccountCertificate");
const integrationAccountMap_1 = require("./integrationAccountMap");
const integrationAccountPartner_1 = require("./integrationAccountPartner");
const integrationAccountSchema_1 = require("./integrationAccountSchema");
const integrationAccountSession_1 = require("./integrationAccountSession");
const integrationServiceEnvironment_1 = require("./integrationServiceEnvironment");
const integrationServiceEnvironmentManagedApi_1 = require("./integrationServiceEnvironmentManagedApi");
const rosettaNetProcessConfiguration_1 = require("./rosettaNetProcessConfiguration");
const workflow_1 = require("./workflow");
const workflowAccessKey_1 = require("./workflowAccessKey");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:logic:IntegrationAccount":
                return new integrationAccount_1.IntegrationAccount(name, undefined, { urn });
            case "azure-native:logic:IntegrationAccountAgreement":
                return new integrationAccountAgreement_1.IntegrationAccountAgreement(name, undefined, { urn });
            case "azure-native:logic:IntegrationAccountAssembly":
                return new integrationAccountAssembly_1.IntegrationAccountAssembly(name, undefined, { urn });
            case "azure-native:logic:IntegrationAccountBatchConfiguration":
                return new integrationAccountBatchConfiguration_1.IntegrationAccountBatchConfiguration(name, undefined, { urn });
            case "azure-native:logic:IntegrationAccountCertificate":
                return new integrationAccountCertificate_1.IntegrationAccountCertificate(name, undefined, { urn });
            case "azure-native:logic:IntegrationAccountMap":
                return new integrationAccountMap_1.IntegrationAccountMap(name, undefined, { urn });
            case "azure-native:logic:IntegrationAccountPartner":
                return new integrationAccountPartner_1.IntegrationAccountPartner(name, undefined, { urn });
            case "azure-native:logic:IntegrationAccountSchema":
                return new integrationAccountSchema_1.IntegrationAccountSchema(name, undefined, { urn });
            case "azure-native:logic:IntegrationAccountSession":
                return new integrationAccountSession_1.IntegrationAccountSession(name, undefined, { urn });
            case "azure-native:logic:IntegrationServiceEnvironment":
                return new integrationServiceEnvironment_1.IntegrationServiceEnvironment(name, undefined, { urn });
            case "azure-native:logic:IntegrationServiceEnvironmentManagedApi":
                return new integrationServiceEnvironmentManagedApi_1.IntegrationServiceEnvironmentManagedApi(name, undefined, { urn });
            case "azure-native:logic:RosettaNetProcessConfiguration":
                return new rosettaNetProcessConfiguration_1.RosettaNetProcessConfiguration(name, undefined, { urn });
            case "azure-native:logic:Workflow":
                return new workflow_1.Workflow(name, undefined, { urn });
            case "azure-native:logic:WorkflowAccessKey":
                return new workflowAccessKey_1.WorkflowAccessKey(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "logic", _module);
