// @pulumi/azure-native/types/enums/redhatopenshift/v20210901preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Visibility = exports.VMSize = exports.SoftwareDefinedNetwork = exports.EncryptionAtHost = void 0;
exports.EncryptionAtHost = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.SoftwareDefinedNetwork = {
    OVNKubernetes: "OVNKubernetes",
    OpenShiftSDN: "OpenShiftSDN",
};
exports.VMSize = {
    Standard_D16as_v4: "Standard_D16as_v4",
    Standard_D16s_v3: "Standard_D16s_v3",
    Standard_D2s_v3: "Standard_D2s_v3",
    Standard_D32as_v4: "Standard_D32as_v4",
    Standard_D32s_v3: "Standard_D32s_v3",
    Standard_D4as_v4: "Standard_D4as_v4",
    Standard_D4s_v3: "Standard_D4s_v3",
    Standard_D8as_v4: "Standard_D8as_v4",
    Standard_D8s_v3: "Standard_D8s_v3",
    Standard_E16s_v3: "Standard_E16s_v3",
    Standard_E32s_v3: "Standard_E32s_v3",
    Standard_E4s_v3: "Standard_E4s_v3",
    Standard_E64i_v3: "Standard_E64i_v3",
    Standard_E64is_v3: "Standard_E64is_v3",
    Standard_E8s_v3: "Standard_E8s_v3",
    Standard_F16s_v2: "Standard_F16s_v2",
    Standard_F32s_v2: "Standard_F32s_v2",
    Standard_F4s_v2: "Standard_F4s_v2",
    Standard_F72s_v2: "Standard_F72s_v2",
    Standard_F8s_v2: "Standard_F8s_v2",
    Standard_G5: "Standard_G5",
    Standard_GS5: "Standard_GS5",
    Standard_M128ms: "Standard_M128ms",
};
exports.Visibility = {
    Private: "Private",
    Public: "Public",
};
