// @pulumi/azure-native/recoveryservices/v20170701/protectionIntent.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProtectionIntent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Base class for backup ProtectionIntent.
 *
 * @deprecated Version v20170701 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.
 */
class ProtectionIntent extends pulumi.CustomResource {
    /**
     * Create a ProtectionIntent resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20170701 will be removed in the next major version of the provider. Upgrade to version v20180710 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ProtectionIntent is deprecated: Version v20170701 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.vaultName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'vaultName'");
            }
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["fabricName"] = args ? args.fabricName : undefined;
            resourceInputs["intentObjectName"] = args ? args.intentObjectName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["vaultName"] = args ? args.vaultName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210201:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210201preview:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210210:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210301:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210401:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210601:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210701:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20210801:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20211001:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20211201:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20220101:ProtectionIntent" }, { type: "azure-native:recoveryservices/v20220201:ProtectionIntent" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ProtectionIntent.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ProtectionIntent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ProtectionIntent is deprecated: Version v20170701 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        return new ProtectionIntent(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ProtectionIntent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ProtectionIntent.__pulumiType;
    }
}
exports.ProtectionIntent = ProtectionIntent;
/** @internal */
ProtectionIntent.__pulumiType = 'azure-native:recoveryservices/v20170701:ProtectionIntent';
