// @pulumi/azure-native/aadiam/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20200701preview = exports.v20200301preview = exports.v20200301 = exports.v20170401preview = exports.v20170401 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./azureADMetric"), exports);
__exportStar(require("./diagnosticSetting"), exports);
__exportStar(require("./getDiagnosticSetting"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getazureADMetric"), exports);
__exportStar(require("./getprivateLinkForAzureAd"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./privateLinkForAzureAd"), exports);
// Export enums:
__exportStar(require("../types/enums/aadiam"), exports);
// Export sub-modules:
const v20170401 = require("./v20170401");
exports.v20170401 = v20170401;
const v20170401preview = require("./v20170401preview");
exports.v20170401preview = v20170401preview;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
// Import resources to register:
const diagnosticSetting_1 = require("./diagnosticSetting");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const azureADMetric_1 = require("./azureADMetric");
const privateLinkForAzureAd_1 = require("./privateLinkForAzureAd");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:aadiam:DiagnosticSetting":
                return new diagnosticSetting_1.DiagnosticSetting(name, undefined, { urn });
            case "azure-native:aadiam:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:aadiam:azureADMetric":
                return new azureADMetric_1.AzureADMetric(name, undefined, { urn });
            case "azure-native:aadiam:privateLinkForAzureAd":
                return new privateLinkForAzureAd_1.PrivateLinkForAzureAd(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "aadiam", _module);
