// @pulumi/azure-native/types/enums/eventgrid/v20211015preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sku = exports.ResourceProvisioningState = exports.ResourceKind = exports.ReadinessState = exports.PublicNetworkAccess = exports.PersistedConnectionStatus = exports.PartnerTopicRoutingMode = exports.PartnerTopicActivationState = exports.PartnerRegistrationVisibilityState = exports.PartnerEndpointType = exports.PartnerDestinationProvisioningState = exports.PartnerDestinationActivationState = exports.PartnerConfigurationProvisioningState = exports.PartnerClientAuthenticationType = exports.IpActionType = exports.InputSchemaMappingType = exports.InputSchema = exports.IdentityType = exports.EventSubscriptionIdentityType = exports.EventDeliverySchema = exports.EventDefinitionKind = exports.EndpointType = exports.DeliveryAttributeMappingType = exports.DeadLetterEndPointType = exports.DataResidencyBoundary = exports.ChannelType = exports.ChannelProvisioningState = exports.AdvancedFilterOperatorType = void 0;
exports.AdvancedFilterOperatorType = {
    NumberIn: "NumberIn",
    NumberNotIn: "NumberNotIn",
    NumberLessThan: "NumberLessThan",
    NumberGreaterThan: "NumberGreaterThan",
    NumberLessThanOrEquals: "NumberLessThanOrEquals",
    NumberGreaterThanOrEquals: "NumberGreaterThanOrEquals",
    BoolEquals: "BoolEquals",
    StringIn: "StringIn",
    StringNotIn: "StringNotIn",
    StringBeginsWith: "StringBeginsWith",
    StringEndsWith: "StringEndsWith",
    StringContains: "StringContains",
    NumberInRange: "NumberInRange",
    NumberNotInRange: "NumberNotInRange",
    StringNotBeginsWith: "StringNotBeginsWith",
    StringNotEndsWith: "StringNotEndsWith",
    StringNotContains: "StringNotContains",
    IsNullOrUndefined: "IsNullOrUndefined",
    IsNotNull: "IsNotNull",
};
exports.ChannelProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.ChannelType = {
    PartnerTopic: "PartnerTopic",
    PartnerDestination: "PartnerDestination",
};
exports.DataResidencyBoundary = {
    WithinGeopair: "WithinGeopair",
    WithinRegion: "WithinRegion",
};
exports.DeadLetterEndPointType = {
    StorageBlob: "StorageBlob",
};
exports.DeliveryAttributeMappingType = {
    Static: "Static",
    Dynamic: "Dynamic",
};
exports.EndpointType = {
    WebHook: "WebHook",
    EventHub: "EventHub",
    StorageQueue: "StorageQueue",
    HybridConnection: "HybridConnection",
    ServiceBusQueue: "ServiceBusQueue",
    ServiceBusTopic: "ServiceBusTopic",
    AzureFunction: "AzureFunction",
    PartnerDestination: "PartnerDestination",
};
exports.EventDefinitionKind = {
    Inline: "Inline",
};
exports.EventDeliverySchema = {
    EventGridSchema: "EventGridSchema",
    CustomInputSchema: "CustomInputSchema",
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.EventSubscriptionIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.InputSchema = {
    EventGridSchema: "EventGridSchema",
    CustomEventSchema: "CustomEventSchema",
    CloudEventSchemaV1_0: "CloudEventSchemaV1_0",
};
exports.InputSchemaMappingType = {
    Json: "Json",
};
exports.IpActionType = {
    Allow: "Allow",
};
exports.PartnerClientAuthenticationType = {
    AzureAD: "AzureAD",
};
exports.PartnerConfigurationProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.PartnerDestinationActivationState = {
    NeverActivated: "NeverActivated",
    Activated: "Activated",
};
exports.PartnerDestinationProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.PartnerEndpointType = {
    WebHook: "WebHook",
};
exports.PartnerRegistrationVisibilityState = {
    Hidden: "Hidden",
    PublicPreview: "PublicPreview",
    GenerallyAvailable: "GenerallyAvailable",
};
exports.PartnerTopicActivationState = {
    NeverActivated: "NeverActivated",
    Activated: "Activated",
    Deactivated: "Deactivated",
};
exports.PartnerTopicRoutingMode = {
    SourceEventAttribute: "SourceEventAttribute",
    ChannelNameHeader: "ChannelNameHeader",
};
exports.PersistedConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ReadinessState = {
    NeverActivated: "NeverActivated",
    Activated: "Activated",
};
exports.ResourceKind = {
    Azure: "Azure",
    AzureArc: "AzureArc",
};
exports.ResourceProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.Sku = {
    Basic: "Basic",
    Premium: "Premium",
};
