// @pulumi/azure-native/dataprotection/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220301 = exports.v20220201preview = exports.v20220101 = exports.v20211201preview = exports.v20211001preview = exports.v20210701 = exports.v20210601preview = exports.v20210201preview = exports.v20210101 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./backupInstance"), exports);
__exportStar(require("./backupPolicy"), exports);
__exportStar(require("./backupVault"), exports);
__exportStar(require("./getBackupInstance"), exports);
__exportStar(require("./getBackupPolicy"), exports);
__exportStar(require("./getBackupVault"), exports);
__exportStar(require("./getResourceGuard"), exports);
__exportStar(require("./resourceGuard"), exports);
// Export enums:
__exportStar(require("../types/enums/dataprotection"), exports);
// Export sub-modules:
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210201preview = require("./v20210201preview");
exports.v20210201preview = v20210201preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
const v20211001preview = require("./v20211001preview");
exports.v20211001preview = v20211001preview;
const v20211201preview = require("./v20211201preview");
exports.v20211201preview = v20211201preview;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
const v20220201preview = require("./v20220201preview");
exports.v20220201preview = v20220201preview;
const v20220301 = require("./v20220301");
exports.v20220301 = v20220301;
// Import resources to register:
const backupInstance_1 = require("./backupInstance");
const backupPolicy_1 = require("./backupPolicy");
const backupVault_1 = require("./backupVault");
const resourceGuard_1 = require("./resourceGuard");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:dataprotection:BackupInstance":
                return new backupInstance_1.BackupInstance(name, undefined, { urn });
            case "azure-native:dataprotection:BackupPolicy":
                return new backupPolicy_1.BackupPolicy(name, undefined, { urn });
            case "azure-native:dataprotection:BackupVault":
                return new backupVault_1.BackupVault(name, undefined, { urn });
            case "azure-native:dataprotection:ResourceGuard":
                return new resourceGuard_1.ResourceGuard(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "dataprotection", _module);
