// @pulumi/azure-native/desktopvirtualization/v20200921preview/msixpackage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MSIXPackage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Schema for MSIX Package properties.
 *
 * @deprecated Version v20200921preview will be removed in the next major version of the provider. Upgrade to version v20210201preview or later.
 */
class MSIXPackage extends pulumi.CustomResource {
    /**
     * Create a MSIXPackage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200921preview will be removed in the next major version of the provider. Upgrade to version v20210201preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("MSIXPackage is deprecated: Version v20200921preview will be removed in the next major version of the provider. Upgrade to version v20210201preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.hostPoolName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostPoolName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["hostPoolName"] = args ? args.hostPoolName : undefined;
            resourceInputs["imagePath"] = args ? args.imagePath : undefined;
            resourceInputs["isActive"] = args ? args.isActive : undefined;
            resourceInputs["isRegularRegistration"] = args ? args.isRegularRegistration : undefined;
            resourceInputs["lastUpdated"] = args ? args.lastUpdated : undefined;
            resourceInputs["msixPackageFullName"] = args ? args.msixPackageFullName : undefined;
            resourceInputs["packageApplications"] = args ? args.packageApplications : undefined;
            resourceInputs["packageDependencies"] = args ? args.packageDependencies : undefined;
            resourceInputs["packageFamilyName"] = args ? args.packageFamilyName : undefined;
            resourceInputs["packageName"] = args ? args.packageName : undefined;
            resourceInputs["packageRelativePath"] = args ? args.packageRelativePath : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["imagePath"] = undefined /*out*/;
            resourceInputs["isActive"] = undefined /*out*/;
            resourceInputs["isRegularRegistration"] = undefined /*out*/;
            resourceInputs["lastUpdated"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["packageApplications"] = undefined /*out*/;
            resourceInputs["packageDependencies"] = undefined /*out*/;
            resourceInputs["packageFamilyName"] = undefined /*out*/;
            resourceInputs["packageName"] = undefined /*out*/;
            resourceInputs["packageRelativePath"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:desktopvirtualization:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20201019preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20201102preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20201110preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210114preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210201preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210309preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210401preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210712:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20210903preview:MSIXPackage" }, { type: "azure-native:desktopvirtualization/v20220210preview:MSIXPackage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MSIXPackage.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MSIXPackage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("MSIXPackage is deprecated: Version v20200921preview will be removed in the next major version of the provider. Upgrade to version v20210201preview or later.");
        return new MSIXPackage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MSIXPackage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MSIXPackage.__pulumiType;
    }
}
exports.MSIXPackage = MSIXPackage;
/** @internal */
MSIXPackage.__pulumiType = 'azure-native:desktopvirtualization/v20200921preview:MSIXPackage';
