// @pulumi/azure-native/healthcareapis/v20210601preview/iotConnectorFhirDestination.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotConnectorFhirDestination = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * IoT Connector FHIR destination definition.
 *
 * @deprecated Version v20210601preview will be removed in the next major version of the provider. Upgrade to version v20211101 or later.
 */
class IotConnectorFhirDestination extends pulumi.CustomResource {
    /**
     * Create a IotConnectorFhirDestination resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20210601preview will be removed in the next major version of the provider. Upgrade to version v20211101 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("IotConnectorFhirDestination is deprecated: Version v20210601preview will be removed in the next major version of the provider. Upgrade to version v20211101 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fhirMapping === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fhirMapping'");
            }
            if ((!args || args.fhirServiceResourceId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fhirServiceResourceId'");
            }
            if ((!args || args.iotConnectorName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iotConnectorName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceIdentityResolutionType === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceIdentityResolutionType'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["fhirDestinationName"] = args ? args.fhirDestinationName : undefined;
            resourceInputs["fhirMapping"] = args ? args.fhirMapping : undefined;
            resourceInputs["fhirServiceResourceId"] = args ? args.fhirServiceResourceId : undefined;
            resourceInputs["iotConnectorName"] = args ? args.iotConnectorName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceIdentityResolutionType"] = args ? args.resourceIdentityResolutionType : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["fhirMapping"] = undefined /*out*/;
            resourceInputs["fhirServiceResourceId"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["resourceIdentityResolutionType"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:healthcareapis:IotConnectorFhirDestination" }, { type: "azure-native:healthcareapis/v20211101:IotConnectorFhirDestination" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IotConnectorFhirDestination.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing IotConnectorFhirDestination resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("IotConnectorFhirDestination is deprecated: Version v20210601preview will be removed in the next major version of the provider. Upgrade to version v20211101 or later.");
        return new IotConnectorFhirDestination(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IotConnectorFhirDestination.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IotConnectorFhirDestination.__pulumiType;
    }
}
exports.IotConnectorFhirDestination = IotConnectorFhirDestination;
/** @internal */
IotConnectorFhirDestination.__pulumiType = 'azure-native:healthcareapis/v20210601preview:IotConnectorFhirDestination';
