// @pulumi/azure-native/types/enums/recoveryservices/v20211101/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlServerLicenseType = exports.SetMultiVmSyncStatus = exports.ReplicationProtectedItemOperation = exports.RecoveryPlanGroupType = exports.RecoveryPlanActionLocation = exports.PossibleOperationsDirections = exports.LicenseType = exports.FailoverDeploymentModel = exports.ExtendedLocationType = exports.DiskAccountType = exports.AutomationAccountAuthenticationType = exports.AgentAutoUpdateStatus = void 0;
exports.AgentAutoUpdateStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.AutomationAccountAuthenticationType = {
    RunAsAccount: "RunAsAccount",
    SystemAssignedIdentity: "SystemAssignedIdentity",
};
exports.DiskAccountType = {
    Standard_LRS: "Standard_LRS",
    Premium_LRS: "Premium_LRS",
    StandardSSD_LRS: "StandardSSD_LRS",
};
exports.ExtendedLocationType = {
    EdgeZone: "EdgeZone",
};
exports.FailoverDeploymentModel = {
    NotApplicable: "NotApplicable",
    Classic: "Classic",
    ResourceManager: "ResourceManager",
};
exports.LicenseType = {
    NotSpecified: "NotSpecified",
    NoLicenseType: "NoLicenseType",
    WindowsServer: "WindowsServer",
};
exports.PossibleOperationsDirections = {
    PrimaryToRecovery: "PrimaryToRecovery",
    RecoveryToPrimary: "RecoveryToPrimary",
};
exports.RecoveryPlanActionLocation = {
    Primary: "Primary",
    Recovery: "Recovery",
};
exports.RecoveryPlanGroupType = {
    Shutdown: "Shutdown",
    Boot: "Boot",
    Failover: "Failover",
};
exports.ReplicationProtectedItemOperation = {
    ReverseReplicate: "ReverseReplicate",
    Commit: "Commit",
    PlannedFailover: "PlannedFailover",
    UnplannedFailover: "UnplannedFailover",
    DisableProtection: "DisableProtection",
    TestFailover: "TestFailover",
    TestFailoverCleanup: "TestFailoverCleanup",
    Failback: "Failback",
    FinalizeFailback: "FinalizeFailback",
    CancelFailover: "CancelFailover",
    ChangePit: "ChangePit",
    RepairReplication: "RepairReplication",
    SwitchProtection: "SwitchProtection",
    CompleteMigration: "CompleteMigration",
};
exports.SetMultiVmSyncStatus = {
    Enable: "Enable",
    Disable: "Disable",
};
exports.SqlServerLicenseType = {
    NotSpecified: "NotSpecified",
    NoLicenseType: "NoLicenseType",
    PAYG: "PAYG",
    AHUB: "AHUB",
};
