// @pulumi/azure-native/elastic/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211001preview = exports.v20210901preview = exports.v20200701preview = exports.v20200701 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getMonitor"), exports);
__exportStar(require("./getTagRule"), exports);
__exportStar(require("./listDeploymentInfo"), exports);
__exportStar(require("./listMonitoredResource"), exports);
__exportStar(require("./listUpgradableVersionDetails"), exports);
__exportStar(require("./listVMHost"), exports);
__exportStar(require("./monitor"), exports);
__exportStar(require("./tagRule"), exports);
// Export enums:
__exportStar(require("../types/enums/elastic"), exports);
// Export sub-modules:
const v20200701 = require("./v20200701");
exports.v20200701 = v20200701;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
const v20211001preview = require("./v20211001preview");
exports.v20211001preview = v20211001preview;
// Import resources to register:
const monitor_1 = require("./monitor");
const tagRule_1 = require("./tagRule");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:elastic:Monitor":
                return new monitor_1.Monitor(name, undefined, { urn });
            case "azure-native:elastic:TagRule":
                return new tagRule_1.TagRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "elastic", _module);
