// @pulumi/azure-native/types/enums/digitaltwins/v20210630preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicNetworkAccess = exports.PrivateLinkServiceConnectionStatus = exports.EndpointType = exports.DigitalTwinsIdentityType = exports.ConnectionType = exports.AuthenticationType = void 0;
exports.AuthenticationType = {
    KeyBased: "KeyBased",
    IdentityBased: "IdentityBased",
};
exports.ConnectionType = {
    AzureDataExplorer: "AzureDataExplorer",
};
exports.DigitalTwinsIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.EndpointType = {
    EventHub: "EventHub",
    EventGrid: "EventGrid",
    ServiceBus: "ServiceBus",
};
exports.PrivateLinkServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
