// @pulumi/azure-native/documentdb/v20160331/databaseAccountMongoDBCollection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseAccountMongoDBCollection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An Azure Cosmos DB MongoDB collection.
 *
 * @deprecated Version v20160331 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.
 */
class DatabaseAccountMongoDBCollection extends pulumi.CustomResource {
    /**
     * Create a DatabaseAccountMongoDBCollection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160331 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("DatabaseAccountMongoDBCollection is deprecated: Version v20160331 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["collectionName"] = args ? args.collectionName : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resource"] = args ? args.resource : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["indexes"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["shardKey"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["indexes"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["shardKey"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20150401:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20150408:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20151106:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20160319:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20190801:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20191212:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20200301:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20200401:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20200601preview:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20200901:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20210115:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20210301preview:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20210315:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20210401preview:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20210415:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20210515:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20210615:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20210701preview:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20211015:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20211015preview:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20211115preview:DatabaseAccountMongoDBCollection" }, { type: "azure-native:documentdb/v20220215preview:DatabaseAccountMongoDBCollection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DatabaseAccountMongoDBCollection.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DatabaseAccountMongoDBCollection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("DatabaseAccountMongoDBCollection is deprecated: Version v20160331 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        return new DatabaseAccountMongoDBCollection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DatabaseAccountMongoDBCollection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DatabaseAccountMongoDBCollection.__pulumiType;
    }
}
exports.DatabaseAccountMongoDBCollection = DatabaseAccountMongoDBCollection;
/** @internal */
DatabaseAccountMongoDBCollection.__pulumiType = 'azure-native:documentdb/v20160331:DatabaseAccountMongoDBCollection';
