// @pulumi/azure-native/network/v20210501/serviceEndpointPolicy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceEndpointPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Service End point policy resource.
 */
class ServiceEndpointPolicy extends pulumi.CustomResource {
    /**
     * Create a ServiceEndpointPolicy resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["contextualServiceEndpointPolicies"] = args ? args.contextualServiceEndpointPolicies : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceAlias"] = args ? args.serviceAlias : undefined;
            resourceInputs["serviceEndpointPolicyDefinitions"] = args ? args.serviceEndpointPolicyDefinitions : undefined;
            resourceInputs["serviceEndpointPolicyName"] = args ? args.serviceEndpointPolicyName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["subnets"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["contextualServiceEndpointPolicies"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["serviceAlias"] = undefined /*out*/;
            resourceInputs["serviceEndpointPolicyDefinitions"] = undefined /*out*/;
            resourceInputs["subnets"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network:ServiceEndpointPolicy" }, { type: "azure-native:network/v20180701:ServiceEndpointPolicy" }, { type: "azure-native:network/v20180801:ServiceEndpointPolicy" }, { type: "azure-native:network/v20181001:ServiceEndpointPolicy" }, { type: "azure-native:network/v20181101:ServiceEndpointPolicy" }, { type: "azure-native:network/v20181201:ServiceEndpointPolicy" }, { type: "azure-native:network/v20190201:ServiceEndpointPolicy" }, { type: "azure-native:network/v20190401:ServiceEndpointPolicy" }, { type: "azure-native:network/v20190601:ServiceEndpointPolicy" }, { type: "azure-native:network/v20190701:ServiceEndpointPolicy" }, { type: "azure-native:network/v20190801:ServiceEndpointPolicy" }, { type: "azure-native:network/v20190901:ServiceEndpointPolicy" }, { type: "azure-native:network/v20191101:ServiceEndpointPolicy" }, { type: "azure-native:network/v20191201:ServiceEndpointPolicy" }, { type: "azure-native:network/v20200301:ServiceEndpointPolicy" }, { type: "azure-native:network/v20200401:ServiceEndpointPolicy" }, { type: "azure-native:network/v20200501:ServiceEndpointPolicy" }, { type: "azure-native:network/v20200601:ServiceEndpointPolicy" }, { type: "azure-native:network/v20200701:ServiceEndpointPolicy" }, { type: "azure-native:network/v20200801:ServiceEndpointPolicy" }, { type: "azure-native:network/v20201101:ServiceEndpointPolicy" }, { type: "azure-native:network/v20210201:ServiceEndpointPolicy" }, { type: "azure-native:network/v20210301:ServiceEndpointPolicy" }, { type: "azure-native:network/v20210801:ServiceEndpointPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServiceEndpointPolicy.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ServiceEndpointPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new ServiceEndpointPolicy(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServiceEndpointPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServiceEndpointPolicy.__pulumiType;
    }
}
exports.ServiceEndpointPolicy = ServiceEndpointPolicy;
/** @internal */
ServiceEndpointPolicy.__pulumiType = 'azure-native:network/v20210501:ServiceEndpointPolicy';
