// @pulumi/azure-native/search/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210401preview = exports.v20200801preview = exports.v20200801 = exports.v20200313 = exports.v20191001preview = exports.v20150819 = exports.v20150228 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getService"), exports);
__exportStar(require("./getSharedPrivateLinkResource"), exports);
__exportStar(require("./listAdminKey"), exports);
__exportStar(require("./listQueryKeyBySearchService"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./service"), exports);
__exportStar(require("./sharedPrivateLinkResource"), exports);
// Export enums:
__exportStar(require("../types/enums/search"), exports);
// Export sub-modules:
const v20150228 = require("./v20150228");
exports.v20150228 = v20150228;
const v20150819 = require("./v20150819");
exports.v20150819 = v20150819;
const v20191001preview = require("./v20191001preview");
exports.v20191001preview = v20191001preview;
const v20200313 = require("./v20200313");
exports.v20200313 = v20200313;
const v20200801 = require("./v20200801");
exports.v20200801 = v20200801;
const v20200801preview = require("./v20200801preview");
exports.v20200801preview = v20200801preview;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
// Import resources to register:
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const service_1 = require("./service");
const sharedPrivateLinkResource_1 = require("./sharedPrivateLinkResource");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:search:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:search:Service":
                return new service_1.Service(name, undefined, { urn });
            case "azure-native:search:SharedPrivateLinkResource":
                return new sharedPrivateLinkResource_1.SharedPrivateLinkResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "search", _module);
