// @pulumi/azure-native/documentdb/v20190801/sqlResourceSqlContainer.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlResourceSqlContainer = void 0;
const pulumi = require("@pulumi/pulumi");
const types_1 = require("../../types");
const utilities = require("../../utilities");
/**
 * An Azure Cosmos DB container.
 *
 * @deprecated Version v20190801 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.
 */
class SqlResourceSqlContainer extends pulumi.CustomResource {
    /**
     * Create a SqlResourceSqlContainer resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20190801 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SqlResourceSqlContainer is deprecated: Version v20190801 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accountName'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.options === undefined) && !opts.urn) {
                throw new Error("Missing required property 'options'");
            }
            if ((!args || args.resource === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resource'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accountName"] = args ? args.accountName : undefined;
            resourceInputs["containerName"] = args ? args.containerName : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["options"] = args ? args.options : undefined;
            resourceInputs["resource"] = args ? (args.resource ? pulumi.output(args.resource).apply(types_1.input.documentdb.v20190801.sqlContainerResourceArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["resource"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:documentdb:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20150401:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20150408:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20151106:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20160319:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20160331:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20191212:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20200301:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20200401:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20200601preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20200901:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20210115:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20210301preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20210315:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20210401preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20210415:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20210515:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20210615:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20210701preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20211015:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20211015preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20211115preview:SqlResourceSqlContainer" }, { type: "azure-native:documentdb/v20220215preview:SqlResourceSqlContainer" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlResourceSqlContainer.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SqlResourceSqlContainer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SqlResourceSqlContainer is deprecated: Version v20190801 will be removed in the next major version of the provider. Upgrade to version v20210301preview or later.");
        return new SqlResourceSqlContainer(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlResourceSqlContainer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlResourceSqlContainer.__pulumiType;
    }
}
exports.SqlResourceSqlContainer = SqlResourceSqlContainer;
/** @internal */
SqlResourceSqlContainer.__pulumiType = 'azure-native:documentdb/v20190801:SqlResourceSqlContainer';
