// @pulumi/azure-native/types/enums/databoxedge/v20190701/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerEventType = exports.SkuTier = exports.SkuName = exports.ShareStatus = exports.ShareAccessType = exports.ShareAccessProtocol = exports.SSLStatus = exports.RoleTypes = exports.RoleStatus = exports.PlatformType = exports.OrderState = exports.MonitoringStatus = exports.EncryptionAlgorithm = exports.DayOfWeek = exports.DataPolicy = exports.DataBoxEdgeDeviceStatus = exports.ClientPermissionType = exports.AzureContainerDataFormat = exports.AccountType = void 0;
exports.AccountType = {
    GeneralPurposeStorage: "GeneralPurposeStorage",
    BlobStorage: "BlobStorage",
};
exports.AzureContainerDataFormat = {
    BlockBlob: "BlockBlob",
    PageBlob: "PageBlob",
    AzureFile: "AzureFile",
};
exports.ClientPermissionType = {
    NoAccess: "NoAccess",
    ReadOnly: "ReadOnly",
    ReadWrite: "ReadWrite",
};
exports.DataBoxEdgeDeviceStatus = {
    ReadyToSetup: "ReadyToSetup",
    Online: "Online",
    Offline: "Offline",
    NeedsAttention: "NeedsAttention",
    Disconnected: "Disconnected",
    PartiallyDisconnected: "PartiallyDisconnected",
    Maintenance: "Maintenance",
};
exports.DataPolicy = {
    Cloud: "Cloud",
    Local: "Local",
};
exports.DayOfWeek = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.EncryptionAlgorithm = {
    None: "None",
    AES256: "AES256",
    RSAES_PKCS1_v_1_5: "RSAES_PKCS1_v_1_5",
};
exports.MonitoringStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.OrderState = {
    Untracked: "Untracked",
    AwaitingFulfilment: "AwaitingFulfilment",
    AwaitingPreparation: "AwaitingPreparation",
    AwaitingShipment: "AwaitingShipment",
    Shipped: "Shipped",
    Arriving: "Arriving",
    Delivered: "Delivered",
    ReplacementRequested: "ReplacementRequested",
    LostDevice: "LostDevice",
    Declined: "Declined",
    ReturnInitiated: "ReturnInitiated",
    AwaitingReturnShipment: "AwaitingReturnShipment",
    ShippedBack: "ShippedBack",
    CollectedAtMicrosoft: "CollectedAtMicrosoft",
};
exports.PlatformType = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.RoleStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RoleTypes = {
    IOT: "IOT",
    ASA: "ASA",
    Functions: "Functions",
    Cognitive: "Cognitive",
};
exports.SSLStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ShareAccessProtocol = {
    SMB: "SMB",
    NFS: "NFS",
};
exports.ShareAccessType = {
    Change: "Change",
    Read: "Read",
    Custom: "Custom",
};
exports.ShareStatus = {
    Offline: "Offline",
    Unknown: "Unknown",
    OK: "OK",
    Updating: "Updating",
    NeedsAttention: "NeedsAttention",
};
exports.SkuName = {
    Gateway: "Gateway",
    Edge: "Edge",
};
exports.SkuTier = {
    Standard: "Standard",
};
exports.TriggerEventType = {
    FileEvent: "FileEvent",
    PeriodicTimerEvent: "PeriodicTimerEvent",
};
