// @pulumi/azure-native/insights/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220401 = exports.v20211014 = exports.v20210901preview = exports.v20210901 = exports.v20210801 = exports.v20210701preview = exports.v20210501preview = exports.v20210401 = exports.v20210308 = exports.v20210303preview = exports.v20210201preview = exports.v20201120 = exports.v20201020 = exports.v20201005preview = exports.v20201001 = exports.v20200602preview = exports.v20200501preview = exports.v20200301preview = exports.v20200210preview = exports.v20200202preview = exports.v20200202 = exports.v20200101preview = exports.v20191101preview = exports.v20191017preview = exports.v20190601 = exports.v20190301 = exports.v20180901 = exports.v20180617preview = exports.v20180601preview = exports.v20180501preview = exports.v20180416 = exports.v20180301 = exports.v20171001 = exports.v20170501preview = exports.v20170401 = exports.v20160301 = exports.v20150501 = exports.v20150401 = exports.v20140401 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./actionGroup"), exports);
__exportStar(require("./activityLogAlert"), exports);
__exportStar(require("./alertRule"), exports);
__exportStar(require("./analyticsItem"), exports);
__exportStar(require("./autoscaleSetting"), exports);
__exportStar(require("./component"), exports);
__exportStar(require("./componentCurrentBillingFeature"), exports);
__exportStar(require("./componentLinkedStorageAccount"), exports);
__exportStar(require("./dataCollectionEndpoint"), exports);
__exportStar(require("./dataCollectionRule"), exports);
__exportStar(require("./dataCollectionRuleAssociation"), exports);
__exportStar(require("./diagnosticSetting"), exports);
__exportStar(require("./exportConfiguration"), exports);
__exportStar(require("./favorite"), exports);
__exportStar(require("./getActionGroup"), exports);
__exportStar(require("./getActivityLogAlert"), exports);
__exportStar(require("./getAlertRule"), exports);
__exportStar(require("./getAnalyticsItem"), exports);
__exportStar(require("./getAutoscaleSetting"), exports);
__exportStar(require("./getComponent"), exports);
__exportStar(require("./getComponentCurrentBillingFeature"), exports);
__exportStar(require("./getComponentLinkedStorageAccount"), exports);
__exportStar(require("./getDataCollectionEndpoint"), exports);
__exportStar(require("./getDataCollectionRule"), exports);
__exportStar(require("./getDataCollectionRuleAssociation"), exports);
__exportStar(require("./getDiagnosticServiceTokenReadOnly"), exports);
__exportStar(require("./getDiagnosticServiceTokenReadWrite"), exports);
__exportStar(require("./getDiagnosticSetting"), exports);
__exportStar(require("./getExportConfiguration"), exports);
__exportStar(require("./getFavorite"), exports);
__exportStar(require("./getGuestDiagnosticsSettingsAssociation"), exports);
__exportStar(require("./getLiveToken"), exports);
__exportStar(require("./getLogProfile"), exports);
__exportStar(require("./getManagementGroupDiagnosticSetting"), exports);
__exportStar(require("./getMetricAlert"), exports);
__exportStar(require("./getMyWorkbook"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getPrivateLinkScope"), exports);
__exportStar(require("./getPrivateLinkScopedResource"), exports);
__exportStar(require("./getProactiveDetectionConfiguration"), exports);
__exportStar(require("./getScheduledQueryRule"), exports);
__exportStar(require("./getSubscriptionDiagnosticSetting"), exports);
__exportStar(require("./getTestResultFile"), exports);
__exportStar(require("./getWebTest"), exports);
__exportStar(require("./getWorkbook"), exports);
__exportStar(require("./getWorkbookTemplate"), exports);
__exportStar(require("./getguestDiagnosticsSetting"), exports);
__exportStar(require("./guestDiagnosticsSetting"), exports);
__exportStar(require("./guestDiagnosticsSettingsAssociation"), exports);
__exportStar(require("./listEASubscriptionListMigrationDatePost"), exports);
__exportStar(require("./logProfile"), exports);
__exportStar(require("./managementGroupDiagnosticSetting"), exports);
__exportStar(require("./metricAlert"), exports);
__exportStar(require("./myWorkbook"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./privateLinkScope"), exports);
__exportStar(require("./privateLinkScopedResource"), exports);
__exportStar(require("./proactiveDetectionConfiguration"), exports);
__exportStar(require("./scheduledQueryRule"), exports);
__exportStar(require("./subscriptionDiagnosticSetting"), exports);
__exportStar(require("./webTest"), exports);
__exportStar(require("./workbook"), exports);
__exportStar(require("./workbookTemplate"), exports);
// Export enums:
__exportStar(require("../types/enums/insights"), exports);
// Export sub-modules:
const v20140401 = require("./v20140401");
exports.v20140401 = v20140401;
const v20150401 = require("./v20150401");
exports.v20150401 = v20150401;
const v20150501 = require("./v20150501");
exports.v20150501 = v20150501;
const v20160301 = require("./v20160301");
exports.v20160301 = v20160301;
const v20170401 = require("./v20170401");
exports.v20170401 = v20170401;
const v20170501preview = require("./v20170501preview");
exports.v20170501preview = v20170501preview;
const v20171001 = require("./v20171001");
exports.v20171001 = v20171001;
const v20180301 = require("./v20180301");
exports.v20180301 = v20180301;
const v20180416 = require("./v20180416");
exports.v20180416 = v20180416;
const v20180501preview = require("./v20180501preview");
exports.v20180501preview = v20180501preview;
const v20180601preview = require("./v20180601preview");
exports.v20180601preview = v20180601preview;
const v20180617preview = require("./v20180617preview");
exports.v20180617preview = v20180617preview;
const v20180901 = require("./v20180901");
exports.v20180901 = v20180901;
const v20190301 = require("./v20190301");
exports.v20190301 = v20190301;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20191017preview = require("./v20191017preview");
exports.v20191017preview = v20191017preview;
const v20191101preview = require("./v20191101preview");
exports.v20191101preview = v20191101preview;
const v20200101preview = require("./v20200101preview");
exports.v20200101preview = v20200101preview;
const v20200202 = require("./v20200202");
exports.v20200202 = v20200202;
const v20200202preview = require("./v20200202preview");
exports.v20200202preview = v20200202preview;
const v20200210preview = require("./v20200210preview");
exports.v20200210preview = v20200210preview;
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20200501preview = require("./v20200501preview");
exports.v20200501preview = v20200501preview;
const v20200602preview = require("./v20200602preview");
exports.v20200602preview = v20200602preview;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20201005preview = require("./v20201005preview");
exports.v20201005preview = v20201005preview;
const v20201020 = require("./v20201020");
exports.v20201020 = v20201020;
const v20201120 = require("./v20201120");
exports.v20201120 = v20201120;
const v20210201preview = require("./v20210201preview");
exports.v20210201preview = v20210201preview;
const v20210303preview = require("./v20210303preview");
exports.v20210303preview = v20210303preview;
const v20210308 = require("./v20210308");
exports.v20210308 = v20210308;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
const v20210501preview = require("./v20210501preview");
exports.v20210501preview = v20210501preview;
const v20210701preview = require("./v20210701preview");
exports.v20210701preview = v20210701preview;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
const v20210901 = require("./v20210901");
exports.v20210901 = v20210901;
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
const v20211014 = require("./v20211014");
exports.v20211014 = v20211014;
const v20220401 = require("./v20220401");
exports.v20220401 = v20220401;
// Import resources to register:
const actionGroup_1 = require("./actionGroup");
const activityLogAlert_1 = require("./activityLogAlert");
const alertRule_1 = require("./alertRule");
const analyticsItem_1 = require("./analyticsItem");
const autoscaleSetting_1 = require("./autoscaleSetting");
const component_1 = require("./component");
const componentCurrentBillingFeature_1 = require("./componentCurrentBillingFeature");
const componentLinkedStorageAccount_1 = require("./componentLinkedStorageAccount");
const dataCollectionEndpoint_1 = require("./dataCollectionEndpoint");
const dataCollectionRule_1 = require("./dataCollectionRule");
const dataCollectionRuleAssociation_1 = require("./dataCollectionRuleAssociation");
const diagnosticSetting_1 = require("./diagnosticSetting");
const exportConfiguration_1 = require("./exportConfiguration");
const favorite_1 = require("./favorite");
const guestDiagnosticsSettingsAssociation_1 = require("./guestDiagnosticsSettingsAssociation");
const logProfile_1 = require("./logProfile");
const managementGroupDiagnosticSetting_1 = require("./managementGroupDiagnosticSetting");
const metricAlert_1 = require("./metricAlert");
const myWorkbook_1 = require("./myWorkbook");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const privateLinkScope_1 = require("./privateLinkScope");
const privateLinkScopedResource_1 = require("./privateLinkScopedResource");
const proactiveDetectionConfiguration_1 = require("./proactiveDetectionConfiguration");
const scheduledQueryRule_1 = require("./scheduledQueryRule");
const subscriptionDiagnosticSetting_1 = require("./subscriptionDiagnosticSetting");
const webTest_1 = require("./webTest");
const workbook_1 = require("./workbook");
const workbookTemplate_1 = require("./workbookTemplate");
const guestDiagnosticsSetting_1 = require("./guestDiagnosticsSetting");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:insights:ActionGroup":
                return new actionGroup_1.ActionGroup(name, undefined, { urn });
            case "azure-native:insights:ActivityLogAlert":
                return new activityLogAlert_1.ActivityLogAlert(name, undefined, { urn });
            case "azure-native:insights:AlertRule":
                return new alertRule_1.AlertRule(name, undefined, { urn });
            case "azure-native:insights:AnalyticsItem":
                return new analyticsItem_1.AnalyticsItem(name, undefined, { urn });
            case "azure-native:insights:AutoscaleSetting":
                return new autoscaleSetting_1.AutoscaleSetting(name, undefined, { urn });
            case "azure-native:insights:Component":
                return new component_1.Component(name, undefined, { urn });
            case "azure-native:insights:ComponentCurrentBillingFeature":
                return new componentCurrentBillingFeature_1.ComponentCurrentBillingFeature(name, undefined, { urn });
            case "azure-native:insights:ComponentLinkedStorageAccount":
                return new componentLinkedStorageAccount_1.ComponentLinkedStorageAccount(name, undefined, { urn });
            case "azure-native:insights:DataCollectionEndpoint":
                return new dataCollectionEndpoint_1.DataCollectionEndpoint(name, undefined, { urn });
            case "azure-native:insights:DataCollectionRule":
                return new dataCollectionRule_1.DataCollectionRule(name, undefined, { urn });
            case "azure-native:insights:DataCollectionRuleAssociation":
                return new dataCollectionRuleAssociation_1.DataCollectionRuleAssociation(name, undefined, { urn });
            case "azure-native:insights:DiagnosticSetting":
                return new diagnosticSetting_1.DiagnosticSetting(name, undefined, { urn });
            case "azure-native:insights:ExportConfiguration":
                return new exportConfiguration_1.ExportConfiguration(name, undefined, { urn });
            case "azure-native:insights:Favorite":
                return new favorite_1.Favorite(name, undefined, { urn });
            case "azure-native:insights:GuestDiagnosticsSettingsAssociation":
                return new guestDiagnosticsSettingsAssociation_1.GuestDiagnosticsSettingsAssociation(name, undefined, { urn });
            case "azure-native:insights:LogProfile":
                return new logProfile_1.LogProfile(name, undefined, { urn });
            case "azure-native:insights:ManagementGroupDiagnosticSetting":
                return new managementGroupDiagnosticSetting_1.ManagementGroupDiagnosticSetting(name, undefined, { urn });
            case "azure-native:insights:MetricAlert":
                return new metricAlert_1.MetricAlert(name, undefined, { urn });
            case "azure-native:insights:MyWorkbook":
                return new myWorkbook_1.MyWorkbook(name, undefined, { urn });
            case "azure-native:insights:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:insights:PrivateLinkScope":
                return new privateLinkScope_1.PrivateLinkScope(name, undefined, { urn });
            case "azure-native:insights:PrivateLinkScopedResource":
                return new privateLinkScopedResource_1.PrivateLinkScopedResource(name, undefined, { urn });
            case "azure-native:insights:ProactiveDetectionConfiguration":
                return new proactiveDetectionConfiguration_1.ProactiveDetectionConfiguration(name, undefined, { urn });
            case "azure-native:insights:ScheduledQueryRule":
                return new scheduledQueryRule_1.ScheduledQueryRule(name, undefined, { urn });
            case "azure-native:insights:SubscriptionDiagnosticSetting":
                return new subscriptionDiagnosticSetting_1.SubscriptionDiagnosticSetting(name, undefined, { urn });
            case "azure-native:insights:WebTest":
                return new webTest_1.WebTest(name, undefined, { urn });
            case "azure-native:insights:Workbook":
                return new workbook_1.Workbook(name, undefined, { urn });
            case "azure-native:insights:WorkbookTemplate":
                return new workbookTemplate_1.WorkbookTemplate(name, undefined, { urn });
            case "azure-native:insights:guestDiagnosticsSetting":
                return new guestDiagnosticsSetting_1.GuestDiagnosticsSetting(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "insights", _module);
