// @pulumi/azure-native/servicefabricmesh/v20180701preview/network.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Network = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * This type describes a network resource.
 *
 * @deprecated Version v20180701preview will be removed in the next major version of the provider. Upgrade to version v20180901preview or later.
 */
class Network extends pulumi.CustomResource {
    /**
     * Create a Network resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180701preview will be removed in the next major version of the provider. Upgrade to version v20180901preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Network is deprecated: Version v20180701preview will be removed in the next major version of the provider. Upgrade to version v20180901preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.addressPrefix === undefined) && !opts.urn) {
                throw new Error("Missing required property 'addressPrefix'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["addressPrefix"] = args ? args.addressPrefix : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["ingressConfig"] = args ? args.ingressConfig : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["networkName"] = args ? args.networkName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["addressPrefix"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["ingressConfig"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:servicefabricmesh:Network" }, { type: "azure-native:servicefabricmesh/v20180901preview:Network" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Network.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Network resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Network is deprecated: Version v20180701preview will be removed in the next major version of the provider. Upgrade to version v20180901preview or later.");
        return new Network(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Network.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Network.__pulumiType;
    }
}
exports.Network = Network;
/** @internal */
Network.__pulumiType = 'azure-native:servicefabricmesh/v20180701preview:Network';
