// @pulumi/azure-native/types/enums/cache/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TlsVersion = exports.SkuName = exports.SkuFamily = exports.ReplicationRole = exports.RdbFrequency = exports.PublicNetworkAccess = exports.Protocol = exports.PrivateEndpointServiceConnectionStatus = exports.EvictionPolicy = exports.DayOfWeek = exports.ClusteringPolicy = exports.AofFrequency = exports.v20220101 = exports.v20210801 = exports.v20210601 = exports.v20210301 = exports.v20210201preview = exports.v20201201 = exports.v20201001preview = exports.v20200601 = exports.v20190701 = exports.v20180301 = exports.v20171001 = exports.v20170201 = exports.v20160401 = exports.v20150801 = void 0;
// Export sub-modules:
const v20150801 = require("./v20150801");
exports.v20150801 = v20150801;
const v20160401 = require("./v20160401");
exports.v20160401 = v20160401;
const v20170201 = require("./v20170201");
exports.v20170201 = v20170201;
const v20171001 = require("./v20171001");
exports.v20171001 = v20171001;
const v20180301 = require("./v20180301");
exports.v20180301 = v20180301;
const v20190701 = require("./v20190701");
exports.v20190701 = v20190701;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20201001preview = require("./v20201001preview");
exports.v20201001preview = v20201001preview;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210201preview = require("./v20210201preview");
exports.v20210201preview = v20210201preview;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
exports.AofFrequency = {
    AofFrequency_1s: "1s",
    Always: "always",
};
exports.ClusteringPolicy = {
    EnterpriseCluster: "EnterpriseCluster",
    OSSCluster: "OSSCluster",
};
exports.DayOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
    Everyday: "Everyday",
    Weekend: "Weekend",
};
exports.EvictionPolicy = {
    AllKeysLFU: "AllKeysLFU",
    AllKeysLRU: "AllKeysLRU",
    AllKeysRandom: "AllKeysRandom",
    VolatileLRU: "VolatileLRU",
    VolatileLFU: "VolatileLFU",
    VolatileTTL: "VolatileTTL",
    VolatileRandom: "VolatileRandom",
    NoEviction: "NoEviction",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.Protocol = {
    Encrypted: "Encrypted",
    Plaintext: "Plaintext",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.RdbFrequency = {
    RdbFrequency_1h: "1h",
    RdbFrequency_6h: "6h",
    RdbFrequency_12h: "12h",
};
exports.ReplicationRole = {
    Primary: "Primary",
    Secondary: "Secondary",
};
exports.SkuFamily = {
    C: "C",
    P: "P",
};
exports.SkuName = {
    Enterprise_E10: "Enterprise_E10",
    Enterprise_E20: "Enterprise_E20",
    Enterprise_E50: "Enterprise_E50",
    Enterprise_E100: "Enterprise_E100",
    EnterpriseFlash_F300: "EnterpriseFlash_F300",
    EnterpriseFlash_F700: "EnterpriseFlash_F700",
    EnterpriseFlash_F1500: "EnterpriseFlash_F1500",
};
exports.TlsVersion = {
    TlsVersion_1_0: "1.0",
    TlsVersion_1_1: "1.1",
    TlsVersion_1_2: "1.2",
};
