// @pulumi/azure-native/aad/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210501 = exports.v20210301 = exports.v20200101 = exports.v20170601 = exports.v20170101 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./domainService"), exports);
__exportStar(require("./getDomainService"), exports);
__exportStar(require("./getOuContainer"), exports);
__exportStar(require("./ouContainer"), exports);
// Export enums:
__exportStar(require("../types/enums/aad"), exports);
// Export sub-modules:
const v20170101 = require("./v20170101");
exports.v20170101 = v20170101;
const v20170601 = require("./v20170601");
exports.v20170601 = v20170601;
const v20200101 = require("./v20200101");
exports.v20200101 = v20200101;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
// Import resources to register:
const domainService_1 = require("./domainService");
const ouContainer_1 = require("./ouContainer");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:aad:DomainService":
                return new domainService_1.DomainService(name, undefined, { urn });
            case "azure-native:aad:OuContainer":
                return new ouContainer_1.OuContainer(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "aad", _module);
