// @pulumi/azure-native/types/enums/solutions/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.JitApproverType = exports.JitApprovalMode = exports.DeploymentMode = exports.ApplicationManagementMode = exports.ApplicationLockLevel = exports.ApplicationDefinitionArtifactName = exports.ApplicationArtifactType = exports.v20210701 = exports.v20200821preview = exports.v20190701 = exports.v20180601 = exports.v20170901 = exports.v20160901preview = void 0;
// Export sub-modules:
const v20160901preview = require("./v20160901preview");
exports.v20160901preview = v20160901preview;
const v20170901 = require("./v20170901");
exports.v20170901 = v20170901;
const v20180601 = require("./v20180601");
exports.v20180601 = v20180601;
const v20190701 = require("./v20190701");
exports.v20190701 = v20190701;
const v20200821preview = require("./v20200821preview");
exports.v20200821preview = v20200821preview;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
exports.ApplicationArtifactType = {
    NotSpecified: "NotSpecified",
    Template: "Template",
    Custom: "Custom",
};
exports.ApplicationDefinitionArtifactName = {
    NotSpecified: "NotSpecified",
    ApplicationResourceTemplate: "ApplicationResourceTemplate",
    CreateUiDefinition: "CreateUiDefinition",
    MainTemplateParameters: "MainTemplateParameters",
};
exports.ApplicationLockLevel = {
    CanNotDelete: "CanNotDelete",
    ReadOnly: "ReadOnly",
    None: "None",
};
exports.ApplicationManagementMode = {
    NotSpecified: "NotSpecified",
    Unmanaged: "Unmanaged",
    Managed: "Managed",
};
exports.DeploymentMode = {
    NotSpecified: "NotSpecified",
    Incremental: "Incremental",
    Complete: "Complete",
};
exports.JitApprovalMode = {
    NotSpecified: "NotSpecified",
    AutoApprove: "AutoApprove",
    ManualApprove: "ManualApprove",
};
exports.JitApproverType = {
    User: "user",
    Group: "group",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
