// @pulumi/azure-native/types/enums/sqlvirtualmachine/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageWorkloadType = exports.SqlWorkloadType = exports.SqlVmGroupImageSku = exports.SqlServerLicenseType = exports.SqlManagementMode = exports.SqlImageSku = exports.IdentityType = exports.FullBackupFrequencyType = exports.DiskConfigurationType = exports.DayOfWeek = exports.ConnectivityType = exports.BackupScheduleType = exports.v20211101preview = exports.v20170301preview = void 0;
// Export sub-modules:
const v20170301preview = require("./v20170301preview");
exports.v20170301preview = v20170301preview;
const v20211101preview = require("./v20211101preview");
exports.v20211101preview = v20211101preview;
exports.BackupScheduleType = {
    Manual: "Manual",
    Automated: "Automated",
};
exports.ConnectivityType = {
    LOCAL: "LOCAL",
    PRIVATE: "PRIVATE",
    PUBLIC: "PUBLIC",
};
exports.DayOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
};
exports.DiskConfigurationType = {
    NEW: "NEW",
    EXTEND: "EXTEND",
    ADD: "ADD",
};
exports.FullBackupFrequencyType = {
    Daily: "Daily",
    Weekly: "Weekly",
};
exports.IdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.SqlImageSku = {
    Developer: "Developer",
    Express: "Express",
    Standard: "Standard",
    Enterprise: "Enterprise",
    Web: "Web",
};
exports.SqlManagementMode = {
    Full: "Full",
    LightWeight: "LightWeight",
    NoAgent: "NoAgent",
};
exports.SqlServerLicenseType = {
    PAYG: "PAYG",
    AHUB: "AHUB",
    DR: "DR",
};
exports.SqlVmGroupImageSku = {
    Developer: "Developer",
    Enterprise: "Enterprise",
};
exports.SqlWorkloadType = {
    GENERAL: "GENERAL",
    OLTP: "OLTP",
    DW: "DW",
};
exports.StorageWorkloadType = {
    GENERAL: "GENERAL",
    OLTP: "OLTP",
    DW: "DW",
};
