// @pulumi/azure-native/digitaltwins/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210630preview = exports.v20201201 = exports.v20201031 = exports.v20200301preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./digitalTwin"), exports);
__exportStar(require("./digitalTwinsEndpoint"), exports);
__exportStar(require("./getDigitalTwin"), exports);
__exportStar(require("./getDigitalTwinsEndpoint"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getTimeSeriesDatabaseConnection"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./timeSeriesDatabaseConnection"), exports);
// Export enums:
__exportStar(require("../types/enums/digitaltwins"), exports);
// Export sub-modules:
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
const v20201031 = require("./v20201031");
exports.v20201031 = v20201031;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210630preview = require("./v20210630preview");
exports.v20210630preview = v20210630preview;
// Import resources to register:
const digitalTwin_1 = require("./digitalTwin");
const digitalTwinsEndpoint_1 = require("./digitalTwinsEndpoint");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const timeSeriesDatabaseConnection_1 = require("./timeSeriesDatabaseConnection");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:digitaltwins:DigitalTwin":
                return new digitalTwin_1.DigitalTwin(name, undefined, { urn });
            case "azure-native:digitaltwins:DigitalTwinsEndpoint":
                return new digitalTwinsEndpoint_1.DigitalTwinsEndpoint(name, undefined, { urn });
            case "azure-native:digitaltwins:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:digitaltwins:TimeSeriesDatabaseConnection":
                return new timeSeriesDatabaseConnection_1.TimeSeriesDatabaseConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "digitaltwins", _module);
