// @pulumi/azure-native/databoxedge/v20200901/device.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Device = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The Data Box Edge/Gateway device.
 *
 * @deprecated Version v20200901 will be removed in the next major version of the provider. Upgrade to version v20201201 or later.
 */
class Device extends pulumi.CustomResource {
    /**
     * Create a Device resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200901 will be removed in the next major version of the provider. Upgrade to version v20201201 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Device is deprecated: Version v20200901 will be removed in the next major version of the provider. Upgrade to version v20201201 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["dataBoxEdgeDeviceStatus"] = args ? args.dataBoxEdgeDeviceStatus : undefined;
            resourceInputs["deviceName"] = args ? args.deviceName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["configuredRoleTypes"] = undefined /*out*/;
            resourceInputs["culture"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["deviceHcsVersion"] = undefined /*out*/;
            resourceInputs["deviceLocalCapacity"] = undefined /*out*/;
            resourceInputs["deviceModel"] = undefined /*out*/;
            resourceInputs["deviceSoftwareVersion"] = undefined /*out*/;
            resourceInputs["deviceType"] = undefined /*out*/;
            resourceInputs["edgeProfile"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["friendlyName"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["modelDescription"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nodeCount"] = undefined /*out*/;
            resourceInputs["resourceMoveDetails"] = undefined /*out*/;
            resourceInputs["serialNumber"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["timeZone"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["configuredRoleTypes"] = undefined /*out*/;
            resourceInputs["culture"] = undefined /*out*/;
            resourceInputs["dataBoxEdgeDeviceStatus"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["deviceHcsVersion"] = undefined /*out*/;
            resourceInputs["deviceLocalCapacity"] = undefined /*out*/;
            resourceInputs["deviceModel"] = undefined /*out*/;
            resourceInputs["deviceSoftwareVersion"] = undefined /*out*/;
            resourceInputs["deviceType"] = undefined /*out*/;
            resourceInputs["edgeProfile"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["friendlyName"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["modelDescription"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["nodeCount"] = undefined /*out*/;
            resourceInputs["resourceMoveDetails"] = undefined /*out*/;
            resourceInputs["serialNumber"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["timeZone"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databoxedge:Device" }, { type: "azure-native:databoxedge/v20190301:Device" }, { type: "azure-native:databoxedge/v20190701:Device" }, { type: "azure-native:databoxedge/v20190801:Device" }, { type: "azure-native:databoxedge/v20200501preview:Device" }, { type: "azure-native:databoxedge/v20200901preview:Device" }, { type: "azure-native:databoxedge/v20201201:Device" }, { type: "azure-native:databoxedge/v20210201:Device" }, { type: "azure-native:databoxedge/v20210201preview:Device" }, { type: "azure-native:databoxedge/v20210601:Device" }, { type: "azure-native:databoxedge/v20210601preview:Device" }, { type: "azure-native:databoxedge/v20220301:Device" }, { type: "azure-native:databoxedge/v20220401preview:Device" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Device.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Device resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Device is deprecated: Version v20200901 will be removed in the next major version of the provider. Upgrade to version v20201201 or later.");
        return new Device(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Device.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Device.__pulumiType;
    }
}
exports.Device = Device;
/** @internal */
Device.__pulumiType = 'azure-native:databoxedge/v20200901:Device';
