// @pulumi/azure-native/solutions/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210701 = exports.v20200821preview = exports.v20190701 = exports.v20180601 = exports.v20170901 = exports.v20160901preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./application"), exports);
__exportStar(require("./applicationDefinition"), exports);
__exportStar(require("./getApplication"), exports);
__exportStar(require("./getApplicationDefinition"), exports);
__exportStar(require("./getJitRequest"), exports);
__exportStar(require("./jitRequest"), exports);
// Export enums:
__exportStar(require("../types/enums/solutions"), exports);
// Export sub-modules:
const v20160901preview = require("./v20160901preview");
exports.v20160901preview = v20160901preview;
const v20170901 = require("./v20170901");
exports.v20170901 = v20170901;
const v20180601 = require("./v20180601");
exports.v20180601 = v20180601;
const v20190701 = require("./v20190701");
exports.v20190701 = v20190701;
const v20200821preview = require("./v20200821preview");
exports.v20200821preview = v20200821preview;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
// Import resources to register:
const application_1 = require("./application");
const applicationDefinition_1 = require("./applicationDefinition");
const jitRequest_1 = require("./jitRequest");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:solutions:Application":
                return new application_1.Application(name, undefined, { urn });
            case "azure-native:solutions:ApplicationDefinition":
                return new applicationDefinition_1.ApplicationDefinition(name, undefined, { urn });
            case "azure-native:solutions:JitRequest":
                return new jitRequest_1.JitRequest(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "solutions", _module);
