// @pulumi/azure-native/types/enums/web/v20210301/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnauthenticatedClientAction = exports.SupportedTlsVersions = exports.StorageType = exports.StagingEnvironmentPolicy = exports.SslState = exports.SiteLoadBalancing = exports.ScmType = exports.RouteType = exports.RedundancyMode = exports.PublicCertificateLocation = exports.ManagedServiceIdentityType = exports.ManagedPipelineMode = exports.LogLevel = exports.LoadBalancingMode = exports.IpFilterTag = exports.IngressTransportMethod = exports.HostType = exports.HostNameType = exports.FtpsState = exports.FrontEndServiceType = exports.FrequencyUnit = exports.EnterpriseGradeCdnStatus = exports.DatabaseType = exports.CustomHostNameDnsRecordType = exports.ConnectionStringType = exports.ClientCertMode = exports.BuiltInAuthenticationProvider = exports.AzureStorageType = exports.AzureResourceType = exports.AutoHealActionType = exports.ActiveRevisionsMode = void 0;
exports.ActiveRevisionsMode = {
    Multiple: "multiple",
    Single: "single",
};
exports.AutoHealActionType = {
    Recycle: "Recycle",
    LogEvent: "LogEvent",
    CustomAction: "CustomAction",
};
exports.AzureResourceType = {
    Website: "Website",
    TrafficManager: "TrafficManager",
};
exports.AzureStorageType = {
    AzureFiles: "AzureFiles",
    AzureBlob: "AzureBlob",
};
exports.BuiltInAuthenticationProvider = {
    AzureActiveDirectory: "AzureActiveDirectory",
    Facebook: "Facebook",
    Google: "Google",
    MicrosoftAccount: "MicrosoftAccount",
    Twitter: "Twitter",
    Github: "Github",
};
exports.ClientCertMode = {
    Required: "Required",
    Optional: "Optional",
    OptionalInteractiveUser: "OptionalInteractiveUser",
};
exports.ConnectionStringType = {
    MySql: "MySql",
    SQLServer: "SQLServer",
    SQLAzure: "SQLAzure",
    Custom: "Custom",
    NotificationHub: "NotificationHub",
    ServiceBus: "ServiceBus",
    EventHub: "EventHub",
    ApiHub: "ApiHub",
    DocDb: "DocDb",
    RedisCache: "RedisCache",
    PostgreSQL: "PostgreSQL",
};
exports.CustomHostNameDnsRecordType = {
    CName: "CName",
    A: "A",
};
exports.DatabaseType = {
    SqlAzure: "SqlAzure",
    MySql: "MySql",
    LocalMySql: "LocalMySql",
    PostgreSql: "PostgreSql",
};
exports.EnterpriseGradeCdnStatus = {
    Enabled: "Enabled",
    Enabling: "Enabling",
    Disabled: "Disabled",
    Disabling: "Disabling",
};
exports.FrequencyUnit = {
    Day: "Day",
    Hour: "Hour",
};
exports.FrontEndServiceType = {
    NodePort: "NodePort",
    LoadBalancer: "LoadBalancer",
};
exports.FtpsState = {
    AllAllowed: "AllAllowed",
    FtpsOnly: "FtpsOnly",
    Disabled: "Disabled",
};
exports.HostNameType = {
    Verified: "Verified",
    Managed: "Managed",
};
exports.HostType = {
    Standard: "Standard",
    Repository: "Repository",
};
exports.IngressTransportMethod = {
    Auto: "auto",
    Http: "http",
    Http2: "http2",
};
exports.IpFilterTag = {
    Default: "Default",
    XffProxy: "XffProxy",
    ServiceTag: "ServiceTag",
};
exports.LoadBalancingMode = {
    None: "None",
    Web: "Web",
    Publishing: "Publishing",
    Web_Publishing: "Web, Publishing",
};
exports.LogLevel = {
    Off: "Off",
    Verbose: "Verbose",
    Information: "Information",
    Warning: "Warning",
    Error: "Error",
};
exports.ManagedPipelineMode = {
    Integrated: "Integrated",
    Classic: "Classic",
};
exports.ManagedServiceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.PublicCertificateLocation = {
    CurrentUserMy: "CurrentUserMy",
    LocalMachineMy: "LocalMachineMy",
    Unknown: "Unknown",
};
exports.RedundancyMode = {
    None: "None",
    Manual: "Manual",
    Failover: "Failover",
    ActiveActive: "ActiveActive",
    GeoRedundant: "GeoRedundant",
};
exports.RouteType = {
    DEFAULT: "DEFAULT",
    INHERITED: "INHERITED",
    STATIC: "STATIC",
};
exports.ScmType = {
    None: "None",
    Dropbox: "Dropbox",
    Tfs: "Tfs",
    LocalGit: "LocalGit",
    GitHub: "GitHub",
    CodePlexGit: "CodePlexGit",
    CodePlexHg: "CodePlexHg",
    BitbucketGit: "BitbucketGit",
    BitbucketHg: "BitbucketHg",
    ExternalGit: "ExternalGit",
    ExternalHg: "ExternalHg",
    OneDrive: "OneDrive",
    VSO: "VSO",
    VSTSRM: "VSTSRM",
};
exports.SiteLoadBalancing = {
    WeightedRoundRobin: "WeightedRoundRobin",
    LeastRequests: "LeastRequests",
    LeastResponseTime: "LeastResponseTime",
    WeightedTotalTraffic: "WeightedTotalTraffic",
    RequestHash: "RequestHash",
    PerSiteRoundRobin: "PerSiteRoundRobin",
};
exports.SslState = {
    Disabled: "Disabled",
    SniEnabled: "SniEnabled",
    IpBasedEnabled: "IpBasedEnabled",
};
exports.StagingEnvironmentPolicy = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.StorageType = {
    LocalNode: "LocalNode",
    NetworkFileSystem: "NetworkFileSystem",
};
exports.SupportedTlsVersions = {
    SupportedTlsVersions_1_0: "1.0",
    SupportedTlsVersions_1_1: "1.1",
    SupportedTlsVersions_1_2: "1.2",
};
exports.UnauthenticatedClientAction = {
    RedirectToLoginPage: "RedirectToLoginPage",
    AllowAnonymous: "AllowAnonymous",
};
