// @pulumi/azure-native/recoveryservices/v20180110/replicationRecoveryServicesProvider.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationRecoveryServicesProvider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Provider details.
 *
 * @deprecated Version v20180110 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.
 */
class ReplicationRecoveryServicesProvider extends pulumi.CustomResource {
    /**
     * Create a ReplicationRecoveryServicesProvider resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180110 will be removed in the next major version of the provider. Upgrade to version v20180710 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ReplicationRecoveryServicesProvider is deprecated: Version v20180110 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if ((!args || args.properties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["fabricName"] = args ? args.fabricName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["providerName"] = args ? args.providerName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20180710:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210210:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210301:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210401:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210601:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210701:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20210801:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20211001:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20211101:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20211201:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20220101:ReplicationRecoveryServicesProvider" }, { type: "azure-native:recoveryservices/v20220201:ReplicationRecoveryServicesProvider" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ReplicationRecoveryServicesProvider.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ReplicationRecoveryServicesProvider resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ReplicationRecoveryServicesProvider is deprecated: Version v20180110 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        return new ReplicationRecoveryServicesProvider(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationRecoveryServicesProvider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationRecoveryServicesProvider.__pulumiType;
    }
}
exports.ReplicationRecoveryServicesProvider = ReplicationRecoveryServicesProvider;
/** @internal */
ReplicationRecoveryServicesProvider.__pulumiType = 'azure-native:recoveryservices/v20180110:ReplicationRecoveryServicesProvider';
