// @pulumi/azure-native/recoveryservices/v20180110/replicationProtectedItem.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationProtectedItem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Replication protected item.
 *
 * @deprecated Version v20180110 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.
 */
class ReplicationProtectedItem extends pulumi.CustomResource {
    /**
     * Create a ReplicationProtectedItem resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180110 will be removed in the next major version of the provider. Upgrade to version v20180710 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ReplicationProtectedItem is deprecated: Version v20180110 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if ((!args || args.protectionContainerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'protectionContainerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["fabricName"] = args ? args.fabricName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["protectionContainerName"] = args ? args.protectionContainerName : undefined;
            resourceInputs["replicatedProtectedItemName"] = args ? args.replicatedProtectedItemName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20160810:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20180710:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210210:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210301:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210401:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210601:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210701:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20210801:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20211001:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20211101:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20211201:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20220101:ReplicationProtectedItem" }, { type: "azure-native:recoveryservices/v20220201:ReplicationProtectedItem" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ReplicationProtectedItem.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ReplicationProtectedItem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ReplicationProtectedItem is deprecated: Version v20180110 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        return new ReplicationProtectedItem(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationProtectedItem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationProtectedItem.__pulumiType;
    }
}
exports.ReplicationProtectedItem = ReplicationProtectedItem;
/** @internal */
ReplicationProtectedItem.__pulumiType = 'azure-native:recoveryservices/v20180110:ReplicationProtectedItem';
