// @pulumi/azure-native/types/enums/peering/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValidationState = exports.Tier = exports.Size = exports.SessionAddressProvider = exports.Role = exports.Kind = exports.Family = exports.DirectPeeringType = exports.v20220101 = exports.v20210601 = exports.v20210101 = exports.v20201001 = exports.v20200401 = exports.v20200101preview = exports.v20190901preview = exports.v20190801preview = void 0;
// Export sub-modules:
const v20190801preview = require("./v20190801preview");
exports.v20190801preview = v20190801preview;
const v20190901preview = require("./v20190901preview");
exports.v20190901preview = v20190901preview;
const v20200101preview = require("./v20200101preview");
exports.v20200101preview = v20200101preview;
const v20200401 = require("./v20200401");
exports.v20200401 = v20200401;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
exports.DirectPeeringType = {
    Edge: "Edge",
    Transit: "Transit",
    Cdn: "Cdn",
    Internal: "Internal",
    Ix: "Ix",
    IxRs: "IxRs",
    Voice: "Voice",
};
exports.Family = {
    Direct: "Direct",
    Exchange: "Exchange",
};
exports.Kind = {
    Direct: "Direct",
    Exchange: "Exchange",
};
exports.Role = {
    Noc: "Noc",
    Policy: "Policy",
    Technical: "Technical",
    Service: "Service",
    Escalation: "Escalation",
    Other: "Other",
};
exports.SessionAddressProvider = {
    Microsoft: "Microsoft",
    Peer: "Peer",
};
exports.Size = {
    Free: "Free",
    Metered: "Metered",
    Unlimited: "Unlimited",
};
exports.Tier = {
    Basic: "Basic",
    Premium: "Premium",
};
exports.ValidationState = {
    None: "None",
    Pending: "Pending",
    Approved: "Approved",
    Failed: "Failed",
};
