// @pulumi/azure-native/types/enums/avs/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SslEnum = exports.ScriptExecutionParameterType = exports.PortMirroringDirectionEnum = exports.PlacementPolicyType = exports.PlacementPolicyState = exports.InternetEnum = exports.DnsServiceLogLevelEnum = exports.DhcpTypeEnum = exports.AffinityType = exports.AddonType = exports.v20211201 = exports.v20210601 = exports.v20210101preview = exports.v20200717preview = exports.v20200320 = void 0;
// Export sub-modules:
const v20200320 = require("./v20200320");
exports.v20200320 = v20200320;
const v20200717preview = require("./v20200717preview");
exports.v20200717preview = v20200717preview;
const v20210101preview = require("./v20210101preview");
exports.v20210101preview = v20210101preview;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20211201 = require("./v20211201");
exports.v20211201 = v20211201;
exports.AddonType = {
    SRM: "SRM",
    VR: "VR",
};
exports.AffinityType = {
    Affinity: "Affinity",
    AntiAffinity: "AntiAffinity",
};
exports.DhcpTypeEnum = {
    SERVER_RELAY: "SERVER, RELAY",
};
exports.DnsServiceLogLevelEnum = {
    DEBUG: "DEBUG",
    INFO: "INFO",
    WARNING: "WARNING",
    ERROR: "ERROR",
    FATAL: "FATAL",
};
exports.InternetEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PlacementPolicyState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PlacementPolicyType = {
    VmVm: "VmVm",
    VmHost: "VmHost",
};
exports.PortMirroringDirectionEnum = {
    INGRESS_EGRESS_BIDIRECTIONAL: "INGRESS, EGRESS, BIDIRECTIONAL",
};
exports.ScriptExecutionParameterType = {
    Value: "Value",
    SecureValue: "SecureValue",
    Credential: "Credential",
};
exports.SslEnum = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
