// @pulumi/azure-native/containerregistry/v20180201preview/buildTask.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BuildTask = void 0;
const pulumi = require("@pulumi/pulumi");
const types_1 = require("../../types");
const utilities = require("../../utilities");
/**
 * The build task that has the resource properties and all build items. The build task will have all information to schedule a build against it.
 */
class BuildTask extends pulumi.CustomResource {
    /**
     * Create a BuildTask resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.alias === undefined) && !opts.urn) {
                throw new Error("Missing required property 'alias'");
            }
            if ((!args || args.platform === undefined) && !opts.urn) {
                throw new Error("Missing required property 'platform'");
            }
            if ((!args || args.registryName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'registryName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.sourceRepository === undefined) && !opts.urn) {
                throw new Error("Missing required property 'sourceRepository'");
            }
            resourceInputs["alias"] = args ? args.alias : undefined;
            resourceInputs["buildTaskName"] = args ? args.buildTaskName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["platform"] = args ? args.platform : undefined;
            resourceInputs["registryName"] = args ? args.registryName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sourceRepository"] = args ? (args.sourceRepository ? pulumi.output(args.sourceRepository).apply(types_1.input.containerregistry.v20180201preview.sourceRepositoryPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["status"] = args ? args.status : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["timeout"] = (_a = (args ? args.timeout : undefined)) !== null && _a !== void 0 ? _a : 3600;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["alias"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["platform"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sourceRepository"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["timeout"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:containerregistry:BuildTask" }, { type: "azure-native:containerregistry/v20180901:BuildTask" }, { type: "azure-native:containerregistry/v20190401:BuildTask" }, { type: "azure-native:containerregistry/v20190601preview:BuildTask" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(BuildTask.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing BuildTask resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new BuildTask(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BuildTask.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BuildTask.__pulumiType;
    }
}
exports.BuildTask = BuildTask;
/** @internal */
BuildTask.__pulumiType = 'azure-native:containerregistry/v20180201preview:BuildTask';
