// @pulumi/azure-native/cache/v20160401/redisFirewallRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RedisFirewallRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A firewall rule on a redis cache has a name, and describes a contiguous range of IP addresses permitted to connect
 *
 * @deprecated Version v20160401 will be removed in the next major version of the provider. Upgrade to version v20200601 or later.
 */
class RedisFirewallRule extends pulumi.CustomResource {
    /**
     * Create a RedisFirewallRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160401 will be removed in the next major version of the provider. Upgrade to version v20200601 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("RedisFirewallRule is deprecated: Version v20160401 will be removed in the next major version of the provider. Upgrade to version v20200601 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.cacheName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'cacheName'");
            }
            if ((!args || args.endIP === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endIP'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.startIP === undefined) && !opts.urn) {
                throw new Error("Missing required property 'startIP'");
            }
            resourceInputs["cacheName"] = args ? args.cacheName : undefined;
            resourceInputs["endIP"] = args ? args.endIP : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["ruleName"] = args ? args.ruleName : undefined;
            resourceInputs["startIP"] = args ? args.startIP : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["endIP"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["startIP"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:cache:RedisFirewallRule" }, { type: "azure-native:cache/v20170201:RedisFirewallRule" }, { type: "azure-native:cache/v20171001:RedisFirewallRule" }, { type: "azure-native:cache/v20180301:RedisFirewallRule" }, { type: "azure-native:cache/v20190701:RedisFirewallRule" }, { type: "azure-native:cache/v20200601:RedisFirewallRule" }, { type: "azure-native:cache/v20201201:RedisFirewallRule" }, { type: "azure-native:cache/v20210601:RedisFirewallRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RedisFirewallRule.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing RedisFirewallRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("RedisFirewallRule is deprecated: Version v20160401 will be removed in the next major version of the provider. Upgrade to version v20200601 or later.");
        return new RedisFirewallRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RedisFirewallRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RedisFirewallRule.__pulumiType;
    }
}
exports.RedisFirewallRule = RedisFirewallRule;
/** @internal */
RedisFirewallRule.__pulumiType = 'azure-native:cache/v20160401:RedisFirewallRule';
