// @pulumi/azure-native/types/enums/deviceupdate/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.ManagedServiceIdentityType = exports.AuthenticationType = exports.v20200301preview = void 0;
// Export sub-modules:
const v20200301preview = require("./v20200301preview");
exports.v20200301preview = v20200301preview;
exports.AuthenticationType = {
    KeyBased: "KeyBased",
};
exports.ManagedServiceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
