// @pulumi/azure-native/deploymentmanager/v20180901preview/rollout.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Rollout = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Defines the PUT rollout request body.
 *
 * @deprecated Version v20180901preview will be removed in the next major version of the provider. Upgrade to version v20191101preview or later.
 */
class Rollout extends pulumi.CustomResource {
    /**
     * Create a Rollout resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180901preview will be removed in the next major version of the provider. Upgrade to version v20191101preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Rollout is deprecated: Version v20180901preview will be removed in the next major version of the provider. Upgrade to version v20191101preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.buildVersion === undefined) && !opts.urn) {
                throw new Error("Missing required property 'buildVersion'");
            }
            if ((!args || args.identity === undefined) && !opts.urn) {
                throw new Error("Missing required property 'identity'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.stepGroups === undefined) && !opts.urn) {
                throw new Error("Missing required property 'stepGroups'");
            }
            if ((!args || args.targetServiceTopologyId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetServiceTopologyId'");
            }
            resourceInputs["artifactSourceId"] = args ? args.artifactSourceId : undefined;
            resourceInputs["buildVersion"] = args ? args.buildVersion : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["rolloutName"] = args ? args.rolloutName : undefined;
            resourceInputs["stepGroups"] = args ? args.stepGroups : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["targetServiceTopologyId"] = args ? args.targetServiceTopologyId : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["artifactSourceId"] = undefined /*out*/;
            resourceInputs["buildVersion"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["stepGroups"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["targetServiceTopologyId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:deploymentmanager:Rollout" }, { type: "azure-native:deploymentmanager/v20191101preview:Rollout" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Rollout.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Rollout resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Rollout is deprecated: Version v20180901preview will be removed in the next major version of the provider. Upgrade to version v20191101preview or later.");
        return new Rollout(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Rollout.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Rollout.__pulumiType;
    }
}
exports.Rollout = Rollout;
/** @internal */
Rollout.__pulumiType = 'azure-native:deploymentmanager/v20180901preview:Rollout';
