// @pulumi/azure-native/deploymentmanager/v20180901preview/service.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The resource representation of a service in a service topology.
 *
 * @deprecated Version v20180901preview will be removed in the next major version of the provider. Upgrade to version v20191101preview or later.
 */
class Service extends pulumi.CustomResource {
    /**
     * Create a Service resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180901preview will be removed in the next major version of the provider. Upgrade to version v20191101preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Service is deprecated: Version v20180901preview will be removed in the next major version of the provider. Upgrade to version v20191101preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.serviceTopologyName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceTopologyName'");
            }
            if ((!args || args.targetLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetLocation'");
            }
            if ((!args || args.targetSubscriptionId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'targetSubscriptionId'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["serviceTopologyName"] = args ? args.serviceTopologyName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["targetLocation"] = args ? args.targetLocation : undefined;
            resourceInputs["targetSubscriptionId"] = args ? args.targetSubscriptionId : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["targetLocation"] = undefined /*out*/;
            resourceInputs["targetSubscriptionId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:deploymentmanager:Service" }, { type: "azure-native:deploymentmanager/v20191101preview:Service" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Service.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Service resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Service is deprecated: Version v20180901preview will be removed in the next major version of the provider. Upgrade to version v20191101preview or later.");
        return new Service(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Service.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Service.__pulumiType;
    }
}
exports.Service = Service;
/** @internal */
Service.__pulumiType = 'azure-native:deploymentmanager/v20180901preview:Service';
