// @pulumi/azure-native/databoxedge/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220401preview = exports.v20220301 = exports.v20210601preview = exports.v20210601 = exports.v20210201preview = exports.v20210201 = exports.v20201201 = exports.v20200901preview = exports.v20200901 = exports.v20200501preview = exports.v20190801 = exports.v20190701 = exports.v20190301 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./addon"), exports);
__exportStar(require("./arcAddon"), exports);
__exportStar(require("./bandwidthSchedule"), exports);
__exportStar(require("./cloudEdgeManagementRole"), exports);
__exportStar(require("./container"), exports);
__exportStar(require("./device"), exports);
__exportStar(require("./fileEventTrigger"), exports);
__exportStar(require("./getAddon"), exports);
__exportStar(require("./getArcAddon"), exports);
__exportStar(require("./getBandwidthSchedule"), exports);
__exportStar(require("./getCloudEdgeManagementRole"), exports);
__exportStar(require("./getContainer"), exports);
__exportStar(require("./getDevice"), exports);
__exportStar(require("./getDeviceExtendedInformation"), exports);
__exportStar(require("./getFileEventTrigger"), exports);
__exportStar(require("./getIoTAddon"), exports);
__exportStar(require("./getIoTRole"), exports);
__exportStar(require("./getKubernetesRole"), exports);
__exportStar(require("./getMECRole"), exports);
__exportStar(require("./getMonitoringConfig"), exports);
__exportStar(require("./getOrder"), exports);
__exportStar(require("./getPeriodicTimerEventTrigger"), exports);
__exportStar(require("./getRole"), exports);
__exportStar(require("./getShare"), exports);
__exportStar(require("./getStorageAccount"), exports);
__exportStar(require("./getStorageAccountCredential"), exports);
__exportStar(require("./getTrigger"), exports);
__exportStar(require("./getUser"), exports);
__exportStar(require("./ioTAddon"), exports);
__exportStar(require("./ioTRole"), exports);
__exportStar(require("./kubernetesRole"), exports);
__exportStar(require("./listOrderDCAccessCode"), exports);
__exportStar(require("./mecrole"), exports);
__exportStar(require("./monitoringConfig"), exports);
__exportStar(require("./order"), exports);
__exportStar(require("./periodicTimerEventTrigger"), exports);
__exportStar(require("./role"), exports);
__exportStar(require("./share"), exports);
__exportStar(require("./storageAccount"), exports);
__exportStar(require("./storageAccountCredential"), exports);
__exportStar(require("./trigger"), exports);
__exportStar(require("./user"), exports);
// Export enums:
__exportStar(require("../types/enums/databoxedge"), exports);
// Export sub-modules:
const v20190301 = require("./v20190301");
exports.v20190301 = v20190301;
const v20190701 = require("./v20190701");
exports.v20190701 = v20190701;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20200501preview = require("./v20200501preview");
exports.v20200501preview = v20200501preview;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20200901preview = require("./v20200901preview");
exports.v20200901preview = v20200901preview;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20210201preview = require("./v20210201preview");
exports.v20210201preview = v20210201preview;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20220301 = require("./v20220301");
exports.v20220301 = v20220301;
const v20220401preview = require("./v20220401preview");
exports.v20220401preview = v20220401preview;
// Import resources to register:
const addon_1 = require("./addon");
const arcAddon_1 = require("./arcAddon");
const bandwidthSchedule_1 = require("./bandwidthSchedule");
const cloudEdgeManagementRole_1 = require("./cloudEdgeManagementRole");
const container_1 = require("./container");
const device_1 = require("./device");
const fileEventTrigger_1 = require("./fileEventTrigger");
const ioTAddon_1 = require("./ioTAddon");
const ioTRole_1 = require("./ioTRole");
const kubernetesRole_1 = require("./kubernetesRole");
const mecrole_1 = require("./mecrole");
const monitoringConfig_1 = require("./monitoringConfig");
const order_1 = require("./order");
const periodicTimerEventTrigger_1 = require("./periodicTimerEventTrigger");
const role_1 = require("./role");
const share_1 = require("./share");
const storageAccount_1 = require("./storageAccount");
const storageAccountCredential_1 = require("./storageAccountCredential");
const trigger_1 = require("./trigger");
const user_1 = require("./user");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:databoxedge:Addon":
                return new addon_1.Addon(name, undefined, { urn });
            case "azure-native:databoxedge:ArcAddon":
                return new arcAddon_1.ArcAddon(name, undefined, { urn });
            case "azure-native:databoxedge:BandwidthSchedule":
                return new bandwidthSchedule_1.BandwidthSchedule(name, undefined, { urn });
            case "azure-native:databoxedge:CloudEdgeManagementRole":
                return new cloudEdgeManagementRole_1.CloudEdgeManagementRole(name, undefined, { urn });
            case "azure-native:databoxedge:Container":
                return new container_1.Container(name, undefined, { urn });
            case "azure-native:databoxedge:Device":
                return new device_1.Device(name, undefined, { urn });
            case "azure-native:databoxedge:FileEventTrigger":
                return new fileEventTrigger_1.FileEventTrigger(name, undefined, { urn });
            case "azure-native:databoxedge:IoTAddon":
                return new ioTAddon_1.IoTAddon(name, undefined, { urn });
            case "azure-native:databoxedge:IoTRole":
                return new ioTRole_1.IoTRole(name, undefined, { urn });
            case "azure-native:databoxedge:KubernetesRole":
                return new kubernetesRole_1.KubernetesRole(name, undefined, { urn });
            case "azure-native:databoxedge:MECRole":
                return new mecrole_1.MECRole(name, undefined, { urn });
            case "azure-native:databoxedge:MonitoringConfig":
                return new monitoringConfig_1.MonitoringConfig(name, undefined, { urn });
            case "azure-native:databoxedge:Order":
                return new order_1.Order(name, undefined, { urn });
            case "azure-native:databoxedge:PeriodicTimerEventTrigger":
                return new periodicTimerEventTrigger_1.PeriodicTimerEventTrigger(name, undefined, { urn });
            case "azure-native:databoxedge:Role":
                return new role_1.Role(name, undefined, { urn });
            case "azure-native:databoxedge:Share":
                return new share_1.Share(name, undefined, { urn });
            case "azure-native:databoxedge:StorageAccount":
                return new storageAccount_1.StorageAccount(name, undefined, { urn });
            case "azure-native:databoxedge:StorageAccountCredential":
                return new storageAccountCredential_1.StorageAccountCredential(name, undefined, { urn });
            case "azure-native:databoxedge:Trigger":
                return new trigger_1.Trigger(name, undefined, { urn });
            case "azure-native:databoxedge:User":
                return new user_1.User(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "databoxedge", _module);
