// @pulumi/azure-native/cdn/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210601 = exports.v20200901 = exports.v20200415 = exports.v20200331 = exports.v20191231 = exports.v20190615preview = exports.v20190615 = exports.v20190415 = exports.v20171012 = exports.v20170402 = exports.v20161002 = exports.v20160402 = exports.v20150601 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./afdcustomDomain"), exports);
__exportStar(require("./afdendpoint"), exports);
__exportStar(require("./afdorigin"), exports);
__exportStar(require("./afdoriginGroup"), exports);
__exportStar(require("./customDomain"), exports);
__exportStar(require("./endpoint"), exports);
__exportStar(require("./getAFDCustomDomain"), exports);
__exportStar(require("./getAFDEndpoint"), exports);
__exportStar(require("./getAFDOrigin"), exports);
__exportStar(require("./getAFDOriginGroup"), exports);
__exportStar(require("./getCustomDomain"), exports);
__exportStar(require("./getEndpoint"), exports);
__exportStar(require("./getOrigin"), exports);
__exportStar(require("./getOriginGroup"), exports);
__exportStar(require("./getPolicy"), exports);
__exportStar(require("./getProfile"), exports);
__exportStar(require("./getProfileSupportedOptimizationTypes"), exports);
__exportStar(require("./getRoute"), exports);
__exportStar(require("./getRule"), exports);
__exportStar(require("./getRuleSet"), exports);
__exportStar(require("./getSecret"), exports);
__exportStar(require("./getSecurityPolicy"), exports);
__exportStar(require("./origin"), exports);
__exportStar(require("./originGroup"), exports);
__exportStar(require("./policy"), exports);
__exportStar(require("./profile"), exports);
__exportStar(require("./route"), exports);
__exportStar(require("./rule"), exports);
__exportStar(require("./ruleSet"), exports);
__exportStar(require("./secret"), exports);
__exportStar(require("./securityPolicy"), exports);
// Export enums:
__exportStar(require("../types/enums/cdn"), exports);
// Export sub-modules:
const v20150601 = require("./v20150601");
exports.v20150601 = v20150601;
const v20160402 = require("./v20160402");
exports.v20160402 = v20160402;
const v20161002 = require("./v20161002");
exports.v20161002 = v20161002;
const v20170402 = require("./v20170402");
exports.v20170402 = v20170402;
const v20171012 = require("./v20171012");
exports.v20171012 = v20171012;
const v20190415 = require("./v20190415");
exports.v20190415 = v20190415;
const v20190615 = require("./v20190615");
exports.v20190615 = v20190615;
const v20190615preview = require("./v20190615preview");
exports.v20190615preview = v20190615preview;
const v20191231 = require("./v20191231");
exports.v20191231 = v20191231;
const v20200331 = require("./v20200331");
exports.v20200331 = v20200331;
const v20200415 = require("./v20200415");
exports.v20200415 = v20200415;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
// Import resources to register:
const afdcustomDomain_1 = require("./afdcustomDomain");
const afdendpoint_1 = require("./afdendpoint");
const afdorigin_1 = require("./afdorigin");
const afdoriginGroup_1 = require("./afdoriginGroup");
const customDomain_1 = require("./customDomain");
const endpoint_1 = require("./endpoint");
const origin_1 = require("./origin");
const originGroup_1 = require("./originGroup");
const policy_1 = require("./policy");
const profile_1 = require("./profile");
const route_1 = require("./route");
const rule_1 = require("./rule");
const ruleSet_1 = require("./ruleSet");
const secret_1 = require("./secret");
const securityPolicy_1 = require("./securityPolicy");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cdn:AFDCustomDomain":
                return new afdcustomDomain_1.AFDCustomDomain(name, undefined, { urn });
            case "azure-native:cdn:AFDEndpoint":
                return new afdendpoint_1.AFDEndpoint(name, undefined, { urn });
            case "azure-native:cdn:AFDOrigin":
                return new afdorigin_1.AFDOrigin(name, undefined, { urn });
            case "azure-native:cdn:AFDOriginGroup":
                return new afdoriginGroup_1.AFDOriginGroup(name, undefined, { urn });
            case "azure-native:cdn:CustomDomain":
                return new customDomain_1.CustomDomain(name, undefined, { urn });
            case "azure-native:cdn:Endpoint":
                return new endpoint_1.Endpoint(name, undefined, { urn });
            case "azure-native:cdn:Origin":
                return new origin_1.Origin(name, undefined, { urn });
            case "azure-native:cdn:OriginGroup":
                return new originGroup_1.OriginGroup(name, undefined, { urn });
            case "azure-native:cdn:Policy":
                return new policy_1.Policy(name, undefined, { urn });
            case "azure-native:cdn:Profile":
                return new profile_1.Profile(name, undefined, { urn });
            case "azure-native:cdn:Route":
                return new route_1.Route(name, undefined, { urn });
            case "azure-native:cdn:Rule":
                return new rule_1.Rule(name, undefined, { urn });
            case "azure-native:cdn:RuleSet":
                return new ruleSet_1.RuleSet(name, undefined, { urn });
            case "azure-native:cdn:Secret":
                return new secret_1.Secret(name, undefined, { urn });
            case "azure-native:cdn:SecurityPolicy":
                return new securityPolicy_1.SecurityPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cdn", _module);
