// @pulumi/azure-native/virtualmachineimages/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211001 = exports.v20200214 = exports.v20190501preview = exports.v20190201preview = exports.v20180201preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getVirtualMachineImageTemplate"), exports);
__exportStar(require("./virtualMachineImageTemplate"), exports);
// Export enums:
__exportStar(require("../types/enums/virtualmachineimages"), exports);
// Export sub-modules:
const v20180201preview = require("./v20180201preview");
exports.v20180201preview = v20180201preview;
const v20190201preview = require("./v20190201preview");
exports.v20190201preview = v20190201preview;
const v20190501preview = require("./v20190501preview");
exports.v20190501preview = v20190501preview;
const v20200214 = require("./v20200214");
exports.v20200214 = v20200214;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
// Import resources to register:
const virtualMachineImageTemplate_1 = require("./virtualMachineImageTemplate");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:virtualmachineimages:VirtualMachineImageTemplate":
                return new virtualMachineImageTemplate_1.VirtualMachineImageTemplate(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "virtualmachineimages", _module);
