// @pulumi/azure-native/servicebus/v20140901/namespaceAuthorizationRule.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NamespaceAuthorizationRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Description of a namespace authorization rule.
 *
 * @deprecated Version v20140901 will be removed in the next major version of the provider. Upgrade to version v20170401 or later.
 */
class NamespaceAuthorizationRule extends pulumi.CustomResource {
    /**
     * Create a NamespaceAuthorizationRule resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20140901 will be removed in the next major version of the provider. Upgrade to version v20170401 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("NamespaceAuthorizationRule is deprecated: Version v20140901 will be removed in the next major version of the provider. Upgrade to version v20170401 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.namespaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'namespaceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.rights === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rights'");
            }
            resourceInputs["authorizationRuleName"] = args ? args.authorizationRuleName : undefined;
            resourceInputs["claimType"] = args ? args.claimType : undefined;
            resourceInputs["claimValue"] = args ? args.claimValue : undefined;
            resourceInputs["keyName"] = args ? args.keyName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["namespaceName"] = args ? args.namespaceName : undefined;
            resourceInputs["primaryKey"] = args ? args.primaryKey : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["rights"] = args ? args.rights : undefined;
            resourceInputs["secondaryKey"] = args ? args.secondaryKey : undefined;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["modifiedTime"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["claimType"] = undefined /*out*/;
            resourceInputs["claimValue"] = undefined /*out*/;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["keyName"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["modifiedTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["primaryKey"] = undefined /*out*/;
            resourceInputs["rights"] = undefined /*out*/;
            resourceInputs["secondaryKey"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:servicebus:NamespaceAuthorizationRule" }, { type: "azure-native:servicebus/v20150801:NamespaceAuthorizationRule" }, { type: "azure-native:servicebus/v20170401:NamespaceAuthorizationRule" }, { type: "azure-native:servicebus/v20180101preview:NamespaceAuthorizationRule" }, { type: "azure-native:servicebus/v20210101preview:NamespaceAuthorizationRule" }, { type: "azure-native:servicebus/v20210601preview:NamespaceAuthorizationRule" }, { type: "azure-native:servicebus/v20211101:NamespaceAuthorizationRule" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NamespaceAuthorizationRule.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NamespaceAuthorizationRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("NamespaceAuthorizationRule is deprecated: Version v20140901 will be removed in the next major version of the provider. Upgrade to version v20170401 or later.");
        return new NamespaceAuthorizationRule(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NamespaceAuthorizationRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NamespaceAuthorizationRule.__pulumiType;
    }
}
exports.NamespaceAuthorizationRule = NamespaceAuthorizationRule;
/** @internal */
NamespaceAuthorizationRule.__pulumiType = 'azure-native:servicebus/v20140901:NamespaceAuthorizationRule';
