// @pulumi/azure-native/synapse/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210601preview = exports.v20210601 = exports.v20210501 = exports.v20210401preview = exports.v20210301 = exports.v20201201 = exports.v20200401preview = exports.v20190601preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./bigDataPool"), exports);
__exportStar(require("./dataConnection"), exports);
__exportStar(require("./database"), exports);
__exportStar(require("./databasePrincipalAssignment"), exports);
__exportStar(require("./eventGridDataConnection"), exports);
__exportStar(require("./eventHubDataConnection"), exports);
__exportStar(require("./getBigDataPool"), exports);
__exportStar(require("./getDataConnection"), exports);
__exportStar(require("./getDatabase"), exports);
__exportStar(require("./getDatabasePrincipalAssignment"), exports);
__exportStar(require("./getEventGridDataConnection"), exports);
__exportStar(require("./getEventHubDataConnection"), exports);
__exportStar(require("./getIntegrationRuntime"), exports);
__exportStar(require("./getIntegrationRuntimeConnectionInfo"), exports);
__exportStar(require("./getIntegrationRuntimeObjectMetadatum"), exports);
__exportStar(require("./getIntegrationRuntimeStatus"), exports);
__exportStar(require("./getIotHubDataConnection"), exports);
__exportStar(require("./getIpFirewallRule"), exports);
__exportStar(require("./getKey"), exports);
__exportStar(require("./getKustoPoolAttachedDatabaseConfiguration"), exports);
__exportStar(require("./getKustoPoolPrincipalAssignment"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getPrivateLinkHub"), exports);
__exportStar(require("./getReadWriteDatabase"), exports);
__exportStar(require("./getSqlPool"), exports);
__exportStar(require("./getSqlPoolSensitivityLabel"), exports);
__exportStar(require("./getSqlPoolTransparentDataEncryption"), exports);
__exportStar(require("./getSqlPoolVulnerabilityAssessment"), exports);
__exportStar(require("./getSqlPoolVulnerabilityAssessmentRuleBaseline"), exports);
__exportStar(require("./getSqlPoolWorkloadClassifier"), exports);
__exportStar(require("./getSqlPoolWorkloadGroup"), exports);
__exportStar(require("./getWorkspace"), exports);
__exportStar(require("./getWorkspaceAadAdmin"), exports);
__exportStar(require("./getWorkspaceManagedSqlServerVulnerabilityAssessment"), exports);
__exportStar(require("./getWorkspaceSqlAadAdmin"), exports);
__exportStar(require("./getkustoPool"), exports);
__exportStar(require("./integrationRuntime"), exports);
__exportStar(require("./iotHubDataConnection"), exports);
__exportStar(require("./ipFirewallRule"), exports);
__exportStar(require("./key"), exports);
__exportStar(require("./kustoPool"), exports);
__exportStar(require("./kustoPoolAttachedDatabaseConfiguration"), exports);
__exportStar(require("./kustoPoolPrincipalAssignment"), exports);
__exportStar(require("./listIntegrationRuntimeAuthKey"), exports);
__exportStar(require("./listKustoPoolFollowerDatabases"), exports);
__exportStar(require("./listKustoPoolLanguageExtensions"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./privateLinkHub"), exports);
__exportStar(require("./readWriteDatabase"), exports);
__exportStar(require("./sqlPool"), exports);
__exportStar(require("./sqlPoolSensitivityLabel"), exports);
__exportStar(require("./sqlPoolTransparentDataEncryption"), exports);
__exportStar(require("./sqlPoolVulnerabilityAssessment"), exports);
__exportStar(require("./sqlPoolVulnerabilityAssessmentRuleBaseline"), exports);
__exportStar(require("./sqlPoolWorkloadClassifier"), exports);
__exportStar(require("./sqlPoolWorkloadGroup"), exports);
__exportStar(require("./workspace"), exports);
__exportStar(require("./workspaceAadAdmin"), exports);
__exportStar(require("./workspaceManagedSqlServerVulnerabilityAssessment"), exports);
__exportStar(require("./workspaceSqlAadAdmin"), exports);
// Export enums:
__exportStar(require("../types/enums/synapse"), exports);
// Export sub-modules:
const v20190601preview = require("./v20190601preview");
exports.v20190601preview = v20190601preview;
const v20200401preview = require("./v20200401preview");
exports.v20200401preview = v20200401preview;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
// Import resources to register:
const bigDataPool_1 = require("./bigDataPool");
const dataConnection_1 = require("./dataConnection");
const database_1 = require("./database");
const databasePrincipalAssignment_1 = require("./databasePrincipalAssignment");
const eventGridDataConnection_1 = require("./eventGridDataConnection");
const eventHubDataConnection_1 = require("./eventHubDataConnection");
const integrationRuntime_1 = require("./integrationRuntime");
const iotHubDataConnection_1 = require("./iotHubDataConnection");
const ipFirewallRule_1 = require("./ipFirewallRule");
const key_1 = require("./key");
const kustoPoolAttachedDatabaseConfiguration_1 = require("./kustoPoolAttachedDatabaseConfiguration");
const kustoPoolPrincipalAssignment_1 = require("./kustoPoolPrincipalAssignment");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const privateLinkHub_1 = require("./privateLinkHub");
const readWriteDatabase_1 = require("./readWriteDatabase");
const sqlPool_1 = require("./sqlPool");
const sqlPoolSensitivityLabel_1 = require("./sqlPoolSensitivityLabel");
const sqlPoolTransparentDataEncryption_1 = require("./sqlPoolTransparentDataEncryption");
const sqlPoolVulnerabilityAssessment_1 = require("./sqlPoolVulnerabilityAssessment");
const sqlPoolVulnerabilityAssessmentRuleBaseline_1 = require("./sqlPoolVulnerabilityAssessmentRuleBaseline");
const sqlPoolWorkloadClassifier_1 = require("./sqlPoolWorkloadClassifier");
const sqlPoolWorkloadGroup_1 = require("./sqlPoolWorkloadGroup");
const workspace_1 = require("./workspace");
const workspaceAadAdmin_1 = require("./workspaceAadAdmin");
const workspaceManagedSqlServerVulnerabilityAssessment_1 = require("./workspaceManagedSqlServerVulnerabilityAssessment");
const workspaceSqlAadAdmin_1 = require("./workspaceSqlAadAdmin");
const kustoPool_1 = require("./kustoPool");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:synapse:BigDataPool":
                return new bigDataPool_1.BigDataPool(name, undefined, { urn });
            case "azure-native:synapse:DataConnection":
                return new dataConnection_1.DataConnection(name, undefined, { urn });
            case "azure-native:synapse:Database":
                return new database_1.Database(name, undefined, { urn });
            case "azure-native:synapse:DatabasePrincipalAssignment":
                return new databasePrincipalAssignment_1.DatabasePrincipalAssignment(name, undefined, { urn });
            case "azure-native:synapse:EventGridDataConnection":
                return new eventGridDataConnection_1.EventGridDataConnection(name, undefined, { urn });
            case "azure-native:synapse:EventHubDataConnection":
                return new eventHubDataConnection_1.EventHubDataConnection(name, undefined, { urn });
            case "azure-native:synapse:IntegrationRuntime":
                return new integrationRuntime_1.IntegrationRuntime(name, undefined, { urn });
            case "azure-native:synapse:IotHubDataConnection":
                return new iotHubDataConnection_1.IotHubDataConnection(name, undefined, { urn });
            case "azure-native:synapse:IpFirewallRule":
                return new ipFirewallRule_1.IpFirewallRule(name, undefined, { urn });
            case "azure-native:synapse:Key":
                return new key_1.Key(name, undefined, { urn });
            case "azure-native:synapse:KustoPoolAttachedDatabaseConfiguration":
                return new kustoPoolAttachedDatabaseConfiguration_1.KustoPoolAttachedDatabaseConfiguration(name, undefined, { urn });
            case "azure-native:synapse:KustoPoolPrincipalAssignment":
                return new kustoPoolPrincipalAssignment_1.KustoPoolPrincipalAssignment(name, undefined, { urn });
            case "azure-native:synapse:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:synapse:PrivateLinkHub":
                return new privateLinkHub_1.PrivateLinkHub(name, undefined, { urn });
            case "azure-native:synapse:ReadWriteDatabase":
                return new readWriteDatabase_1.ReadWriteDatabase(name, undefined, { urn });
            case "azure-native:synapse:SqlPool":
                return new sqlPool_1.SqlPool(name, undefined, { urn });
            case "azure-native:synapse:SqlPoolSensitivityLabel":
                return new sqlPoolSensitivityLabel_1.SqlPoolSensitivityLabel(name, undefined, { urn });
            case "azure-native:synapse:SqlPoolTransparentDataEncryption":
                return new sqlPoolTransparentDataEncryption_1.SqlPoolTransparentDataEncryption(name, undefined, { urn });
            case "azure-native:synapse:SqlPoolVulnerabilityAssessment":
                return new sqlPoolVulnerabilityAssessment_1.SqlPoolVulnerabilityAssessment(name, undefined, { urn });
            case "azure-native:synapse:SqlPoolVulnerabilityAssessmentRuleBaseline":
                return new sqlPoolVulnerabilityAssessmentRuleBaseline_1.SqlPoolVulnerabilityAssessmentRuleBaseline(name, undefined, { urn });
            case "azure-native:synapse:SqlPoolWorkloadClassifier":
                return new sqlPoolWorkloadClassifier_1.SqlPoolWorkloadClassifier(name, undefined, { urn });
            case "azure-native:synapse:SqlPoolWorkloadGroup":
                return new sqlPoolWorkloadGroup_1.SqlPoolWorkloadGroup(name, undefined, { urn });
            case "azure-native:synapse:Workspace":
                return new workspace_1.Workspace(name, undefined, { urn });
            case "azure-native:synapse:WorkspaceAadAdmin":
                return new workspaceAadAdmin_1.WorkspaceAadAdmin(name, undefined, { urn });
            case "azure-native:synapse:WorkspaceManagedSqlServerVulnerabilityAssessment":
                return new workspaceManagedSqlServerVulnerabilityAssessment_1.WorkspaceManagedSqlServerVulnerabilityAssessment(name, undefined, { urn });
            case "azure-native:synapse:WorkspaceSqlAadAdmin":
                return new workspaceSqlAadAdmin_1.WorkspaceSqlAadAdmin(name, undefined, { urn });
            case "azure-native:synapse:kustoPool":
                return new kustoPool_1.KustoPool(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "synapse", _module);
