// @pulumi/azure-native/types/enums/migrate/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneRedundant = exports.TimeRange = exports.TargetAvailabilityZone = exports.ResourceIdentityType = exports.ReservedInstance = exports.ProvisioningState = exports.ProjectStatus = exports.Percentile = exports.Currency = exports.AzureVmFamily = exports.AzureStorageRedundancy = exports.AzurePricingTier = exports.AzureOfferCode = exports.AzureLocation = exports.AzureHybridUseBenefit = exports.AzureDiskType = exports.AssessmentStage = exports.AssessmentSizingCriterion = exports.v20210801 = exports.v20210101 = exports.v20200501 = exports.v20191001preview = exports.v20191001 = exports.v20180901preview = exports.v20180202 = exports.v20171111preview = void 0;
// Export sub-modules:
const v20171111preview = require("./v20171111preview");
exports.v20171111preview = v20171111preview;
const v20180202 = require("./v20180202");
exports.v20180202 = v20180202;
const v20180901preview = require("./v20180901preview");
exports.v20180901preview = v20180901preview;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191001preview = require("./v20191001preview");
exports.v20191001preview = v20191001preview;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
exports.AssessmentSizingCriterion = {
    PerformanceBased: "PerformanceBased",
    AsOnPremises: "AsOnPremises",
};
exports.AssessmentStage = {
    InProgress: "InProgress",
    UnderReview: "UnderReview",
    Approved: "Approved",
};
exports.AzureDiskType = {
    Unknown: "Unknown",
    Standard: "Standard",
    Premium: "Premium",
    StandardSSD: "StandardSSD",
    StandardOrPremium: "StandardOrPremium",
};
exports.AzureHybridUseBenefit = {
    Unknown: "Unknown",
    Yes: "Yes",
    No: "No",
};
exports.AzureLocation = {
    Unknown: "Unknown",
    EastAsia: "EastAsia",
    SoutheastAsia: "SoutheastAsia",
    AustraliaEast: "AustraliaEast",
    AustraliaSoutheast: "AustraliaSoutheast",
    BrazilSouth: "BrazilSouth",
    CanadaCentral: "CanadaCentral",
    CanadaEast: "CanadaEast",
    WestEurope: "WestEurope",
    NorthEurope: "NorthEurope",
    CentralIndia: "CentralIndia",
    SouthIndia: "SouthIndia",
    WestIndia: "WestIndia",
    JapanEast: "JapanEast",
    JapanWest: "JapanWest",
    KoreaCentral: "KoreaCentral",
    KoreaSouth: "KoreaSouth",
    UkWest: "UkWest",
    UkSouth: "UkSouth",
    NorthCentralUs: "NorthCentralUs",
    EastUs: "EastUs",
    WestUs2: "WestUs2",
    SouthCentralUs: "SouthCentralUs",
    CentralUs: "CentralUs",
    EastUs2: "EastUs2",
    WestUs: "WestUs",
    WestCentralUs: "WestCentralUs",
    GermanyCentral: "GermanyCentral",
    GermanyNortheast: "GermanyNortheast",
    ChinaNorth: "ChinaNorth",
    ChinaEast: "ChinaEast",
    USGovArizona: "USGovArizona",
    USGovTexas: "USGovTexas",
    USGovIowa: "USGovIowa",
    USGovVirginia: "USGovVirginia",
    USDoDCentral: "USDoDCentral",
    USDoDEast: "USDoDEast",
};
exports.AzureOfferCode = {
    Unknown: "Unknown",
    MSAZR0003P: "MSAZR0003P",
    MSAZR0044P: "MSAZR0044P",
    MSAZR0059P: "MSAZR0059P",
    MSAZR0060P: "MSAZR0060P",
    MSAZR0062P: "MSAZR0062P",
    MSAZR0063P: "MSAZR0063P",
    MSAZR0064P: "MSAZR0064P",
    MSAZR0029P: "MSAZR0029P",
    MSAZR0022P: "MSAZR0022P",
    MSAZR0023P: "MSAZR0023P",
    MSAZR0148P: "MSAZR0148P",
    MSAZR0025P: "MSAZR0025P",
    MSAZR0036P: "MSAZR0036P",
    MSAZR0120P: "MSAZR0120P",
    MSAZR0121P: "MSAZR0121P",
    MSAZR0122P: "MSAZR0122P",
    MSAZR0123P: "MSAZR0123P",
    MSAZR0124P: "MSAZR0124P",
    MSAZR0125P: "MSAZR0125P",
    MSAZR0126P: "MSAZR0126P",
    MSAZR0127P: "MSAZR0127P",
    MSAZR0128P: "MSAZR0128P",
    MSAZR0129P: "MSAZR0129P",
    MSAZR0130P: "MSAZR0130P",
    MSAZR0111P: "MSAZR0111P",
    MSAZR0144P: "MSAZR0144P",
    MSAZR0149P: "MSAZR0149P",
    MSMCAZR0044P: "MSMCAZR0044P",
    MSMCAZR0059P: "MSMCAZR0059P",
    MSMCAZR0060P: "MSMCAZR0060P",
    MSMCAZR0063P: "MSMCAZR0063P",
    MSMCAZR0120P: "MSMCAZR0120P",
    MSMCAZR0121P: "MSMCAZR0121P",
    MSMCAZR0125P: "MSMCAZR0125P",
    MSMCAZR0128P: "MSMCAZR0128P",
    MSAZRDE0003P: "MSAZRDE0003P",
    MSAZRDE0044P: "MSAZRDE0044P",
    MSAZRUSGOV0003P: "MSAZRUSGOV0003P",
    EA: "EA",
};
exports.AzurePricingTier = {
    Standard: "Standard",
    Basic: "Basic",
};
exports.AzureStorageRedundancy = {
    Unknown: "Unknown",
    LocallyRedundant: "LocallyRedundant",
    ZoneRedundant: "ZoneRedundant",
    GeoRedundant: "GeoRedundant",
    ReadAccessGeoRedundant: "ReadAccessGeoRedundant",
};
exports.AzureVmFamily = {
    Unknown: "Unknown",
    Basic_A0_A4: "Basic_A0_A4",
    Standard_A0_A7: "Standard_A0_A7",
    Standard_A8_A11: "Standard_A8_A11",
    Av2_series: "Av2_series",
    D_series: "D_series",
    Dv2_series: "Dv2_series",
    DS_series: "DS_series",
    DSv2_series: "DSv2_series",
    F_series: "F_series",
    Fs_series: "Fs_series",
    G_series: "G_series",
    GS_series: "GS_series",
    H_series: "H_series",
    Ls_series: "Ls_series",
    Dsv3_series: "Dsv3_series",
    Dv3_series: "Dv3_series",
    Fsv2_series: "Fsv2_series",
    Ev3_series: "Ev3_series",
    Esv3_series: "Esv3_series",
    M_series: "M_series",
    DC_Series: "DC_Series",
};
exports.Currency = {
    Unknown: "Unknown",
    USD: "USD",
    DKK: "DKK",
    CAD: "CAD",
    IDR: "IDR",
    JPY: "JPY",
    KRW: "KRW",
    NZD: "NZD",
    NOK: "NOK",
    RUB: "RUB",
    SAR: "SAR",
    ZAR: "ZAR",
    SEK: "SEK",
    TRY: "TRY",
    GBP: "GBP",
    MXN: "MXN",
    MYR: "MYR",
    INR: "INR",
    HKD: "HKD",
    BRL: "BRL",
    TWD: "TWD",
    EUR: "EUR",
    CHF: "CHF",
    ARS: "ARS",
    AUD: "AUD",
    CNY: "CNY",
};
exports.Percentile = {
    Percentile50: "Percentile50",
    Percentile90: "Percentile90",
    Percentile95: "Percentile95",
    Percentile99: "Percentile99",
};
exports.ProjectStatus = {
    Active: "Active",
    Inactive: "Inactive",
};
exports.ProvisioningState = {
    Accepted: "Accepted",
    Creating: "Creating",
    Deleting: "Deleting",
    Failed: "Failed",
    Moving: "Moving",
    Succeeded: "Succeeded",
};
exports.ReservedInstance = {
    None: "None",
    RI1Year: "RI1Year",
    RI3Year: "RI3Year",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
};
exports.TargetAvailabilityZone = {
    One: "1",
    Two: "2",
    Three: "3",
    NA: "NA",
};
exports.TimeRange = {
    Day: "Day",
    Week: "Week",
    Month: "Month",
    Custom: "Custom",
};
exports.ZoneRedundant = {
    Enable: "Enable",
    Disable: "Disable",
};
