// @pulumi/azure-native/web/v20160801/webAppVnetConnectionSlot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppVnetConnectionSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Virtual Network information contract.
 */
class WebAppVnetConnectionSlot extends pulumi.CustomResource {
    /**
     * Create a WebAppVnetConnectionSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["certBlob"] = args ? args.certBlob : undefined;
            resourceInputs["dnsServers"] = args ? args.dnsServers : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["slot"] = args ? args.slot : undefined;
            resourceInputs["vnetName"] = args ? args.vnetName : undefined;
            resourceInputs["vnetResourceId"] = args ? args.vnetResourceId : undefined;
            resourceInputs["certThumbprint"] = undefined /*out*/;
            resourceInputs["resyncRequired"] = undefined /*out*/;
            resourceInputs["routes"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["certBlob"] = undefined /*out*/;
            resourceInputs["certThumbprint"] = undefined /*out*/;
            resourceInputs["dnsServers"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["resyncRequired"] = undefined /*out*/;
            resourceInputs["routes"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vnetResourceId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20150801:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20180201:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20181101:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20190801:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20200601:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20200901:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20201001:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20201201:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20210101:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20210115:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20210201:WebAppVnetConnectionSlot" }, { type: "azure-native:web/v20210301:WebAppVnetConnectionSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppVnetConnectionSlot.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing WebAppVnetConnectionSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppVnetConnectionSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppVnetConnectionSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppVnetConnectionSlot.__pulumiType;
    }
}
exports.WebAppVnetConnectionSlot = WebAppVnetConnectionSlot;
/** @internal */
WebAppVnetConnectionSlot.__pulumiType = 'azure-native:web/v20160801:WebAppVnetConnectionSlot';
