// @pulumi/azure-native/edgeorder/v20201201preview/addressByName.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressByName = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Address Resource.
 *
 * @deprecated Version v20201201preview will be removed in the next major version of the provider. Upgrade to version v20211201 or later.
 */
class AddressByName extends pulumi.CustomResource {
    /**
     * Create a AddressByName resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20201201preview will be removed in the next major version of the provider. Upgrade to version v20211201 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("AddressByName is deprecated: Version v20201201preview will be removed in the next major version of the provider. Upgrade to version v20211201 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.contactDetails === undefined) && !opts.urn) {
                throw new Error("Missing required property 'contactDetails'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["addressName"] = args ? args.addressName : undefined;
            resourceInputs["contactDetails"] = args ? args.contactDetails : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["shippingAddress"] = args ? args.shippingAddress : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["contactDetails"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["shippingAddress"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:edgeorder:AddressByName" }, { type: "azure-native:edgeorder/v20211201:AddressByName" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AddressByName.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing AddressByName resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("AddressByName is deprecated: Version v20201201preview will be removed in the next major version of the provider. Upgrade to version v20211201 or later.");
        return new AddressByName(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AddressByName.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AddressByName.__pulumiType;
    }
}
exports.AddressByName = AddressByName;
/** @internal */
AddressByName.__pulumiType = 'azure-native:edgeorder/v20201201preview:AddressByName';
