// @pulumi/azure-native/types/enums/network/v20210201preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserRuleKind = exports.UseHubGateway = exports.SecurityType = exports.SecurityConfigurationRuleProtocol = exports.SecurityConfigurationRuleDirection = exports.SecurityConfigurationRuleAccess = exports.IsGlobal = exports.GroupConnectivity = exports.DeleteExistingPeering = exports.DeleteExistingNSGs = exports.ConnectivityTopology = exports.ConfigurationType = exports.AssociationAccessMode = exports.AdminRuleKind = exports.AddressPrefixType = exports.AccessRuleDirection = void 0;
exports.AccessRuleDirection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.AddressPrefixType = {
    IPPrefix: "IPPrefix",
    ServiceTag: "ServiceTag",
};
exports.AdminRuleKind = {
    Custom: "Custom",
    Default: "Default",
};
exports.AssociationAccessMode = {
    Learning: "Learning",
    Enforced: "Enforced",
    Audit: "Audit",
};
exports.ConfigurationType = {
    SecurityAdmin: "SecurityAdmin",
    SecurityUser: "SecurityUser",
    Connectivity: "Connectivity",
};
exports.ConnectivityTopology = {
    HubAndSpoke: "HubAndSpoke",
    Mesh: "Mesh",
};
exports.DeleteExistingNSGs = {
    False: "False",
    True: "True",
};
exports.DeleteExistingPeering = {
    False: "False",
    True: "True",
};
exports.GroupConnectivity = {
    None: "None",
    DirectlyConnected: "DirectlyConnected",
};
exports.IsGlobal = {
    False: "False",
    True: "True",
};
exports.SecurityConfigurationRuleAccess = {
    Allow: "Allow",
    Deny: "Deny",
    AlwaysAllow: "AlwaysAllow",
};
exports.SecurityConfigurationRuleDirection = {
    Inbound: "Inbound",
    Outbound: "Outbound",
};
exports.SecurityConfigurationRuleProtocol = {
    Tcp: "Tcp",
    Udp: "Udp",
    Icmp: "Icmp",
    Esp: "Esp",
    Any: "Any",
    Ah: "Ah",
};
exports.SecurityType = {
    AdminPolicy: "AdminPolicy",
    UserPolicy: "UserPolicy",
};
exports.UseHubGateway = {
    False: "False",
    True: "True",
};
exports.UserRuleKind = {
    Custom: "Custom",
    Default: "Default",
};
