// @pulumi/azure-native/healthcareapis/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211101 = exports.v20210601preview = exports.v20210111 = exports.v20200330 = exports.v20200315 = exports.v20190916 = exports.v20180820preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./dicomService"), exports);
__exportStar(require("./fhirService"), exports);
__exportStar(require("./getDicomService"), exports);
__exportStar(require("./getFhirService"), exports);
__exportStar(require("./getIotConnector"), exports);
__exportStar(require("./getIotConnectorFhirDestination"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getService"), exports);
__exportStar(require("./getWorkspace"), exports);
__exportStar(require("./getWorkspacePrivateEndpointConnection"), exports);
__exportStar(require("./iotConnector"), exports);
__exportStar(require("./iotConnectorFhirDestination"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./service"), exports);
__exportStar(require("./workspace"), exports);
__exportStar(require("./workspacePrivateEndpointConnection"), exports);
// Export enums:
__exportStar(require("../types/enums/healthcareapis"), exports);
// Export sub-modules:
const v20180820preview = require("./v20180820preview");
exports.v20180820preview = v20180820preview;
const v20190916 = require("./v20190916");
exports.v20190916 = v20190916;
const v20200315 = require("./v20200315");
exports.v20200315 = v20200315;
const v20200330 = require("./v20200330");
exports.v20200330 = v20200330;
const v20210111 = require("./v20210111");
exports.v20210111 = v20210111;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20211101 = require("./v20211101");
exports.v20211101 = v20211101;
// Import resources to register:
const dicomService_1 = require("./dicomService");
const fhirService_1 = require("./fhirService");
const iotConnector_1 = require("./iotConnector");
const iotConnectorFhirDestination_1 = require("./iotConnectorFhirDestination");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const service_1 = require("./service");
const workspace_1 = require("./workspace");
const workspacePrivateEndpointConnection_1 = require("./workspacePrivateEndpointConnection");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:healthcareapis:DicomService":
                return new dicomService_1.DicomService(name, undefined, { urn });
            case "azure-native:healthcareapis:FhirService":
                return new fhirService_1.FhirService(name, undefined, { urn });
            case "azure-native:healthcareapis:IotConnector":
                return new iotConnector_1.IotConnector(name, undefined, { urn });
            case "azure-native:healthcareapis:IotConnectorFhirDestination":
                return new iotConnectorFhirDestination_1.IotConnectorFhirDestination(name, undefined, { urn });
            case "azure-native:healthcareapis:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:healthcareapis:Service":
                return new service_1.Service(name, undefined, { urn });
            case "azure-native:healthcareapis:Workspace":
                return new workspace_1.Workspace(name, undefined, { urn });
            case "azure-native:healthcareapis:WorkspacePrivateEndpointConnection":
                return new workspacePrivateEndpointConnection_1.WorkspacePrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "healthcareapis", _module);
