// @pulumi/azure-native/sqlvirtualmachine/v20211101preview/sqlVirtualMachine.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlVirtualMachine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A SQL virtual machine.
 */
class SqlVirtualMachine extends pulumi.CustomResource {
    /**
     * Create a SqlVirtualMachine resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["assessmentSettings"] = args ? args.assessmentSettings : undefined;
            resourceInputs["autoBackupSettings"] = args ? args.autoBackupSettings : undefined;
            resourceInputs["autoPatchingSettings"] = args ? args.autoPatchingSettings : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["keyVaultCredentialSettings"] = args ? args.keyVaultCredentialSettings : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["serverConfigurationsManagementSettings"] = args ? args.serverConfigurationsManagementSettings : undefined;
            resourceInputs["sqlImageOffer"] = args ? args.sqlImageOffer : undefined;
            resourceInputs["sqlImageSku"] = args ? args.sqlImageSku : undefined;
            resourceInputs["sqlManagement"] = args ? args.sqlManagement : undefined;
            resourceInputs["sqlServerLicenseType"] = args ? args.sqlServerLicenseType : undefined;
            resourceInputs["sqlVirtualMachineGroupResourceId"] = args ? args.sqlVirtualMachineGroupResourceId : undefined;
            resourceInputs["sqlVirtualMachineName"] = args ? args.sqlVirtualMachineName : undefined;
            resourceInputs["storageConfigurationSettings"] = args ? args.storageConfigurationSettings : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["virtualMachineResourceId"] = args ? args.virtualMachineResourceId : undefined;
            resourceInputs["wsfcDomainCredentials"] = args ? args.wsfcDomainCredentials : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["assessmentSettings"] = undefined /*out*/;
            resourceInputs["autoBackupSettings"] = undefined /*out*/;
            resourceInputs["autoPatchingSettings"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["keyVaultCredentialSettings"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["serverConfigurationsManagementSettings"] = undefined /*out*/;
            resourceInputs["sqlImageOffer"] = undefined /*out*/;
            resourceInputs["sqlImageSku"] = undefined /*out*/;
            resourceInputs["sqlManagement"] = undefined /*out*/;
            resourceInputs["sqlServerLicenseType"] = undefined /*out*/;
            resourceInputs["sqlVirtualMachineGroupResourceId"] = undefined /*out*/;
            resourceInputs["storageConfigurationSettings"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualMachineResourceId"] = undefined /*out*/;
            resourceInputs["wsfcDomainCredentials"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:sqlvirtualmachine:SqlVirtualMachine" }, { type: "azure-native:sqlvirtualmachine/v20170301preview:SqlVirtualMachine" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SqlVirtualMachine.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SqlVirtualMachine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new SqlVirtualMachine(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlVirtualMachine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlVirtualMachine.__pulumiType;
    }
}
exports.SqlVirtualMachine = SqlVirtualMachine;
/** @internal */
SqlVirtualMachine.__pulumiType = 'azure-native:sqlvirtualmachine/v20211101preview:SqlVirtualMachine';
