// @pulumi/azure-native/desktopvirtualization/v20201110preview/application.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Schema for Application properties.
 *
 * @deprecated Version v20201110preview will be removed in the next major version of the provider. Upgrade to version v20210201preview or later.
 */
class Application extends pulumi.CustomResource {
    /**
     * Create a Application resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20201110preview will be removed in the next major version of the provider. Upgrade to version v20210201preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Application is deprecated: Version v20201110preview will be removed in the next major version of the provider. Upgrade to version v20210201preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.applicationGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'applicationGroupName'");
            }
            if ((!args || args.commandLineSetting === undefined) && !opts.urn) {
                throw new Error("Missing required property 'commandLineSetting'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["applicationGroupName"] = args ? args.applicationGroupName : undefined;
            resourceInputs["applicationName"] = args ? args.applicationName : undefined;
            resourceInputs["applicationType"] = args ? args.applicationType : undefined;
            resourceInputs["commandLineArguments"] = args ? args.commandLineArguments : undefined;
            resourceInputs["commandLineSetting"] = args ? args.commandLineSetting : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["filePath"] = args ? args.filePath : undefined;
            resourceInputs["friendlyName"] = args ? args.friendlyName : undefined;
            resourceInputs["iconIndex"] = args ? args.iconIndex : undefined;
            resourceInputs["iconPath"] = args ? args.iconPath : undefined;
            resourceInputs["msixPackageApplicationId"] = args ? args.msixPackageApplicationId : undefined;
            resourceInputs["msixPackageFamilyName"] = args ? args.msixPackageFamilyName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["showInPortal"] = args ? args.showInPortal : undefined;
            resourceInputs["iconContent"] = undefined /*out*/;
            resourceInputs["iconHash"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["applicationType"] = undefined /*out*/;
            resourceInputs["commandLineArguments"] = undefined /*out*/;
            resourceInputs["commandLineSetting"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["filePath"] = undefined /*out*/;
            resourceInputs["friendlyName"] = undefined /*out*/;
            resourceInputs["iconContent"] = undefined /*out*/;
            resourceInputs["iconHash"] = undefined /*out*/;
            resourceInputs["iconIndex"] = undefined /*out*/;
            resourceInputs["iconPath"] = undefined /*out*/;
            resourceInputs["msixPackageApplicationId"] = undefined /*out*/;
            resourceInputs["msixPackageFamilyName"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["showInPortal"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:desktopvirtualization:Application" }, { type: "azure-native:desktopvirtualization/v20190123preview:Application" }, { type: "azure-native:desktopvirtualization/v20190924preview:Application" }, { type: "azure-native:desktopvirtualization/v20191210preview:Application" }, { type: "azure-native:desktopvirtualization/v20200921preview:Application" }, { type: "azure-native:desktopvirtualization/v20201019preview:Application" }, { type: "azure-native:desktopvirtualization/v20201102preview:Application" }, { type: "azure-native:desktopvirtualization/v20210114preview:Application" }, { type: "azure-native:desktopvirtualization/v20210201preview:Application" }, { type: "azure-native:desktopvirtualization/v20210309preview:Application" }, { type: "azure-native:desktopvirtualization/v20210401preview:Application" }, { type: "azure-native:desktopvirtualization/v20210712:Application" }, { type: "azure-native:desktopvirtualization/v20210903preview:Application" }, { type: "azure-native:desktopvirtualization/v20220210preview:Application" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Application.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Application is deprecated: Version v20201110preview will be removed in the next major version of the provider. Upgrade to version v20210201preview or later.");
        return new Application(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
}
exports.Application = Application;
/** @internal */
Application.__pulumiType = 'azure-native:desktopvirtualization/v20201110preview:Application';
