// @pulumi/azure-native/education/getLab.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetLab = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Lab details.
 * API Version: 2021-12-01-preview.
 */
class GetLab extends pulumi.CustomResource {
    /**
     * Create a GetLab resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.billingAccountName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'billingAccountName'");
            }
            if ((!args || args.billingProfileName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'billingProfileName'");
            }
            if ((!args || args.budgetPerStudent === undefined) && !opts.urn) {
                throw new Error("Missing required property 'budgetPerStudent'");
            }
            if ((!args || args.description === undefined) && !opts.urn) {
                throw new Error("Missing required property 'description'");
            }
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.expirationDate === undefined) && !opts.urn) {
                throw new Error("Missing required property 'expirationDate'");
            }
            if ((!args || args.invoiceSectionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'invoiceSectionName'");
            }
            resourceInputs["billingAccountName"] = args ? args.billingAccountName : undefined;
            resourceInputs["billingProfileName"] = args ? args.billingProfileName : undefined;
            resourceInputs["budgetPerStudent"] = args ? args.budgetPerStudent : undefined;
            resourceInputs["currency"] = args ? args.currency : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["expirationDate"] = args ? args.expirationDate : undefined;
            resourceInputs["invoiceSectionName"] = args ? args.invoiceSectionName : undefined;
            resourceInputs["value"] = args ? args.value : undefined;
            resourceInputs["effectiveDate"] = undefined /*out*/;
            resourceInputs["invitationCode"] = undefined /*out*/;
            resourceInputs["maxStudentCount"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["budgetPerStudent"] = undefined /*out*/;
            resourceInputs["currency"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["effectiveDate"] = undefined /*out*/;
            resourceInputs["expirationDate"] = undefined /*out*/;
            resourceInputs["invitationCode"] = undefined /*out*/;
            resourceInputs["maxStudentCount"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["value"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:education/v20211201preview:GetLab" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GetLab.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing GetLab resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GetLab(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GetLab.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GetLab.__pulumiType;
    }
}
exports.GetLab = GetLab;
/** @internal */
GetLab.__pulumiType = 'azure-native:education:GetLab';
