// @pulumi/azure-native/types/enums/cache/v20200601/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TlsVersion = exports.SkuName = exports.SkuFamily = exports.ReplicationRole = exports.PublicNetworkAccess = exports.PrivateEndpointServiceConnectionStatus = exports.DayOfWeek = void 0;
exports.DayOfWeek = {
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
    Sunday: "Sunday",
    Everyday: "Everyday",
    Weekend: "Weekend",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ReplicationRole = {
    Primary: "Primary",
    Secondary: "Secondary",
};
exports.SkuFamily = {
    C: "C",
    P: "P",
};
exports.SkuName = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.TlsVersion = {
    TlsVersion_1_0: "1.0",
    TlsVersion_1_1: "1.1",
    TlsVersion_1_2: "1.2",
};
