// @pulumi/azure-native/maintenance/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210901preview = exports.v20210501 = exports.v20210401preview = exports.v20200701preview = exports.v20200401 = exports.v20180601preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./configurationAssignment"), exports);
__exportStar(require("./configurationAssignmentParent"), exports);
__exportStar(require("./getConfigurationAssignment"), exports);
__exportStar(require("./getConfigurationAssignmentParent"), exports);
__exportStar(require("./getMaintenanceConfiguration"), exports);
__exportStar(require("./maintenanceConfiguration"), exports);
// Export enums:
__exportStar(require("../types/enums/maintenance"), exports);
// Export sub-modules:
const v20180601preview = require("./v20180601preview");
exports.v20180601preview = v20180601preview;
const v20200401 = require("./v20200401");
exports.v20200401 = v20200401;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
// Import resources to register:
const configurationAssignment_1 = require("./configurationAssignment");
const configurationAssignmentParent_1 = require("./configurationAssignmentParent");
const maintenanceConfiguration_1 = require("./maintenanceConfiguration");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:maintenance:ConfigurationAssignment":
                return new configurationAssignment_1.ConfigurationAssignment(name, undefined, { urn });
            case "azure-native:maintenance:ConfigurationAssignmentParent":
                return new configurationAssignmentParent_1.ConfigurationAssignmentParent(name, undefined, { urn });
            case "azure-native:maintenance:MaintenanceConfiguration":
                return new maintenanceConfiguration_1.MaintenanceConfiguration(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "maintenance", _module);
