// @pulumi/azure-native/network/v20180401/vpnSite.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpnSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * VpnSite Resource.
 *
 * @deprecated Version v20180401 will be removed in the next major version of the provider. Upgrade to version v20180501 or later.
 */
class VpnSite extends pulumi.CustomResource {
    /**
     * Create a VpnSite resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180401 will be removed in the next major version of the provider. Upgrade to version v20180501 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("VpnSite is deprecated: Version v20180401 will be removed in the next major version of the provider. Upgrade to version v20180501 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["addressSpace"] = args ? args.addressSpace : undefined;
            resourceInputs["bgpProperties"] = args ? args.bgpProperties : undefined;
            resourceInputs["deviceProperties"] = args ? args.deviceProperties : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["ipAddress"] = args ? args.ipAddress : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["siteKey"] = args ? args.siteKey : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["virtualWAN"] = args ? args.virtualWAN : undefined;
            resourceInputs["vpnSiteName"] = args ? args.vpnSiteName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["addressSpace"] = undefined /*out*/;
            resourceInputs["bgpProperties"] = undefined /*out*/;
            resourceInputs["deviceProperties"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["ipAddress"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["siteKey"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["virtualWAN"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network:VpnSite" }, { type: "azure-native:network/v20180601:VpnSite" }, { type: "azure-native:network/v20180701:VpnSite" }, { type: "azure-native:network/v20180801:VpnSite" }, { type: "azure-native:network/v20181001:VpnSite" }, { type: "azure-native:network/v20181101:VpnSite" }, { type: "azure-native:network/v20181201:VpnSite" }, { type: "azure-native:network/v20190201:VpnSite" }, { type: "azure-native:network/v20190401:VpnSite" }, { type: "azure-native:network/v20190601:VpnSite" }, { type: "azure-native:network/v20190701:VpnSite" }, { type: "azure-native:network/v20190801:VpnSite" }, { type: "azure-native:network/v20190901:VpnSite" }, { type: "azure-native:network/v20191101:VpnSite" }, { type: "azure-native:network/v20191201:VpnSite" }, { type: "azure-native:network/v20200301:VpnSite" }, { type: "azure-native:network/v20200401:VpnSite" }, { type: "azure-native:network/v20200501:VpnSite" }, { type: "azure-native:network/v20200601:VpnSite" }, { type: "azure-native:network/v20200701:VpnSite" }, { type: "azure-native:network/v20200801:VpnSite" }, { type: "azure-native:network/v20201101:VpnSite" }, { type: "azure-native:network/v20210201:VpnSite" }, { type: "azure-native:network/v20210301:VpnSite" }, { type: "azure-native:network/v20210501:VpnSite" }, { type: "azure-native:network/v20210801:VpnSite" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VpnSite.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing VpnSite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("VpnSite is deprecated: Version v20180401 will be removed in the next major version of the provider. Upgrade to version v20180501 or later.");
        return new VpnSite(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VpnSite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VpnSite.__pulumiType;
    }
}
exports.VpnSite = VpnSite;
/** @internal */
VpnSite.__pulumiType = 'azure-native:network/v20180401:VpnSite';
