// @pulumi/azure-native/portal/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20200901preview = exports.v20190101preview = exports.v20181001preview = exports.v20181001 = exports.v20150801preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./console"), exports);
__exportStar(require("./consoleWithLocation"), exports);
__exportStar(require("./dashboard"), exports);
__exportStar(require("./getConsole"), exports);
__exportStar(require("./getConsoleWithLocation"), exports);
__exportStar(require("./getDashboard"), exports);
__exportStar(require("./getTenantConfiguration"), exports);
__exportStar(require("./getUserSettings"), exports);
__exportStar(require("./getUserSettingsWithLocation"), exports);
__exportStar(require("./listListTenantConfigurationViolation"), exports);
__exportStar(require("./tenantConfiguration"), exports);
__exportStar(require("./userSettings"), exports);
__exportStar(require("./userSettingsWithLocation"), exports);
// Export enums:
__exportStar(require("../types/enums/portal"), exports);
// Export sub-modules:
const v20150801preview = require("./v20150801preview");
exports.v20150801preview = v20150801preview;
const v20181001 = require("./v20181001");
exports.v20181001 = v20181001;
const v20181001preview = require("./v20181001preview");
exports.v20181001preview = v20181001preview;
const v20190101preview = require("./v20190101preview");
exports.v20190101preview = v20190101preview;
const v20200901preview = require("./v20200901preview");
exports.v20200901preview = v20200901preview;
// Import resources to register:
const console_1 = require("./console");
const consoleWithLocation_1 = require("./consoleWithLocation");
const dashboard_1 = require("./dashboard");
const tenantConfiguration_1 = require("./tenantConfiguration");
const userSettings_1 = require("./userSettings");
const userSettingsWithLocation_1 = require("./userSettingsWithLocation");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:portal:Console":
                return new console_1.Console(name, undefined, { urn });
            case "azure-native:portal:ConsoleWithLocation":
                return new consoleWithLocation_1.ConsoleWithLocation(name, undefined, { urn });
            case "azure-native:portal:Dashboard":
                return new dashboard_1.Dashboard(name, undefined, { urn });
            case "azure-native:portal:TenantConfiguration":
                return new tenantConfiguration_1.TenantConfiguration(name, undefined, { urn });
            case "azure-native:portal:UserSettings":
                return new userSettings_1.UserSettings(name, undefined, { urn });
            case "azure-native:portal:UserSettingsWithLocation":
                return new userSettingsWithLocation_1.UserSettingsWithLocation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "portal", _module);
