// @pulumi/azure-native/certificateregistration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210301 = exports.v20210201 = exports.v20210115 = exports.v20210101 = exports.v20201201 = exports.v20201001 = exports.v20200901 = exports.v20200601 = exports.v20190801 = exports.v20180201 = exports.v20150801 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./appServiceCertificateOrder"), exports);
__exportStar(require("./appServiceCertificateOrderCertificate"), exports);
__exportStar(require("./getAppServiceCertificateOrder"), exports);
__exportStar(require("./getAppServiceCertificateOrderCertificate"), exports);
// Export enums:
__exportStar(require("../types/enums/certificateregistration"), exports);
// Export sub-modules:
const v20150801 = require("./v20150801");
exports.v20150801 = v20150801;
const v20180201 = require("./v20180201");
exports.v20180201 = v20180201;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210115 = require("./v20210115");
exports.v20210115 = v20210115;
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
// Import resources to register:
const appServiceCertificateOrder_1 = require("./appServiceCertificateOrder");
const appServiceCertificateOrderCertificate_1 = require("./appServiceCertificateOrderCertificate");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:certificateregistration:AppServiceCertificateOrder":
                return new appServiceCertificateOrder_1.AppServiceCertificateOrder(name, undefined, { urn });
            case "azure-native:certificateregistration:AppServiceCertificateOrderCertificate":
                return new appServiceCertificateOrderCertificate_1.AppServiceCertificateOrderCertificate(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "certificateregistration", _module);
