// @pulumi/azure-native/devices/v20190701preview/iotHubResourceEventHubConsumerGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotHubResourceEventHubConsumerGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The properties of the EventHubConsumerGroupInfo object.
 *
 * @deprecated Version v20190701preview will be removed in the next major version of the provider. Upgrade to version v20200301 or later.
 */
class IotHubResourceEventHubConsumerGroup extends pulumi.CustomResource {
    /**
     * Create a IotHubResourceEventHubConsumerGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20190701preview will be removed in the next major version of the provider. Upgrade to version v20200301 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("IotHubResourceEventHubConsumerGroup is deprecated: Version v20190701preview will be removed in the next major version of the provider. Upgrade to version v20200301 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.eventHubEndpointName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'eventHubEndpointName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["eventHubEndpointName"] = args ? args.eventHubEndpointName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devices:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20160203:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20170119:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20170701:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20180122:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20180401:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20181201preview:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20190322:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20190322preview:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20191104:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20200301:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20200401:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20200615:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20200710preview:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20200801:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20200831:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20200831preview:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20210201preview:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20210303preview:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20210331:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20210701:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20210701preview:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20210702:IotHubResourceEventHubConsumerGroup" }, { type: "azure-native:devices/v20210702preview:IotHubResourceEventHubConsumerGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IotHubResourceEventHubConsumerGroup.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing IotHubResourceEventHubConsumerGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("IotHubResourceEventHubConsumerGroup is deprecated: Version v20190701preview will be removed in the next major version of the provider. Upgrade to version v20200301 or later.");
        return new IotHubResourceEventHubConsumerGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IotHubResourceEventHubConsumerGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IotHubResourceEventHubConsumerGroup.__pulumiType;
    }
}
exports.IotHubResourceEventHubConsumerGroup = IotHubResourceEventHubConsumerGroup;
/** @internal */
IotHubResourceEventHubConsumerGroup.__pulumiType = 'azure-native:devices/v20190701preview:IotHubResourceEventHubConsumerGroup';
