// @pulumi/azure-native/types/enums/insights/v20210401/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KnownWindowsEventLogDataSourceStreams = exports.KnownSyslogDataSourceStreams = exports.KnownSyslogDataSourceLogLevels = exports.KnownSyslogDataSourceFacilityNames = exports.KnownPublicNetworkAccessOptions = exports.KnownPerfCounterDataSourceStreams = exports.KnownExtensionDataSourceStreams = exports.KnownDataFlowStreams = exports.KnownDataCollectionRuleResourceKind = exports.KnownDataCollectionEndpointResourceKind = void 0;
exports.KnownDataCollectionEndpointResourceKind = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.KnownDataCollectionRuleResourceKind = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.KnownDataFlowStreams = {
    Microsoft_Event: "Microsoft-Event",
    Microsoft_InsightsMetrics: "Microsoft-InsightsMetrics",
    Microsoft_Perf: "Microsoft-Perf",
    Microsoft_Syslog: "Microsoft-Syslog",
    Microsoft_WindowsEvent: "Microsoft-WindowsEvent",
};
exports.KnownExtensionDataSourceStreams = {
    Microsoft_Event: "Microsoft-Event",
    Microsoft_InsightsMetrics: "Microsoft-InsightsMetrics",
    Microsoft_Perf: "Microsoft-Perf",
    Microsoft_Syslog: "Microsoft-Syslog",
    Microsoft_WindowsEvent: "Microsoft-WindowsEvent",
};
exports.KnownPerfCounterDataSourceStreams = {
    Microsoft_Perf: "Microsoft-Perf",
    Microsoft_InsightsMetrics: "Microsoft-InsightsMetrics",
};
exports.KnownPublicNetworkAccessOptions = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.KnownSyslogDataSourceFacilityNames = {
    Auth: "auth",
    Authpriv: "authpriv",
    Cron: "cron",
    Daemon: "daemon",
    Kern: "kern",
    Lpr: "lpr",
    Mail: "mail",
    Mark: "mark",
    News: "news",
    Syslog: "syslog",
    User: "user",
    Uucp: "uucp",
    Local0: "local0",
    Local1: "local1",
    Local2: "local2",
    Local3: "local3",
    Local4: "local4",
    Local5: "local5",
    Local6: "local6",
    Local7: "local7",
    Asterisk: "*",
};
exports.KnownSyslogDataSourceLogLevels = {
    Debug: "Debug",
    Info: "Info",
    Notice: "Notice",
    Warning: "Warning",
    Error: "Error",
    Critical: "Critical",
    Alert: "Alert",
    Emergency: "Emergency",
    Asterisk: "*",
};
exports.KnownSyslogDataSourceStreams = {
    Microsoft_Syslog: "Microsoft-Syslog",
};
exports.KnownWindowsEventLogDataSourceStreams = {
    Microsoft_WindowsEvent: "Microsoft-WindowsEvent",
    Microsoft_Event: "Microsoft-Event",
};
