// @pulumi/azure-native/policyinsights/v20180701preview/remediationAtResourceGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemediationAtResourceGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The remediation definition.
 *
 * @deprecated Version v20180701preview will be removed in the next major version of the provider. Upgrade to version v20190701 or later.
 */
class RemediationAtResourceGroup extends pulumi.CustomResource {
    /**
     * Create a RemediationAtResourceGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180701preview will be removed in the next major version of the provider. Upgrade to version v20190701 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("RemediationAtResourceGroup is deprecated: Version v20180701preview will be removed in the next major version of the provider. Upgrade to version v20190701 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["deploymentStatus"] = args ? args.deploymentStatus : undefined;
            resourceInputs["filters"] = args ? args.filters : undefined;
            resourceInputs["policyAssignmentId"] = args ? args.policyAssignmentId : undefined;
            resourceInputs["policyDefinitionReferenceId"] = args ? args.policyDefinitionReferenceId : undefined;
            resourceInputs["remediationName"] = args ? args.remediationName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["lastUpdatedOn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["deploymentStatus"] = undefined /*out*/;
            resourceInputs["filters"] = undefined /*out*/;
            resourceInputs["lastUpdatedOn"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["policyAssignmentId"] = undefined /*out*/;
            resourceInputs["policyDefinitionReferenceId"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:policyinsights:RemediationAtResourceGroup" }, { type: "azure-native:policyinsights/v20190701:RemediationAtResourceGroup" }, { type: "azure-native:policyinsights/v20211001:RemediationAtResourceGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(RemediationAtResourceGroup.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing RemediationAtResourceGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("RemediationAtResourceGroup is deprecated: Version v20180701preview will be removed in the next major version of the provider. Upgrade to version v20190701 or later.");
        return new RemediationAtResourceGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of RemediationAtResourceGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RemediationAtResourceGroup.__pulumiType;
    }
}
exports.RemediationAtResourceGroup = RemediationAtResourceGroup;
/** @internal */
RemediationAtResourceGroup.__pulumiType = 'azure-native:policyinsights/v20180701preview:RemediationAtResourceGroup';
