// @pulumi/azure-native/guestconfiguration/guestConfigurationHCRPAssignment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GuestConfigurationHCRPAssignment = void 0;
const pulumi = require("@pulumi/pulumi");
const types_1 = require("../types");
const utilities = require("../utilities");
/**
 * Guest configuration assignment is an association between a machine and guest configuration.
 * API Version: 2020-06-25.
 */
class GuestConfigurationHCRPAssignment extends pulumi.CustomResource {
    /**
     * Create a GuestConfigurationHCRPAssignment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.machineName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'machineName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["guestConfigurationAssignmentName"] = args ? args.guestConfigurationAssignmentName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["machineName"] = args ? args.machineName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["properties"] = args ? (args.properties ? pulumi.output(args.properties).apply(types_1.input.guestconfiguration.guestConfigurationAssignmentPropertiesArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:guestconfiguration/v20181120:GuestConfigurationHCRPAssignment" }, { type: "azure-native:guestconfiguration/v20200625:GuestConfigurationHCRPAssignment" }, { type: "azure-native:guestconfiguration/v20210125:GuestConfigurationHCRPAssignment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(GuestConfigurationHCRPAssignment.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing GuestConfigurationHCRPAssignment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new GuestConfigurationHCRPAssignment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GuestConfigurationHCRPAssignment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GuestConfigurationHCRPAssignment.__pulumiType;
    }
}
exports.GuestConfigurationHCRPAssignment = GuestConfigurationHCRPAssignment;
/** @internal */
GuestConfigurationHCRPAssignment.__pulumiType = 'azure-native:guestconfiguration:GuestConfigurationHCRPAssignment';
