// @pulumi/azure-native/guestconfiguration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210125 = exports.v20200625 = exports.v20181120 = exports.v20180630preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getGuestConfigurationAssignment"), exports);
__exportStar(require("./getGuestConfigurationHCRPAssignment"), exports);
__exportStar(require("./guestConfigurationAssignment"), exports);
__exportStar(require("./guestConfigurationHCRPAssignment"), exports);
// Export enums:
__exportStar(require("../types/enums/guestconfiguration"), exports);
// Export sub-modules:
const v20180630preview = require("./v20180630preview");
exports.v20180630preview = v20180630preview;
const v20181120 = require("./v20181120");
exports.v20181120 = v20181120;
const v20200625 = require("./v20200625");
exports.v20200625 = v20200625;
const v20210125 = require("./v20210125");
exports.v20210125 = v20210125;
// Import resources to register:
const guestConfigurationAssignment_1 = require("./guestConfigurationAssignment");
const guestConfigurationHCRPAssignment_1 = require("./guestConfigurationHCRPAssignment");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:guestconfiguration:GuestConfigurationAssignment":
                return new guestConfigurationAssignment_1.GuestConfigurationAssignment(name, undefined, { urn });
            case "azure-native:guestconfiguration:GuestConfigurationHCRPAssignment":
                return new guestConfigurationHCRPAssignment_1.GuestConfigurationHCRPAssignment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "guestconfiguration", _module);
