// @pulumi/azure-native/databoxedge/v20200501preview/order.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Order = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The order details.
 *
 * @deprecated Version v20200501preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later.
 */
class Order extends pulumi.CustomResource {
    /**
     * Create a Order resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200501preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Order is deprecated: Version v20200501preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.contactInformation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'contactInformation'");
            }
            if ((!args || args.deviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.shippingAddress === undefined) && !opts.urn) {
                throw new Error("Missing required property 'shippingAddress'");
            }
            resourceInputs["contactInformation"] = args ? args.contactInformation : undefined;
            resourceInputs["currentStatus"] = args ? args.currentStatus : undefined;
            resourceInputs["deviceName"] = args ? args.deviceName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["shippingAddress"] = args ? args.shippingAddress : undefined;
            resourceInputs["deliveryTrackingInfo"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["orderHistory"] = undefined /*out*/;
            resourceInputs["returnTrackingInfo"] = undefined /*out*/;
            resourceInputs["serialNumber"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["contactInformation"] = undefined /*out*/;
            resourceInputs["currentStatus"] = undefined /*out*/;
            resourceInputs["deliveryTrackingInfo"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["orderHistory"] = undefined /*out*/;
            resourceInputs["returnTrackingInfo"] = undefined /*out*/;
            resourceInputs["serialNumber"] = undefined /*out*/;
            resourceInputs["shippingAddress"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databoxedge:Order" }, { type: "azure-native:databoxedge/v20190301:Order" }, { type: "azure-native:databoxedge/v20190701:Order" }, { type: "azure-native:databoxedge/v20190801:Order" }, { type: "azure-native:databoxedge/v20200901:Order" }, { type: "azure-native:databoxedge/v20200901preview:Order" }, { type: "azure-native:databoxedge/v20201201:Order" }, { type: "azure-native:databoxedge/v20210201:Order" }, { type: "azure-native:databoxedge/v20210201preview:Order" }, { type: "azure-native:databoxedge/v20210601:Order" }, { type: "azure-native:databoxedge/v20210601preview:Order" }, { type: "azure-native:databoxedge/v20220301:Order" }, { type: "azure-native:databoxedge/v20220401preview:Order" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Order.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Order resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Order is deprecated: Version v20200501preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later.");
        return new Order(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Order.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Order.__pulumiType;
    }
}
exports.Order = Order;
/** @internal */
Order.__pulumiType = 'azure-native:databoxedge/v20200501preview:Order';
