// @pulumi/azure-native/databoxedge/v20200501preview/role.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Role = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Compute role.
 *
 * @deprecated Version v20200501preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later.
 */
class Role extends pulumi.CustomResource {
    /**
     * Create a Role resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200501preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Role is deprecated: Version v20200501preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.deviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["deviceName"] = args ? args.deviceName : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:databoxedge:Role" }, { type: "azure-native:databoxedge/v20190301:Role" }, { type: "azure-native:databoxedge/v20190701:Role" }, { type: "azure-native:databoxedge/v20190801:Role" }, { type: "azure-native:databoxedge/v20200901:Role" }, { type: "azure-native:databoxedge/v20200901preview:Role" }, { type: "azure-native:databoxedge/v20201201:Role" }, { type: "azure-native:databoxedge/v20210201:Role" }, { type: "azure-native:databoxedge/v20210201preview:Role" }, { type: "azure-native:databoxedge/v20210601:Role" }, { type: "azure-native:databoxedge/v20210601preview:Role" }, { type: "azure-native:databoxedge/v20220301:Role" }, { type: "azure-native:databoxedge/v20220401preview:Role" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Role.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Role resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Role is deprecated: Version v20200501preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later.");
        return new Role(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Role.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Role.__pulumiType;
    }
}
exports.Role = Role;
/** @internal */
Role.__pulumiType = 'azure-native:databoxedge/v20200501preview:Role';
