// @pulumi/azure-native/types/enums/appconfiguration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicNetworkAccess = exports.IdentityType = exports.ConnectionStatus = exports.v20211001preview = exports.v20210301preview = exports.v20200701preview = exports.v20200601 = exports.v20191101preview = exports.v20191001 = void 0;
// Export sub-modules:
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191101preview = require("./v20191101preview");
exports.v20191101preview = v20191101preview;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20210301preview = require("./v20210301preview");
exports.v20210301preview = v20210301preview;
const v20211001preview = require("./v20211001preview");
exports.v20211001preview = v20211001preview;
exports.ConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
