// @pulumi/azure-native/addons/v20170515/supportPlanType.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SupportPlanType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The status of the Canonical support plan.
 *
 * @deprecated Version v20170515 will be removed in the next major version of the provider. Upgrade to version v20180301 or later.
 */
class SupportPlanType extends pulumi.CustomResource {
    /**
     * Create a SupportPlanType resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20170515 will be removed in the next major version of the provider. Upgrade to version v20180301 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SupportPlanType is deprecated: Version v20170515 will be removed in the next major version of the provider. Upgrade to version v20180301 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.providerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'providerName'");
            }
            resourceInputs["planTypeName"] = args ? args.planTypeName : undefined;
            resourceInputs["providerName"] = args ? args.providerName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:addons:SupportPlanType" }, { type: "azure-native:addons/v20180301:SupportPlanType" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SupportPlanType.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SupportPlanType resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SupportPlanType is deprecated: Version v20170515 will be removed in the next major version of the provider. Upgrade to version v20180301 or later.");
        return new SupportPlanType(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SupportPlanType.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SupportPlanType.__pulumiType;
    }
}
exports.SupportPlanType = SupportPlanType;
/** @internal */
SupportPlanType.__pulumiType = 'azure-native:addons/v20170515:SupportPlanType';
