// @pulumi/azure-native/machinelearningservices/v20191101/machineLearningCompute.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MachineLearningCompute = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Machine Learning compute object wrapped into ARM resource envelope.
 *
 * @deprecated Version v20191101 will be removed in the next major version of the provider. Upgrade to version v20200301 or later.
 */
class MachineLearningCompute extends pulumi.CustomResource {
    /**
     * Create a MachineLearningCompute resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20191101 will be removed in the next major version of the provider. Upgrade to version v20200301 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("MachineLearningCompute is deprecated: Version v20191101 will be removed in the next major version of the provider. Upgrade to version v20200301 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["computeName"] = args ? args.computeName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:machinelearningservices:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20180301preview:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20181119:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20190501:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20190601:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20200101:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20200218preview:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20200301:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20200401:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20200501preview:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20200515preview:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20200601:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20200801:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20200901preview:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20210101:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20210301preview:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20210401:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20210701:MachineLearningCompute" }, { type: "azure-native:machinelearningservices/v20220101preview:MachineLearningCompute" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MachineLearningCompute.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MachineLearningCompute resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("MachineLearningCompute is deprecated: Version v20191101 will be removed in the next major version of the provider. Upgrade to version v20200301 or later.");
        return new MachineLearningCompute(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MachineLearningCompute.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MachineLearningCompute.__pulumiType;
    }
}
exports.MachineLearningCompute = MachineLearningCompute;
/** @internal */
MachineLearningCompute.__pulumiType = 'azure-native:machinelearningservices/v20191101:MachineLearningCompute';
