// @pulumi/azure-native/network/v20150501preview/expressRouteCircuitAuthorization.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpressRouteCircuitAuthorization = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Authorization in a ExpressRouteCircuit resource
 *
 * @deprecated Version v20150501preview will be removed in the next major version of the provider. Upgrade to version v20180501 or later.
 */
class ExpressRouteCircuitAuthorization extends pulumi.CustomResource {
    /**
     * Create a ExpressRouteCircuitAuthorization resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150501preview will be removed in the next major version of the provider. Upgrade to version v20180501 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ExpressRouteCircuitAuthorization is deprecated: Version v20150501preview will be removed in the next major version of the provider. Upgrade to version v20180501 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.circuitName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'circuitName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["authorizationKey"] = args ? args.authorizationKey : undefined;
            resourceInputs["authorizationName"] = args ? args.authorizationName : undefined;
            resourceInputs["authorizationUseStatus"] = args ? args.authorizationUseStatus : undefined;
            resourceInputs["circuitName"] = args ? args.circuitName : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["provisioningState"] = args ? args.provisioningState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
        }
        else {
            resourceInputs["authorizationKey"] = undefined /*out*/;
            resourceInputs["authorizationUseStatus"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20150615:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20160330:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20160601:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20160901:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20161201:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20170301:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20170601:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20170801:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20170901:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20171001:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20171101:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20180101:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20180201:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20180401:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20180601:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20180701:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20180801:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20181001:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20181101:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20181201:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20190201:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20190401:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20190601:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20190701:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20190801:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20190901:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20191101:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20191201:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20200301:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20200401:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20200501:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20200601:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20200701:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20200801:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20201101:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20210201:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20210301:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20210501:ExpressRouteCircuitAuthorization" }, { type: "azure-native:network/v20210801:ExpressRouteCircuitAuthorization" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ExpressRouteCircuitAuthorization.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ExpressRouteCircuitAuthorization resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ExpressRouteCircuitAuthorization is deprecated: Version v20150501preview will be removed in the next major version of the provider. Upgrade to version v20180501 or later.");
        return new ExpressRouteCircuitAuthorization(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ExpressRouteCircuitAuthorization.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ExpressRouteCircuitAuthorization.__pulumiType;
    }
}
exports.ExpressRouteCircuitAuthorization = ExpressRouteCircuitAuthorization;
/** @internal */
ExpressRouteCircuitAuthorization.__pulumiType = 'azure-native:network/v20150501preview:ExpressRouteCircuitAuthorization';
