// @pulumi/azure-native/recoveryservices/v20160810/replicationProtectionContainerMapping.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicationProtectionContainerMapping = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Protection container mapping object.
 *
 * @deprecated Version v20160810 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.
 */
class ReplicationProtectionContainerMapping extends pulumi.CustomResource {
    /**
     * Create a ReplicationProtectionContainerMapping resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160810 will be removed in the next major version of the provider. Upgrade to version v20180710 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ReplicationProtectionContainerMapping is deprecated: Version v20160810 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.fabricName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'fabricName'");
            }
            if ((!args || args.protectionContainerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'protectionContainerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["fabricName"] = args ? args.fabricName : undefined;
            resourceInputs["mappingName"] = args ? args.mappingName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["protectionContainerName"] = args ? args.protectionContainerName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20180110:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20180710:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20210210:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20210301:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20210401:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20210601:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20210701:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20210801:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20211001:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20211101:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20211201:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20220101:ReplicationProtectionContainerMapping" }, { type: "azure-native:recoveryservices/v20220201:ReplicationProtectionContainerMapping" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ReplicationProtectionContainerMapping.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ReplicationProtectionContainerMapping resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ReplicationProtectionContainerMapping is deprecated: Version v20160810 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        return new ReplicationProtectionContainerMapping(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ReplicationProtectionContainerMapping.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicationProtectionContainerMapping.__pulumiType;
    }
}
exports.ReplicationProtectionContainerMapping = ReplicationProtectionContainerMapping;
/** @internal */
ReplicationProtectionContainerMapping.__pulumiType = 'azure-native:recoveryservices/v20160810:ReplicationProtectionContainerMapping';
