// @pulumi/azure-native/devtestlab/v20160515/formula.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Formula = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A formula for creating a VM, specifying an image base and other parameters
 *
 * @deprecated Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later.
 */
class Formula extends pulumi.CustomResource {
    /**
     * Create a Formula resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Formula is deprecated: Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.labName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'labName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["author"] = args ? args.author : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["formulaContent"] = args ? args.formulaContent : undefined;
            resourceInputs["labName"] = args ? args.labName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["osType"] = args ? args.osType : undefined;
            resourceInputs["provisioningState"] = args ? args.provisioningState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["uniqueIdentifier"] = args ? args.uniqueIdentifier : undefined;
            resourceInputs["vm"] = args ? args.vm : undefined;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["author"] = undefined /*out*/;
            resourceInputs["creationDate"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["formulaContent"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["osType"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueIdentifier"] = undefined /*out*/;
            resourceInputs["vm"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devtestlab:Formula" }, { type: "azure-native:devtestlab/v20150521preview:Formula" }, { type: "azure-native:devtestlab/v20180915:Formula" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Formula.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Formula resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Formula is deprecated: Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later.");
        return new Formula(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Formula.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Formula.__pulumiType;
    }
}
exports.Formula = Formula;
/** @internal */
Formula.__pulumiType = 'azure-native:devtestlab/v20160515:Formula';
