// @pulumi/azure-native/devtestlab/v20160515/lab.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Lab = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A lab.
 *
 * @deprecated Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later.
 */
class Lab extends pulumi.CustomResource {
    /**
     * Create a Lab resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Lab is deprecated: Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["labStorageType"] = args ? args.labStorageType : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["premiumDataDisks"] = args ? args.premiumDataDisks : undefined;
            resourceInputs["provisioningState"] = args ? args.provisioningState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["uniqueIdentifier"] = args ? args.uniqueIdentifier : undefined;
            resourceInputs["artifactsStorageAccount"] = undefined /*out*/;
            resourceInputs["createdDate"] = undefined /*out*/;
            resourceInputs["defaultPremiumStorageAccount"] = undefined /*out*/;
            resourceInputs["defaultStorageAccount"] = undefined /*out*/;
            resourceInputs["premiumDataDiskStorageAccount"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vaultName"] = undefined /*out*/;
        }
        else {
            resourceInputs["artifactsStorageAccount"] = undefined /*out*/;
            resourceInputs["createdDate"] = undefined /*out*/;
            resourceInputs["defaultPremiumStorageAccount"] = undefined /*out*/;
            resourceInputs["defaultStorageAccount"] = undefined /*out*/;
            resourceInputs["labStorageType"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["premiumDataDiskStorageAccount"] = undefined /*out*/;
            resourceInputs["premiumDataDisks"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueIdentifier"] = undefined /*out*/;
            resourceInputs["vaultName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devtestlab:Lab" }, { type: "azure-native:devtestlab/v20150521preview:Lab" }, { type: "azure-native:devtestlab/v20180915:Lab" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Lab.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Lab resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Lab is deprecated: Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later.");
        return new Lab(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Lab.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Lab.__pulumiType;
    }
}
exports.Lab = Lab;
/** @internal */
Lab.__pulumiType = 'azure-native:devtestlab/v20160515:Lab';
