// @pulumi/azure-native/devtestlab/v20160515/virtualMachine.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VirtualMachine = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A virtual machine.
 *
 * @deprecated Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later.
 */
class VirtualMachine extends pulumi.CustomResource {
    /**
     * Create a VirtualMachine resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("VirtualMachine is deprecated: Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.labName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'labName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["allowClaim"] = args ? args.allowClaim : undefined;
            resourceInputs["applicableSchedule"] = args ? args.applicableSchedule : undefined;
            resourceInputs["artifactDeploymentStatus"] = args ? args.artifactDeploymentStatus : undefined;
            resourceInputs["artifacts"] = args ? args.artifacts : undefined;
            resourceInputs["computeVm"] = args ? args.computeVm : undefined;
            resourceInputs["createdByUser"] = args ? args.createdByUser : undefined;
            resourceInputs["createdByUserId"] = args ? args.createdByUserId : undefined;
            resourceInputs["createdDate"] = args ? args.createdDate : undefined;
            resourceInputs["customImageId"] = args ? args.customImageId : undefined;
            resourceInputs["disallowPublicIpAddress"] = args ? args.disallowPublicIpAddress : undefined;
            resourceInputs["environmentId"] = args ? args.environmentId : undefined;
            resourceInputs["expirationDate"] = args ? args.expirationDate : undefined;
            resourceInputs["fqdn"] = args ? args.fqdn : undefined;
            resourceInputs["galleryImageReference"] = args ? args.galleryImageReference : undefined;
            resourceInputs["isAuthenticationWithSshKey"] = args ? args.isAuthenticationWithSshKey : undefined;
            resourceInputs["labName"] = args ? args.labName : undefined;
            resourceInputs["labSubnetName"] = args ? args.labSubnetName : undefined;
            resourceInputs["labVirtualNetworkId"] = args ? args.labVirtualNetworkId : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["networkInterface"] = args ? args.networkInterface : undefined;
            resourceInputs["notes"] = args ? args.notes : undefined;
            resourceInputs["osType"] = args ? args.osType : undefined;
            resourceInputs["ownerObjectId"] = args ? args.ownerObjectId : undefined;
            resourceInputs["ownerUserPrincipalName"] = args ? args.ownerUserPrincipalName : undefined;
            resourceInputs["password"] = args ? args.password : undefined;
            resourceInputs["provisioningState"] = args ? args.provisioningState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["size"] = args ? args.size : undefined;
            resourceInputs["sshKey"] = args ? args.sshKey : undefined;
            resourceInputs["storageType"] = args ? args.storageType : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["uniqueIdentifier"] = args ? args.uniqueIdentifier : undefined;
            resourceInputs["userName"] = args ? args.userName : undefined;
            resourceInputs["virtualMachineCreationSource"] = args ? args.virtualMachineCreationSource : undefined;
            resourceInputs["computeId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["allowClaim"] = undefined /*out*/;
            resourceInputs["applicableSchedule"] = undefined /*out*/;
            resourceInputs["artifactDeploymentStatus"] = undefined /*out*/;
            resourceInputs["artifacts"] = undefined /*out*/;
            resourceInputs["computeId"] = undefined /*out*/;
            resourceInputs["computeVm"] = undefined /*out*/;
            resourceInputs["createdByUser"] = undefined /*out*/;
            resourceInputs["createdByUserId"] = undefined /*out*/;
            resourceInputs["createdDate"] = undefined /*out*/;
            resourceInputs["customImageId"] = undefined /*out*/;
            resourceInputs["disallowPublicIpAddress"] = undefined /*out*/;
            resourceInputs["environmentId"] = undefined /*out*/;
            resourceInputs["expirationDate"] = undefined /*out*/;
            resourceInputs["fqdn"] = undefined /*out*/;
            resourceInputs["galleryImageReference"] = undefined /*out*/;
            resourceInputs["isAuthenticationWithSshKey"] = undefined /*out*/;
            resourceInputs["labSubnetName"] = undefined /*out*/;
            resourceInputs["labVirtualNetworkId"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkInterface"] = undefined /*out*/;
            resourceInputs["notes"] = undefined /*out*/;
            resourceInputs["osType"] = undefined /*out*/;
            resourceInputs["ownerObjectId"] = undefined /*out*/;
            resourceInputs["ownerUserPrincipalName"] = undefined /*out*/;
            resourceInputs["password"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["size"] = undefined /*out*/;
            resourceInputs["sshKey"] = undefined /*out*/;
            resourceInputs["storageType"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["uniqueIdentifier"] = undefined /*out*/;
            resourceInputs["userName"] = undefined /*out*/;
            resourceInputs["virtualMachineCreationSource"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:devtestlab:VirtualMachine" }, { type: "azure-native:devtestlab/v20150521preview:VirtualMachine" }, { type: "azure-native:devtestlab/v20180915:VirtualMachine" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(VirtualMachine.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing VirtualMachine resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("VirtualMachine is deprecated: Version v20160515 will be removed in the next major version of the provider. Upgrade to version v20180915 or later.");
        return new VirtualMachine(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of VirtualMachine.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VirtualMachine.__pulumiType;
    }
}
exports.VirtualMachine = VirtualMachine;
/** @internal */
VirtualMachine.__pulumiType = 'azure-native:devtestlab/v20160515:VirtualMachine';
