// @pulumi/azure-native/types/enums/signalrservice/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignalRSkuTier = exports.SignalRRequestType = exports.ServiceKind = exports.PrivateLinkServiceConnectionStatus = exports.FeatureFlags = exports.ACLAction = exports.v20220201 = exports.v20211001 = exports.v20210901preview = exports.v20210601preview = exports.v20210401preview = exports.v20200701preview = exports.v20200501 = exports.v20181001 = exports.v20180301preview = void 0;
// Export sub-modules:
const v20180301preview = require("./v20180301preview");
exports.v20180301preview = v20180301preview;
const v20181001 = require("./v20181001");
exports.v20181001 = v20181001;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
const v20220201 = require("./v20220201");
exports.v20220201 = v20220201;
exports.ACLAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.FeatureFlags = {
    ServiceMode: "ServiceMode",
    EnableConnectivityLogs: "EnableConnectivityLogs",
    EnableMessagingLogs: "EnableMessagingLogs",
};
exports.PrivateLinkServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.ServiceKind = {
    SignalR: "SignalR",
    RawWebSockets: "RawWebSockets",
};
exports.SignalRRequestType = {
    ClientConnection: "ClientConnection",
    ServerConnection: "ServerConnection",
    RESTAPI: "RESTAPI",
    Trace: "Trace",
};
exports.SignalRSkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
