// @pulumi/azure-native/types/enums/appplatform/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserSourceType = exports.StorageType = exports.RuntimeVersion = exports.ManagedIdentityType = exports.BindingType = exports.v20220301preview = exports.v20220101preview = exports.v20210901preview = exports.v20210601preview = exports.v20201101preview = exports.v20200701 = void 0;
// Export sub-modules:
const v20200701 = require("./v20200701");
exports.v20200701 = v20200701;
const v20201101preview = require("./v20201101preview");
exports.v20201101preview = v20201101preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
const v20220101preview = require("./v20220101preview");
exports.v20220101preview = v20220101preview;
const v20220301preview = require("./v20220301preview");
exports.v20220301preview = v20220301preview;
exports.BindingType = {
    ApplicationInsights: "ApplicationInsights",
    ApacheSkyWalking: "ApacheSkyWalking",
    AppDynamics: "AppDynamics",
    Dynatrace: "Dynatrace",
    NewRelic: "NewRelic",
    ElasticAPM: "ElasticAPM",
};
exports.ManagedIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
};
exports.RuntimeVersion = {
    Java_8: "Java_8",
    Java_11: "Java_11",
    NetCore_31: "NetCore_31",
};
exports.StorageType = {
    StorageAccount: "StorageAccount",
};
exports.UserSourceType = {
    Jar: "Jar",
    NetCoreZip: "NetCoreZip",
    Source: "Source",
};
