// @pulumi/azure-native/cognitiveservices/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220301 = exports.v20211001 = exports.v20210430 = exports.v20170418 = exports.v20160201preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./account"), exports);
__exportStar(require("./commitmentPlan"), exports);
__exportStar(require("./deployment"), exports);
__exportStar(require("./getAccount"), exports);
__exportStar(require("./getCommitmentPlan"), exports);
__exportStar(require("./getDeployment"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./listAccountKeys"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
// Export enums:
__exportStar(require("../types/enums/cognitiveservices"), exports);
// Export sub-modules:
const v20160201preview = require("./v20160201preview");
exports.v20160201preview = v20160201preview;
const v20170418 = require("./v20170418");
exports.v20170418 = v20170418;
const v20210430 = require("./v20210430");
exports.v20210430 = v20210430;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
const v20220301 = require("./v20220301");
exports.v20220301 = v20220301;
// Import resources to register:
const account_1 = require("./account");
const commitmentPlan_1 = require("./commitmentPlan");
const deployment_1 = require("./deployment");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:cognitiveservices:Account":
                return new account_1.Account(name, undefined, { urn });
            case "azure-native:cognitiveservices:CommitmentPlan":
                return new commitmentPlan_1.CommitmentPlan(name, undefined, { urn });
            case "azure-native:cognitiveservices:Deployment":
                return new deployment_1.Deployment(name, undefined, { urn });
            case "azure-native:cognitiveservices:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "cognitiveservices", _module);
