// @pulumi/azure-native/customerinsights/v20170101/hub.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Hub = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Hub resource.
 *
 * @deprecated Version v20170101 will be removed in the next major version of the provider. Upgrade to version v20170426 or later.
 */
class Hub extends pulumi.CustomResource {
    /**
     * Create a Hub resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20170101 will be removed in the next major version of the provider. Upgrade to version v20170426 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Hub is deprecated: Version v20170101 will be removed in the next major version of the provider. Upgrade to version v20170426 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["hubBillingInfo"] = args ? args.hubBillingInfo : undefined;
            resourceInputs["hubName"] = args ? args.hubName : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["tenantFeatures"] = args ? args.tenantFeatures : undefined;
            resourceInputs["apiEndpoint"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["webEndpoint"] = undefined /*out*/;
        }
        else {
            resourceInputs["apiEndpoint"] = undefined /*out*/;
            resourceInputs["hubBillingInfo"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["tenantFeatures"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["webEndpoint"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:customerinsights:Hub" }, { type: "azure-native:customerinsights/v20170426:Hub" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Hub.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Hub resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Hub is deprecated: Version v20170101 will be removed in the next major version of the provider. Upgrade to version v20170426 or later.");
        return new Hub(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Hub.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Hub.__pulumiType;
    }
}
exports.Hub = Hub;
/** @internal */
Hub.__pulumiType = 'azure-native:customerinsights/v20170101:Hub';
