// @pulumi/azure-native/types/enums/fluidrelay/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisioningState = exports.v20220215 = exports.v20210910preview = exports.v20210830preview = exports.v20210615preview = exports.v20210312preview = void 0;
// Export sub-modules:
const v20210312preview = require("./v20210312preview");
exports.v20210312preview = v20210312preview;
const v20210615preview = require("./v20210615preview");
exports.v20210615preview = v20210615preview;
const v20210830preview = require("./v20210830preview");
exports.v20210830preview = v20210830preview;
const v20210910preview = require("./v20210910preview");
exports.v20210910preview = v20210910preview;
const v20220215 = require("./v20220215");
exports.v20220215 = v20220215;
exports.ProvisioningState = {
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
};
