// @pulumi/azure-native/types/enums/labservices/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeekDay = exports.ShutdownOnIdleMode = exports.RecurrenceFrequency = exports.ManagedLabVmSize = exports.LabUserAccessMode = exports.EnableState = exports.ConnectionType = exports.ConfigurationState = exports.AddRemove = exports.v20211115preview = exports.v20211001preview = exports.v20181015 = void 0;
// Export sub-modules:
const v20181015 = require("./v20181015");
exports.v20181015 = v20181015;
const v20211001preview = require("./v20211001preview");
exports.v20211001preview = v20211001preview;
const v20211115preview = require("./v20211115preview");
exports.v20211115preview = v20211115preview;
exports.AddRemove = {
    /**
     * Indicates that a user is adding a favorite lab
     */
    Add: "Add",
    /**
     * Indicates that a user is removing a favorite lab
     */
    Remove: "Remove",
};
exports.ConfigurationState = {
    /**
     * User either hasn't started configuring their template
     * or they haven't started the configuration process.
     */
    NotApplicable: "NotApplicable",
    /**
     * User is finished modifying the template.
     */
    Completed: "Completed",
};
exports.ConnectionType = {
    Public: "Public",
    Private: "Private",
    None: "None",
};
exports.EnableState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.LabUserAccessMode = {
    /**
     * Only users registered with the lab can access VMs.
     */
    Restricted: "Restricted",
    /**
     * Any user can register with the lab and access its VMs.
     */
    Open: "Open",
};
exports.ManagedLabVmSize = {
    /**
     * The base VM size
     */
    Basic: "Basic",
    /**
     * The standard or default VM size
     */
    Standard: "Standard",
    /**
     * The most performant VM size
     */
    Performance: "Performance",
};
exports.RecurrenceFrequency = {
    /**
     * Schedule will run every days.
     */
    Daily: "Daily",
    /**
     * Schedule will run every week on days specified in weekDays.
     */
    Weekly: "Weekly",
};
exports.ShutdownOnIdleMode = {
    /**
     * The VM won't be shut down when it is idle.
     */
    None: "None",
    /**
     * The VM will be considered as idle when there is no keyboard or mouse input.
     */
    UserAbsence: "UserAbsence",
    /**
     * The VM will be considered as idle when user is absent and the resource (CPU and disk) consumption is low.
     */
    LowUsage: "LowUsage",
};
exports.WeekDay = {
    /**
     * Schedule will run on Sunday
     */
    Sunday: "Sunday",
    /**
     * Schedule will run on Monday
     */
    Monday: "Monday",
    /**
     * Schedule will run on Tuesday
     */
    Tuesday: "Tuesday",
    /**
     * Schedule will run on Wednesday
     */
    Wednesday: "Wednesday",
    /**
     * Schedule will run on Thursday
     */
    Thursday: "Thursday",
    /**
     * Schedule will run on Friday
     */
    Friday: "Friday",
    /**
     * Schedule will run on Saturday
     */
    Saturday: "Saturday",
};
