// @pulumi/azure-native/web/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210301 = exports.v20210201 = exports.v20210115 = exports.v20210101 = exports.v20201201 = exports.v20201001 = exports.v20200901 = exports.v20200601 = exports.v20190801 = exports.v20181101 = exports.v20180201 = exports.v20160901 = exports.v20160801 = exports.v20160601 = exports.v20160301 = exports.v20150801preview = exports.v20150801 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./appServiceEnvironment"), exports);
__exportStar(require("./appServiceEnvironmentPrivateEndpointConnection"), exports);
__exportStar(require("./appServicePlan"), exports);
__exportStar(require("./appServicePlanRouteForVnet"), exports);
__exportStar(require("./certificate"), exports);
__exportStar(require("./connection"), exports);
__exportStar(require("./connectionGateway"), exports);
__exportStar(require("./containerApp"), exports);
__exportStar(require("./customApi"), exports);
__exportStar(require("./getAppServiceEnvironment"), exports);
__exportStar(require("./getAppServiceEnvironmentPrivateEndpointConnection"), exports);
__exportStar(require("./getAppServicePlan"), exports);
__exportStar(require("./getCertificate"), exports);
__exportStar(require("./getConnection"), exports);
__exportStar(require("./getConnectionGateway"), exports);
__exportStar(require("./getContainerApp"), exports);
__exportStar(require("./getCustomApi"), exports);
__exportStar(require("./getKubeEnvironment"), exports);
__exportStar(require("./getStaticSite"), exports);
__exportStar(require("./getStaticSiteCustomDomain"), exports);
__exportStar(require("./getStaticSitePrivateEndpointConnection"), exports);
__exportStar(require("./getStaticSiteUserProvidedFunctionAppForStaticSite"), exports);
__exportStar(require("./getStaticSiteUserProvidedFunctionAppForStaticSiteBuild"), exports);
__exportStar(require("./getWebApp"), exports);
__exportStar(require("./getWebAppDeployment"), exports);
__exportStar(require("./getWebAppDeploymentSlot"), exports);
__exportStar(require("./getWebAppDiagnosticLogsConfiguration"), exports);
__exportStar(require("./getWebAppDomainOwnershipIdentifier"), exports);
__exportStar(require("./getWebAppDomainOwnershipIdentifierSlot"), exports);
__exportStar(require("./getWebAppFunction"), exports);
__exportStar(require("./getWebAppHostNameBinding"), exports);
__exportStar(require("./getWebAppHostNameBindingSlot"), exports);
__exportStar(require("./getWebAppHybridConnection"), exports);
__exportStar(require("./getWebAppHybridConnectionSlot"), exports);
__exportStar(require("./getWebAppInstanceFunctionSlot"), exports);
__exportStar(require("./getWebAppPremierAddOn"), exports);
__exportStar(require("./getWebAppPremierAddOnSlot"), exports);
__exportStar(require("./getWebAppPrivateEndpointConnection"), exports);
__exportStar(require("./getWebAppPrivateEndpointConnectionSlot"), exports);
__exportStar(require("./getWebAppPublicCertificate"), exports);
__exportStar(require("./getWebAppPublicCertificateSlot"), exports);
__exportStar(require("./getWebAppRelayServiceConnection"), exports);
__exportStar(require("./getWebAppRelayServiceConnectionSlot"), exports);
__exportStar(require("./getWebAppSiteExtension"), exports);
__exportStar(require("./getWebAppSiteExtensionSlot"), exports);
__exportStar(require("./getWebAppSlot"), exports);
__exportStar(require("./getWebAppSlotConfigurationNames"), exports);
__exportStar(require("./getWebAppSourceControl"), exports);
__exportStar(require("./getWebAppSourceControlSlot"), exports);
__exportStar(require("./getWebAppSwiftVirtualNetworkConnection"), exports);
__exportStar(require("./getWebAppSwiftVirtualNetworkConnectionSlot"), exports);
__exportStar(require("./getWebAppVnetConnection"), exports);
__exportStar(require("./getWebAppVnetConnectionSlot"), exports);
__exportStar(require("./kubeEnvironment"), exports);
__exportStar(require("./listAppServicePlanHybridConnectionKeys"), exports);
__exportStar(require("./listConnectionConsentLinks"), exports);
__exportStar(require("./listConnectionKeys"), exports);
__exportStar(require("./listContainerAppSecrets"), exports);
__exportStar(require("./listCustomApiWsdlInterfaces"), exports);
__exportStar(require("./listSiteIdentifiersAssignedToHostName"), exports);
__exportStar(require("./listStaticSiteAppSettings"), exports);
__exportStar(require("./listStaticSiteBuildAppSettings"), exports);
__exportStar(require("./listStaticSiteBuildFunctionAppSettings"), exports);
__exportStar(require("./listStaticSiteConfiguredRoles"), exports);
__exportStar(require("./listStaticSiteFunctionAppSettings"), exports);
__exportStar(require("./listStaticSiteSecrets"), exports);
__exportStar(require("./listStaticSiteUsers"), exports);
__exportStar(require("./listWebAppApplicationSettings"), exports);
__exportStar(require("./listWebAppApplicationSettingsSlot"), exports);
__exportStar(require("./listWebAppAuthSettings"), exports);
__exportStar(require("./listWebAppAuthSettingsSlot"), exports);
__exportStar(require("./listWebAppAzureStorageAccounts"), exports);
__exportStar(require("./listWebAppAzureStorageAccountsSlot"), exports);
__exportStar(require("./listWebAppBackupConfiguration"), exports);
__exportStar(require("./listWebAppBackupConfigurationSlot"), exports);
__exportStar(require("./listWebAppBackupStatusSecrets"), exports);
__exportStar(require("./listWebAppBackupStatusSecretsSlot"), exports);
__exportStar(require("./listWebAppConnectionStrings"), exports);
__exportStar(require("./listWebAppConnectionStringsSlot"), exports);
__exportStar(require("./listWebAppFunctionKeys"), exports);
__exportStar(require("./listWebAppFunctionKeysSlot"), exports);
__exportStar(require("./listWebAppFunctionSecrets"), exports);
__exportStar(require("./listWebAppFunctionSecretsSlot"), exports);
__exportStar(require("./listWebAppHostKeys"), exports);
__exportStar(require("./listWebAppHostKeysSlot"), exports);
__exportStar(require("./listWebAppHybridConnectionKeys"), exports);
__exportStar(require("./listWebAppHybridConnectionKeysSlot"), exports);
__exportStar(require("./listWebAppMetadata"), exports);
__exportStar(require("./listWebAppMetadataSlot"), exports);
__exportStar(require("./listWebAppPublishingCredentials"), exports);
__exportStar(require("./listWebAppPublishingCredentialsSlot"), exports);
__exportStar(require("./listWebAppSiteBackups"), exports);
__exportStar(require("./listWebAppSiteBackupsSlot"), exports);
__exportStar(require("./listWebAppSitePushSettings"), exports);
__exportStar(require("./listWebAppSitePushSettingsSlot"), exports);
__exportStar(require("./listWebAppSyncFunctionTriggers"), exports);
__exportStar(require("./listWebAppSyncFunctionTriggersSlot"), exports);
__exportStar(require("./staticSite"), exports);
__exportStar(require("./staticSiteCustomDomain"), exports);
__exportStar(require("./staticSitePrivateEndpointConnection"), exports);
__exportStar(require("./staticSiteUserProvidedFunctionAppForStaticSite"), exports);
__exportStar(require("./staticSiteUserProvidedFunctionAppForStaticSiteBuild"), exports);
__exportStar(require("./webApp"), exports);
__exportStar(require("./webAppApplicationSettings"), exports);
__exportStar(require("./webAppApplicationSettingsSlot"), exports);
__exportStar(require("./webAppAuthSettings"), exports);
__exportStar(require("./webAppAuthSettingsSlot"), exports);
__exportStar(require("./webAppAuthSettingsV2"), exports);
__exportStar(require("./webAppAuthSettingsV2Slot"), exports);
__exportStar(require("./webAppAzureStorageAccounts"), exports);
__exportStar(require("./webAppAzureStorageAccountsSlot"), exports);
__exportStar(require("./webAppBackupConfiguration"), exports);
__exportStar(require("./webAppBackupConfigurationSlot"), exports);
__exportStar(require("./webAppConnectionStrings"), exports);
__exportStar(require("./webAppConnectionStringsSlot"), exports);
__exportStar(require("./webAppDeployment"), exports);
__exportStar(require("./webAppDeploymentSlot"), exports);
__exportStar(require("./webAppDiagnosticLogsConfiguration"), exports);
__exportStar(require("./webAppDomainOwnershipIdentifier"), exports);
__exportStar(require("./webAppDomainOwnershipIdentifierSlot"), exports);
__exportStar(require("./webAppFunction"), exports);
__exportStar(require("./webAppHostNameBinding"), exports);
__exportStar(require("./webAppHostNameBindingSlot"), exports);
__exportStar(require("./webAppHybridConnection"), exports);
__exportStar(require("./webAppHybridConnectionSlot"), exports);
__exportStar(require("./webAppInstanceFunctionSlot"), exports);
__exportStar(require("./webAppMetadata"), exports);
__exportStar(require("./webAppMetadataSlot"), exports);
__exportStar(require("./webAppPremierAddOn"), exports);
__exportStar(require("./webAppPremierAddOnSlot"), exports);
__exportStar(require("./webAppPrivateEndpointConnection"), exports);
__exportStar(require("./webAppPrivateEndpointConnectionSlot"), exports);
__exportStar(require("./webAppPublicCertificate"), exports);
__exportStar(require("./webAppPublicCertificateSlot"), exports);
__exportStar(require("./webAppRelayServiceConnection"), exports);
__exportStar(require("./webAppRelayServiceConnectionSlot"), exports);
__exportStar(require("./webAppSiteExtension"), exports);
__exportStar(require("./webAppSiteExtensionSlot"), exports);
__exportStar(require("./webAppSitePushSettings"), exports);
__exportStar(require("./webAppSitePushSettingsSlot"), exports);
__exportStar(require("./webAppSlot"), exports);
__exportStar(require("./webAppSlotConfigurationNames"), exports);
__exportStar(require("./webAppSourceControl"), exports);
__exportStar(require("./webAppSourceControlSlot"), exports);
__exportStar(require("./webAppSwiftVirtualNetworkConnection"), exports);
__exportStar(require("./webAppSwiftVirtualNetworkConnectionSlot"), exports);
__exportStar(require("./webAppVnetConnection"), exports);
__exportStar(require("./webAppVnetConnectionSlot"), exports);
// Export enums:
__exportStar(require("../types/enums/web"), exports);
// Export sub-modules:
const v20150801 = require("./v20150801");
exports.v20150801 = v20150801;
const v20150801preview = require("./v20150801preview");
exports.v20150801preview = v20150801preview;
const v20160301 = require("./v20160301");
exports.v20160301 = v20160301;
const v20160601 = require("./v20160601");
exports.v20160601 = v20160601;
const v20160801 = require("./v20160801");
exports.v20160801 = v20160801;
const v20160901 = require("./v20160901");
exports.v20160901 = v20160901;
const v20180201 = require("./v20180201");
exports.v20180201 = v20180201;
const v20181101 = require("./v20181101");
exports.v20181101 = v20181101;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210115 = require("./v20210115");
exports.v20210115 = v20210115;
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
// Import resources to register:
const appServiceEnvironment_1 = require("./appServiceEnvironment");
const appServiceEnvironmentPrivateEndpointConnection_1 = require("./appServiceEnvironmentPrivateEndpointConnection");
const appServicePlan_1 = require("./appServicePlan");
const appServicePlanRouteForVnet_1 = require("./appServicePlanRouteForVnet");
const certificate_1 = require("./certificate");
const connection_1 = require("./connection");
const connectionGateway_1 = require("./connectionGateway");
const containerApp_1 = require("./containerApp");
const customApi_1 = require("./customApi");
const kubeEnvironment_1 = require("./kubeEnvironment");
const staticSite_1 = require("./staticSite");
const staticSiteCustomDomain_1 = require("./staticSiteCustomDomain");
const staticSitePrivateEndpointConnection_1 = require("./staticSitePrivateEndpointConnection");
const staticSiteUserProvidedFunctionAppForStaticSite_1 = require("./staticSiteUserProvidedFunctionAppForStaticSite");
const staticSiteUserProvidedFunctionAppForStaticSiteBuild_1 = require("./staticSiteUserProvidedFunctionAppForStaticSiteBuild");
const webApp_1 = require("./webApp");
const webAppApplicationSettings_1 = require("./webAppApplicationSettings");
const webAppApplicationSettingsSlot_1 = require("./webAppApplicationSettingsSlot");
const webAppAuthSettings_1 = require("./webAppAuthSettings");
const webAppAuthSettingsSlot_1 = require("./webAppAuthSettingsSlot");
const webAppAuthSettingsV2_1 = require("./webAppAuthSettingsV2");
const webAppAuthSettingsV2Slot_1 = require("./webAppAuthSettingsV2Slot");
const webAppAzureStorageAccounts_1 = require("./webAppAzureStorageAccounts");
const webAppAzureStorageAccountsSlot_1 = require("./webAppAzureStorageAccountsSlot");
const webAppBackupConfiguration_1 = require("./webAppBackupConfiguration");
const webAppBackupConfigurationSlot_1 = require("./webAppBackupConfigurationSlot");
const webAppConnectionStrings_1 = require("./webAppConnectionStrings");
const webAppConnectionStringsSlot_1 = require("./webAppConnectionStringsSlot");
const webAppDeployment_1 = require("./webAppDeployment");
const webAppDeploymentSlot_1 = require("./webAppDeploymentSlot");
const webAppDiagnosticLogsConfiguration_1 = require("./webAppDiagnosticLogsConfiguration");
const webAppDomainOwnershipIdentifier_1 = require("./webAppDomainOwnershipIdentifier");
const webAppDomainOwnershipIdentifierSlot_1 = require("./webAppDomainOwnershipIdentifierSlot");
const webAppFunction_1 = require("./webAppFunction");
const webAppHostNameBinding_1 = require("./webAppHostNameBinding");
const webAppHostNameBindingSlot_1 = require("./webAppHostNameBindingSlot");
const webAppHybridConnection_1 = require("./webAppHybridConnection");
const webAppHybridConnectionSlot_1 = require("./webAppHybridConnectionSlot");
const webAppInstanceFunctionSlot_1 = require("./webAppInstanceFunctionSlot");
const webAppMetadata_1 = require("./webAppMetadata");
const webAppMetadataSlot_1 = require("./webAppMetadataSlot");
const webAppPremierAddOn_1 = require("./webAppPremierAddOn");
const webAppPremierAddOnSlot_1 = require("./webAppPremierAddOnSlot");
const webAppPrivateEndpointConnection_1 = require("./webAppPrivateEndpointConnection");
const webAppPrivateEndpointConnectionSlot_1 = require("./webAppPrivateEndpointConnectionSlot");
const webAppPublicCertificate_1 = require("./webAppPublicCertificate");
const webAppPublicCertificateSlot_1 = require("./webAppPublicCertificateSlot");
const webAppRelayServiceConnection_1 = require("./webAppRelayServiceConnection");
const webAppRelayServiceConnectionSlot_1 = require("./webAppRelayServiceConnectionSlot");
const webAppSiteExtension_1 = require("./webAppSiteExtension");
const webAppSiteExtensionSlot_1 = require("./webAppSiteExtensionSlot");
const webAppSitePushSettings_1 = require("./webAppSitePushSettings");
const webAppSitePushSettingsSlot_1 = require("./webAppSitePushSettingsSlot");
const webAppSlot_1 = require("./webAppSlot");
const webAppSlotConfigurationNames_1 = require("./webAppSlotConfigurationNames");
const webAppSourceControl_1 = require("./webAppSourceControl");
const webAppSourceControlSlot_1 = require("./webAppSourceControlSlot");
const webAppSwiftVirtualNetworkConnection_1 = require("./webAppSwiftVirtualNetworkConnection");
const webAppSwiftVirtualNetworkConnectionSlot_1 = require("./webAppSwiftVirtualNetworkConnectionSlot");
const webAppVnetConnection_1 = require("./webAppVnetConnection");
const webAppVnetConnectionSlot_1 = require("./webAppVnetConnectionSlot");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:web:AppServiceEnvironment":
                return new appServiceEnvironment_1.AppServiceEnvironment(name, undefined, { urn });
            case "azure-native:web:AppServiceEnvironmentPrivateEndpointConnection":
                return new appServiceEnvironmentPrivateEndpointConnection_1.AppServiceEnvironmentPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:web:AppServicePlan":
                return new appServicePlan_1.AppServicePlan(name, undefined, { urn });
            case "azure-native:web:AppServicePlanRouteForVnet":
                return new appServicePlanRouteForVnet_1.AppServicePlanRouteForVnet(name, undefined, { urn });
            case "azure-native:web:Certificate":
                return new certificate_1.Certificate(name, undefined, { urn });
            case "azure-native:web:Connection":
                return new connection_1.Connection(name, undefined, { urn });
            case "azure-native:web:ConnectionGateway":
                return new connectionGateway_1.ConnectionGateway(name, undefined, { urn });
            case "azure-native:web:ContainerApp":
                return new containerApp_1.ContainerApp(name, undefined, { urn });
            case "azure-native:web:CustomApi":
                return new customApi_1.CustomApi(name, undefined, { urn });
            case "azure-native:web:KubeEnvironment":
                return new kubeEnvironment_1.KubeEnvironment(name, undefined, { urn });
            case "azure-native:web:StaticSite":
                return new staticSite_1.StaticSite(name, undefined, { urn });
            case "azure-native:web:StaticSiteCustomDomain":
                return new staticSiteCustomDomain_1.StaticSiteCustomDomain(name, undefined, { urn });
            case "azure-native:web:StaticSitePrivateEndpointConnection":
                return new staticSitePrivateEndpointConnection_1.StaticSitePrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:web:StaticSiteUserProvidedFunctionAppForStaticSite":
                return new staticSiteUserProvidedFunctionAppForStaticSite_1.StaticSiteUserProvidedFunctionAppForStaticSite(name, undefined, { urn });
            case "azure-native:web:StaticSiteUserProvidedFunctionAppForStaticSiteBuild":
                return new staticSiteUserProvidedFunctionAppForStaticSiteBuild_1.StaticSiteUserProvidedFunctionAppForStaticSiteBuild(name, undefined, { urn });
            case "azure-native:web:WebApp":
                return new webApp_1.WebApp(name, undefined, { urn });
            case "azure-native:web:WebAppApplicationSettings":
                return new webAppApplicationSettings_1.WebAppApplicationSettings(name, undefined, { urn });
            case "azure-native:web:WebAppApplicationSettingsSlot":
                return new webAppApplicationSettingsSlot_1.WebAppApplicationSettingsSlot(name, undefined, { urn });
            case "azure-native:web:WebAppAuthSettings":
                return new webAppAuthSettings_1.WebAppAuthSettings(name, undefined, { urn });
            case "azure-native:web:WebAppAuthSettingsSlot":
                return new webAppAuthSettingsSlot_1.WebAppAuthSettingsSlot(name, undefined, { urn });
            case "azure-native:web:WebAppAuthSettingsV2":
                return new webAppAuthSettingsV2_1.WebAppAuthSettingsV2(name, undefined, { urn });
            case "azure-native:web:WebAppAuthSettingsV2Slot":
                return new webAppAuthSettingsV2Slot_1.WebAppAuthSettingsV2Slot(name, undefined, { urn });
            case "azure-native:web:WebAppAzureStorageAccounts":
                return new webAppAzureStorageAccounts_1.WebAppAzureStorageAccounts(name, undefined, { urn });
            case "azure-native:web:WebAppAzureStorageAccountsSlot":
                return new webAppAzureStorageAccountsSlot_1.WebAppAzureStorageAccountsSlot(name, undefined, { urn });
            case "azure-native:web:WebAppBackupConfiguration":
                return new webAppBackupConfiguration_1.WebAppBackupConfiguration(name, undefined, { urn });
            case "azure-native:web:WebAppBackupConfigurationSlot":
                return new webAppBackupConfigurationSlot_1.WebAppBackupConfigurationSlot(name, undefined, { urn });
            case "azure-native:web:WebAppConnectionStrings":
                return new webAppConnectionStrings_1.WebAppConnectionStrings(name, undefined, { urn });
            case "azure-native:web:WebAppConnectionStringsSlot":
                return new webAppConnectionStringsSlot_1.WebAppConnectionStringsSlot(name, undefined, { urn });
            case "azure-native:web:WebAppDeployment":
                return new webAppDeployment_1.WebAppDeployment(name, undefined, { urn });
            case "azure-native:web:WebAppDeploymentSlot":
                return new webAppDeploymentSlot_1.WebAppDeploymentSlot(name, undefined, { urn });
            case "azure-native:web:WebAppDiagnosticLogsConfiguration":
                return new webAppDiagnosticLogsConfiguration_1.WebAppDiagnosticLogsConfiguration(name, undefined, { urn });
            case "azure-native:web:WebAppDomainOwnershipIdentifier":
                return new webAppDomainOwnershipIdentifier_1.WebAppDomainOwnershipIdentifier(name, undefined, { urn });
            case "azure-native:web:WebAppDomainOwnershipIdentifierSlot":
                return new webAppDomainOwnershipIdentifierSlot_1.WebAppDomainOwnershipIdentifierSlot(name, undefined, { urn });
            case "azure-native:web:WebAppFunction":
                return new webAppFunction_1.WebAppFunction(name, undefined, { urn });
            case "azure-native:web:WebAppHostNameBinding":
                return new webAppHostNameBinding_1.WebAppHostNameBinding(name, undefined, { urn });
            case "azure-native:web:WebAppHostNameBindingSlot":
                return new webAppHostNameBindingSlot_1.WebAppHostNameBindingSlot(name, undefined, { urn });
            case "azure-native:web:WebAppHybridConnection":
                return new webAppHybridConnection_1.WebAppHybridConnection(name, undefined, { urn });
            case "azure-native:web:WebAppHybridConnectionSlot":
                return new webAppHybridConnectionSlot_1.WebAppHybridConnectionSlot(name, undefined, { urn });
            case "azure-native:web:WebAppInstanceFunctionSlot":
                return new webAppInstanceFunctionSlot_1.WebAppInstanceFunctionSlot(name, undefined, { urn });
            case "azure-native:web:WebAppMetadata":
                return new webAppMetadata_1.WebAppMetadata(name, undefined, { urn });
            case "azure-native:web:WebAppMetadataSlot":
                return new webAppMetadataSlot_1.WebAppMetadataSlot(name, undefined, { urn });
            case "azure-native:web:WebAppPremierAddOn":
                return new webAppPremierAddOn_1.WebAppPremierAddOn(name, undefined, { urn });
            case "azure-native:web:WebAppPremierAddOnSlot":
                return new webAppPremierAddOnSlot_1.WebAppPremierAddOnSlot(name, undefined, { urn });
            case "azure-native:web:WebAppPrivateEndpointConnection":
                return new webAppPrivateEndpointConnection_1.WebAppPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:web:WebAppPrivateEndpointConnectionSlot":
                return new webAppPrivateEndpointConnectionSlot_1.WebAppPrivateEndpointConnectionSlot(name, undefined, { urn });
            case "azure-native:web:WebAppPublicCertificate":
                return new webAppPublicCertificate_1.WebAppPublicCertificate(name, undefined, { urn });
            case "azure-native:web:WebAppPublicCertificateSlot":
                return new webAppPublicCertificateSlot_1.WebAppPublicCertificateSlot(name, undefined, { urn });
            case "azure-native:web:WebAppRelayServiceConnection":
                return new webAppRelayServiceConnection_1.WebAppRelayServiceConnection(name, undefined, { urn });
            case "azure-native:web:WebAppRelayServiceConnectionSlot":
                return new webAppRelayServiceConnectionSlot_1.WebAppRelayServiceConnectionSlot(name, undefined, { urn });
            case "azure-native:web:WebAppSiteExtension":
                return new webAppSiteExtension_1.WebAppSiteExtension(name, undefined, { urn });
            case "azure-native:web:WebAppSiteExtensionSlot":
                return new webAppSiteExtensionSlot_1.WebAppSiteExtensionSlot(name, undefined, { urn });
            case "azure-native:web:WebAppSitePushSettings":
                return new webAppSitePushSettings_1.WebAppSitePushSettings(name, undefined, { urn });
            case "azure-native:web:WebAppSitePushSettingsSlot":
                return new webAppSitePushSettingsSlot_1.WebAppSitePushSettingsSlot(name, undefined, { urn });
            case "azure-native:web:WebAppSlot":
                return new webAppSlot_1.WebAppSlot(name, undefined, { urn });
            case "azure-native:web:WebAppSlotConfigurationNames":
                return new webAppSlotConfigurationNames_1.WebAppSlotConfigurationNames(name, undefined, { urn });
            case "azure-native:web:WebAppSourceControl":
                return new webAppSourceControl_1.WebAppSourceControl(name, undefined, { urn });
            case "azure-native:web:WebAppSourceControlSlot":
                return new webAppSourceControlSlot_1.WebAppSourceControlSlot(name, undefined, { urn });
            case "azure-native:web:WebAppSwiftVirtualNetworkConnection":
                return new webAppSwiftVirtualNetworkConnection_1.WebAppSwiftVirtualNetworkConnection(name, undefined, { urn });
            case "azure-native:web:WebAppSwiftVirtualNetworkConnectionSlot":
                return new webAppSwiftVirtualNetworkConnectionSlot_1.WebAppSwiftVirtualNetworkConnectionSlot(name, undefined, { urn });
            case "azure-native:web:WebAppVnetConnection":
                return new webAppVnetConnection_1.WebAppVnetConnection(name, undefined, { urn });
            case "azure-native:web:WebAppVnetConnectionSlot":
                return new webAppVnetConnectionSlot_1.WebAppVnetConnectionSlot(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "web", _module);
