// @pulumi/azure-native/delegatednetwork/v20200808preview/orchestratorInstanceServiceDetails.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrchestratorInstanceServiceDetails = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents an instance of a orchestrator.
 *
 * @deprecated Version v20200808preview will be removed in the next major version of the provider. Upgrade to version v20210315 or later.
 */
class OrchestratorInstanceServiceDetails extends pulumi.CustomResource {
    /**
     * Create a OrchestratorInstanceServiceDetails resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200808preview will be removed in the next major version of the provider. Upgrade to version v20210315 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("OrchestratorInstanceServiceDetails is deprecated: Version v20200808preview will be removed in the next major version of the provider. Upgrade to version v20210315 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.controllerDetails === undefined) && !opts.urn) {
                throw new Error("Missing required property 'controllerDetails'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["apiServerEndpoint"] = args ? args.apiServerEndpoint : undefined;
            resourceInputs["clusterRootCA"] = args ? args.clusterRootCA : undefined;
            resourceInputs["controllerDetails"] = args ? args.controllerDetails : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["orchestratorAppId"] = args ? args.orchestratorAppId : undefined;
            resourceInputs["orchestratorTenantId"] = args ? args.orchestratorTenantId : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["apiServerEndpoint"] = undefined /*out*/;
            resourceInputs["clusterRootCA"] = undefined /*out*/;
            resourceInputs["controllerDetails"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["orchestratorAppId"] = undefined /*out*/;
            resourceInputs["orchestratorTenantId"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:delegatednetwork:OrchestratorInstanceServiceDetails" }, { type: "azure-native:delegatednetwork/v20210315:OrchestratorInstanceServiceDetails" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(OrchestratorInstanceServiceDetails.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing OrchestratorInstanceServiceDetails resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("OrchestratorInstanceServiceDetails is deprecated: Version v20200808preview will be removed in the next major version of the provider. Upgrade to version v20210315 or later.");
        return new OrchestratorInstanceServiceDetails(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of OrchestratorInstanceServiceDetails.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === OrchestratorInstanceServiceDetails.__pulumiType;
    }
}
exports.OrchestratorInstanceServiceDetails = OrchestratorInstanceServiceDetails;
/** @internal */
OrchestratorInstanceServiceDetails.__pulumiType = 'azure-native:delegatednetwork/v20200808preview:OrchestratorInstanceServiceDetails';
