// @pulumi/azure-native/types/enums/signalrservice/v20200501/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SignalRSkuTier = exports.SignalRRequestType = exports.ServiceKind = exports.PrivateLinkServiceConnectionStatus = exports.FeatureFlags = exports.ACLAction = void 0;
exports.ACLAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.FeatureFlags = {
    ServiceMode: "ServiceMode",
    EnableConnectivityLogs: "EnableConnectivityLogs",
    EnableMessagingLogs: "EnableMessagingLogs",
};
exports.PrivateLinkServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.ServiceKind = {
    SignalR: "SignalR",
    RawWebSockets: "RawWebSockets",
};
exports.SignalRRequestType = {
    ClientConnection: "ClientConnection",
    ServerConnection: "ServerConnection",
    RESTAPI: "RESTAPI",
    Trace: "Trace",
};
exports.SignalRSkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
