// @pulumi/azure-native/types/enums/kubernetesconfiguration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceKindType = exports.ScopeType = exports.ResourceIdentityType = exports.PublicNetworkAccessType = exports.PrivateEndpointServiceConnectionStatus = exports.OperatorType = exports.OperatorScopeType = exports.LevelType = exports.KustomizationValidationType = exports.v20220402preview = exports.v20220301 = exports.v20220101preview = exports.v20211101preview = exports.v20210901 = exports.v20210501preview = exports.v20210301 = exports.v20201001preview = exports.v20200701preview = exports.v20191101preview = void 0;
// Export sub-modules:
const v20191101preview = require("./v20191101preview");
exports.v20191101preview = v20191101preview;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20201001preview = require("./v20201001preview");
exports.v20201001preview = v20201001preview;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210501preview = require("./v20210501preview");
exports.v20210501preview = v20210501preview;
const v20210901 = require("./v20210901");
exports.v20210901 = v20210901;
const v20211101preview = require("./v20211101preview");
exports.v20211101preview = v20211101preview;
const v20220101preview = require("./v20220101preview");
exports.v20220101preview = v20220101preview;
const v20220301 = require("./v20220301");
exports.v20220301 = v20220301;
const v20220402preview = require("./v20220402preview");
exports.v20220402preview = v20220402preview;
exports.KustomizationValidationType = {
    None: "none",
    Client: "client",
    Server: "server",
};
exports.LevelType = {
    Error: "Error",
    Warning: "Warning",
    Information: "Information",
};
exports.OperatorScopeType = {
    Cluster: "cluster",
    Namespace: "namespace",
};
exports.OperatorType = {
    Flux: "Flux",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.PublicNetworkAccessType = {
    /**
     * Allows Azure Arc agents to communicate with Azure Arc services over both public (internet) and private endpoints.
     */
    Enabled: "Enabled",
    /**
     * Does not allow Azure Arc agents to communicate with Azure Arc services over public (internet) endpoints. The agents must use the private link.
     */
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
};
exports.ScopeType = {
    Cluster: "cluster",
    Namespace: "namespace",
};
exports.SourceKindType = {
    GitRepository: "GitRepository",
};
