// @pulumi/azure-native/types/enums/compute/v20211001/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageAccountType = exports.ReplicationMode = exports.OperatingSystemTypes = exports.OperatingSystemStateTypes = exports.HyperVGeneration = exports.HostCaching = exports.GallerySharingPermissionTypes = exports.GalleryExtendedLocationType = exports.ConfidentialVMEncryptionType = exports.Architecture = void 0;
exports.Architecture = {
    X64: "x64",
    Arm64: "Arm64",
};
exports.ConfidentialVMEncryptionType = {
    EncryptedVMGuestStateOnlyWithPmk: "EncryptedVMGuestStateOnlyWithPmk",
    EncryptedWithPmk: "EncryptedWithPmk",
    EncryptedWithCmk: "EncryptedWithCmk",
};
exports.GalleryExtendedLocationType = {
    EdgeZone: "EdgeZone",
    Unknown: "Unknown",
};
exports.GallerySharingPermissionTypes = {
    Private: "Private",
    Groups: "Groups",
};
exports.HostCaching = {
    None: "None",
    ReadOnly: "ReadOnly",
    ReadWrite: "ReadWrite",
};
exports.HyperVGeneration = {
    V1: "V1",
    V2: "V2",
};
exports.OperatingSystemStateTypes = {
    Generalized: "Generalized",
    Specialized: "Specialized",
};
exports.OperatingSystemTypes = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.ReplicationMode = {
    Full: "Full",
    Shallow: "Shallow",
};
exports.StorageAccountType = {
    Standard_LRS: "Standard_LRS",
    Standard_ZRS: "Standard_ZRS",
    Premium_LRS: "Premium_LRS",
};
