// @pulumi/azure-native/types/enums/azurearcdata/v20211101/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlVersion = exports.SqlManagedInstanceSkuTier = exports.Infrastructure = exports.ExtendedLocationTypes = exports.EditionType = exports.DefenderStatus = exports.ConnectionStatus = exports.ArcSqlServerLicenseType = exports.ArcSqlManagedInstanceLicenseType = void 0;
exports.ArcSqlManagedInstanceLicenseType = {
    BasePrice: "BasePrice",
    LicenseIncluded: "LicenseIncluded",
    DisasterRecovery: "DisasterRecovery",
};
exports.ArcSqlServerLicenseType = {
    Paid: "Paid",
    Free: "Free",
    HADR: "HADR",
    Undefined: "Undefined",
};
exports.ConnectionStatus = {
    Connected: "Connected",
    Disconnected: "Disconnected",
    Unknown: "Unknown",
};
exports.DefenderStatus = {
    Protected: "Protected",
    Unprotected: "Unprotected",
    Unknown: "Unknown",
};
exports.EditionType = {
    Evaluation: "Evaluation",
    Enterprise: "Enterprise",
    Standard: "Standard",
    Web: "Web",
    Developer: "Developer",
    Express: "Express",
};
exports.ExtendedLocationTypes = {
    CustomLocation: "CustomLocation",
};
exports.Infrastructure = {
    Azure: "azure",
    Gcp: "gcp",
    Aws: "aws",
    Alibaba: "alibaba",
    Onpremises: "onpremises",
    Other: "other",
};
exports.SqlManagedInstanceSkuTier = {
    GeneralPurpose: "GeneralPurpose",
    BusinessCritical: "BusinessCritical",
};
exports.SqlVersion = {
    SQL_Server_2019: "SQL Server 2019",
    SQL_Server_2017: "SQL Server 2017",
    SQL_Server_2016: "SQL Server 2016",
};
