// @pulumi/azure-native/synapse/v20200401preview/sqlDatabase.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlDatabase = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * A sql database resource.
 *
 * @deprecated Version v20200401preview will be removed in the next major version of the provider. Upgrade to version v20210301 or later.
 */
class SqlDatabase extends pulumi.CustomResource {
    /**
     * Create a SqlDatabase resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200401preview will be removed in the next major version of the provider. Upgrade to version v20210301 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SqlDatabase is deprecated: Version v20200401preview will be removed in the next major version of the provider. Upgrade to version v20210301 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["collation"] = args ? args.collation : undefined;
            resourceInputs["dataRetention"] = args ? args.dataRetention : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sqlDatabaseName"] = args ? args.sqlDatabaseName : undefined;
            resourceInputs["storageRedundancy"] = args ? args.storageRedundancy : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["databaseGuid"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["collation"] = undefined /*out*/;
            resourceInputs["dataRetention"] = undefined /*out*/;
            resourceInputs["databaseGuid"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["storageRedundancy"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SqlDatabase.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SqlDatabase resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SqlDatabase is deprecated: Version v20200401preview will be removed in the next major version of the provider. Upgrade to version v20210301 or later.");
        return new SqlDatabase(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SqlDatabase.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SqlDatabase.__pulumiType;
    }
}
exports.SqlDatabase = SqlDatabase;
/** @internal */
SqlDatabase.__pulumiType = 'azure-native:synapse/v20200401preview:SqlDatabase';
