// @pulumi/azure-native/azurestack/v20160101/registration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Registration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Registration information.
 *
 * @deprecated Version v20160101 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.
 */
class Registration extends pulumi.CustomResource {
    /**
     * Create a Registration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160101 will be removed in the next major version of the provider. Upgrade to version v20170601 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Registration is deprecated: Version v20160101 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.registrationToken === undefined) && !opts.urn) {
                throw new Error("Missing required property 'registrationToken'");
            }
            if ((!args || args.resourceGroup === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroup'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["registrationName"] = args ? args.registrationName : undefined;
            resourceInputs["registrationToken"] = args ? args.registrationToken : undefined;
            resourceInputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            resourceInputs["billingModel"] = undefined /*out*/;
            resourceInputs["cloudId"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["objectId"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["billingModel"] = undefined /*out*/;
            resourceInputs["cloudId"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["objectId"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:azurestack:Registration" }, { type: "azure-native:azurestack/v20170601:Registration" }, { type: "azure-native:azurestack/v20200601preview:Registration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Registration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Registration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Registration is deprecated: Version v20160101 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        return new Registration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Registration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Registration.__pulumiType;
    }
}
exports.Registration = Registration;
/** @internal */
Registration.__pulumiType = 'azure-native:azurestack/v20160101:Registration';
