// @pulumi/azure-native/azurearcdata/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220301preview = exports.v20211101 = exports.v20210801 = exports.v20210701preview = exports.v20210601preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./activeDirectoryConnector"), exports);
__exportStar(require("./dataController"), exports);
__exportStar(require("./getActiveDirectoryConnector"), exports);
__exportStar(require("./getDataController"), exports);
__exportStar(require("./getPostgresInstance"), exports);
__exportStar(require("./getSqlManagedInstance"), exports);
__exportStar(require("./getSqlServerInstance"), exports);
__exportStar(require("./postgresInstance"), exports);
__exportStar(require("./sqlManagedInstance"), exports);
__exportStar(require("./sqlServerInstance"), exports);
// Export enums:
__exportStar(require("../types/enums/azurearcdata"), exports);
// Export sub-modules:
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20210701preview = require("./v20210701preview");
exports.v20210701preview = v20210701preview;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
const v20211101 = require("./v20211101");
exports.v20211101 = v20211101;
const v20220301preview = require("./v20220301preview");
exports.v20220301preview = v20220301preview;
// Import resources to register:
const activeDirectoryConnector_1 = require("./activeDirectoryConnector");
const dataController_1 = require("./dataController");
const postgresInstance_1 = require("./postgresInstance");
const sqlManagedInstance_1 = require("./sqlManagedInstance");
const sqlServerInstance_1 = require("./sqlServerInstance");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:azurearcdata:ActiveDirectoryConnector":
                return new activeDirectoryConnector_1.ActiveDirectoryConnector(name, undefined, { urn });
            case "azure-native:azurearcdata:DataController":
                return new dataController_1.DataController(name, undefined, { urn });
            case "azure-native:azurearcdata:PostgresInstance":
                return new postgresInstance_1.PostgresInstance(name, undefined, { urn });
            case "azure-native:azurearcdata:SqlManagedInstance":
                return new sqlManagedInstance_1.SqlManagedInstance(name, undefined, { urn });
            case "azure-native:azurearcdata:SqlServerInstance":
                return new sqlServerInstance_1.SqlServerInstance(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "azurearcdata", _module);
