// @pulumi/tls/certRequest.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
class CertRequest extends pulumi.CustomResource {
    constructor(name, argsOrState, opts) {
        let inputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            inputs["certRequestPem"] = state ? state.certRequestPem : undefined;
            inputs["dnsNames"] = state ? state.dnsNames : undefined;
            inputs["ipAddresses"] = state ? state.ipAddresses : undefined;
            inputs["keyAlgorithm"] = state ? state.keyAlgorithm : undefined;
            inputs["privateKeyPem"] = state ? state.privateKeyPem : undefined;
            inputs["subjects"] = state ? state.subjects : undefined;
            inputs["uris"] = state ? state.uris : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.keyAlgorithm === undefined) && !opts.urn) {
                throw new Error("Missing required property 'keyAlgorithm'");
            }
            if ((!args || args.privateKeyPem === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateKeyPem'");
            }
            if ((!args || args.subjects === undefined) && !opts.urn) {
                throw new Error("Missing required property 'subjects'");
            }
            inputs["dnsNames"] = args ? args.dnsNames : undefined;
            inputs["ipAddresses"] = args ? args.ipAddresses : undefined;
            inputs["keyAlgorithm"] = args ? args.keyAlgorithm : undefined;
            inputs["privateKeyPem"] = args ? args.privateKeyPem : undefined;
            inputs["subjects"] = args ? args.subjects : undefined;
            inputs["uris"] = args ? args.uris : undefined;
            inputs["certRequestPem"] = undefined /*out*/;
        }
        if (!opts.version) {
            opts = pulumi.mergeOptions(opts, { version: utilities.getVersion() });
        }
        super(CertRequest.__pulumiType, name, inputs, opts);
    }
    /**
     * Get an existing CertRequest resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CertRequest(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of CertRequest.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CertRequest.__pulumiType;
    }
}
exports.CertRequest = CertRequest;
/** @internal */
CertRequest.__pulumiType = 'tls:index/certRequest:CertRequest';
