// @pulumi/azure-native/web/v20200901/webAppDiagnosticLogsConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebAppDiagnosticLogsConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const types_1 = require("../../types");
const utilities = require("../../utilities");
/**
 * Configuration of App Service site logs.
 */
class WebAppDiagnosticLogsConfiguration extends pulumi.CustomResource {
    /**
     * Create a WebAppDiagnosticLogsConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["applicationLogs"] = args ? (args.applicationLogs ? pulumi.output(args.applicationLogs).apply(types_1.input.web.v20200901.applicationLogsConfigArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["detailedErrorMessages"] = args ? args.detailedErrorMessages : undefined;
            resourceInputs["failedRequestsTracing"] = args ? args.failedRequestsTracing : undefined;
            resourceInputs["httpLogs"] = args ? args.httpLogs : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["applicationLogs"] = undefined /*out*/;
            resourceInputs["detailedErrorMessages"] = undefined /*out*/;
            resourceInputs["failedRequestsTracing"] = undefined /*out*/;
            resourceInputs["httpLogs"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20150801:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20160801:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20180201:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20181101:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20190801:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20200601:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20201001:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20201201:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20210101:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20210115:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20210201:WebAppDiagnosticLogsConfiguration" }, { type: "azure-native:web/v20210301:WebAppDiagnosticLogsConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WebAppDiagnosticLogsConfiguration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing WebAppDiagnosticLogsConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new WebAppDiagnosticLogsConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of WebAppDiagnosticLogsConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WebAppDiagnosticLogsConfiguration.__pulumiType;
    }
}
exports.WebAppDiagnosticLogsConfiguration = WebAppDiagnosticLogsConfiguration;
/** @internal */
WebAppDiagnosticLogsConfiguration.__pulumiType = 'azure-native:web/v20200901:WebAppDiagnosticLogsConfiguration';
