// @pulumi/azure-native/types/enums/maintenance/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceScope = exports.v20210901preview = exports.v20210501 = exports.v20210401preview = exports.v20200701preview = exports.v20200401 = exports.v20180601preview = void 0;
// Export sub-modules:
const v20180601preview = require("./v20180601preview");
exports.v20180601preview = v20180601preview;
const v20200401 = require("./v20200401");
exports.v20200401 = v20200401;
const v20200701preview = require("./v20200701preview");
exports.v20200701preview = v20200701preview;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
exports.MaintenanceScope = {
    All: "All",
    Host: "Host",
    Resource: "Resource",
    InResource: "InResource",
};
