// @pulumi/azure-native/types/enums/documentdb/v20211015/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerType = exports.TriggerOperation = exports.SpatialType = exports.ServerVersion = exports.RoleDefinitionType = exports.RestoreMode = exports.ResourceIdentityType = exports.PublicNetworkAccess = exports.PartitionKind = exports.NetworkAclBypass = exports.ManagedCassandraResourceIdentityType = exports.ManagedCassandraProvisioningState = exports.IndexingMode = exports.IndexKind = exports.DefaultConsistencyLevel = exports.DatabaseAccountOfferType = exports.DatabaseAccountKind = exports.DataType = exports.CreateMode = exports.ConnectorOffer = exports.ConflictResolutionMode = exports.CompositePathSortOrder = exports.BackupStorageRedundancy = exports.BackupPolicyType = exports.BackupPolicyMigrationStatus = exports.AuthenticationMethod = exports.AnalyticalStorageSchemaType = void 0;
exports.AnalyticalStorageSchemaType = {
    WellDefined: "WellDefined",
    FullFidelity: "FullFidelity",
};
exports.AuthenticationMethod = {
    None: "None",
    Cassandra: "Cassandra",
};
exports.BackupPolicyMigrationStatus = {
    Invalid: "Invalid",
    InProgress: "InProgress",
    Completed: "Completed",
    Failed: "Failed",
};
exports.BackupPolicyType = {
    Periodic: "Periodic",
    Continuous: "Continuous",
};
exports.BackupStorageRedundancy = {
    Geo: "Geo",
    Local: "Local",
    Zone: "Zone",
};
exports.CompositePathSortOrder = {
    Ascending: "ascending",
    Descending: "descending",
};
exports.ConflictResolutionMode = {
    LastWriterWins: "LastWriterWins",
    Custom: "Custom",
};
exports.ConnectorOffer = {
    Small: "Small",
};
exports.CreateMode = {
    Default: "Default",
    Restore: "Restore",
};
exports.DataType = {
    String: "String",
    Number: "Number",
    Point: "Point",
    Polygon: "Polygon",
    LineString: "LineString",
    MultiPolygon: "MultiPolygon",
};
exports.DatabaseAccountKind = {
    GlobalDocumentDB: "GlobalDocumentDB",
    MongoDB: "MongoDB",
    Parse: "Parse",
};
exports.DatabaseAccountOfferType = {
    Standard: "Standard",
};
exports.DefaultConsistencyLevel = {
    Eventual: "Eventual",
    Session: "Session",
    BoundedStaleness: "BoundedStaleness",
    Strong: "Strong",
    ConsistentPrefix: "ConsistentPrefix",
};
exports.IndexKind = {
    Hash: "Hash",
    Range: "Range",
    Spatial: "Spatial",
};
exports.IndexingMode = {
    Consistent: "consistent",
    Lazy: "lazy",
    None: "none",
};
exports.ManagedCassandraProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
};
exports.ManagedCassandraResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
};
exports.NetworkAclBypass = {
    None: "None",
    AzureServices: "AzureServices",
};
exports.PartitionKind = {
    Hash: "Hash",
    Range: "Range",
    MultiHash: "MultiHash",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
    None: "None",
};
exports.RestoreMode = {
    PointInTime: "PointInTime",
};
exports.RoleDefinitionType = {
    BuiltInRole: "BuiltInRole",
    CustomRole: "CustomRole",
};
exports.ServerVersion = {
    ServerVersion_3_2: "3.2",
    ServerVersion_3_6: "3.6",
    ServerVersion_4_0: "4.0",
    ServerVersion_4_2: "4.2",
};
exports.SpatialType = {
    Point: "Point",
    LineString: "LineString",
    Polygon: "Polygon",
    MultiPolygon: "MultiPolygon",
};
exports.TriggerOperation = {
    All: "All",
    Create: "Create",
    Update: "Update",
    Delete: "Delete",
    Replace: "Replace",
};
exports.TriggerType = {
    Pre: "Pre",
    Post: "Post",
};
