// @pulumi/azure-native/authorization/v20150101/managementLock.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagementLock = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Management lock information.
 *
 * @deprecated Version v20150101 will be removed in the next major version of the provider. Upgrade to version v20170401 or later.
 */
class ManagementLock extends pulumi.CustomResource {
    /**
     * Create a ManagementLock resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150101 will be removed in the next major version of the provider. Upgrade to version v20170401 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ManagementLock is deprecated: Version v20150101 will be removed in the next major version of the provider. Upgrade to version v20170401 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["level"] = args ? args.level : undefined;
            resourceInputs["lockName"] = args ? args.lockName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["notes"] = args ? args.notes : undefined;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["level"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["notes"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:authorization:ManagementLock" }, { type: "azure-native:authorization/v20160901:ManagementLock" }, { type: "azure-native:authorization/v20170401:ManagementLock" }, { type: "azure-native:authorization/v20200501:ManagementLock" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagementLock.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ManagementLock resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ManagementLock is deprecated: Version v20150101 will be removed in the next major version of the provider. Upgrade to version v20170401 or later.");
        return new ManagementLock(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagementLock.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagementLock.__pulumiType;
    }
}
exports.ManagementLock = ManagementLock;
/** @internal */
ManagementLock.__pulumiType = 'azure-native:authorization/v20150101:ManagementLock';
