// @pulumi/azure-native/storsimple/v20161001/accessControlRecord.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessControlRecord = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The access control record
 *
 * @deprecated Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.
 */
class AccessControlRecord extends pulumi.CustomResource {
    /**
     * Create a AccessControlRecord resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("AccessControlRecord is deprecated: Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.initiatorName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'initiatorName'");
            }
            if ((!args || args.managerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accessControlRecordName"] = args ? args.accessControlRecordName : undefined;
            resourceInputs["initiatorName"] = args ? args.initiatorName : undefined;
            resourceInputs["managerName"] = args ? args.managerName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["initiatorName"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storsimple:AccessControlRecord" }, { type: "azure-native:storsimple/v20170601:AccessControlRecord" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AccessControlRecord.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing AccessControlRecord resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("AccessControlRecord is deprecated: Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        return new AccessControlRecord(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AccessControlRecord.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccessControlRecord.__pulumiType;
    }
}
exports.AccessControlRecord = AccessControlRecord;
/** @internal */
AccessControlRecord.__pulumiType = 'azure-native:storsimple/v20161001:AccessControlRecord';
