// @pulumi/azure-native/storsimple/v20161001/backupScheduleGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupScheduleGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The Backup Schedule Group
 *
 * @deprecated Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.
 */
class BackupScheduleGroup extends pulumi.CustomResource {
    /**
     * Create a BackupScheduleGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("BackupScheduleGroup is deprecated: Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.deviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if ((!args || args.managerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.startTime === undefined) && !opts.urn) {
                throw new Error("Missing required property 'startTime'");
            }
            resourceInputs["deviceName"] = args ? args.deviceName : undefined;
            resourceInputs["managerName"] = args ? args.managerName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scheduleGroupName"] = args ? args.scheduleGroupName : undefined;
            resourceInputs["startTime"] = args ? args.startTime : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["startTime"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BackupScheduleGroup.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing BackupScheduleGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("BackupScheduleGroup is deprecated: Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        return new BackupScheduleGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of BackupScheduleGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BackupScheduleGroup.__pulumiType;
    }
}
exports.BackupScheduleGroup = BackupScheduleGroup;
/** @internal */
BackupScheduleGroup.__pulumiType = 'azure-native:storsimple/v20161001:BackupScheduleGroup';
