// @pulumi/azure-native/storsimple/v20161001/iscsiDisk.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IscsiDisk = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The iSCSI disk.
 *
 * @deprecated Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.
 */
class IscsiDisk extends pulumi.CustomResource {
    /**
     * Create a IscsiDisk resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("IscsiDisk is deprecated: Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.accessControlRecords === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accessControlRecords'");
            }
            if ((!args || args.dataPolicy === undefined) && !opts.urn) {
                throw new Error("Missing required property 'dataPolicy'");
            }
            if ((!args || args.deviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'deviceName'");
            }
            if ((!args || args.diskStatus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'diskStatus'");
            }
            if ((!args || args.iscsiServerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iscsiServerName'");
            }
            if ((!args || args.managerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managerName'");
            }
            if ((!args || args.monitoringStatus === undefined) && !opts.urn) {
                throw new Error("Missing required property 'monitoringStatus'");
            }
            if ((!args || args.provisionedCapacityInBytes === undefined) && !opts.urn) {
                throw new Error("Missing required property 'provisionedCapacityInBytes'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["accessControlRecords"] = args ? args.accessControlRecords : undefined;
            resourceInputs["dataPolicy"] = args ? args.dataPolicy : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["deviceName"] = args ? args.deviceName : undefined;
            resourceInputs["diskName"] = args ? args.diskName : undefined;
            resourceInputs["diskStatus"] = args ? args.diskStatus : undefined;
            resourceInputs["iscsiServerName"] = args ? args.iscsiServerName : undefined;
            resourceInputs["managerName"] = args ? args.managerName : undefined;
            resourceInputs["monitoringStatus"] = args ? args.monitoringStatus : undefined;
            resourceInputs["provisionedCapacityInBytes"] = args ? args.provisionedCapacityInBytes : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["localUsedCapacityInBytes"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["usedCapacityInBytes"] = undefined /*out*/;
        }
        else {
            resourceInputs["accessControlRecords"] = undefined /*out*/;
            resourceInputs["dataPolicy"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["diskStatus"] = undefined /*out*/;
            resourceInputs["localUsedCapacityInBytes"] = undefined /*out*/;
            resourceInputs["monitoringStatus"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisionedCapacityInBytes"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["usedCapacityInBytes"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(IscsiDisk.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing IscsiDisk resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("IscsiDisk is deprecated: Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        return new IscsiDisk(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IscsiDisk.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IscsiDisk.__pulumiType;
    }
}
exports.IscsiDisk = IscsiDisk;
/** @internal */
IscsiDisk.__pulumiType = 'azure-native:storsimple/v20161001:IscsiDisk';
