// @pulumi/azure-native/storsimple/v20161001/managerExtendedInfo.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagerExtendedInfo = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The extended info of the manager.
 *
 * @deprecated Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.
 */
class ManagerExtendedInfo extends pulumi.CustomResource {
    /**
     * Create a ManagerExtendedInfo resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ManagerExtendedInfo is deprecated: Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.algorithm === undefined) && !opts.urn) {
                throw new Error("Missing required property 'algorithm'");
            }
            if ((!args || args.integrityKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'integrityKey'");
            }
            if ((!args || args.managerName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'managerName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["algorithm"] = args ? args.algorithm : undefined;
            resourceInputs["encryptionKey"] = args ? args.encryptionKey : undefined;
            resourceInputs["encryptionKeyThumbprint"] = args ? args.encryptionKeyThumbprint : undefined;
            resourceInputs["integrityKey"] = args ? args.integrityKey : undefined;
            resourceInputs["managerName"] = args ? args.managerName : undefined;
            resourceInputs["portalCertificateThumbprint"] = args ? args.portalCertificateThumbprint : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["algorithm"] = undefined /*out*/;
            resourceInputs["encryptionKey"] = undefined /*out*/;
            resourceInputs["encryptionKeyThumbprint"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["integrityKey"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["portalCertificateThumbprint"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:storsimple:ManagerExtendedInfo" }, { type: "azure-native:storsimple/v20170601:ManagerExtendedInfo" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagerExtendedInfo.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ManagerExtendedInfo resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ManagerExtendedInfo is deprecated: Version v20161001 will be removed in the next major version of the provider. Upgrade to version v20170601 or later.");
        return new ManagerExtendedInfo(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagerExtendedInfo.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagerExtendedInfo.__pulumiType;
    }
}
exports.ManagerExtendedInfo = ManagerExtendedInfo;
/** @internal */
ManagerExtendedInfo.__pulumiType = 'azure-native:storsimple/v20161001:ManagerExtendedInfo';
