// @pulumi/azure-native/types/enums/sql/v20140401/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransparentDataEncryptionStatus = exports.ServiceObjectiveName = exports.ServerVersion = exports.SecurityAlertPolicyUseServerDefault = exports.SecurityAlertPolicyState = exports.SecurityAlertPolicyEmailAccountAdmins = exports.SampleName = exports.ReadScale = exports.GeoBackupPolicyState = exports.ElasticPoolEdition = exports.DatabaseEdition = exports.DataMaskingState = exports.CreateMode = exports.AutoExecuteStatus = exports.AdministratorType = void 0;
exports.AdministratorType = {
    ActiveDirectory: "ActiveDirectory",
};
exports.AutoExecuteStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
    Default: "Default",
};
exports.CreateMode = {
    Copy: "Copy",
    Default: "Default",
    NonReadableSecondary: "NonReadableSecondary",
    OnlineSecondary: "OnlineSecondary",
    PointInTimeRestore: "PointInTimeRestore",
    Recovery: "Recovery",
    Restore: "Restore",
    RestoreLongTermRetentionBackup: "RestoreLongTermRetentionBackup",
};
exports.DataMaskingState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.DatabaseEdition = {
    Web: "Web",
    Business: "Business",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
    PremiumRS: "PremiumRS",
    Free: "Free",
    Stretch: "Stretch",
    DataWarehouse: "DataWarehouse",
    System: "System",
    System2: "System2",
    GeneralPurpose: "GeneralPurpose",
    BusinessCritical: "BusinessCritical",
    Hyperscale: "Hyperscale",
};
exports.ElasticPoolEdition = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
    GeneralPurpose: "GeneralPurpose",
    BusinessCritical: "BusinessCritical",
};
exports.GeoBackupPolicyState = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.ReadScale = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SampleName = {
    AdventureWorksLT: "AdventureWorksLT",
};
exports.SecurityAlertPolicyEmailAccountAdmins = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SecurityAlertPolicyState = {
    New: "New",
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.SecurityAlertPolicyUseServerDefault = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ServerVersion = {
    ServerVersion_2_0: "2.0",
    ServerVersion_12_0: "12.0",
};
exports.ServiceObjectiveName = {
    System: "System",
    System0: "System0",
    System1: "System1",
    System2: "System2",
    System3: "System3",
    System4: "System4",
    System2L: "System2L",
    System3L: "System3L",
    System4L: "System4L",
    Free: "Free",
    Basic: "Basic",
    S0: "S0",
    S1: "S1",
    S2: "S2",
    S3: "S3",
    S4: "S4",
    S6: "S6",
    S7: "S7",
    S9: "S9",
    S12: "S12",
    P1: "P1",
    P2: "P2",
    P3: "P3",
    P4: "P4",
    P6: "P6",
    P11: "P11",
    P15: "P15",
    PRS1: "PRS1",
    PRS2: "PRS2",
    PRS4: "PRS4",
    PRS6: "PRS6",
    DW100: "DW100",
    DW200: "DW200",
    DW300: "DW300",
    DW400: "DW400",
    DW500: "DW500",
    DW600: "DW600",
    DW1000: "DW1000",
    DW1200: "DW1200",
    DW1000c: "DW1000c",
    DW1500: "DW1500",
    DW1500c: "DW1500c",
    DW2000: "DW2000",
    DW2000c: "DW2000c",
    DW3000: "DW3000",
    DW2500c: "DW2500c",
    DW3000c: "DW3000c",
    DW6000: "DW6000",
    DW5000c: "DW5000c",
    DW6000c: "DW6000c",
    DW7500c: "DW7500c",
    DW10000c: "DW10000c",
    DW15000c: "DW15000c",
    DW30000c: "DW30000c",
    DS100: "DS100",
    DS200: "DS200",
    DS300: "DS300",
    DS400: "DS400",
    DS500: "DS500",
    DS600: "DS600",
    DS1000: "DS1000",
    DS1200: "DS1200",
    DS1500: "DS1500",
    DS2000: "DS2000",
    ElasticPool: "ElasticPool",
};
exports.TransparentDataEncryptionStatus = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
