// @pulumi/azure-native/hybridcompute/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220310 = exports.v20211210preview = exports.v20210610preview = exports.v20210520 = exports.v20210517preview = exports.v20210422preview = exports.v20210325preview = exports.v20210128preview = exports.v20200815preview = exports.v20200802 = exports.v20200730preview = exports.v20191212 = exports.v20190802preview = exports.v20190318preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getMachine"), exports);
__exportStar(require("./getMachineExtension"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getPrivateLinkScope"), exports);
__exportStar(require("./getPrivateLinkScopedResource"), exports);
__exportStar(require("./machine"), exports);
__exportStar(require("./machineExtension"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./privateLinkScope"), exports);
__exportStar(require("./privateLinkScopedResource"), exports);
// Export enums:
__exportStar(require("../types/enums/hybridcompute"), exports);
// Export sub-modules:
const v20190318preview = require("./v20190318preview");
exports.v20190318preview = v20190318preview;
const v20190802preview = require("./v20190802preview");
exports.v20190802preview = v20190802preview;
const v20191212 = require("./v20191212");
exports.v20191212 = v20191212;
const v20200730preview = require("./v20200730preview");
exports.v20200730preview = v20200730preview;
const v20200802 = require("./v20200802");
exports.v20200802 = v20200802;
const v20200815preview = require("./v20200815preview");
exports.v20200815preview = v20200815preview;
const v20210128preview = require("./v20210128preview");
exports.v20210128preview = v20210128preview;
const v20210325preview = require("./v20210325preview");
exports.v20210325preview = v20210325preview;
const v20210422preview = require("./v20210422preview");
exports.v20210422preview = v20210422preview;
const v20210517preview = require("./v20210517preview");
exports.v20210517preview = v20210517preview;
const v20210520 = require("./v20210520");
exports.v20210520 = v20210520;
const v20210610preview = require("./v20210610preview");
exports.v20210610preview = v20210610preview;
const v20211210preview = require("./v20211210preview");
exports.v20211210preview = v20211210preview;
const v20220310 = require("./v20220310");
exports.v20220310 = v20220310;
// Import resources to register:
const machine_1 = require("./machine");
const machineExtension_1 = require("./machineExtension");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const privateLinkScope_1 = require("./privateLinkScope");
const privateLinkScopedResource_1 = require("./privateLinkScopedResource");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:hybridcompute:Machine":
                return new machine_1.Machine(name, undefined, { urn });
            case "azure-native:hybridcompute:MachineExtension":
                return new machineExtension_1.MachineExtension(name, undefined, { urn });
            case "azure-native:hybridcompute:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:hybridcompute:PrivateLinkScope":
                return new privateLinkScope_1.PrivateLinkScope(name, undefined, { urn });
            case "azure-native:hybridcompute:PrivateLinkScopedResource":
                return new privateLinkScopedResource_1.PrivateLinkScopedResource(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "hybridcompute", _module);
