// @pulumi/azure-native/security/v20150601preview/adaptiveApplicationControl.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdaptiveApplicationControl = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * @deprecated Version v20150601preview will be removed in the next major version of the provider. Upgrade to version v20170801preview or later.
 */
class AdaptiveApplicationControl extends pulumi.CustomResource {
    /**
     * Create a AdaptiveApplicationControl resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150601preview will be removed in the next major version of the provider. Upgrade to version v20170801preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("AdaptiveApplicationControl is deprecated: Version v20150601preview will be removed in the next major version of the provider. Upgrade to version v20170801preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.ascLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'ascLocation'");
            }
            resourceInputs["ascLocation"] = args ? args.ascLocation : undefined;
            resourceInputs["enforcementMode"] = args ? args.enforcementMode : undefined;
            resourceInputs["groupName"] = args ? args.groupName : undefined;
            resourceInputs["pathRecommendations"] = args ? args.pathRecommendations : undefined;
            resourceInputs["protectionMode"] = args ? args.protectionMode : undefined;
            resourceInputs["vmRecommendations"] = args ? args.vmRecommendations : undefined;
            resourceInputs["configurationStatus"] = undefined /*out*/;
            resourceInputs["issues"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["recommendationStatus"] = undefined /*out*/;
            resourceInputs["sourceSystem"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["configurationStatus"] = undefined /*out*/;
            resourceInputs["enforcementMode"] = undefined /*out*/;
            resourceInputs["issues"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["pathRecommendations"] = undefined /*out*/;
            resourceInputs["protectionMode"] = undefined /*out*/;
            resourceInputs["recommendationStatus"] = undefined /*out*/;
            resourceInputs["sourceSystem"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vmRecommendations"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:security:AdaptiveApplicationControl" }, { type: "azure-native:security/v20200101:AdaptiveApplicationControl" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(AdaptiveApplicationControl.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing AdaptiveApplicationControl resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("AdaptiveApplicationControl is deprecated: Version v20150601preview will be removed in the next major version of the provider. Upgrade to version v20170801preview or later.");
        return new AdaptiveApplicationControl(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AdaptiveApplicationControl.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AdaptiveApplicationControl.__pulumiType;
    }
}
exports.AdaptiveApplicationControl = AdaptiveApplicationControl;
/** @internal */
AdaptiveApplicationControl.__pulumiType = 'azure-native:security/v20150601preview:AdaptiveApplicationControl';
