// @pulumi/azure-native/servicebus/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211101 = exports.v20210601preview = exports.v20210101preview = exports.v20180101preview = exports.v20170401 = exports.v20150801 = exports.v20140901 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./disasterRecoveryConfig"), exports);
__exportStar(require("./getDisasterRecoveryConfig"), exports);
__exportStar(require("./getMigrationConfig"), exports);
__exportStar(require("./getNamespace"), exports);
__exportStar(require("./getNamespaceAuthorizationRule"), exports);
__exportStar(require("./getNamespaceIpFilterRule"), exports);
__exportStar(require("./getNamespaceNetworkRuleSet"), exports);
__exportStar(require("./getNamespaceVirtualNetworkRule"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getQueue"), exports);
__exportStar(require("./getQueueAuthorizationRule"), exports);
__exportStar(require("./getRule"), exports);
__exportStar(require("./getSubscription"), exports);
__exportStar(require("./getTopic"), exports);
__exportStar(require("./getTopicAuthorizationRule"), exports);
__exportStar(require("./listDisasterRecoveryConfigKeys"), exports);
__exportStar(require("./listNamespaceKeys"), exports);
__exportStar(require("./listQueueKeys"), exports);
__exportStar(require("./listTopicKeys"), exports);
__exportStar(require("./migrationConfig"), exports);
__exportStar(require("./namespace"), exports);
__exportStar(require("./namespaceAuthorizationRule"), exports);
__exportStar(require("./namespaceIpFilterRule"), exports);
__exportStar(require("./namespaceNetworkRuleSet"), exports);
__exportStar(require("./namespaceVirtualNetworkRule"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./queue"), exports);
__exportStar(require("./queueAuthorizationRule"), exports);
__exportStar(require("./rule"), exports);
__exportStar(require("./subscription"), exports);
__exportStar(require("./topic"), exports);
__exportStar(require("./topicAuthorizationRule"), exports);
// Export enums:
__exportStar(require("../types/enums/servicebus"), exports);
// Export sub-modules:
const v20140901 = require("./v20140901");
exports.v20140901 = v20140901;
const v20150801 = require("./v20150801");
exports.v20150801 = v20150801;
const v20170401 = require("./v20170401");
exports.v20170401 = v20170401;
const v20180101preview = require("./v20180101preview");
exports.v20180101preview = v20180101preview;
const v20210101preview = require("./v20210101preview");
exports.v20210101preview = v20210101preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20211101 = require("./v20211101");
exports.v20211101 = v20211101;
// Import resources to register:
const disasterRecoveryConfig_1 = require("./disasterRecoveryConfig");
const migrationConfig_1 = require("./migrationConfig");
const namespace_1 = require("./namespace");
const namespaceAuthorizationRule_1 = require("./namespaceAuthorizationRule");
const namespaceIpFilterRule_1 = require("./namespaceIpFilterRule");
const namespaceNetworkRuleSet_1 = require("./namespaceNetworkRuleSet");
const namespaceVirtualNetworkRule_1 = require("./namespaceVirtualNetworkRule");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const queue_1 = require("./queue");
const queueAuthorizationRule_1 = require("./queueAuthorizationRule");
const rule_1 = require("./rule");
const subscription_1 = require("./subscription");
const topic_1 = require("./topic");
const topicAuthorizationRule_1 = require("./topicAuthorizationRule");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:servicebus:DisasterRecoveryConfig":
                return new disasterRecoveryConfig_1.DisasterRecoveryConfig(name, undefined, { urn });
            case "azure-native:servicebus:MigrationConfig":
                return new migrationConfig_1.MigrationConfig(name, undefined, { urn });
            case "azure-native:servicebus:Namespace":
                return new namespace_1.Namespace(name, undefined, { urn });
            case "azure-native:servicebus:NamespaceAuthorizationRule":
                return new namespaceAuthorizationRule_1.NamespaceAuthorizationRule(name, undefined, { urn });
            case "azure-native:servicebus:NamespaceIpFilterRule":
                return new namespaceIpFilterRule_1.NamespaceIpFilterRule(name, undefined, { urn });
            case "azure-native:servicebus:NamespaceNetworkRuleSet":
                return new namespaceNetworkRuleSet_1.NamespaceNetworkRuleSet(name, undefined, { urn });
            case "azure-native:servicebus:NamespaceVirtualNetworkRule":
                return new namespaceVirtualNetworkRule_1.NamespaceVirtualNetworkRule(name, undefined, { urn });
            case "azure-native:servicebus:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:servicebus:Queue":
                return new queue_1.Queue(name, undefined, { urn });
            case "azure-native:servicebus:QueueAuthorizationRule":
                return new queueAuthorizationRule_1.QueueAuthorizationRule(name, undefined, { urn });
            case "azure-native:servicebus:Rule":
                return new rule_1.Rule(name, undefined, { urn });
            case "azure-native:servicebus:Subscription":
                return new subscription_1.Subscription(name, undefined, { urn });
            case "azure-native:servicebus:Topic":
                return new topic_1.Topic(name, undefined, { urn });
            case "azure-native:servicebus:TopicAuthorizationRule":
                return new topicAuthorizationRule_1.TopicAuthorizationRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "servicebus", _module);
