// @pulumi/azure-native/machinelearning/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20191001 = exports.v20170101 = exports.v20160501preview = exports.v20160401 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./commitmentPlan"), exports);
__exportStar(require("./getCommitmentPlan"), exports);
__exportStar(require("./getWebService"), exports);
__exportStar(require("./getWorkspace"), exports);
__exportStar(require("./listWorkspaceKeys"), exports);
__exportStar(require("./webService"), exports);
__exportStar(require("./workspace"), exports);
// Export enums:
__exportStar(require("../types/enums/machinelearning"), exports);
// Export sub-modules:
const v20160401 = require("./v20160401");
exports.v20160401 = v20160401;
const v20160501preview = require("./v20160501preview");
exports.v20160501preview = v20160501preview;
const v20170101 = require("./v20170101");
exports.v20170101 = v20170101;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
// Import resources to register:
const commitmentPlan_1 = require("./commitmentPlan");
const webService_1 = require("./webService");
const workspace_1 = require("./workspace");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearning:CommitmentPlan":
                return new commitmentPlan_1.CommitmentPlan(name, undefined, { urn });
            case "azure-native:machinelearning:WebService":
                return new webService_1.WebService(name, undefined, { urn });
            case "azure-native:machinelearning:Workspace":
                return new workspace_1.Workspace(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearning", _module);
