// @pulumi/azure-native/servicefabric/v20191101preview/application.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const types_1 = require("../../types");
const utilities = require("../../utilities");
/**
 * The application resource.
 *
 * @deprecated Version v20191101preview will be removed in the next major version of the provider. Upgrade to version v20200101preview or later.
 */
class Application extends pulumi.CustomResource {
    /**
     * Create a Application resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20191101preview will be removed in the next major version of the provider. Upgrade to version v20200101preview or later. */
    constructor(name, args, opts) {
        var _a;
        pulumi.log.warn("Application is deprecated: Version v20191101preview will be removed in the next major version of the provider. Upgrade to version v20200101preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["applicationName"] = args ? args.applicationName : undefined;
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["managedIdentities"] = args ? args.managedIdentities : undefined;
            resourceInputs["maximumNodes"] = (_a = (args ? args.maximumNodes : undefined)) !== null && _a !== void 0 ? _a : 0;
            resourceInputs["metrics"] = args ? args.metrics : undefined;
            resourceInputs["minimumNodes"] = args ? args.minimumNodes : undefined;
            resourceInputs["parameters"] = args ? args.parameters : undefined;
            resourceInputs["removeApplicationCapacity"] = args ? args.removeApplicationCapacity : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["typeName"] = args ? args.typeName : undefined;
            resourceInputs["typeVersion"] = args ? args.typeVersion : undefined;
            resourceInputs["upgradePolicy"] = args ? (args.upgradePolicy ? pulumi.output(args.upgradePolicy).apply(types_1.input.servicefabric.v20191101preview.applicationUpgradePolicyArgsProvideDefaults) : undefined) : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["managedIdentities"] = undefined /*out*/;
            resourceInputs["maximumNodes"] = undefined /*out*/;
            resourceInputs["metrics"] = undefined /*out*/;
            resourceInputs["minimumNodes"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["parameters"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["removeApplicationCapacity"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["typeName"] = undefined /*out*/;
            resourceInputs["typeVersion"] = undefined /*out*/;
            resourceInputs["upgradePolicy"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:servicefabric:Application" }, { type: "azure-native:servicefabric/v20170701preview:Application" }, { type: "azure-native:servicefabric/v20190301:Application" }, { type: "azure-native:servicefabric/v20190301preview:Application" }, { type: "azure-native:servicefabric/v20190601preview:Application" }, { type: "azure-native:servicefabric/v20200301:Application" }, { type: "azure-native:servicefabric/v20201201preview:Application" }, { type: "azure-native:servicefabric/v20210601:Application" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Application.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Application is deprecated: Version v20191101preview will be removed in the next major version of the provider. Upgrade to version v20200101preview or later.");
        return new Application(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
}
exports.Application = Application;
/** @internal */
Application.__pulumiType = 'azure-native:servicefabric/v20191101preview:Application';
