// @pulumi/azure-native/types/enums/containerinstance/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Scheme = exports.ResourceIdentityType = exports.OperatingSystemTypes = exports.LogAnalyticsLogType = exports.GpuSku = exports.ContainerNetworkProtocol = exports.ContainerGroupSku = exports.ContainerGroupRestartPolicy = exports.ContainerGroupNetworkProtocol = exports.ContainerGroupIpAddressType = exports.v20211001 = exports.v20210901 = exports.v20210701 = exports.v20210301 = exports.v20201101 = exports.v20191201 = exports.v20181001 = exports.v20180901 = exports.v20180601 = exports.v20180401 = exports.v20180201preview = exports.v20171201preview = exports.v20171001preview = exports.v20170801preview = void 0;
// Export sub-modules:
const v20170801preview = require("./v20170801preview");
exports.v20170801preview = v20170801preview;
const v20171001preview = require("./v20171001preview");
exports.v20171001preview = v20171001preview;
const v20171201preview = require("./v20171201preview");
exports.v20171201preview = v20171201preview;
const v20180201preview = require("./v20180201preview");
exports.v20180201preview = v20180201preview;
const v20180401 = require("./v20180401");
exports.v20180401 = v20180401;
const v20180601 = require("./v20180601");
exports.v20180601 = v20180601;
const v20180901 = require("./v20180901");
exports.v20180901 = v20180901;
const v20181001 = require("./v20181001");
exports.v20181001 = v20181001;
const v20191201 = require("./v20191201");
exports.v20191201 = v20191201;
const v20201101 = require("./v20201101");
exports.v20201101 = v20201101;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
const v20210901 = require("./v20210901");
exports.v20210901 = v20210901;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
exports.ContainerGroupIpAddressType = {
    Public: "Public",
    Private: "Private",
};
exports.ContainerGroupNetworkProtocol = {
    TCP: "TCP",
    UDP: "UDP",
};
exports.ContainerGroupRestartPolicy = {
    Always: "Always",
    OnFailure: "OnFailure",
    Never: "Never",
};
exports.ContainerGroupSku = {
    Standard: "Standard",
    Dedicated: "Dedicated",
};
exports.ContainerNetworkProtocol = {
    TCP: "TCP",
    UDP: "UDP",
};
exports.GpuSku = {
    K80: "K80",
    P100: "P100",
    V100: "V100",
};
exports.LogAnalyticsLogType = {
    ContainerInsights: "ContainerInsights",
    ContainerInstanceLogs: "ContainerInstanceLogs",
};
exports.OperatingSystemTypes = {
    Windows: "Windows",
    Linux: "Linux",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.Scheme = {
    Http: "http",
    Https: "https",
};
