// @pulumi/azure-native/synapse/v20190601preview/integrationRuntime.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntegrationRuntime = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Integration runtime resource type.
 *
 * @deprecated Version v20190601preview will be removed in the next major version of the provider. Upgrade to version v20210301 or later.
 */
class IntegrationRuntime extends pulumi.CustomResource {
    /**
     * Create a IntegrationRuntime resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20190601preview will be removed in the next major version of the provider. Upgrade to version v20210301 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("IntegrationRuntime is deprecated: Version v20190601preview will be removed in the next major version of the provider. Upgrade to version v20210301 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.properties === undefined) && !opts.urn) {
                throw new Error("Missing required property 'properties'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["integrationRuntimeName"] = args ? args.integrationRuntimeName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:synapse:IntegrationRuntime" }, { type: "azure-native:synapse/v20201201:IntegrationRuntime" }, { type: "azure-native:synapse/v20210301:IntegrationRuntime" }, { type: "azure-native:synapse/v20210401preview:IntegrationRuntime" }, { type: "azure-native:synapse/v20210501:IntegrationRuntime" }, { type: "azure-native:synapse/v20210601:IntegrationRuntime" }, { type: "azure-native:synapse/v20210601preview:IntegrationRuntime" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IntegrationRuntime.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing IntegrationRuntime resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("IntegrationRuntime is deprecated: Version v20190601preview will be removed in the next major version of the provider. Upgrade to version v20210301 or later.");
        return new IntegrationRuntime(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IntegrationRuntime.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IntegrationRuntime.__pulumiType;
    }
}
exports.IntegrationRuntime = IntegrationRuntime;
/** @internal */
IntegrationRuntime.__pulumiType = 'azure-native:synapse/v20190601preview:IntegrationRuntime';
