// @pulumi/azure-native/kusto/v20190121/dataConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Class representing an data connection.
 *
 * @deprecated Please use one of the variants: EventGridDataConnection, EventHubDataConnection.
 */
class DataConnection extends pulumi.CustomResource {
    /**
     * Create a DataConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Please use one of the variants: EventGridDataConnection, EventHubDataConnection. */
    constructor(name, args, opts) {
        pulumi.log.warn("DataConnection is deprecated: Please use one of the variants: EventGridDataConnection, EventHubDataConnection.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.clusterName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if ((!args || args.databaseName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'databaseName'");
            }
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["clusterName"] = args ? args.clusterName : undefined;
            resourceInputs["dataConnectionName"] = args ? args.dataConnectionName : undefined;
            resourceInputs["databaseName"] = args ? args.databaseName : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:kusto:DataConnection" }, { type: "azure-native:kusto/v20190515:DataConnection" }, { type: "azure-native:kusto/v20190907:DataConnection" }, { type: "azure-native:kusto/v20191109:DataConnection" }, { type: "azure-native:kusto/v20200215:DataConnection" }, { type: "azure-native:kusto/v20200614:DataConnection" }, { type: "azure-native:kusto/v20200918:DataConnection" }, { type: "azure-native:kusto/v20210101:DataConnection" }, { type: "azure-native:kusto/v20210827:DataConnection" }, { type: "azure-native:kusto/v20220201:DataConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DataConnection.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DataConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("DataConnection is deprecated: Please use one of the variants: EventGridDataConnection, EventHubDataConnection.");
        return new DataConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DataConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataConnection.__pulumiType;
    }
}
exports.DataConnection = DataConnection;
/** @internal */
DataConnection.__pulumiType = 'azure-native:kusto/v20190121:DataConnection';
