// @pulumi/azure-native/resources/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220201 = exports.v20210501 = exports.v20210401 = exports.v20210301preview = exports.v20210101 = exports.v20201001 = exports.v20200801 = exports.v20200601 = exports.v20191001preview = exports.v20191001 = exports.v20190801 = exports.v20190701 = exports.v20190601preview = exports.v20190510 = exports.v20190501 = exports.v20190301 = exports.v20180501 = exports.v20180201 = exports.v20170510 = exports.v20160901 = exports.v20160701 = exports.v20160201 = exports.v20151101 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./azureCliScript"), exports);
__exportStar(require("./azurePowerShellScript"), exports);
__exportStar(require("./deployment"), exports);
__exportStar(require("./deploymentAtManagementGroupScope"), exports);
__exportStar(require("./deploymentAtScope"), exports);
__exportStar(require("./deploymentAtSubscriptionScope"), exports);
__exportStar(require("./deploymentAtTenantScope"), exports);
__exportStar(require("./deploymentScript"), exports);
__exportStar(require("./getAzureCliScript"), exports);
__exportStar(require("./getAzurePowerShellScript"), exports);
__exportStar(require("./getDeployment"), exports);
__exportStar(require("./getDeploymentAtManagementGroupScope"), exports);
__exportStar(require("./getDeploymentAtScope"), exports);
__exportStar(require("./getDeploymentAtSubscriptionScope"), exports);
__exportStar(require("./getDeploymentAtTenantScope"), exports);
__exportStar(require("./getDeploymentScript"), exports);
__exportStar(require("./getResource"), exports);
__exportStar(require("./getResourceGroup"), exports);
__exportStar(require("./getTagAtScope"), exports);
__exportStar(require("./getTemplateSpec"), exports);
__exportStar(require("./getTemplateSpecVersion"), exports);
__exportStar(require("./resource"), exports);
__exportStar(require("./resourceGroup"), exports);
__exportStar(require("./tagAtScope"), exports);
__exportStar(require("./templateSpec"), exports);
__exportStar(require("./templateSpecVersion"), exports);
// Export enums:
__exportStar(require("../types/enums/resources"), exports);
// Export sub-modules:
const v20151101 = require("./v20151101");
exports.v20151101 = v20151101;
const v20160201 = require("./v20160201");
exports.v20160201 = v20160201;
const v20160701 = require("./v20160701");
exports.v20160701 = v20160701;
const v20160901 = require("./v20160901");
exports.v20160901 = v20160901;
const v20170510 = require("./v20170510");
exports.v20170510 = v20170510;
const v20180201 = require("./v20180201");
exports.v20180201 = v20180201;
const v20180501 = require("./v20180501");
exports.v20180501 = v20180501;
const v20190301 = require("./v20190301");
exports.v20190301 = v20190301;
const v20190501 = require("./v20190501");
exports.v20190501 = v20190501;
const v20190510 = require("./v20190510");
exports.v20190510 = v20190510;
const v20190601preview = require("./v20190601preview");
exports.v20190601preview = v20190601preview;
const v20190701 = require("./v20190701");
exports.v20190701 = v20190701;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191001preview = require("./v20191001preview");
exports.v20191001preview = v20191001preview;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200801 = require("./v20200801");
exports.v20200801 = v20200801;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210301preview = require("./v20210301preview");
exports.v20210301preview = v20210301preview;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
const v20220201 = require("./v20220201");
exports.v20220201 = v20220201;
// Import resources to register:
const azureCliScript_1 = require("./azureCliScript");
const azurePowerShellScript_1 = require("./azurePowerShellScript");
const deployment_1 = require("./deployment");
const deploymentAtManagementGroupScope_1 = require("./deploymentAtManagementGroupScope");
const deploymentAtScope_1 = require("./deploymentAtScope");
const deploymentAtSubscriptionScope_1 = require("./deploymentAtSubscriptionScope");
const deploymentAtTenantScope_1 = require("./deploymentAtTenantScope");
const deploymentScript_1 = require("./deploymentScript");
const resource_1 = require("./resource");
const resourceGroup_1 = require("./resourceGroup");
const tagAtScope_1 = require("./tagAtScope");
const templateSpec_1 = require("./templateSpec");
const templateSpecVersion_1 = require("./templateSpecVersion");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:resources:AzureCliScript":
                return new azureCliScript_1.AzureCliScript(name, undefined, { urn });
            case "azure-native:resources:AzurePowerShellScript":
                return new azurePowerShellScript_1.AzurePowerShellScript(name, undefined, { urn });
            case "azure-native:resources:Deployment":
                return new deployment_1.Deployment(name, undefined, { urn });
            case "azure-native:resources:DeploymentAtManagementGroupScope":
                return new deploymentAtManagementGroupScope_1.DeploymentAtManagementGroupScope(name, undefined, { urn });
            case "azure-native:resources:DeploymentAtScope":
                return new deploymentAtScope_1.DeploymentAtScope(name, undefined, { urn });
            case "azure-native:resources:DeploymentAtSubscriptionScope":
                return new deploymentAtSubscriptionScope_1.DeploymentAtSubscriptionScope(name, undefined, { urn });
            case "azure-native:resources:DeploymentAtTenantScope":
                return new deploymentAtTenantScope_1.DeploymentAtTenantScope(name, undefined, { urn });
            case "azure-native:resources:DeploymentScript":
                return new deploymentScript_1.DeploymentScript(name, undefined, { urn });
            case "azure-native:resources:Resource":
                return new resource_1.Resource(name, undefined, { urn });
            case "azure-native:resources:ResourceGroup":
                return new resourceGroup_1.ResourceGroup(name, undefined, { urn });
            case "azure-native:resources:TagAtScope":
                return new tagAtScope_1.TagAtScope(name, undefined, { urn });
            case "azure-native:resources:TemplateSpec":
                return new templateSpec_1.TemplateSpec(name, undefined, { urn });
            case "azure-native:resources:TemplateSpecVersion":
                return new templateSpecVersion_1.TemplateSpecVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "resources", _module);
