// @pulumi/azure-native/types/enums/netapp/v20210801/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevel = exports.SecurityStyle = exports.ReplicationSchedule = exports.QosType = exports.NetworkFeatures = exports.EndpointType = exports.EncryptionType = exports.ChownMode = exports.AvsDataStore = exports.ApplicationType = void 0;
exports.ApplicationType = {
    SAP_HANA: "SAP-HANA",
};
exports.AvsDataStore = {
    /**
     * avsDataStore is enabled
     */
    Enabled: "Enabled",
    /**
     * avsDataStore is disabled
     */
    Disabled: "Disabled",
};
exports.ChownMode = {
    Restricted: "Restricted",
    Unrestricted: "Unrestricted",
};
exports.EncryptionType = {
    /**
     * EncryptionType Single, volumes will use single encryption at rest
     */
    Single: "Single",
    /**
     * EncryptionType Double, volumes will use double encryption at rest
     */
    Double: "Double",
};
exports.EndpointType = {
    Src: "src",
    Dst: "dst",
};
exports.NetworkFeatures = {
    /**
     * Basic network feature.
     */
    Basic: "Basic",
    /**
     * Standard network feature.
     */
    Standard: "Standard",
};
exports.QosType = {
    /**
     * qos type Auto
     */
    Auto: "Auto",
    /**
     * qos type Manual
     */
    Manual: "Manual",
};
exports.ReplicationSchedule = {
    ReplicationSchedule_10minutely: "_10minutely",
    Hourly: "hourly",
    Daily: "daily",
};
exports.SecurityStyle = {
    Ntfs: "ntfs",
    Unix: "unix",
};
exports.ServiceLevel = {
    /**
     * Standard service level
     */
    Standard: "Standard",
    /**
     * Premium service level
     */
    Premium: "Premium",
    /**
     * Ultra service level
     */
    Ultra: "Ultra",
    /**
     * Zone redundant storage service level
     */
    StandardZRS: "StandardZRS",
};
