// @pulumi/azure-native/resources/v20180501/resourceGroup.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Resource group information.
 *
 * @deprecated Version v20180501 will be removed in the next major version of the provider. Upgrade to version v20190501 or later.
 */
class ResourceGroup extends pulumi.CustomResource {
    /**
     * Create a ResourceGroup resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20180501 will be removed in the next major version of the provider. Upgrade to version v20190501 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ResourceGroup is deprecated: Version v20180501 will be removed in the next major version of the provider. Upgrade to version v20190501 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["managedBy"] = args ? args.managedBy : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["managedBy"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:resources:ResourceGroup" }, { type: "azure-native:resources/v20151101:ResourceGroup" }, { type: "azure-native:resources/v20160201:ResourceGroup" }, { type: "azure-native:resources/v20160701:ResourceGroup" }, { type: "azure-native:resources/v20160901:ResourceGroup" }, { type: "azure-native:resources/v20170510:ResourceGroup" }, { type: "azure-native:resources/v20180201:ResourceGroup" }, { type: "azure-native:resources/v20190301:ResourceGroup" }, { type: "azure-native:resources/v20190501:ResourceGroup" }, { type: "azure-native:resources/v20190510:ResourceGroup" }, { type: "azure-native:resources/v20190701:ResourceGroup" }, { type: "azure-native:resources/v20190801:ResourceGroup" }, { type: "azure-native:resources/v20191001:ResourceGroup" }, { type: "azure-native:resources/v20200601:ResourceGroup" }, { type: "azure-native:resources/v20200801:ResourceGroup" }, { type: "azure-native:resources/v20201001:ResourceGroup" }, { type: "azure-native:resources/v20210101:ResourceGroup" }, { type: "azure-native:resources/v20210401:ResourceGroup" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ResourceGroup.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ResourceGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ResourceGroup is deprecated: Version v20180501 will be removed in the next major version of the provider. Upgrade to version v20190501 or later.");
        return new ResourceGroup(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ResourceGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ResourceGroup.__pulumiType;
    }
}
exports.ResourceGroup = ResourceGroup;
/** @internal */
ResourceGroup.__pulumiType = 'azure-native:resources/v20180501:ResourceGroup';
