// @pulumi/azure-native/types/enums/dashboard/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZoneRedundancy = exports.ProvisioningState = exports.IdentityType = exports.AutoGeneratedDomainNameLabelScope = exports.v20210901preview = void 0;
// Export sub-modules:
const v20210901preview = require("./v20210901preview");
exports.v20210901preview = v20210901preview;
exports.AutoGeneratedDomainNameLabelScope = {
    TenantReuse: "TenantReuse",
};
exports.IdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
exports.ProvisioningState = {
    Accepted: "Accepted",
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
    Deleted: "Deleted",
    NotSpecified: "NotSpecified",
};
exports.ZoneRedundancy = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
