// @pulumi/azure-native/peering/v20190801preview/peeringServicePrefix.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PeeringServicePrefix = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The peering service prefix class.
 *
 * @deprecated Version v20190801preview will be removed in the next major version of the provider. Upgrade to version v20210101 or later.
 */
class PeeringServicePrefix extends pulumi.CustomResource {
    /**
     * Create a PeeringServicePrefix resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20190801preview will be removed in the next major version of the provider. Upgrade to version v20210101 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("PeeringServicePrefix is deprecated: Version v20190801preview will be removed in the next major version of the provider. Upgrade to version v20210101 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.peeringServiceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'peeringServiceName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["learnedType"] = args ? args.learnedType : undefined;
            resourceInputs["peeringServiceName"] = args ? args.peeringServiceName : undefined;
            resourceInputs["prefix"] = args ? args.prefix : undefined;
            resourceInputs["prefixName"] = args ? args.prefixName : undefined;
            resourceInputs["prefixValidationState"] = args ? args.prefixValidationState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["learnedType"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["prefix"] = undefined /*out*/;
            resourceInputs["prefixValidationState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:peering:PeeringServicePrefix" }, { type: "azure-native:peering/v20190901preview:PeeringServicePrefix" }, { type: "azure-native:peering/v20200101preview:PeeringServicePrefix" }, { type: "azure-native:peering/v20200401:PeeringServicePrefix" }, { type: "azure-native:peering/v20201001:PeeringServicePrefix" }, { type: "azure-native:peering/v20210101:PeeringServicePrefix" }, { type: "azure-native:peering/v20210601:PeeringServicePrefix" }, { type: "azure-native:peering/v20220101:PeeringServicePrefix" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PeeringServicePrefix.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing PeeringServicePrefix resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("PeeringServicePrefix is deprecated: Version v20190801preview will be removed in the next major version of the provider. Upgrade to version v20210101 or later.");
        return new PeeringServicePrefix(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PeeringServicePrefix.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PeeringServicePrefix.__pulumiType;
    }
}
exports.PeeringServicePrefix = PeeringServicePrefix;
/** @internal */
PeeringServicePrefix.__pulumiType = 'azure-native:peering/v20190801preview:PeeringServicePrefix';
