// @pulumi/azure-native/types/enums/mobilenetwork/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrafficControlPermission = exports.SdfDirection = exports.PreemptionVulnerability = exports.PreemptionCapability = exports.PduSessionType = exports.NaptEnabled = exports.CreatedByType = exports.CoreNetworkType = exports.v20220301preview = exports.v20220101preview = void 0;
// Export sub-modules:
const v20220101preview = require("./v20220101preview");
exports.v20220101preview = v20220101preview;
const v20220301preview = require("./v20220301preview");
exports.v20220301preview = v20220301preview;
exports.CoreNetworkType = {
    CoreNetworkType_5GC: "5GC",
    EPC: "EPC",
};
exports.CreatedByType = {
    User: "User",
    Application: "Application",
    ManagedIdentity: "ManagedIdentity",
    Key: "Key",
};
exports.NaptEnabled = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.PduSessionType = {
    IPv4: "IPv4",
    IPv6: "IPv6",
};
exports.PreemptionCapability = {
    NotPreempt: "NotPreempt",
    MayPreempt: "MayPreempt",
};
exports.PreemptionVulnerability = {
    NotPreemptable: "NotPreemptable",
    Preemptable: "Preemptable",
};
exports.SdfDirection = {
    Uplink: "Uplink",
    Downlink: "Downlink",
    Bidirectional: "Bidirectional",
};
exports.TrafficControlPermission = {
    Enabled: "Enabled",
    Blocked: "Blocked",
};
