// @pulumi/azure-native/securityandcompliance/v20210111/privateEndpointConnectionsSec.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateEndpointConnectionsSec = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The Private Endpoint Connection resource.
 *
 * @deprecated Version v20210111 will be removed in the next major version of the provider. Upgrade to version v20210308 or later.
 */
class PrivateEndpointConnectionsSec extends pulumi.CustomResource {
    /**
     * Create a PrivateEndpointConnectionsSec resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20210111 will be removed in the next major version of the provider. Upgrade to version v20210308 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("PrivateEndpointConnectionsSec is deprecated: Version v20210111 will be removed in the next major version of the provider. Upgrade to version v20210308 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.privateLinkServiceConnectionState === undefined) && !opts.urn) {
                throw new Error("Missing required property 'privateLinkServiceConnectionState'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.resourceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceName'");
            }
            resourceInputs["privateEndpointConnectionName"] = args ? args.privateEndpointConnectionName : undefined;
            resourceInputs["privateLinkServiceConnectionState"] = args ? args.privateLinkServiceConnectionState : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpoint"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["privateEndpoint"] = undefined /*out*/;
            resourceInputs["privateLinkServiceConnectionState"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityandcompliance:PrivateEndpointConnectionsSec" }, { type: "azure-native:securityandcompliance/v20210308:PrivateEndpointConnectionsSec" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(PrivateEndpointConnectionsSec.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing PrivateEndpointConnectionsSec resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("PrivateEndpointConnectionsSec is deprecated: Version v20210111 will be removed in the next major version of the provider. Upgrade to version v20210308 or later.");
        return new PrivateEndpointConnectionsSec(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of PrivateEndpointConnectionsSec.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateEndpointConnectionsSec.__pulumiType;
    }
}
exports.PrivateEndpointConnectionsSec = PrivateEndpointConnectionsSec;
/** @internal */
PrivateEndpointConnectionsSec.__pulumiType = 'azure-native:securityandcompliance/v20210111:PrivateEndpointConnectionsSec';
