// @pulumi/azure-native/types/enums/netapp/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevel = exports.SecurityStyle = exports.ReplicationSchedule = exports.QosType = exports.NetworkFeatures = exports.EndpointType = exports.EnableSubvolumes = exports.ChownMode = exports.AvsDataStore = exports.ApplicationType = exports.v20211001 = exports.v20210801 = exports.v20210601 = exports.v20210401preview = exports.v20210401 = exports.v20210201 = exports.v20201201 = exports.v20201101 = exports.v20200901 = exports.v20200801 = exports.v20200701 = exports.v20200601 = exports.v20200501 = exports.v20200301 = exports.v20200201 = exports.v20191101 = exports.v20191001 = exports.v20190801 = exports.v20190701 = exports.v20190601 = exports.v20190501 = exports.v20170815 = void 0;
// Export sub-modules:
const v20170815 = require("./v20170815");
exports.v20170815 = v20170815;
const v20190501 = require("./v20190501");
exports.v20190501 = v20190501;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20190701 = require("./v20190701");
exports.v20190701 = v20190701;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20200201 = require("./v20200201");
exports.v20200201 = v20200201;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200701 = require("./v20200701");
exports.v20200701 = v20200701;
const v20200801 = require("./v20200801");
exports.v20200801 = v20200801;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20201101 = require("./v20201101");
exports.v20201101 = v20201101;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
exports.ApplicationType = {
    SAP_HANA: "SAP-HANA",
};
exports.AvsDataStore = {
    /**
     * avsDataStore is enabled
     */
    Enabled: "Enabled",
    /**
     * avsDataStore is disabled
     */
    Disabled: "Disabled",
};
exports.ChownMode = {
    Restricted: "Restricted",
    Unrestricted: "Unrestricted",
};
exports.EnableSubvolumes = {
    /**
     * subvolumes are enabled
     */
    Enabled: "Enabled",
    /**
     * subvolumes are not enabled
     */
    Disabled: "Disabled",
};
exports.EndpointType = {
    Src: "src",
    Dst: "dst",
};
exports.NetworkFeatures = {
    /**
     * Basic network feature.
     */
    Basic: "Basic",
    /**
     * Standard network feature.
     */
    Standard: "Standard",
};
exports.QosType = {
    /**
     * qos type Auto
     */
    Auto: "Auto",
    /**
     * qos type Manual
     */
    Manual: "Manual",
};
exports.ReplicationSchedule = {
    ReplicationSchedule_10minutely: "_10minutely",
    Hourly: "hourly",
    Daily: "daily",
};
exports.SecurityStyle = {
    Ntfs: "ntfs",
    Unix: "unix",
};
exports.ServiceLevel = {
    /**
     * Standard service level
     */
    Standard: "Standard",
    /**
     * Premium service level
     */
    Premium: "Premium",
    /**
     * Ultra service level
     */
    Ultra: "Ultra",
    /**
     * Zone redundant storage service level
     */
    StandardZRS: "StandardZRS",
};
