// @pulumi/azure-native/security/v20170801preview/iotSecuritySolution.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotSecuritySolution = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Security Solution
 */
class IotSecuritySolution extends pulumi.CustomResource {
    /**
     * Create a IotSecuritySolution resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        var _a;
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            if ((!args || args.iotHubs === undefined) && !opts.urn) {
                throw new Error("Missing required property 'iotHubs'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspace === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspace'");
            }
            resourceInputs["disabledDataSources"] = args ? args.disabledDataSources : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["export"] = args ? args.export : undefined;
            resourceInputs["iotHubs"] = args ? args.iotHubs : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["recommendationsConfiguration"] = args ? args.recommendationsConfiguration : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["solutionName"] = args ? args.solutionName : undefined;
            resourceInputs["status"] = (_a = (args ? args.status : undefined)) !== null && _a !== void 0 ? _a : "Enabled";
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["userDefinedResources"] = args ? args.userDefinedResources : undefined;
            resourceInputs["workspace"] = args ? args.workspace : undefined;
            resourceInputs["autoDiscoveredResources"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["autoDiscoveredResources"] = undefined /*out*/;
            resourceInputs["disabledDataSources"] = undefined /*out*/;
            resourceInputs["displayName"] = undefined /*out*/;
            resourceInputs["export"] = undefined /*out*/;
            resourceInputs["iotHubs"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["recommendationsConfiguration"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["userDefinedResources"] = undefined /*out*/;
            resourceInputs["workspace"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:security:IotSecuritySolution" }, { type: "azure-native:security/v20190801:IotSecuritySolution" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(IotSecuritySolution.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing IotSecuritySolution resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new IotSecuritySolution(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of IotSecuritySolution.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IotSecuritySolution.__pulumiType;
    }
}
exports.IotSecuritySolution = IotSecuritySolution;
/** @internal */
IotSecuritySolution.__pulumiType = 'azure-native:security/v20170801preview:IotSecuritySolution';
