// @pulumi/azure-native/subscription/v20191001preview/subscriptionAlias.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionAlias = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Subscription Information with the alias.
 *
 * @deprecated Version v20191001preview will be removed in the next major version of the provider. Upgrade to version v20200901 or later.
 */
class SubscriptionAlias extends pulumi.CustomResource {
    /**
     * Create a SubscriptionAlias resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20191001preview will be removed in the next major version of the provider. Upgrade to version v20200901 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SubscriptionAlias is deprecated: Version v20191001preview will be removed in the next major version of the provider. Upgrade to version v20200901 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            resourceInputs["aliasName"] = args ? args.aliasName : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:subscription:SubscriptionAlias" }, { type: "azure-native:subscription/v20200901:SubscriptionAlias" }, { type: "azure-native:subscription/v20211001:SubscriptionAlias" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SubscriptionAlias.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SubscriptionAlias resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SubscriptionAlias is deprecated: Version v20191001preview will be removed in the next major version of the provider. Upgrade to version v20200901 or later.");
        return new SubscriptionAlias(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SubscriptionAlias.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SubscriptionAlias.__pulumiType;
    }
}
exports.SubscriptionAlias = SubscriptionAlias;
/** @internal */
SubscriptionAlias.__pulumiType = 'azure-native:subscription/v20191001preview:SubscriptionAlias';
