// @pulumi/azure-native/securityinsights/v20210901preview/entityQuery.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntityQuery = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Specific entity query.
 *
 * @deprecated Please use one of the variants: ActivityCustomEntityQuery.
 */
class EntityQuery extends pulumi.CustomResource {
    /**
     * Create a EntityQuery resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Please use one of the variants: ActivityCustomEntityQuery. */
    constructor(name, args, opts) {
        pulumi.log.warn("EntityQuery is deprecated: Please use one of the variants: ActivityCustomEntityQuery.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["entityQueryId"] = args ? args.entityQueryId : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights:EntityQuery" }, { type: "azure-native:securityinsights/v20210301preview:EntityQuery" }, { type: "azure-native:securityinsights/v20211001preview:EntityQuery" }, { type: "azure-native:securityinsights/v20220101preview:EntityQuery" }, { type: "azure-native:securityinsights/v20220401preview:EntityQuery" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(EntityQuery.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing EntityQuery resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("EntityQuery is deprecated: Please use one of the variants: ActivityCustomEntityQuery.");
        return new EntityQuery(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EntityQuery.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EntityQuery.__pulumiType;
    }
}
exports.EntityQuery = EntityQuery;
/** @internal */
EntityQuery.__pulumiType = 'azure-native:securityinsights/v20210901preview:EntityQuery';
