// @pulumi/azure-native/types/enums/documentdb/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TriggerType = exports.TriggerOperation = exports.SpatialType = exports.ServiceType = exports.ServiceSize = exports.ServerVersion = exports.RoleDefinitionType = exports.ResourceIdentityType = exports.PublicNetworkAccess = exports.PartitionKind = exports.NetworkAclBypass = exports.MongoRoleDefinitionType = exports.ManagedCassandraProvisioningState = exports.IndexingMode = exports.IndexKind = exports.DefaultConsistencyLevel = exports.DatabaseAccountOfferType = exports.DatabaseAccountKind = exports.DataType = exports.ConnectorOffer = exports.ConflictResolutionMode = exports.CompositePathSortOrder = exports.BackupPolicyType = exports.AuthenticationMethod = exports.v20220215preview = exports.v20211115preview = exports.v20211015preview = exports.v20211015 = exports.v20210701preview = exports.v20210615 = exports.v20210515 = exports.v20210415 = exports.v20210401preview = exports.v20210315 = exports.v20210301preview = exports.v20210115 = exports.v20200901 = exports.v20200601preview = exports.v20200401 = exports.v20200301 = exports.v20191212 = exports.v20190801 = exports.v20160331 = exports.v20160319 = exports.v20151106 = exports.v20150408 = exports.v20150401 = void 0;
// Export sub-modules:
const v20150401 = require("./v20150401");
exports.v20150401 = v20150401;
const v20150408 = require("./v20150408");
exports.v20150408 = v20150408;
const v20151106 = require("./v20151106");
exports.v20151106 = v20151106;
const v20160319 = require("./v20160319");
exports.v20160319 = v20160319;
const v20160331 = require("./v20160331");
exports.v20160331 = v20160331;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20191212 = require("./v20191212");
exports.v20191212 = v20191212;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200401 = require("./v20200401");
exports.v20200401 = v20200401;
const v20200601preview = require("./v20200601preview");
exports.v20200601preview = v20200601preview;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20210115 = require("./v20210115");
exports.v20210115 = v20210115;
const v20210301preview = require("./v20210301preview");
exports.v20210301preview = v20210301preview;
const v20210315 = require("./v20210315");
exports.v20210315 = v20210315;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210415 = require("./v20210415");
exports.v20210415 = v20210415;
const v20210515 = require("./v20210515");
exports.v20210515 = v20210515;
const v20210615 = require("./v20210615");
exports.v20210615 = v20210615;
const v20210701preview = require("./v20210701preview");
exports.v20210701preview = v20210701preview;
const v20211015 = require("./v20211015");
exports.v20211015 = v20211015;
const v20211015preview = require("./v20211015preview");
exports.v20211015preview = v20211015preview;
const v20211115preview = require("./v20211115preview");
exports.v20211115preview = v20211115preview;
const v20220215preview = require("./v20220215preview");
exports.v20220215preview = v20220215preview;
exports.AuthenticationMethod = {
    None: "None",
    Cassandra: "Cassandra",
};
exports.BackupPolicyType = {
    Periodic: "Periodic",
    Continuous: "Continuous",
};
exports.CompositePathSortOrder = {
    Ascending: "ascending",
    Descending: "descending",
};
exports.ConflictResolutionMode = {
    LastWriterWins: "LastWriterWins",
    Custom: "Custom",
};
exports.ConnectorOffer = {
    Small: "Small",
};
exports.DataType = {
    String: "String",
    Number: "Number",
    Point: "Point",
    Polygon: "Polygon",
    LineString: "LineString",
    MultiPolygon: "MultiPolygon",
};
exports.DatabaseAccountKind = {
    GlobalDocumentDB: "GlobalDocumentDB",
    MongoDB: "MongoDB",
    Parse: "Parse",
};
exports.DatabaseAccountOfferType = {
    Standard: "Standard",
};
exports.DefaultConsistencyLevel = {
    Eventual: "Eventual",
    Session: "Session",
    BoundedStaleness: "BoundedStaleness",
    Strong: "Strong",
    ConsistentPrefix: "ConsistentPrefix",
};
exports.IndexKind = {
    Hash: "Hash",
    Range: "Range",
    Spatial: "Spatial",
};
exports.IndexingMode = {
    Consistent: "consistent",
    Lazy: "lazy",
    None: "none",
};
exports.ManagedCassandraProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
};
exports.MongoRoleDefinitionType = {
    BuiltInRole: "BuiltInRole",
    CustomRole: "CustomRole",
};
exports.NetworkAclBypass = {
    None: "None",
    AzureServices: "AzureServices",
};
exports.PartitionKind = {
    Hash: "Hash",
    Range: "Range",
    MultiHash: "MultiHash",
};
exports.PublicNetworkAccess = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned,UserAssigned",
    None: "None",
};
exports.RoleDefinitionType = {
    BuiltInRole: "BuiltInRole",
    CustomRole: "CustomRole",
};
exports.ServerVersion = {
    ServerVersion_3_2: "3.2",
    ServerVersion_3_6: "3.6",
    ServerVersion_4_0: "4.0",
};
exports.ServiceSize = {
    Cosmos_D4s: "Cosmos.D4s",
    Cosmos_D8s: "Cosmos.D8s",
    Cosmos_D16s: "Cosmos.D16s",
};
exports.ServiceType = {
    SqlDedicatedGateway: "SqlDedicatedGateway",
    DataTransfer: "DataTransfer",
};
exports.SpatialType = {
    Point: "Point",
    LineString: "LineString",
    Polygon: "Polygon",
    MultiPolygon: "MultiPolygon",
};
exports.TriggerOperation = {
    All: "All",
    Create: "Create",
    Update: "Update",
    Delete: "Delete",
    Replace: "Replace",
};
exports.TriggerType = {
    Pre: "Pre",
    Post: "Post",
};
