// @pulumi/azure-native/types/enums/containerservice/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedClusterSKUName = exports.LoadBalancerSku = exports.LicenseType = exports.KubeletDiskType = exports.GPUInstanceProfile = exports.ExtendedLocationTypes = exports.Expander = exports.ConnectionStatus = exports.AgentPoolType = exports.AgentPoolMode = exports.v20220302preview = exports.v20220301 = exports.v20220202preview = exports.v20220201 = exports.v20220102preview = exports.v20220101 = exports.v20211101preview = exports.v20211001 = exports.v20210901 = exports.v20210801 = exports.v20210701 = exports.v20210501 = exports.v20210301 = exports.v20210201 = exports.v20201201 = exports.v20201101 = exports.v20200901 = exports.v20200701 = exports.v20200601 = exports.v20200401 = exports.v20200301 = exports.v20200201 = exports.v20200101 = exports.v20191101 = exports.v20191027preview = exports.v20191001 = exports.v20190930preview = exports.v20190801 = exports.v20190601 = exports.v20190430 = exports.v20190401 = exports.v20190201 = exports.v20180930preview = exports.v20180801preview = exports.v20180331 = exports.v20170831 = exports.v20170131 = exports.v20160930 = exports.v20160330 = exports.v20151101preview = void 0;
exports.WeekDay = exports.UpgradeChannel = exports.SnapshotType = exports.ScaleSetPriority = exports.ScaleSetEvictionPolicy = exports.ResourceIdentityType = exports.OutboundType = exports.OpenShiftContainerServiceVMSize = exports.OpenShiftAgentPoolProfileRole = exports.OSType = exports.OSSKU = exports.OSDiskType = exports.NetworkPolicy = exports.NetworkPlugin = exports.NetworkMode = exports.ManagedClusterSKUTier = void 0;
// Export sub-modules:
const v20151101preview = require("./v20151101preview");
exports.v20151101preview = v20151101preview;
const v20160330 = require("./v20160330");
exports.v20160330 = v20160330;
const v20160930 = require("./v20160930");
exports.v20160930 = v20160930;
const v20170131 = require("./v20170131");
exports.v20170131 = v20170131;
const v20170831 = require("./v20170831");
exports.v20170831 = v20170831;
const v20180331 = require("./v20180331");
exports.v20180331 = v20180331;
const v20180801preview = require("./v20180801preview");
exports.v20180801preview = v20180801preview;
const v20180930preview = require("./v20180930preview");
exports.v20180930preview = v20180930preview;
const v20190201 = require("./v20190201");
exports.v20190201 = v20190201;
const v20190401 = require("./v20190401");
exports.v20190401 = v20190401;
const v20190430 = require("./v20190430");
exports.v20190430 = v20190430;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20190801 = require("./v20190801");
exports.v20190801 = v20190801;
const v20190930preview = require("./v20190930preview");
exports.v20190930preview = v20190930preview;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191027preview = require("./v20191027preview");
exports.v20191027preview = v20191027preview;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20200101 = require("./v20200101");
exports.v20200101 = v20200101;
const v20200201 = require("./v20200201");
exports.v20200201 = v20200201;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200401 = require("./v20200401");
exports.v20200401 = v20200401;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200701 = require("./v20200701");
exports.v20200701 = v20200701;
const v20200901 = require("./v20200901");
exports.v20200901 = v20200901;
const v20201101 = require("./v20201101");
exports.v20201101 = v20201101;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
const v20210901 = require("./v20210901");
exports.v20210901 = v20210901;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
const v20211101preview = require("./v20211101preview");
exports.v20211101preview = v20211101preview;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
const v20220102preview = require("./v20220102preview");
exports.v20220102preview = v20220102preview;
const v20220201 = require("./v20220201");
exports.v20220201 = v20220201;
const v20220202preview = require("./v20220202preview");
exports.v20220202preview = v20220202preview;
const v20220301 = require("./v20220301");
exports.v20220301 = v20220301;
const v20220302preview = require("./v20220302preview");
exports.v20220302preview = v20220302preview;
exports.AgentPoolMode = {
    System: "System",
    User: "User",
};
exports.AgentPoolType = {
    VirtualMachineScaleSets: "VirtualMachineScaleSets",
    AvailabilitySet: "AvailabilitySet",
};
exports.ConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.Expander = {
    Least_waste: "least-waste",
    Most_pods: "most-pods",
    Priority: "priority",
    Random: "random",
};
exports.ExtendedLocationTypes = {
    EdgeZone: "EdgeZone",
};
exports.GPUInstanceProfile = {
    MIG1g: "MIG1g",
    MIG2g: "MIG2g",
    MIG3g: "MIG3g",
    MIG4g: "MIG4g",
    MIG7g: "MIG7g",
};
exports.KubeletDiskType = {
    OS: "OS",
    Temporary: "Temporary",
};
exports.LicenseType = {
    None: "None",
    Windows_Server: "Windows_Server",
};
exports.LoadBalancerSku = {
    Standard: "standard",
    Basic: "basic",
};
exports.ManagedClusterSKUName = {
    Basic: "Basic",
};
exports.ManagedClusterSKUTier = {
    Paid: "Paid",
    Free: "Free",
};
exports.NetworkMode = {
    Transparent: "transparent",
    Bridge: "bridge",
};
exports.NetworkPlugin = {
    Azure: "azure",
    Kubenet: "kubenet",
};
exports.NetworkPolicy = {
    Calico: "calico",
    Azure: "azure",
};
exports.OSDiskType = {
    Managed: "Managed",
    Ephemeral: "Ephemeral",
};
exports.OSSKU = {
    Ubuntu: "Ubuntu",
    CBLMariner: "CBLMariner",
};
exports.OSType = {
    Linux: "Linux",
    Windows: "Windows",
};
exports.OpenShiftAgentPoolProfileRole = {
    Compute: "compute",
    Infra: "infra",
};
exports.OpenShiftContainerServiceVMSize = {
    Standard_D2s_v3: "Standard_D2s_v3",
    Standard_D4s_v3: "Standard_D4s_v3",
    Standard_D8s_v3: "Standard_D8s_v3",
    Standard_D16s_v3: "Standard_D16s_v3",
    Standard_D32s_v3: "Standard_D32s_v3",
    Standard_D64s_v3: "Standard_D64s_v3",
    Standard_DS4_v2: "Standard_DS4_v2",
    Standard_DS5_v2: "Standard_DS5_v2",
    Standard_F8s_v2: "Standard_F8s_v2",
    Standard_F16s_v2: "Standard_F16s_v2",
    Standard_F32s_v2: "Standard_F32s_v2",
    Standard_F64s_v2: "Standard_F64s_v2",
    Standard_F72s_v2: "Standard_F72s_v2",
    Standard_F8s: "Standard_F8s",
    Standard_F16s: "Standard_F16s",
    Standard_E4s_v3: "Standard_E4s_v3",
    Standard_E8s_v3: "Standard_E8s_v3",
    Standard_E16s_v3: "Standard_E16s_v3",
    Standard_E20s_v3: "Standard_E20s_v3",
    Standard_E32s_v3: "Standard_E32s_v3",
    Standard_E64s_v3: "Standard_E64s_v3",
    Standard_GS2: "Standard_GS2",
    Standard_GS3: "Standard_GS3",
    Standard_GS4: "Standard_GS4",
    Standard_GS5: "Standard_GS5",
    Standard_DS12_v2: "Standard_DS12_v2",
    Standard_DS13_v2: "Standard_DS13_v2",
    Standard_DS14_v2: "Standard_DS14_v2",
    Standard_DS15_v2: "Standard_DS15_v2",
    Standard_L4s: "Standard_L4s",
    Standard_L8s: "Standard_L8s",
    Standard_L16s: "Standard_L16s",
    Standard_L32s: "Standard_L32s",
};
exports.OutboundType = {
    LoadBalancer: "loadBalancer",
    UserDefinedRouting: "userDefinedRouting",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    None: "None",
};
exports.ScaleSetEvictionPolicy = {
    Delete: "Delete",
    Deallocate: "Deallocate",
};
exports.ScaleSetPriority = {
    Spot: "Spot",
    Regular: "Regular",
};
exports.SnapshotType = {
    /**
     * The snapshot is a snapshot of a node pool.
     */
    NodePool: "NodePool",
};
exports.UpgradeChannel = {
    Rapid: "rapid",
    Stable: "stable",
    Patch: "patch",
    Node_image: "node-image",
    None: "none",
};
exports.WeekDay = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
