// @pulumi/azure-native/types/enums/desktopvirtualization/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StopHostsWhen = exports.SkuTier = exports.SessionHostLoadBalancingAlgorithm = exports.SSOSecretType = exports.ResourceIdentityType = exports.RemoteApplicationType = exports.RegistrationTokenOperation = exports.PrivateEndpointServiceConnectionStatus = exports.PreferredAppGroupType = exports.PersonalDesktopAssignmentType = exports.Operation = exports.LoadBalancerType = exports.HostPoolType = exports.CommandLineSetting = exports.ApplicationGroupType = exports.v20220210preview = exports.v20210903preview = exports.v20210712 = exports.v20210401preview = exports.v20210309preview = exports.v20210201preview = exports.v20210114preview = exports.v20201110preview = exports.v20201102preview = exports.v20201019preview = exports.v20200921preview = exports.v20191210preview = exports.v20190924preview = exports.v20190123preview = void 0;
// Export sub-modules:
const v20190123preview = require("./v20190123preview");
exports.v20190123preview = v20190123preview;
const v20190924preview = require("./v20190924preview");
exports.v20190924preview = v20190924preview;
const v20191210preview = require("./v20191210preview");
exports.v20191210preview = v20191210preview;
const v20200921preview = require("./v20200921preview");
exports.v20200921preview = v20200921preview;
const v20201019preview = require("./v20201019preview");
exports.v20201019preview = v20201019preview;
const v20201102preview = require("./v20201102preview");
exports.v20201102preview = v20201102preview;
const v20201110preview = require("./v20201110preview");
exports.v20201110preview = v20201110preview;
const v20210114preview = require("./v20210114preview");
exports.v20210114preview = v20210114preview;
const v20210201preview = require("./v20210201preview");
exports.v20210201preview = v20210201preview;
const v20210309preview = require("./v20210309preview");
exports.v20210309preview = v20210309preview;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20210712 = require("./v20210712");
exports.v20210712 = v20210712;
const v20210903preview = require("./v20210903preview");
exports.v20210903preview = v20210903preview;
const v20220210preview = require("./v20220210preview");
exports.v20220210preview = v20220210preview;
exports.ApplicationGroupType = {
    RemoteApp: "RemoteApp",
    Desktop: "Desktop",
};
exports.CommandLineSetting = {
    DoNotAllow: "DoNotAllow",
    Allow: "Allow",
    Require: "Require",
};
exports.HostPoolType = {
    /**
     * Users will be assigned a SessionHost either by administrators (PersonalDesktopAssignmentType = Direct) or upon connecting to the pool (PersonalDesktopAssignmentType = Automatic). They will always be redirected to their assigned SessionHost.
     */
    Personal: "Personal",
    /**
     * Users get a new (random) SessionHost every time it connects to the HostPool.
     */
    Pooled: "Pooled",
    /**
     * Users assign their own machines, load balancing logic remains the same as Personal. PersonalDesktopAssignmentType must be Direct.
     */
    BYODesktop: "BYODesktop",
};
exports.LoadBalancerType = {
    BreadthFirst: "BreadthFirst",
    DepthFirst: "DepthFirst",
    Persistent: "Persistent",
};
exports.Operation = {
    /**
     * Start the migration.
     */
    Start: "Start",
    /**
     * Revoke the migration.
     */
    Revoke: "Revoke",
    /**
     * Complete the migration.
     */
    Complete: "Complete",
    /**
     * Hide the hostpool.
     */
    Hide: "Hide",
    /**
     * Unhide the hostpool.
     */
    Unhide: "Unhide",
};
exports.PersonalDesktopAssignmentType = {
    Automatic: "Automatic",
    Direct: "Direct",
};
exports.PreferredAppGroupType = {
    None: "None",
    Desktop: "Desktop",
    RailApplications: "RailApplications",
};
exports.PrivateEndpointServiceConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
};
exports.RegistrationTokenOperation = {
    Delete: "Delete",
    None: "None",
    Update: "Update",
};
exports.RemoteApplicationType = {
    InBuilt: "InBuilt",
    MsixApplication: "MsixApplication",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
};
exports.SSOSecretType = {
    SharedKey: "SharedKey",
    Certificate: "Certificate",
    SharedKeyInKeyVault: "SharedKeyInKeyVault",
    CertificateInKeyVault: "CertificateInKeyVault",
};
exports.SessionHostLoadBalancingAlgorithm = {
    BreadthFirst: "BreadthFirst",
    DepthFirst: "DepthFirst",
};
exports.SkuTier = {
    Free: "Free",
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.StopHostsWhen = {
    ZeroSessions: "ZeroSessions",
    ZeroActiveSessions: "ZeroActiveSessions",
};
