// @pulumi/azure-native/digitaltwins/v20200301preview/digitalTwin.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DigitalTwin = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * The description of the DigitalTwins service.
 *
 * @deprecated Version v20200301preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later.
 */
class DigitalTwin extends pulumi.CustomResource {
    /**
     * Create a DigitalTwin resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20200301preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("DigitalTwin is deprecated: Version v20200301preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceName"] = args ? args.resourceName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["hostName"] = undefined /*out*/;
            resourceInputs["lastUpdatedTime"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["hostName"] = undefined /*out*/;
            resourceInputs["lastUpdatedTime"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:digitaltwins:DigitalTwin" }, { type: "azure-native:digitaltwins/v20201031:DigitalTwin" }, { type: "azure-native:digitaltwins/v20201201:DigitalTwin" }, { type: "azure-native:digitaltwins/v20210630preview:DigitalTwin" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DigitalTwin.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DigitalTwin resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("DigitalTwin is deprecated: Version v20200301preview will be removed in the next major version of the provider. Upgrade to version v20201201 or later.");
        return new DigitalTwin(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DigitalTwin.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DigitalTwin.__pulumiType;
    }
}
exports.DigitalTwin = DigitalTwin;
/** @internal */
DigitalTwin.__pulumiType = 'azure-native:digitaltwins/v20200301preview:DigitalTwin';
