// @pulumi/azure-native/types/enums/web/v20180201/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnauthenticatedClientAction = exports.SupportedTlsVersions = exports.SslState = exports.SiteLoadBalancing = exports.ScmType = exports.RouteType = exports.RedundancyMode = exports.PublicCertificateLocation = exports.ManagedServiceIdentityType = exports.ManagedPipelineMode = exports.LogLevel = exports.IpFilterTag = exports.InternalLoadBalancingMode = exports.HostType = exports.HostNameType = exports.FtpsState = exports.FrequencyUnit = exports.DatabaseType = exports.CustomHostNameDnsRecordType = exports.ConnectionStringType = exports.ComputeModeOptions = exports.BuiltInAuthenticationProvider = exports.AzureStorageType = exports.AzureResourceType = exports.AutoHealActionType = exports.AccessControlEntryAction = void 0;
exports.AccessControlEntryAction = {
    Permit: "Permit",
    Deny: "Deny",
};
exports.AutoHealActionType = {
    Recycle: "Recycle",
    LogEvent: "LogEvent",
    CustomAction: "CustomAction",
};
exports.AzureResourceType = {
    Website: "Website",
    TrafficManager: "TrafficManager",
};
exports.AzureStorageType = {
    AzureFiles: "AzureFiles",
    AzureBlob: "AzureBlob",
};
exports.BuiltInAuthenticationProvider = {
    AzureActiveDirectory: "AzureActiveDirectory",
    Facebook: "Facebook",
    Google: "Google",
    MicrosoftAccount: "MicrosoftAccount",
    Twitter: "Twitter",
};
exports.ComputeModeOptions = {
    Shared: "Shared",
    Dedicated: "Dedicated",
    Dynamic: "Dynamic",
};
exports.ConnectionStringType = {
    MySql: "MySql",
    SQLServer: "SQLServer",
    SQLAzure: "SQLAzure",
    Custom: "Custom",
    NotificationHub: "NotificationHub",
    ServiceBus: "ServiceBus",
    EventHub: "EventHub",
    ApiHub: "ApiHub",
    DocDb: "DocDb",
    RedisCache: "RedisCache",
    PostgreSQL: "PostgreSQL",
};
exports.CustomHostNameDnsRecordType = {
    CName: "CName",
    A: "A",
};
exports.DatabaseType = {
    SqlAzure: "SqlAzure",
    MySql: "MySql",
    LocalMySql: "LocalMySql",
    PostgreSql: "PostgreSql",
};
exports.FrequencyUnit = {
    Day: "Day",
    Hour: "Hour",
};
exports.FtpsState = {
    AllAllowed: "AllAllowed",
    FtpsOnly: "FtpsOnly",
    Disabled: "Disabled",
};
exports.HostNameType = {
    Verified: "Verified",
    Managed: "Managed",
};
exports.HostType = {
    Standard: "Standard",
    Repository: "Repository",
};
exports.InternalLoadBalancingMode = {
    None: "None",
    Web: "Web",
    Publishing: "Publishing",
};
exports.IpFilterTag = {
    Default: "Default",
    XffProxy: "XffProxy",
};
exports.LogLevel = {
    Off: "Off",
    Verbose: "Verbose",
    Information: "Information",
    Warning: "Warning",
    Error: "Error",
};
exports.ManagedPipelineMode = {
    Integrated: "Integrated",
    Classic: "Classic",
};
exports.ManagedServiceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.PublicCertificateLocation = {
    CurrentUserMy: "CurrentUserMy",
    LocalMachineMy: "LocalMachineMy",
    Unknown: "Unknown",
};
exports.RedundancyMode = {
    None: "None",
    Manual: "Manual",
    Failover: "Failover",
    ActiveActive: "ActiveActive",
    GeoRedundant: "GeoRedundant",
};
exports.RouteType = {
    DEFAULT: "DEFAULT",
    INHERITED: "INHERITED",
    STATIC: "STATIC",
};
exports.ScmType = {
    None: "None",
    Dropbox: "Dropbox",
    Tfs: "Tfs",
    LocalGit: "LocalGit",
    GitHub: "GitHub",
    CodePlexGit: "CodePlexGit",
    CodePlexHg: "CodePlexHg",
    BitbucketGit: "BitbucketGit",
    BitbucketHg: "BitbucketHg",
    ExternalGit: "ExternalGit",
    ExternalHg: "ExternalHg",
    OneDrive: "OneDrive",
    VSO: "VSO",
};
exports.SiteLoadBalancing = {
    WeightedRoundRobin: "WeightedRoundRobin",
    LeastRequests: "LeastRequests",
    LeastResponseTime: "LeastResponseTime",
    WeightedTotalTraffic: "WeightedTotalTraffic",
    RequestHash: "RequestHash",
};
exports.SslState = {
    Disabled: "Disabled",
    SniEnabled: "SniEnabled",
    IpBasedEnabled: "IpBasedEnabled",
};
exports.SupportedTlsVersions = {
    SupportedTlsVersions_1_0: "1.0",
    SupportedTlsVersions_1_1: "1.1",
    SupportedTlsVersions_1_2: "1.2",
};
exports.UnauthenticatedClientAction = {
    RedirectToLoginPage: "RedirectToLoginPage",
    AllowAnonymous: "AllowAnonymous",
};
