// @pulumi/azure-native/network/v20200601/networkProfile.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Network profile resource.
 */
class NetworkProfile extends pulumi.CustomResource {
    /**
     * Create a NetworkProfile resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["containerNetworkInterfaceConfigurations"] = args ? args.containerNetworkInterfaceConfigurations : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["networkProfileName"] = args ? args.networkProfileName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["containerNetworkInterfaces"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["containerNetworkInterfaceConfigurations"] = undefined /*out*/;
            resourceInputs["containerNetworkInterfaces"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGuid"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:network:NetworkProfile" }, { type: "azure-native:network/v20180801:NetworkProfile" }, { type: "azure-native:network/v20181001:NetworkProfile" }, { type: "azure-native:network/v20181101:NetworkProfile" }, { type: "azure-native:network/v20181201:NetworkProfile" }, { type: "azure-native:network/v20190201:NetworkProfile" }, { type: "azure-native:network/v20190401:NetworkProfile" }, { type: "azure-native:network/v20190601:NetworkProfile" }, { type: "azure-native:network/v20190701:NetworkProfile" }, { type: "azure-native:network/v20190801:NetworkProfile" }, { type: "azure-native:network/v20190901:NetworkProfile" }, { type: "azure-native:network/v20191101:NetworkProfile" }, { type: "azure-native:network/v20191201:NetworkProfile" }, { type: "azure-native:network/v20200301:NetworkProfile" }, { type: "azure-native:network/v20200401:NetworkProfile" }, { type: "azure-native:network/v20200501:NetworkProfile" }, { type: "azure-native:network/v20200701:NetworkProfile" }, { type: "azure-native:network/v20200801:NetworkProfile" }, { type: "azure-native:network/v20201101:NetworkProfile" }, { type: "azure-native:network/v20210201:NetworkProfile" }, { type: "azure-native:network/v20210301:NetworkProfile" }, { type: "azure-native:network/v20210501:NetworkProfile" }, { type: "azure-native:network/v20210801:NetworkProfile" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(NetworkProfile.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing NetworkProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new NetworkProfile(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of NetworkProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === NetworkProfile.__pulumiType;
    }
}
exports.NetworkProfile = NetworkProfile;
/** @internal */
NetworkProfile.__pulumiType = 'azure-native:network/v20200601:NetworkProfile';
