// @pulumi/azure-native/consumption/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211001 = exports.v20210501 = exports.v20191101 = exports.v20191001 = exports.v20190601 = exports.v20190501preview = exports.v20190501 = exports.v20190401preview = exports.v20190101 = exports.v20181001 = exports.v20180831 = exports.v20180630 = exports.v20180331 = exports.v20180131 = exports.v20171230preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./budget"), exports);
__exportStar(require("./getBudget"), exports);
// Export enums:
__exportStar(require("../types/enums/consumption"), exports);
// Export sub-modules:
const v20171230preview = require("./v20171230preview");
exports.v20171230preview = v20171230preview;
const v20180131 = require("./v20180131");
exports.v20180131 = v20180131;
const v20180331 = require("./v20180331");
exports.v20180331 = v20180331;
const v20180630 = require("./v20180630");
exports.v20180630 = v20180630;
const v20180831 = require("./v20180831");
exports.v20180831 = v20180831;
const v20181001 = require("./v20181001");
exports.v20181001 = v20181001;
const v20190101 = require("./v20190101");
exports.v20190101 = v20190101;
const v20190401preview = require("./v20190401preview");
exports.v20190401preview = v20190401preview;
const v20190501 = require("./v20190501");
exports.v20190501 = v20190501;
const v20190501preview = require("./v20190501preview");
exports.v20190501preview = v20190501preview;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20191001 = require("./v20191001");
exports.v20191001 = v20191001;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20210501 = require("./v20210501");
exports.v20210501 = v20210501;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
// Import resources to register:
const budget_1 = require("./budget");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:consumption:Budget":
                return new budget_1.Budget(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "consumption", _module);
