// @pulumi/azure-native/compute/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211201 = exports.v20211101 = exports.v20211001 = exports.v20210801 = exports.v20210701 = exports.v20210401 = exports.v20210301 = exports.v20201201 = exports.v20201001preview = exports.v20200930 = exports.v20200630 = exports.v20200601 = exports.v20200501 = exports.v20191201 = exports.v20191101 = exports.v20190701 = exports.v20190301 = exports.v20181001 = exports.v20180930 = exports.v20180601 = exports.v20180401 = exports.v20171201 = exports.v20170330 = exports.v20160430preview = exports.v20160330 = exports.v20150615 = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./availabilitySet"), exports);
__exportStar(require("./capacityReservation"), exports);
__exportStar(require("./capacityReservationGroup"), exports);
__exportStar(require("./cloudService"), exports);
__exportStar(require("./dedicatedHost"), exports);
__exportStar(require("./dedicatedHostGroup"), exports);
__exportStar(require("./disk"), exports);
__exportStar(require("./diskAccess"), exports);
__exportStar(require("./diskAccessAPrivateEndpointConnection"), exports);
__exportStar(require("./diskEncryptionSet"), exports);
__exportStar(require("./gallery"), exports);
__exportStar(require("./galleryApplication"), exports);
__exportStar(require("./galleryApplicationVersion"), exports);
__exportStar(require("./galleryImage"), exports);
__exportStar(require("./galleryImageVersion"), exports);
__exportStar(require("./getAvailabilitySet"), exports);
__exportStar(require("./getCapacityReservation"), exports);
__exportStar(require("./getCapacityReservationGroup"), exports);
__exportStar(require("./getCloudService"), exports);
__exportStar(require("./getDedicatedHost"), exports);
__exportStar(require("./getDedicatedHostGroup"), exports);
__exportStar(require("./getDisk"), exports);
__exportStar(require("./getDiskAccess"), exports);
__exportStar(require("./getDiskAccessAPrivateEndpointConnection"), exports);
__exportStar(require("./getDiskEncryptionSet"), exports);
__exportStar(require("./getGallery"), exports);
__exportStar(require("./getGalleryApplication"), exports);
__exportStar(require("./getGalleryApplicationVersion"), exports);
__exportStar(require("./getGalleryImage"), exports);
__exportStar(require("./getGalleryImageVersion"), exports);
__exportStar(require("./getImage"), exports);
__exportStar(require("./getLogAnalyticExportRequestRateByInterval"), exports);
__exportStar(require("./getLogAnalyticExportThrottledRequests"), exports);
__exportStar(require("./getProximityPlacementGroup"), exports);
__exportStar(require("./getRestorePoint"), exports);
__exportStar(require("./getRestorePointCollection"), exports);
__exportStar(require("./getSnapshot"), exports);
__exportStar(require("./getSshPublicKey"), exports);
__exportStar(require("./getVirtualMachine"), exports);
__exportStar(require("./getVirtualMachineExtension"), exports);
__exportStar(require("./getVirtualMachineRunCommandByVirtualMachine"), exports);
__exportStar(require("./getVirtualMachineScaleSet"), exports);
__exportStar(require("./getVirtualMachineScaleSetExtension"), exports);
__exportStar(require("./getVirtualMachineScaleSetVM"), exports);
__exportStar(require("./getVirtualMachineScaleSetVMExtension"), exports);
__exportStar(require("./getVirtualMachineScaleSetVMRunCommand"), exports);
__exportStar(require("./image"), exports);
__exportStar(require("./proximityPlacementGroup"), exports);
__exportStar(require("./restorePoint"), exports);
__exportStar(require("./restorePointCollection"), exports);
__exportStar(require("./snapshot"), exports);
__exportStar(require("./sshPublicKey"), exports);
__exportStar(require("./virtualMachine"), exports);
__exportStar(require("./virtualMachineExtension"), exports);
__exportStar(require("./virtualMachineRunCommandByVirtualMachine"), exports);
__exportStar(require("./virtualMachineScaleSet"), exports);
__exportStar(require("./virtualMachineScaleSetExtension"), exports);
__exportStar(require("./virtualMachineScaleSetVM"), exports);
__exportStar(require("./virtualMachineScaleSetVMExtension"), exports);
__exportStar(require("./virtualMachineScaleSetVMRunCommand"), exports);
// Export enums:
__exportStar(require("../types/enums/compute"), exports);
// Export sub-modules:
const v20150615 = require("./v20150615");
exports.v20150615 = v20150615;
const v20160330 = require("./v20160330");
exports.v20160330 = v20160330;
const v20160430preview = require("./v20160430preview");
exports.v20160430preview = v20160430preview;
const v20170330 = require("./v20170330");
exports.v20170330 = v20170330;
const v20171201 = require("./v20171201");
exports.v20171201 = v20171201;
const v20180401 = require("./v20180401");
exports.v20180401 = v20180401;
const v20180601 = require("./v20180601");
exports.v20180601 = v20180601;
const v20180930 = require("./v20180930");
exports.v20180930 = v20180930;
const v20181001 = require("./v20181001");
exports.v20181001 = v20181001;
const v20190301 = require("./v20190301");
exports.v20190301 = v20190301;
const v20190701 = require("./v20190701");
exports.v20190701 = v20190701;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20191201 = require("./v20191201");
exports.v20191201 = v20191201;
const v20200501 = require("./v20200501");
exports.v20200501 = v20200501;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200630 = require("./v20200630");
exports.v20200630 = v20200630;
const v20200930 = require("./v20200930");
exports.v20200930 = v20200930;
const v20201001preview = require("./v20201001preview");
exports.v20201001preview = v20201001preview;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
const v20211101 = require("./v20211101");
exports.v20211101 = v20211101;
const v20211201 = require("./v20211201");
exports.v20211201 = v20211201;
// Import resources to register:
const availabilitySet_1 = require("./availabilitySet");
const capacityReservation_1 = require("./capacityReservation");
const capacityReservationGroup_1 = require("./capacityReservationGroup");
const cloudService_1 = require("./cloudService");
const dedicatedHost_1 = require("./dedicatedHost");
const dedicatedHostGroup_1 = require("./dedicatedHostGroup");
const disk_1 = require("./disk");
const diskAccess_1 = require("./diskAccess");
const diskAccessAPrivateEndpointConnection_1 = require("./diskAccessAPrivateEndpointConnection");
const diskEncryptionSet_1 = require("./diskEncryptionSet");
const gallery_1 = require("./gallery");
const galleryApplication_1 = require("./galleryApplication");
const galleryApplicationVersion_1 = require("./galleryApplicationVersion");
const galleryImage_1 = require("./galleryImage");
const galleryImageVersion_1 = require("./galleryImageVersion");
const image_1 = require("./image");
const proximityPlacementGroup_1 = require("./proximityPlacementGroup");
const restorePoint_1 = require("./restorePoint");
const restorePointCollection_1 = require("./restorePointCollection");
const snapshot_1 = require("./snapshot");
const sshPublicKey_1 = require("./sshPublicKey");
const virtualMachine_1 = require("./virtualMachine");
const virtualMachineExtension_1 = require("./virtualMachineExtension");
const virtualMachineRunCommandByVirtualMachine_1 = require("./virtualMachineRunCommandByVirtualMachine");
const virtualMachineScaleSet_1 = require("./virtualMachineScaleSet");
const virtualMachineScaleSetExtension_1 = require("./virtualMachineScaleSetExtension");
const virtualMachineScaleSetVM_1 = require("./virtualMachineScaleSetVM");
const virtualMachineScaleSetVMExtension_1 = require("./virtualMachineScaleSetVMExtension");
const virtualMachineScaleSetVMRunCommand_1 = require("./virtualMachineScaleSetVMRunCommand");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:compute:AvailabilitySet":
                return new availabilitySet_1.AvailabilitySet(name, undefined, { urn });
            case "azure-native:compute:CapacityReservation":
                return new capacityReservation_1.CapacityReservation(name, undefined, { urn });
            case "azure-native:compute:CapacityReservationGroup":
                return new capacityReservationGroup_1.CapacityReservationGroup(name, undefined, { urn });
            case "azure-native:compute:CloudService":
                return new cloudService_1.CloudService(name, undefined, { urn });
            case "azure-native:compute:DedicatedHost":
                return new dedicatedHost_1.DedicatedHost(name, undefined, { urn });
            case "azure-native:compute:DedicatedHostGroup":
                return new dedicatedHostGroup_1.DedicatedHostGroup(name, undefined, { urn });
            case "azure-native:compute:Disk":
                return new disk_1.Disk(name, undefined, { urn });
            case "azure-native:compute:DiskAccess":
                return new diskAccess_1.DiskAccess(name, undefined, { urn });
            case "azure-native:compute:DiskAccessAPrivateEndpointConnection":
                return new diskAccessAPrivateEndpointConnection_1.DiskAccessAPrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:compute:DiskEncryptionSet":
                return new diskEncryptionSet_1.DiskEncryptionSet(name, undefined, { urn });
            case "azure-native:compute:Gallery":
                return new gallery_1.Gallery(name, undefined, { urn });
            case "azure-native:compute:GalleryApplication":
                return new galleryApplication_1.GalleryApplication(name, undefined, { urn });
            case "azure-native:compute:GalleryApplicationVersion":
                return new galleryApplicationVersion_1.GalleryApplicationVersion(name, undefined, { urn });
            case "azure-native:compute:GalleryImage":
                return new galleryImage_1.GalleryImage(name, undefined, { urn });
            case "azure-native:compute:GalleryImageVersion":
                return new galleryImageVersion_1.GalleryImageVersion(name, undefined, { urn });
            case "azure-native:compute:Image":
                return new image_1.Image(name, undefined, { urn });
            case "azure-native:compute:ProximityPlacementGroup":
                return new proximityPlacementGroup_1.ProximityPlacementGroup(name, undefined, { urn });
            case "azure-native:compute:RestorePoint":
                return new restorePoint_1.RestorePoint(name, undefined, { urn });
            case "azure-native:compute:RestorePointCollection":
                return new restorePointCollection_1.RestorePointCollection(name, undefined, { urn });
            case "azure-native:compute:Snapshot":
                return new snapshot_1.Snapshot(name, undefined, { urn });
            case "azure-native:compute:SshPublicKey":
                return new sshPublicKey_1.SshPublicKey(name, undefined, { urn });
            case "azure-native:compute:VirtualMachine":
                return new virtualMachine_1.VirtualMachine(name, undefined, { urn });
            case "azure-native:compute:VirtualMachineExtension":
                return new virtualMachineExtension_1.VirtualMachineExtension(name, undefined, { urn });
            case "azure-native:compute:VirtualMachineRunCommandByVirtualMachine":
                return new virtualMachineRunCommandByVirtualMachine_1.VirtualMachineRunCommandByVirtualMachine(name, undefined, { urn });
            case "azure-native:compute:VirtualMachineScaleSet":
                return new virtualMachineScaleSet_1.VirtualMachineScaleSet(name, undefined, { urn });
            case "azure-native:compute:VirtualMachineScaleSetExtension":
                return new virtualMachineScaleSetExtension_1.VirtualMachineScaleSetExtension(name, undefined, { urn });
            case "azure-native:compute:VirtualMachineScaleSetVM":
                return new virtualMachineScaleSetVM_1.VirtualMachineScaleSetVM(name, undefined, { urn });
            case "azure-native:compute:VirtualMachineScaleSetVMExtension":
                return new virtualMachineScaleSetVMExtension_1.VirtualMachineScaleSetVMExtension(name, undefined, { urn });
            case "azure-native:compute:VirtualMachineScaleSetVMRunCommand":
                return new virtualMachineScaleSetVMRunCommand_1.VirtualMachineScaleSetVMRunCommand(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "compute", _module);
