// @pulumi/azure-native/web/v20150801/managedHostingEnvironment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedHostingEnvironment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Description of an hostingEnvironment (App Service Environment)
 *
 * @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.
 */
class ManagedHostingEnvironment extends pulumi.CustomResource {
    /**
     * Create a ManagedHostingEnvironment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ManagedHostingEnvironment is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.status === undefined) && !opts.urn) {
                throw new Error("Missing required property 'status'");
            }
            resourceInputs["allowedMultiSizes"] = args ? args.allowedMultiSizes : undefined;
            resourceInputs["allowedWorkerSizes"] = args ? args.allowedWorkerSizes : undefined;
            resourceInputs["apiManagementAccountId"] = args ? args.apiManagementAccountId : undefined;
            resourceInputs["clusterSettings"] = args ? args.clusterSettings : undefined;
            resourceInputs["databaseEdition"] = args ? args.databaseEdition : undefined;
            resourceInputs["databaseServiceObjective"] = args ? args.databaseServiceObjective : undefined;
            resourceInputs["dnsSuffix"] = args ? args.dnsSuffix : undefined;
            resourceInputs["environmentCapacities"] = args ? args.environmentCapacities : undefined;
            resourceInputs["environmentIsHealthy"] = args ? args.environmentIsHealthy : undefined;
            resourceInputs["environmentStatus"] = args ? args.environmentStatus : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["internalLoadBalancingMode"] = args ? args.internalLoadBalancingMode : undefined;
            resourceInputs["ipsslAddressCount"] = args ? args.ipsslAddressCount : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["lastAction"] = args ? args.lastAction : undefined;
            resourceInputs["lastActionResult"] = args ? args.lastActionResult : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["maximumNumberOfMachines"] = args ? args.maximumNumberOfMachines : undefined;
            resourceInputs["multiRoleCount"] = args ? args.multiRoleCount : undefined;
            resourceInputs["multiSize"] = args ? args.multiSize : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["networkAccessControlList"] = args ? args.networkAccessControlList : undefined;
            resourceInputs["provisioningState"] = args ? args.provisioningState : undefined;
            resourceInputs["resourceGroup"] = args ? args.resourceGroup : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["status"] = args ? args.status : undefined;
            resourceInputs["subscriptionId"] = args ? args.subscriptionId : undefined;
            resourceInputs["suspended"] = args ? args.suspended : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["upgradeDomains"] = args ? args.upgradeDomains : undefined;
            resourceInputs["vipMappings"] = args ? args.vipMappings : undefined;
            resourceInputs["virtualNetwork"] = args ? args.virtualNetwork : undefined;
            resourceInputs["vnetName"] = args ? args.vnetName : undefined;
            resourceInputs["vnetResourceGroupName"] = args ? args.vnetResourceGroupName : undefined;
            resourceInputs["vnetSubnetName"] = args ? args.vnetSubnetName : undefined;
            resourceInputs["workerPools"] = args ? args.workerPools : undefined;
        }
        else {
            resourceInputs["allowedMultiSizes"] = undefined /*out*/;
            resourceInputs["allowedWorkerSizes"] = undefined /*out*/;
            resourceInputs["apiManagementAccountId"] = undefined /*out*/;
            resourceInputs["clusterSettings"] = undefined /*out*/;
            resourceInputs["databaseEdition"] = undefined /*out*/;
            resourceInputs["databaseServiceObjective"] = undefined /*out*/;
            resourceInputs["dnsSuffix"] = undefined /*out*/;
            resourceInputs["environmentCapacities"] = undefined /*out*/;
            resourceInputs["environmentIsHealthy"] = undefined /*out*/;
            resourceInputs["environmentStatus"] = undefined /*out*/;
            resourceInputs["internalLoadBalancingMode"] = undefined /*out*/;
            resourceInputs["ipsslAddressCount"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["lastAction"] = undefined /*out*/;
            resourceInputs["lastActionResult"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maximumNumberOfMachines"] = undefined /*out*/;
            resourceInputs["multiRoleCount"] = undefined /*out*/;
            resourceInputs["multiSize"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["networkAccessControlList"] = undefined /*out*/;
            resourceInputs["provisioningState"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["subscriptionId"] = undefined /*out*/;
            resourceInputs["suspended"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["upgradeDomains"] = undefined /*out*/;
            resourceInputs["vipMappings"] = undefined /*out*/;
            resourceInputs["virtualNetwork"] = undefined /*out*/;
            resourceInputs["vnetName"] = undefined /*out*/;
            resourceInputs["vnetResourceGroupName"] = undefined /*out*/;
            resourceInputs["vnetSubnetName"] = undefined /*out*/;
            resourceInputs["workerPools"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:ManagedHostingEnvironment" }, { type: "azure-native:web/v20160901:ManagedHostingEnvironment" }, { type: "azure-native:web/v20180201:ManagedHostingEnvironment" }, { type: "azure-native:web/v20190801:ManagedHostingEnvironment" }, { type: "azure-native:web/v20200601:ManagedHostingEnvironment" }, { type: "azure-native:web/v20200901:ManagedHostingEnvironment" }, { type: "azure-native:web/v20201001:ManagedHostingEnvironment" }, { type: "azure-native:web/v20201201:ManagedHostingEnvironment" }, { type: "azure-native:web/v20210101:ManagedHostingEnvironment" }, { type: "azure-native:web/v20210115:ManagedHostingEnvironment" }, { type: "azure-native:web/v20210201:ManagedHostingEnvironment" }, { type: "azure-native:web/v20210301:ManagedHostingEnvironment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagedHostingEnvironment.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ManagedHostingEnvironment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ManagedHostingEnvironment is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        return new ManagedHostingEnvironment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagedHostingEnvironment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedHostingEnvironment.__pulumiType;
    }
}
exports.ManagedHostingEnvironment = ManagedHostingEnvironment;
/** @internal */
ManagedHostingEnvironment.__pulumiType = 'azure-native:web/v20150801:ManagedHostingEnvironment';
