// @pulumi/azure-native/web/v20150801/serverFarm.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerFarm = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * App Service Plan Model
 *
 * @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.
 */
class ServerFarm extends pulumi.CustomResource {
    /**
     * Create a ServerFarm resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ServerFarm is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["adminSiteName"] = args ? args.adminSiteName : undefined;
            resourceInputs["allowPendingState"] = args ? args.allowPendingState : undefined;
            resourceInputs["hostingEnvironmentProfile"] = args ? args.hostingEnvironmentProfile : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["maximumNumberOfWorkers"] = args ? args.maximumNumberOfWorkers : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["perSiteScaling"] = args ? args.perSiteScaling : undefined;
            resourceInputs["reserved"] = args ? args.reserved : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["workerTierName"] = args ? args.workerTierName : undefined;
            resourceInputs["geoRegion"] = undefined /*out*/;
            resourceInputs["numberOfSites"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["subscription"] = undefined /*out*/;
        }
        else {
            resourceInputs["adminSiteName"] = undefined /*out*/;
            resourceInputs["geoRegion"] = undefined /*out*/;
            resourceInputs["hostingEnvironmentProfile"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["maximumNumberOfWorkers"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numberOfSites"] = undefined /*out*/;
            resourceInputs["perSiteScaling"] = undefined /*out*/;
            resourceInputs["reserved"] = undefined /*out*/;
            resourceInputs["resourceGroup"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["subscription"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["workerTierName"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:ServerFarm" }, { type: "azure-native:web/v20160901:ServerFarm" }, { type: "azure-native:web/v20180201:ServerFarm" }, { type: "azure-native:web/v20190801:ServerFarm" }, { type: "azure-native:web/v20200601:ServerFarm" }, { type: "azure-native:web/v20200901:ServerFarm" }, { type: "azure-native:web/v20201001:ServerFarm" }, { type: "azure-native:web/v20201201:ServerFarm" }, { type: "azure-native:web/v20210101:ServerFarm" }, { type: "azure-native:web/v20210115:ServerFarm" }, { type: "azure-native:web/v20210201:ServerFarm" }, { type: "azure-native:web/v20210301:ServerFarm" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ServerFarm.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ServerFarm resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ServerFarm is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        return new ServerFarm(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ServerFarm.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServerFarm.__pulumiType;
    }
}
exports.ServerFarm = ServerFarm;
/** @internal */
ServerFarm.__pulumiType = 'azure-native:web/v20150801:ServerFarm';
