// @pulumi/azure-native/web/v20150801/siteLogsConfig.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteLogsConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Configuration of Azure web site
 *
 * @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.
 */
class SiteLogsConfig extends pulumi.CustomResource {
    /**
     * Create a SiteLogsConfig resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SiteLogsConfig is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["applicationLogs"] = args ? args.applicationLogs : undefined;
            resourceInputs["detailedErrorMessages"] = args ? args.detailedErrorMessages : undefined;
            resourceInputs["failedRequestsTracing"] = args ? args.failedRequestsTracing : undefined;
            resourceInputs["httpLogs"] = args ? args.httpLogs : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
        }
        else {
            resourceInputs["applicationLogs"] = undefined /*out*/;
            resourceInputs["detailedErrorMessages"] = undefined /*out*/;
            resourceInputs["failedRequestsTracing"] = undefined /*out*/;
            resourceInputs["httpLogs"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:SiteLogsConfig" }, { type: "azure-native:web/v20160801:SiteLogsConfig" }, { type: "azure-native:web/v20180201:SiteLogsConfig" }, { type: "azure-native:web/v20181101:SiteLogsConfig" }, { type: "azure-native:web/v20190801:SiteLogsConfig" }, { type: "azure-native:web/v20200601:SiteLogsConfig" }, { type: "azure-native:web/v20200901:SiteLogsConfig" }, { type: "azure-native:web/v20201001:SiteLogsConfig" }, { type: "azure-native:web/v20201201:SiteLogsConfig" }, { type: "azure-native:web/v20210101:SiteLogsConfig" }, { type: "azure-native:web/v20210115:SiteLogsConfig" }, { type: "azure-native:web/v20210201:SiteLogsConfig" }, { type: "azure-native:web/v20210301:SiteLogsConfig" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteLogsConfig.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SiteLogsConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SiteLogsConfig is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        return new SiteLogsConfig(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteLogsConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteLogsConfig.__pulumiType;
    }
}
exports.SiteLogsConfig = SiteLogsConfig;
/** @internal */
SiteLogsConfig.__pulumiType = 'azure-native:web/v20150801:SiteLogsConfig';
