// @pulumi/azure-native/web/v20150801/siteDeploymentSlot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteDeploymentSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Represents user credentials used for publishing activity
 *
 * @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.
 */
class SiteDeploymentSlot extends pulumi.CustomResource {
    /**
     * Create a SiteDeploymentSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SiteDeploymentSlot is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["active"] = args ? args.active : undefined;
            resourceInputs["author"] = args ? args.author : undefined;
            resourceInputs["authorEmail"] = args ? args.authorEmail : undefined;
            resourceInputs["deployer"] = args ? args.deployer : undefined;
            resourceInputs["details"] = args ? args.details : undefined;
            resourceInputs["endTime"] = args ? args.endTime : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["message"] = args ? args.message : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["slot"] = args ? args.slot : undefined;
            resourceInputs["startTime"] = args ? args.startTime : undefined;
            resourceInputs["status"] = args ? args.status : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
        }
        else {
            resourceInputs["active"] = undefined /*out*/;
            resourceInputs["author"] = undefined /*out*/;
            resourceInputs["authorEmail"] = undefined /*out*/;
            resourceInputs["deployer"] = undefined /*out*/;
            resourceInputs["details"] = undefined /*out*/;
            resourceInputs["endTime"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["message"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["startTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:SiteDeploymentSlot" }, { type: "azure-native:web/v20160801:SiteDeploymentSlot" }, { type: "azure-native:web/v20180201:SiteDeploymentSlot" }, { type: "azure-native:web/v20181101:SiteDeploymentSlot" }, { type: "azure-native:web/v20190801:SiteDeploymentSlot" }, { type: "azure-native:web/v20200601:SiteDeploymentSlot" }, { type: "azure-native:web/v20200901:SiteDeploymentSlot" }, { type: "azure-native:web/v20201001:SiteDeploymentSlot" }, { type: "azure-native:web/v20201201:SiteDeploymentSlot" }, { type: "azure-native:web/v20210101:SiteDeploymentSlot" }, { type: "azure-native:web/v20210115:SiteDeploymentSlot" }, { type: "azure-native:web/v20210201:SiteDeploymentSlot" }, { type: "azure-native:web/v20210301:SiteDeploymentSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteDeploymentSlot.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SiteDeploymentSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SiteDeploymentSlot is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        return new SiteDeploymentSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteDeploymentSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteDeploymentSlot.__pulumiType;
    }
}
exports.SiteDeploymentSlot = SiteDeploymentSlot;
/** @internal */
SiteDeploymentSlot.__pulumiType = 'azure-native:web/v20150801:SiteDeploymentSlot';
