// @pulumi/azure-native/web/v20150801/siteConnectionStrings.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteConnectionStrings = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * String dictionary resource
 *
 * @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.
 */
class SiteConnectionStrings extends pulumi.CustomResource {
    /**
     * Create a SiteConnectionStrings resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SiteConnectionStrings is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
        }
        else {
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:SiteConnectionStrings" }, { type: "azure-native:web/v20160801:SiteConnectionStrings" }, { type: "azure-native:web/v20180201:SiteConnectionStrings" }, { type: "azure-native:web/v20181101:SiteConnectionStrings" }, { type: "azure-native:web/v20190801:SiteConnectionStrings" }, { type: "azure-native:web/v20200601:SiteConnectionStrings" }, { type: "azure-native:web/v20200901:SiteConnectionStrings" }, { type: "azure-native:web/v20201001:SiteConnectionStrings" }, { type: "azure-native:web/v20201201:SiteConnectionStrings" }, { type: "azure-native:web/v20210101:SiteConnectionStrings" }, { type: "azure-native:web/v20210115:SiteConnectionStrings" }, { type: "azure-native:web/v20210201:SiteConnectionStrings" }, { type: "azure-native:web/v20210301:SiteConnectionStrings" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteConnectionStrings.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SiteConnectionStrings resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SiteConnectionStrings is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        return new SiteConnectionStrings(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteConnectionStrings.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteConnectionStrings.__pulumiType;
    }
}
exports.SiteConnectionStrings = SiteConnectionStrings;
/** @internal */
SiteConnectionStrings.__pulumiType = 'azure-native:web/v20150801:SiteConnectionStrings';
