// @pulumi/azure-native/web/v20150801/siteVNETConnection.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteVNETConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * VNETInfo contract. This contract is public and is a stripped down version of VNETInfoInternal
 *
 * @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.
 */
class SiteVNETConnection extends pulumi.CustomResource {
    /**
     * Create a SiteVNETConnection resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SiteVNETConnection is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["certBlob"] = args ? args.certBlob : undefined;
            resourceInputs["certThumbprint"] = args ? args.certThumbprint : undefined;
            resourceInputs["dnsServers"] = args ? args.dnsServers : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resyncRequired"] = args ? args.resyncRequired : undefined;
            resourceInputs["routes"] = args ? args.routes : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["vnetName"] = args ? args.vnetName : undefined;
            resourceInputs["vnetResourceId"] = args ? args.vnetResourceId : undefined;
        }
        else {
            resourceInputs["certBlob"] = undefined /*out*/;
            resourceInputs["certThumbprint"] = undefined /*out*/;
            resourceInputs["dnsServers"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["resyncRequired"] = undefined /*out*/;
            resourceInputs["routes"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["vnetResourceId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:SiteVNETConnection" }, { type: "azure-native:web/v20160801:SiteVNETConnection" }, { type: "azure-native:web/v20180201:SiteVNETConnection" }, { type: "azure-native:web/v20181101:SiteVNETConnection" }, { type: "azure-native:web/v20190801:SiteVNETConnection" }, { type: "azure-native:web/v20200601:SiteVNETConnection" }, { type: "azure-native:web/v20200901:SiteVNETConnection" }, { type: "azure-native:web/v20201001:SiteVNETConnection" }, { type: "azure-native:web/v20201201:SiteVNETConnection" }, { type: "azure-native:web/v20210101:SiteVNETConnection" }, { type: "azure-native:web/v20210115:SiteVNETConnection" }, { type: "azure-native:web/v20210201:SiteVNETConnection" }, { type: "azure-native:web/v20210301:SiteVNETConnection" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteVNETConnection.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SiteVNETConnection resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SiteVNETConnection is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        return new SiteVNETConnection(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteVNETConnection.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteVNETConnection.__pulumiType;
    }
}
exports.SiteVNETConnection = SiteVNETConnection;
/** @internal */
SiteVNETConnection.__pulumiType = 'azure-native:web/v20150801:SiteVNETConnection';
