// @pulumi/azure-native/web/v20150801/siteBackupConfiguration.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteBackupConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Description of a backup which will be performed
 *
 * @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.
 */
class SiteBackupConfiguration extends pulumi.CustomResource {
    /**
     * Create a SiteBackupConfiguration resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SiteBackupConfiguration is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["backupSchedule"] = args ? args.backupSchedule : undefined;
            resourceInputs["databases"] = args ? args.databases : undefined;
            resourceInputs["enabled"] = args ? args.enabled : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["storageAccountUrl"] = args ? args.storageAccountUrl : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
        }
        else {
            resourceInputs["backupSchedule"] = undefined /*out*/;
            resourceInputs["databases"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["storageAccountUrl"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:SiteBackupConfiguration" }, { type: "azure-native:web/v20160801:SiteBackupConfiguration" }, { type: "azure-native:web/v20180201:SiteBackupConfiguration" }, { type: "azure-native:web/v20181101:SiteBackupConfiguration" }, { type: "azure-native:web/v20190801:SiteBackupConfiguration" }, { type: "azure-native:web/v20200601:SiteBackupConfiguration" }, { type: "azure-native:web/v20200901:SiteBackupConfiguration" }, { type: "azure-native:web/v20201001:SiteBackupConfiguration" }, { type: "azure-native:web/v20201201:SiteBackupConfiguration" }, { type: "azure-native:web/v20210101:SiteBackupConfiguration" }, { type: "azure-native:web/v20210115:SiteBackupConfiguration" }, { type: "azure-native:web/v20210201:SiteBackupConfiguration" }, { type: "azure-native:web/v20210301:SiteBackupConfiguration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteBackupConfiguration.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SiteBackupConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SiteBackupConfiguration is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        return new SiteBackupConfiguration(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteBackupConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteBackupConfiguration.__pulumiType;
    }
}
exports.SiteBackupConfiguration = SiteBackupConfiguration;
/** @internal */
SiteBackupConfiguration.__pulumiType = 'azure-native:web/v20150801:SiteBackupConfiguration';
