// @pulumi/azure-native/web/v20150801/siteRelayServiceConnectionSlot.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SiteRelayServiceConnectionSlot = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Class that represents a BizTalk Hybrid Connection
 *
 * @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.
 */
class SiteRelayServiceConnectionSlot extends pulumi.CustomResource {
    /**
     * Create a SiteRelayServiceConnectionSlot resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("SiteRelayServiceConnectionSlot is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.name === undefined) && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.slot === undefined) && !opts.urn) {
                throw new Error("Missing required property 'slot'");
            }
            resourceInputs["biztalkUri"] = args ? args.biztalkUri : undefined;
            resourceInputs["entityConnectionString"] = args ? args.entityConnectionString : undefined;
            resourceInputs["entityName"] = args ? args.entityName : undefined;
            resourceInputs["hostname"] = args ? args.hostname : undefined;
            resourceInputs["id"] = args ? args.id : undefined;
            resourceInputs["kind"] = args ? args.kind : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["port"] = args ? args.port : undefined;
            resourceInputs["resourceConnectionString"] = args ? args.resourceConnectionString : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["resourceType"] = args ? args.resourceType : undefined;
            resourceInputs["slot"] = args ? args.slot : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
        }
        else {
            resourceInputs["biztalkUri"] = undefined /*out*/;
            resourceInputs["entityConnectionString"] = undefined /*out*/;
            resourceInputs["entityName"] = undefined /*out*/;
            resourceInputs["hostname"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["port"] = undefined /*out*/;
            resourceInputs["resourceConnectionString"] = undefined /*out*/;
            resourceInputs["resourceType"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:web:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20160801:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20180201:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20181101:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20190801:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20200601:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20200901:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20201001:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20201201:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20210101:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20210115:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20210201:SiteRelayServiceConnectionSlot" }, { type: "azure-native:web/v20210301:SiteRelayServiceConnectionSlot" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(SiteRelayServiceConnectionSlot.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing SiteRelayServiceConnectionSlot resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("SiteRelayServiceConnectionSlot is deprecated: Version v20150801 will be removed in the next major version of the provider. Upgrade to version v20150801preview or later.");
        return new SiteRelayServiceConnectionSlot(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of SiteRelayServiceConnectionSlot.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SiteRelayServiceConnectionSlot.__pulumiType;
    }
}
exports.SiteRelayServiceConnectionSlot = SiteRelayServiceConnectionSlot;
/** @internal */
SiteRelayServiceConnectionSlot.__pulumiType = 'azure-native:web/v20150801:SiteRelayServiceConnectionSlot';
