// @pulumi/azure-native/migrate/v20171111preview/assessment.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Assessment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * An assessment created for a group in the Migration project.
 *
 * @deprecated Version v20171111preview will be removed in the next major version of the provider. Upgrade to version v20180202 or later.
 */
class Assessment extends pulumi.CustomResource {
    /**
     * Create a Assessment resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20171111preview will be removed in the next major version of the provider. Upgrade to version v20180202 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Assessment is deprecated: Version v20171111preview will be removed in the next major version of the provider. Upgrade to version v20180202 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.azureHybridUseBenefit === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureHybridUseBenefit'");
            }
            if ((!args || args.azureLocation === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureLocation'");
            }
            if ((!args || args.azureOfferCode === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureOfferCode'");
            }
            if ((!args || args.azurePricingTier === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azurePricingTier'");
            }
            if ((!args || args.azureStorageRedundancy === undefined) && !opts.urn) {
                throw new Error("Missing required property 'azureStorageRedundancy'");
            }
            if ((!args || args.currency === undefined) && !opts.urn) {
                throw new Error("Missing required property 'currency'");
            }
            if ((!args || args.discountPercentage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'discountPercentage'");
            }
            if ((!args || args.groupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupName'");
            }
            if ((!args || args.percentile === undefined) && !opts.urn) {
                throw new Error("Missing required property 'percentile'");
            }
            if ((!args || args.projectName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.scalingFactor === undefined) && !opts.urn) {
                throw new Error("Missing required property 'scalingFactor'");
            }
            if ((!args || args.stage === undefined) && !opts.urn) {
                throw new Error("Missing required property 'stage'");
            }
            if ((!args || args.timeRange === undefined) && !opts.urn) {
                throw new Error("Missing required property 'timeRange'");
            }
            resourceInputs["assessmentName"] = args ? args.assessmentName : undefined;
            resourceInputs["azureHybridUseBenefit"] = args ? args.azureHybridUseBenefit : undefined;
            resourceInputs["azureLocation"] = args ? args.azureLocation : undefined;
            resourceInputs["azureOfferCode"] = args ? args.azureOfferCode : undefined;
            resourceInputs["azurePricingTier"] = args ? args.azurePricingTier : undefined;
            resourceInputs["azureStorageRedundancy"] = args ? args.azureStorageRedundancy : undefined;
            resourceInputs["currency"] = args ? args.currency : undefined;
            resourceInputs["discountPercentage"] = args ? args.discountPercentage : undefined;
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["groupName"] = args ? args.groupName : undefined;
            resourceInputs["percentile"] = args ? args.percentile : undefined;
            resourceInputs["projectName"] = args ? args.projectName : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scalingFactor"] = args ? args.scalingFactor : undefined;
            resourceInputs["stage"] = args ? args.stage : undefined;
            resourceInputs["timeRange"] = args ? args.timeRange : undefined;
            resourceInputs["createdTimestamp"] = undefined /*out*/;
            resourceInputs["monthlyBandwidthCost"] = undefined /*out*/;
            resourceInputs["monthlyComputeCost"] = undefined /*out*/;
            resourceInputs["monthlyStorageCost"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numberOfMachines"] = undefined /*out*/;
            resourceInputs["pricesTimestamp"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updatedTimestamp"] = undefined /*out*/;
        }
        else {
            resourceInputs["azureHybridUseBenefit"] = undefined /*out*/;
            resourceInputs["azureLocation"] = undefined /*out*/;
            resourceInputs["azureOfferCode"] = undefined /*out*/;
            resourceInputs["azurePricingTier"] = undefined /*out*/;
            resourceInputs["azureStorageRedundancy"] = undefined /*out*/;
            resourceInputs["createdTimestamp"] = undefined /*out*/;
            resourceInputs["currency"] = undefined /*out*/;
            resourceInputs["discountPercentage"] = undefined /*out*/;
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["monthlyBandwidthCost"] = undefined /*out*/;
            resourceInputs["monthlyComputeCost"] = undefined /*out*/;
            resourceInputs["monthlyStorageCost"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["numberOfMachines"] = undefined /*out*/;
            resourceInputs["percentile"] = undefined /*out*/;
            resourceInputs["pricesTimestamp"] = undefined /*out*/;
            resourceInputs["scalingFactor"] = undefined /*out*/;
            resourceInputs["stage"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["timeRange"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updatedTimestamp"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:migrate/v20180202:Assessment" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Assessment.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Assessment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Assessment is deprecated: Version v20171111preview will be removed in the next major version of the provider. Upgrade to version v20180202 or later.");
        return new Assessment(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Assessment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Assessment.__pulumiType;
    }
}
exports.Assessment = Assessment;
/** @internal */
Assessment.__pulumiType = 'azure-native:migrate/v20171111preview:Assessment';
