// @pulumi/azure-native/containerregistry/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20211201preview = exports.v20210901 = exports.v20210801preview = exports.v20210601preview = exports.v20201101preview = exports.v20191201preview = exports.v20190601preview = exports.v20190501preview = exports.v20190501 = exports.v20190401 = exports.v20180901 = exports.v20180201preview = exports.v20171001 = exports.v20170601preview = exports.v20170301 = exports.v20160627preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./agentPool"), exports);
__exportStar(require("./connectedRegistry"), exports);
__exportStar(require("./exportPipeline"), exports);
__exportStar(require("./getAgentPool"), exports);
__exportStar(require("./getBuildLogLink"), exports);
__exportStar(require("./getConnectedRegistry"), exports);
__exportStar(require("./getExportPipeline"), exports);
__exportStar(require("./getImportPipeline"), exports);
__exportStar(require("./getPipelineRun"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getRegistry"), exports);
__exportStar(require("./getRegistryBuildSourceUploadUrl"), exports);
__exportStar(require("./getRegistryCredentials"), exports);
__exportStar(require("./getReplication"), exports);
__exportStar(require("./getScopeMap"), exports);
__exportStar(require("./getTask"), exports);
__exportStar(require("./getTaskRun"), exports);
__exportStar(require("./getToken"), exports);
__exportStar(require("./getWebhook"), exports);
__exportStar(require("./getWebhookCallbackConfig"), exports);
__exportStar(require("./importPipeline"), exports);
__exportStar(require("./listAgentPoolQueueStatus"), exports);
__exportStar(require("./listBuildStepBuildArguments"), exports);
__exportStar(require("./listBuildTaskSourceRepositoryProperties"), exports);
__exportStar(require("./listRegistryBuildSourceUploadUrl"), exports);
__exportStar(require("./listRegistryCredentials"), exports);
__exportStar(require("./listRunLogSasUrl"), exports);
__exportStar(require("./listTaskDetails"), exports);
__exportStar(require("./listTaskRunDetails"), exports);
__exportStar(require("./listWebhookEvents"), exports);
__exportStar(require("./pipelineRun"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./registry"), exports);
__exportStar(require("./replication"), exports);
__exportStar(require("./scopeMap"), exports);
__exportStar(require("./task"), exports);
__exportStar(require("./taskRun"), exports);
__exportStar(require("./token"), exports);
__exportStar(require("./webhook"), exports);
// Export enums:
__exportStar(require("../types/enums/containerregistry"), exports);
// Export sub-modules:
const v20160627preview = require("./v20160627preview");
exports.v20160627preview = v20160627preview;
const v20170301 = require("./v20170301");
exports.v20170301 = v20170301;
const v20170601preview = require("./v20170601preview");
exports.v20170601preview = v20170601preview;
const v20171001 = require("./v20171001");
exports.v20171001 = v20171001;
const v20180201preview = require("./v20180201preview");
exports.v20180201preview = v20180201preview;
const v20180901 = require("./v20180901");
exports.v20180901 = v20180901;
const v20190401 = require("./v20190401");
exports.v20190401 = v20190401;
const v20190501 = require("./v20190501");
exports.v20190501 = v20190501;
const v20190501preview = require("./v20190501preview");
exports.v20190501preview = v20190501preview;
const v20190601preview = require("./v20190601preview");
exports.v20190601preview = v20190601preview;
const v20191201preview = require("./v20191201preview");
exports.v20191201preview = v20191201preview;
const v20201101preview = require("./v20201101preview");
exports.v20201101preview = v20201101preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20210801preview = require("./v20210801preview");
exports.v20210801preview = v20210801preview;
const v20210901 = require("./v20210901");
exports.v20210901 = v20210901;
const v20211201preview = require("./v20211201preview");
exports.v20211201preview = v20211201preview;
// Import resources to register:
const agentPool_1 = require("./agentPool");
const connectedRegistry_1 = require("./connectedRegistry");
const exportPipeline_1 = require("./exportPipeline");
const importPipeline_1 = require("./importPipeline");
const pipelineRun_1 = require("./pipelineRun");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const registry_1 = require("./registry");
const replication_1 = require("./replication");
const scopeMap_1 = require("./scopeMap");
const task_1 = require("./task");
const taskRun_1 = require("./taskRun");
const token_1 = require("./token");
const webhook_1 = require("./webhook");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:containerregistry:AgentPool":
                return new agentPool_1.AgentPool(name, undefined, { urn });
            case "azure-native:containerregistry:ConnectedRegistry":
                return new connectedRegistry_1.ConnectedRegistry(name, undefined, { urn });
            case "azure-native:containerregistry:ExportPipeline":
                return new exportPipeline_1.ExportPipeline(name, undefined, { urn });
            case "azure-native:containerregistry:ImportPipeline":
                return new importPipeline_1.ImportPipeline(name, undefined, { urn });
            case "azure-native:containerregistry:PipelineRun":
                return new pipelineRun_1.PipelineRun(name, undefined, { urn });
            case "azure-native:containerregistry:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:containerregistry:Registry":
                return new registry_1.Registry(name, undefined, { urn });
            case "azure-native:containerregistry:Replication":
                return new replication_1.Replication(name, undefined, { urn });
            case "azure-native:containerregistry:ScopeMap":
                return new scopeMap_1.ScopeMap(name, undefined, { urn });
            case "azure-native:containerregistry:Task":
                return new task_1.Task(name, undefined, { urn });
            case "azure-native:containerregistry:TaskRun":
                return new taskRun_1.TaskRun(name, undefined, { urn });
            case "azure-native:containerregistry:Token":
                return new token_1.Token(name, undefined, { urn });
            case "azure-native:containerregistry:Webhook":
                return new webhook_1.Webhook(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "containerregistry", _module);
