// @pulumi/azure-native/types/enums/insights/v20180501preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebTestKindEnum = exports.WebTestKind = exports.RequestSource = exports.PublicNetworkAccessType = exports.IngestionMode = exports.FlowType = exports.ApplicationType = void 0;
exports.ApplicationType = {
    Web: "web",
    Other: "other",
};
exports.FlowType = {
    Bluefield: "Bluefield",
};
exports.IngestionMode = {
    ApplicationInsights: "ApplicationInsights",
    ApplicationInsightsWithDiagnosticSettings: "ApplicationInsightsWithDiagnosticSettings",
    LogAnalytics: "LogAnalytics",
};
exports.PublicNetworkAccessType = {
    /**
     * Enables connectivity to Application Insights through public DNS.
     */
    Enabled: "Enabled",
    /**
     * Disables public connectivity to Application Insights through public DNS.
     */
    Disabled: "Disabled",
};
exports.RequestSource = {
    Rest: "rest",
};
exports.WebTestKind = {
    Ping: "ping",
    Multistep: "multistep",
};
exports.WebTestKindEnum = {
    Ping: "ping",
    Multistep: "multistep",
    Basic: "basic",
    Standard: "standard",
};
