// @pulumi/azure-native/alertsmanagement/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210808preview = exports.v20210808 = exports.v20210401 = exports.v20190601 = exports.v20190505preview = exports.v20190301 = exports.v20181102privatepreview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./actionRuleByName"), exports);
__exportStar(require("./getActionRuleByName"), exports);
__exportStar(require("./getSmartDetectorAlertRule"), exports);
__exportStar(require("./smartDetectorAlertRule"), exports);
// Export enums:
__exportStar(require("../types/enums/alertsmanagement"), exports);
// Export sub-modules:
const v20181102privatepreview = require("./v20181102privatepreview");
exports.v20181102privatepreview = v20181102privatepreview;
const v20190301 = require("./v20190301");
exports.v20190301 = v20190301;
const v20190505preview = require("./v20190505preview");
exports.v20190505preview = v20190505preview;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
const v20210808 = require("./v20210808");
exports.v20210808 = v20210808;
const v20210808preview = require("./v20210808preview");
exports.v20210808preview = v20210808preview;
// Import resources to register:
const actionRuleByName_1 = require("./actionRuleByName");
const smartDetectorAlertRule_1 = require("./smartDetectorAlertRule");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:alertsmanagement:ActionRuleByName":
                return new actionRuleByName_1.ActionRuleByName(name, undefined, { urn });
            case "azure-native:alertsmanagement:SmartDetectorAlertRule":
                return new smartDetectorAlertRule_1.SmartDetectorAlertRule(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "alertsmanagement", _module);
