// @pulumi/azure-native/types/enums/servicebus/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuTier = exports.SkuName = exports.PrivateLinkConnectionStatus = exports.NetworkRuleIPAction = exports.IPAction = exports.FilterType = exports.EntityStatus = exports.EndPointProvisioningState = exports.DefaultAction = exports.AccessRights = exports.v20211101 = exports.v20210601preview = exports.v20210101preview = exports.v20180101preview = exports.v20170401 = exports.v20150801 = exports.v20140901 = void 0;
// Export sub-modules:
const v20140901 = require("./v20140901");
exports.v20140901 = v20140901;
const v20150801 = require("./v20150801");
exports.v20150801 = v20150801;
const v20170401 = require("./v20170401");
exports.v20170401 = v20170401;
const v20180101preview = require("./v20180101preview");
exports.v20180101preview = v20180101preview;
const v20210101preview = require("./v20210101preview");
exports.v20210101preview = v20210101preview;
const v20210601preview = require("./v20210601preview");
exports.v20210601preview = v20210601preview;
const v20211101 = require("./v20211101");
exports.v20211101 = v20211101;
exports.AccessRights = {
    Manage: "Manage",
    Send: "Send",
    Listen: "Listen",
};
exports.DefaultAction = {
    Allow: "Allow",
    Deny: "Deny",
};
exports.EndPointProvisioningState = {
    Creating: "Creating",
    Updating: "Updating",
    Deleting: "Deleting",
    Succeeded: "Succeeded",
    Canceled: "Canceled",
    Failed: "Failed",
};
exports.EntityStatus = {
    Active: "Active",
    Disabled: "Disabled",
    Restoring: "Restoring",
    SendDisabled: "SendDisabled",
    ReceiveDisabled: "ReceiveDisabled",
    Creating: "Creating",
    Deleting: "Deleting",
    Renaming: "Renaming",
    Unknown: "Unknown",
};
exports.FilterType = {
    SqlFilter: "SqlFilter",
    CorrelationFilter: "CorrelationFilter",
};
exports.IPAction = {
    Accept: "Accept",
    Reject: "Reject",
};
exports.NetworkRuleIPAction = {
    Allow: "Allow",
};
exports.PrivateLinkConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.SkuName = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
exports.SkuTier = {
    Basic: "Basic",
    Standard: "Standard",
    Premium: "Premium",
};
