// @pulumi/azure-native/securityinsights/mcasdataConnector.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MCASDataConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Represents MCAS (Microsoft Cloud App Security) data connector.
 * API Version: 2020-01-01.
 */
class MCASDataConnector extends pulumi.CustomResource {
    /**
     * Create a MCASDataConnector resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    constructor(name, args, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.kind === undefined) && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            if ((!args || args.workspaceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'workspaceName'");
            }
            resourceInputs["dataConnectorId"] = args ? args.dataConnectorId : undefined;
            resourceInputs["dataTypes"] = args ? args.dataTypes : undefined;
            resourceInputs["kind"] = "MicrosoftCloudAppSecurity";
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["tenantId"] = args ? args.tenantId : undefined;
            resourceInputs["workspaceName"] = args ? args.workspaceName : undefined;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["dataTypes"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["kind"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["tenantId"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:securityinsights/v20190101preview:MCASDataConnector" }, { type: "azure-native:securityinsights/v20200101:MCASDataConnector" }, { type: "azure-native:securityinsights/v20210301preview:MCASDataConnector" }, { type: "azure-native:securityinsights/v20210901preview:MCASDataConnector" }, { type: "azure-native:securityinsights/v20211001:MCASDataConnector" }, { type: "azure-native:securityinsights/v20211001preview:MCASDataConnector" }, { type: "azure-native:securityinsights/v20220101preview:MCASDataConnector" }, { type: "azure-native:securityinsights/v20220401preview:MCASDataConnector" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MCASDataConnector.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing MCASDataConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        return new MCASDataConnector(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of MCASDataConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MCASDataConnector.__pulumiType;
    }
}
exports.MCASDataConnector = MCASDataConnector;
/** @internal */
MCASDataConnector.__pulumiType = 'azure-native:securityinsights:MCASDataConnector';
