// @pulumi/azure-native/app/v20220101preview/daprComponent.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DaprComponent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Dapr Component.
 *
 * @deprecated Version v20220101preview will be removed in the next major version of the provider. Upgrade to version v20220301 or later.
 */
class DaprComponent extends pulumi.CustomResource {
    /**
     * Create a DaprComponent resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20220101preview will be removed in the next major version of the provider. Upgrade to version v20220301 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("DaprComponent is deprecated: Version v20220101preview will be removed in the next major version of the provider. Upgrade to version v20220301 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.environmentName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'environmentName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["componentType"] = args ? args.componentType : undefined;
            resourceInputs["environmentName"] = args ? args.environmentName : undefined;
            resourceInputs["ignoreErrors"] = args ? args.ignoreErrors : undefined;
            resourceInputs["initTimeout"] = args ? args.initTimeout : undefined;
            resourceInputs["metadata"] = args ? args.metadata : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["scopes"] = args ? args.scopes : undefined;
            resourceInputs["secrets"] = args ? args.secrets : undefined;
            resourceInputs["version"] = args ? args.version : undefined;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["componentType"] = undefined /*out*/;
            resourceInputs["ignoreErrors"] = undefined /*out*/;
            resourceInputs["initTimeout"] = undefined /*out*/;
            resourceInputs["metadata"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["scopes"] = undefined /*out*/;
            resourceInputs["secrets"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:app:DaprComponent" }, { type: "azure-native:app/v20220301:DaprComponent" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DaprComponent.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing DaprComponent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("DaprComponent is deprecated: Version v20220101preview will be removed in the next major version of the provider. Upgrade to version v20220301 or later.");
        return new DaprComponent(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of DaprComponent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DaprComponent.__pulumiType;
    }
}
exports.DaprComponent = DaprComponent;
/** @internal */
DaprComponent.__pulumiType = 'azure-native:app/v20220101preview:DaprComponent';
