// @pulumi/azure-native/app/v20220101preview/managedEnvironmentsStorage.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ManagedEnvironmentsStorage = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Storage resource for managedEnvironment.
 *
 * @deprecated Version v20220101preview will be removed in the next major version of the provider. Upgrade to version v20220301 or later.
 */
class ManagedEnvironmentsStorage extends pulumi.CustomResource {
    /**
     * Create a ManagedEnvironmentsStorage resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20220101preview will be removed in the next major version of the provider. Upgrade to version v20220301 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("ManagedEnvironmentsStorage is deprecated: Version v20220101preview will be removed in the next major version of the provider. Upgrade to version v20220301 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.envName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'envName'");
            }
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["envName"] = args ? args.envName : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["properties"] = args ? args.properties : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["systemData"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:app:ManagedEnvironmentsStorage" }, { type: "azure-native:app/v20220301:ManagedEnvironmentsStorage" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ManagedEnvironmentsStorage.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing ManagedEnvironmentsStorage resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("ManagedEnvironmentsStorage is deprecated: Version v20220101preview will be removed in the next major version of the provider. Upgrade to version v20220301 or later.");
        return new ManagedEnvironmentsStorage(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of ManagedEnvironmentsStorage.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ManagedEnvironmentsStorage.__pulumiType;
    }
}
exports.ManagedEnvironmentsStorage = ManagedEnvironmentsStorage;
/** @internal */
ManagedEnvironmentsStorage.__pulumiType = 'azure-native:app/v20220101preview:ManagedEnvironmentsStorage';
