// @pulumi/azure-native/types/enums/recoveryservices/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecoveryPlanGroupType = exports.ProvisioningState = exports.ProtectionStatus = exports.ProtectionState = exports.ProtectedItemState = exports.ProtectedItemHealthStatus = exports.PrivateEndpointConnectionStatus = exports.PossibleOperationsDirections = exports.PolicyType = exports.OperationType = exports.MonthOfYear = exports.LicenseType = exports.LastBackupStatus = exports.InfrastructureEncryptionState = exports.HealthStatus = exports.FailoverDeploymentModel = exports.DiskAccountType = exports.DayOfWeek = exports.DataSourceType = exports.CreateMode = exports.ContainerType = exports.BackupManagementType = exports.BackupItemType = exports.AgentAutoUpdateStatus = exports.v20220201 = exports.v20220101 = exports.v20211201 = exports.v20211101preview = exports.v20211101 = exports.v20211001 = exports.v20210801 = exports.v20210701 = exports.v20210601 = exports.v20210401 = exports.v20210301 = exports.v20210210 = exports.v20210201preview = exports.v20210201 = exports.v20210101 = exports.v20201201 = exports.v20201001 = exports.v20200202 = exports.v20190615 = exports.v20190513 = exports.v20180710 = exports.v20180110 = exports.v20170701 = exports.v20161201 = exports.v20160810 = exports.v20160601 = void 0;
exports.WorkloadType = exports.WorkloadItemType = exports.WeekOfMonth = exports.SkuName = exports.SetMultiVmSyncStatus = exports.ScheduleRunType = exports.RetentionScheduleFormat = exports.RetentionDurationType = exports.ResourceIdentityType = exports.ResourceHealthStatus = exports.ReplicationProtectedItemOperation = void 0;
// Export sub-modules:
const v20160601 = require("./v20160601");
exports.v20160601 = v20160601;
const v20160810 = require("./v20160810");
exports.v20160810 = v20160810;
const v20161201 = require("./v20161201");
exports.v20161201 = v20161201;
const v20170701 = require("./v20170701");
exports.v20170701 = v20170701;
const v20180110 = require("./v20180110");
exports.v20180110 = v20180110;
const v20180710 = require("./v20180710");
exports.v20180710 = v20180710;
const v20190513 = require("./v20190513");
exports.v20190513 = v20190513;
const v20190615 = require("./v20190615");
exports.v20190615 = v20190615;
const v20200202 = require("./v20200202");
exports.v20200202 = v20200202;
const v20201001 = require("./v20201001");
exports.v20201001 = v20201001;
const v20201201 = require("./v20201201");
exports.v20201201 = v20201201;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210201 = require("./v20210201");
exports.v20210201 = v20210201;
const v20210201preview = require("./v20210201preview");
exports.v20210201preview = v20210201preview;
const v20210210 = require("./v20210210");
exports.v20210210 = v20210210;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
const v20210601 = require("./v20210601");
exports.v20210601 = v20210601;
const v20210701 = require("./v20210701");
exports.v20210701 = v20210701;
const v20210801 = require("./v20210801");
exports.v20210801 = v20210801;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
const v20211101 = require("./v20211101");
exports.v20211101 = v20211101;
const v20211101preview = require("./v20211101preview");
exports.v20211101preview = v20211101preview;
const v20211201 = require("./v20211201");
exports.v20211201 = v20211201;
const v20220101 = require("./v20220101");
exports.v20220101 = v20220101;
const v20220201 = require("./v20220201");
exports.v20220201 = v20220201;
exports.AgentAutoUpdateStatus = {
    Disabled: "Disabled",
    Enabled: "Enabled",
};
exports.BackupItemType = {
    Invalid: "Invalid",
    VM: "VM",
    FileFolder: "FileFolder",
    AzureSqlDb: "AzureSqlDb",
    SQLDB: "SQLDB",
    Exchange: "Exchange",
    Sharepoint: "Sharepoint",
    VMwareVM: "VMwareVM",
    SystemState: "SystemState",
    Client: "Client",
    GenericDataSource: "GenericDataSource",
    SQLDataBase: "SQLDataBase",
    AzureFileShare: "AzureFileShare",
    SAPHanaDatabase: "SAPHanaDatabase",
    SAPAseDatabase: "SAPAseDatabase",
};
exports.BackupManagementType = {
    Invalid: "Invalid",
    AzureIaasVM: "AzureIaasVM",
    MAB: "MAB",
    DPM: "DPM",
    AzureBackupServer: "AzureBackupServer",
    AzureSql: "AzureSql",
    AzureStorage: "AzureStorage",
    AzureWorkload: "AzureWorkload",
    DefaultBackup: "DefaultBackup",
};
exports.ContainerType = {
    Invalid: "Invalid",
    Unknown: "Unknown",
    IaasVMContainer: "IaasVMContainer",
    IaasVMServiceContainer: "IaasVMServiceContainer",
    DPMContainer: "DPMContainer",
    AzureBackupServerContainer: "AzureBackupServerContainer",
    MABContainer: "MABContainer",
    Cluster: "Cluster",
    AzureSqlContainer: "AzureSqlContainer",
    Windows: "Windows",
    VCenter: "VCenter",
    VMAppContainer: "VMAppContainer",
    SQLAGWorkLoadContainer: "SQLAGWorkLoadContainer",
    StorageContainer: "StorageContainer",
    GenericContainer: "GenericContainer",
};
exports.CreateMode = {
    Invalid: "Invalid",
    Default: "Default",
    Recover: "Recover",
};
exports.DataSourceType = {
    Invalid: "Invalid",
    VM: "VM",
    FileFolder: "FileFolder",
    AzureSqlDb: "AzureSqlDb",
    SQLDB: "SQLDB",
    Exchange: "Exchange",
    Sharepoint: "Sharepoint",
    VMwareVM: "VMwareVM",
    SystemState: "SystemState",
    Client: "Client",
    GenericDataSource: "GenericDataSource",
    SQLDataBase: "SQLDataBase",
    AzureFileShare: "AzureFileShare",
    SAPHanaDatabase: "SAPHanaDatabase",
    SAPAseDatabase: "SAPAseDatabase",
};
exports.DayOfWeek = {
    Sunday: "Sunday",
    Monday: "Monday",
    Tuesday: "Tuesday",
    Wednesday: "Wednesday",
    Thursday: "Thursday",
    Friday: "Friday",
    Saturday: "Saturday",
};
exports.DiskAccountType = {
    Standard_LRS: "Standard_LRS",
    Premium_LRS: "Premium_LRS",
    StandardSSD_LRS: "StandardSSD_LRS",
};
exports.FailoverDeploymentModel = {
    NotApplicable: "NotApplicable",
    Classic: "Classic",
    ResourceManager: "ResourceManager",
};
exports.HealthStatus = {
    Passed: "Passed",
    ActionRequired: "ActionRequired",
    ActionSuggested: "ActionSuggested",
    Invalid: "Invalid",
};
exports.InfrastructureEncryptionState = {
    Enabled: "Enabled",
    Disabled: "Disabled",
};
exports.LastBackupStatus = {
    Invalid: "Invalid",
    Healthy: "Healthy",
    Unhealthy: "Unhealthy",
    IRPending: "IRPending",
};
exports.LicenseType = {
    NotSpecified: "NotSpecified",
    NoLicenseType: "NoLicenseType",
    WindowsServer: "WindowsServer",
};
exports.MonthOfYear = {
    Invalid: "Invalid",
    January: "January",
    February: "February",
    March: "March",
    April: "April",
    May: "May",
    June: "June",
    July: "July",
    August: "August",
    September: "September",
    October: "October",
    November: "November",
    December: "December",
};
exports.OperationType = {
    Invalid: "Invalid",
    Register: "Register",
    Reregister: "Reregister",
};
exports.PolicyType = {
    Invalid: "Invalid",
    Full: "Full",
    Differential: "Differential",
    Log: "Log",
    CopyOnlyFull: "CopyOnlyFull",
    Incremental: "Incremental",
};
exports.PossibleOperationsDirections = {
    PrimaryToRecovery: "PrimaryToRecovery",
    RecoveryToPrimary: "RecoveryToPrimary",
};
exports.PrivateEndpointConnectionStatus = {
    Pending: "Pending",
    Approved: "Approved",
    Rejected: "Rejected",
    Disconnected: "Disconnected",
};
exports.ProtectedItemHealthStatus = {
    Invalid: "Invalid",
    Healthy: "Healthy",
    Unhealthy: "Unhealthy",
    NotReachable: "NotReachable",
    IRPending: "IRPending",
};
exports.ProtectedItemState = {
    Invalid: "Invalid",
    IRPending: "IRPending",
    Protected: "Protected",
    ProtectionError: "ProtectionError",
    ProtectionStopped: "ProtectionStopped",
    ProtectionPaused: "ProtectionPaused",
};
exports.ProtectionState = {
    Invalid: "Invalid",
    IRPending: "IRPending",
    Protected: "Protected",
    ProtectionError: "ProtectionError",
    ProtectionStopped: "ProtectionStopped",
    ProtectionPaused: "ProtectionPaused",
};
exports.ProtectionStatus = {
    Invalid: "Invalid",
    NotProtected: "NotProtected",
    Protecting: "Protecting",
    Protected: "Protected",
    ProtectionFailed: "ProtectionFailed",
};
exports.ProvisioningState = {
    Succeeded: "Succeeded",
    Deleting: "Deleting",
    Failed: "Failed",
    Pending: "Pending",
};
exports.RecoveryPlanGroupType = {
    Shutdown: "Shutdown",
    Boot: "Boot",
    Failover: "Failover",
};
exports.ReplicationProtectedItemOperation = {
    ReverseReplicate: "ReverseReplicate",
    Commit: "Commit",
    PlannedFailover: "PlannedFailover",
    UnplannedFailover: "UnplannedFailover",
    DisableProtection: "DisableProtection",
    TestFailover: "TestFailover",
    TestFailoverCleanup: "TestFailoverCleanup",
    Failback: "Failback",
    FinalizeFailback: "FinalizeFailback",
    ChangePit: "ChangePit",
    RepairReplication: "RepairReplication",
    SwitchProtection: "SwitchProtection",
    CompleteMigration: "CompleteMigration",
};
exports.ResourceHealthStatus = {
    Healthy: "Healthy",
    TransientDegraded: "TransientDegraded",
    PersistentDegraded: "PersistentDegraded",
    TransientUnhealthy: "TransientUnhealthy",
    PersistentUnhealthy: "PersistentUnhealthy",
    Invalid: "Invalid",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    None: "None",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
};
exports.RetentionDurationType = {
    Invalid: "Invalid",
    Days: "Days",
    Weeks: "Weeks",
    Months: "Months",
    Years: "Years",
};
exports.RetentionScheduleFormat = {
    Invalid: "Invalid",
    Daily: "Daily",
    Weekly: "Weekly",
};
exports.ScheduleRunType = {
    Invalid: "Invalid",
    Daily: "Daily",
    Weekly: "Weekly",
};
exports.SetMultiVmSyncStatus = {
    Enable: "Enable",
    Disable: "Disable",
};
exports.SkuName = {
    Standard: "Standard",
    RS0: "RS0",
};
exports.WeekOfMonth = {
    First: "First",
    Second: "Second",
    Third: "Third",
    Fourth: "Fourth",
    Last: "Last",
    Invalid: "Invalid",
};
exports.WorkloadItemType = {
    Invalid: "Invalid",
    SQLInstance: "SQLInstance",
    SQLDataBase: "SQLDataBase",
    SAPHanaSystem: "SAPHanaSystem",
    SAPHanaDatabase: "SAPHanaDatabase",
    SAPAseSystem: "SAPAseSystem",
    SAPAseDatabase: "SAPAseDatabase",
};
exports.WorkloadType = {
    Invalid: "Invalid",
    VM: "VM",
    FileFolder: "FileFolder",
    AzureSqlDb: "AzureSqlDb",
    SQLDB: "SQLDB",
    Exchange: "Exchange",
    Sharepoint: "Sharepoint",
    VMwareVM: "VMwareVM",
    SystemState: "SystemState",
    Client: "Client",
    GenericDataSource: "GenericDataSource",
    SQLDataBase: "SQLDataBase",
    AzureFileShare: "AzureFileShare",
    SAPHanaDatabase: "SAPHanaDatabase",
    SAPAseDatabase: "SAPAseDatabase",
};
