// @pulumi/azure-native/managedservices/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20220101preview = exports.v20200201preview = exports.v20190901 = exports.v20190601 = exports.v20190401preview = exports.v20180601preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getRegistrationAssignment"), exports);
__exportStar(require("./getRegistrationDefinition"), exports);
__exportStar(require("./registrationAssignment"), exports);
__exportStar(require("./registrationDefinition"), exports);
// Export sub-modules:
const v20180601preview = require("./v20180601preview");
exports.v20180601preview = v20180601preview;
const v20190401preview = require("./v20190401preview");
exports.v20190401preview = v20190401preview;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20190901 = require("./v20190901");
exports.v20190901 = v20190901;
const v20200201preview = require("./v20200201preview");
exports.v20200201preview = v20200201preview;
const v20220101preview = require("./v20220101preview");
exports.v20220101preview = v20220101preview;
// Import resources to register:
const registrationAssignment_1 = require("./registrationAssignment");
const registrationDefinition_1 = require("./registrationDefinition");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:managedservices:RegistrationAssignment":
                return new registrationAssignment_1.RegistrationAssignment(name, undefined, { urn });
            case "azure-native:managedservices:RegistrationDefinition":
                return new registrationDefinition_1.RegistrationDefinition(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "managedservices", _module);
