// @pulumi/azure-native/types/enums/kubernetes/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceIdentityType = exports.ProvisioningState = exports.AuthenticationMethod = exports.v20211001 = exports.v20210401preview = exports.v20210301 = exports.v20200101preview = void 0;
// Export sub-modules:
const v20200101preview = require("./v20200101preview");
exports.v20200101preview = v20200101preview;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
const v20210401preview = require("./v20210401preview");
exports.v20210401preview = v20210401preview;
const v20211001 = require("./v20211001");
exports.v20211001 = v20211001;
exports.AuthenticationMethod = {
    Token: "Token",
    AAD: "AAD",
};
exports.ProvisioningState = {
    Succeeded: "Succeeded",
    Failed: "Failed",
    Canceled: "Canceled",
    Provisioning: "Provisioning",
    Updating: "Updating",
    Deleting: "Deleting",
    Accepted: "Accepted",
};
exports.ResourceIdentityType = {
    None: "None",
    SystemAssigned: "SystemAssigned",
};
