// @pulumi/azure-native/recoveryservices/v20160601/vault.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vault = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../../utilities");
/**
 * Resource information, as returned by the resource provider.
 *
 * @deprecated Version v20160601 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.
 */
class Vault extends pulumi.CustomResource {
    /**
     * Create a Vault resource with the given unique name, arguments, and options.
     *
     * @param name The _unique_ name of the resource.
     * @param args The arguments to use to populate this resource's properties.
     * @param opts A bag of options that control this resource's behavior.
     */
    /** @deprecated Version v20160601 will be removed in the next major version of the provider. Upgrade to version v20180710 or later. */
    constructor(name, args, opts) {
        pulumi.log.warn("Vault is deprecated: Version v20160601 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        let resourceInputs = {};
        opts = opts || {};
        if (!opts.id) {
            if ((!args || args.resourceGroupName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'resourceGroupName'");
            }
            resourceInputs["eTag"] = args ? args.eTag : undefined;
            resourceInputs["identity"] = args ? args.identity : undefined;
            resourceInputs["location"] = args ? args.location : undefined;
            resourceInputs["resourceGroupName"] = args ? args.resourceGroupName : undefined;
            resourceInputs["sku"] = args ? args.sku : undefined;
            resourceInputs["tags"] = args ? args.tags : undefined;
            resourceInputs["vaultName"] = args ? args.vaultName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        else {
            resourceInputs["eTag"] = undefined /*out*/;
            resourceInputs["identity"] = undefined /*out*/;
            resourceInputs["location"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["properties"] = undefined /*out*/;
            resourceInputs["sku"] = undefined /*out*/;
            resourceInputs["tags"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "azure-native:recoveryservices:Vault" }, { type: "azure-native:recoveryservices/v20200202:Vault" }, { type: "azure-native:recoveryservices/v20201001:Vault" }, { type: "azure-native:recoveryservices/v20210101:Vault" }, { type: "azure-native:recoveryservices/v20210210:Vault" }, { type: "azure-native:recoveryservices/v20210301:Vault" }, { type: "azure-native:recoveryservices/v20210401:Vault" }, { type: "azure-native:recoveryservices/v20210601:Vault" }, { type: "azure-native:recoveryservices/v20210701:Vault" }, { type: "azure-native:recoveryservices/v20210801:Vault" }, { type: "azure-native:recoveryservices/v20211101preview:Vault" }, { type: "azure-native:recoveryservices/v20211201:Vault" }, { type: "azure-native:recoveryservices/v20220101:Vault" }, { type: "azure-native:recoveryservices/v20220201:Vault" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(Vault.__pulumiType, name, resourceInputs, opts);
    }
    /**
     * Get an existing Vault resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, opts) {
        pulumi.log.warn("Vault is deprecated: Version v20160601 will be removed in the next major version of the provider. Upgrade to version v20180710 or later.");
        return new Vault(name, undefined, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Vault.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Vault.__pulumiType;
    }
}
exports.Vault = Vault;
/** @internal */
Vault.__pulumiType = 'azure-native:recoveryservices/v20160601:Vault';
