// @pulumi/azure-native/types/enums/datamigration/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlSourcePlatform = exports.ServerLevelPermissionsGroup = exports.ResourceType = exports.ProjectTargetPlatform = exports.ProjectSourcePlatform = exports.BackupMode = exports.AuthenticationType = exports.v20220330preview = exports.v20220130preview = exports.v20211030preview = exports.v20210630 = exports.v20180715preview = exports.v20180419 = exports.v20180331preview = exports.v20180315preview = exports.v20171115preview = void 0;
// Export sub-modules:
const v20171115preview = require("./v20171115preview");
exports.v20171115preview = v20171115preview;
const v20180315preview = require("./v20180315preview");
exports.v20180315preview = v20180315preview;
const v20180331preview = require("./v20180331preview");
exports.v20180331preview = v20180331preview;
const v20180419 = require("./v20180419");
exports.v20180419 = v20180419;
const v20180715preview = require("./v20180715preview");
exports.v20180715preview = v20180715preview;
const v20210630 = require("./v20210630");
exports.v20210630 = v20210630;
const v20211030preview = require("./v20211030preview");
exports.v20211030preview = v20211030preview;
const v20220130preview = require("./v20220130preview");
exports.v20220130preview = v20220130preview;
const v20220330preview = require("./v20220330preview");
exports.v20220330preview = v20220330preview;
exports.AuthenticationType = {
    None: "None",
    WindowsAuthentication: "WindowsAuthentication",
    SqlAuthentication: "SqlAuthentication",
    ActiveDirectoryIntegrated: "ActiveDirectoryIntegrated",
    ActiveDirectoryPassword: "ActiveDirectoryPassword",
};
exports.BackupMode = {
    CreateBackup: "CreateBackup",
    ExistingBackup: "ExistingBackup",
};
exports.ProjectSourcePlatform = {
    SQL: "SQL",
    Unknown: "Unknown",
};
exports.ProjectTargetPlatform = {
    SQLDB: "SQLDB",
    Unknown: "Unknown",
};
exports.ResourceType = {
    SqlMi: "SqlMi",
    SqlVm: "SqlVm",
    SqlDb: "SqlDb",
};
exports.ServerLevelPermissionsGroup = {
    Default: "Default",
    MigrationFromSqlServerToAzureDB: "MigrationFromSqlServerToAzureDB",
    MigrationFromSqlServerToAzureMI: "MigrationFromSqlServerToAzureMI",
    MigrationFromMySQLToAzureDBForMySQL: "MigrationFromMySQLToAzureDBForMySQL",
};
exports.SqlSourcePlatform = {
    SqlOnPrem: "SqlOnPrem",
};
